<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link rel="shortcut icon" href="data:image/x-icon;base64,AAABAAMAEBAAAAEAIABoBAAANgAAABgYAAABACAAiAkAAJ4EAAAgIAAAAQAgAKgQAAAmDgAAKAAAABAAAAAgAAAAAQAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wCAgIAC////AP///wC0tKJHlZWSqI6OmuRxcXn9koaK9J2Ym8uNm5V73PPoFv///wAAAAAD////AP///wCAgIAC////AP//7xCHh4OodHSi/2xqw/9nZ9H/XVuU/9eFqv/njb7/yYiq/5GAiOifqKJV////AAAAAAL///8A////AO3t2w56eoHPdHLH/3Vz3vx4dtz7dXTb/mVimv/XhKn/8ZPG/fSVx/vvlcT/kG+E/32Himb///8AAAAAA////wB0dHageHbO/3585fl5d9r/dnTX/3V02/5mYpz80X2i/eaIuv/ukcD/wXak/I6Maf+OkXD9iIKjL////wCEhFo+c3Kv/4F+6P18etz/e3nd/3l33P57eeT/aWag/9F5oP/sib3+tmuZ/3uEXf/P1YX7zdKG/3x9dbX///8Ac3J9nYKA4f+CgOb7fnzf/3174P58et//bm2v8Vtgc7SdbIHNqWmM/4aLZP/L0YT/0dWH/tfdiP+go3P6YlyWJ3BvmduGg+n/g4Hk/oF+4/6AfuX/bm2k7pCUUUX///8Ajv/jCVBmXqPBxoH/1NmI/c7Uhf/b4Yn9wMWB/2BdcGBraqX6iYbs/4SB5P+Eguj7hYPm/2Vlcov///8AACRtB////wAAAIAWrLB29uHojv/X3on+3eSK+9PYh/9XWVN7bWum+ouI7/+Gg+f/hoTq+4iF6f9lZXKL////ACQAbQf///8AAABoFoF+Xva0snL/y818/uDmiPvZ34n/WVlTe3Jxm9uMiu//iofr/oiG6f6Jhu3/cnGo7oyQQ0X///8A/znjCXF8dKNotpL/Za2O/WuVe/+FkGv9mJ5w/2pqaGBzc3+djYrr/42K8fuKh+v/i4js/oqH6/93dbfxYmJztGaTec1yxJr/gOGu/4nptf+K6Lf+i+q8/2eXhfqDNG8ngIBKPnt6uP+Rjvf9jInr/4yJ7f+Miu3+jor0/25xoP920pz/gOGw/oHerv+H4rL/kPC9+5Ttvf98ioC1////AP///wB0dHSgioff/5SR+/mOi+7/jYvt/4+L8P5scJ78etKe/YXis/+H47P/kO+8/JLxv/92m4f9nXKNL////wD///8A7e3bDnt7hM+Ihdv/ko/4/JSS+PuUkPb+bnKg/4Lapv+Q7739kfO/+43puf9/qJH/mYeUZv///wAAAAADgICAAv///wD//+8QhoaBqIB/sf+Iht7/jYjw/2tvnf+D3Kf/ieS1/4bIpP+AkYfoqJOfVf///wAAAAAC////AP///wCAgIAC////AP///wCwsJdHlZWQqJORn+RzdHv9hZOK9Jecmcuej5l7/+jzFv///wAAAAAD////AP///wD4PwAA4A8AAMAHAACAAwAAgAEAAAABAAADgQAAA8EAAAPBAAADgQAAAAEAAIABAACAAwAAwAcAAOAPAAD4PwAAKAAAABgAAAAwAAAAAQAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wD///8A////AP///wEAAAAB////AP///wi6urdDubmxj6ennsuQkIzvc3Nz/YSIh/WWoZzWrrezoLe6t1X///8T////AP///wCAgIAC////AP///wD///8A////AP///wD///8A////Af///wD///8Ax8fDQJeYkLV4eH39WlmM/19dpP9dX6b/WFRr/611j/+/fJ//om2I/4N3ff+Jko7Ntri2Xv///wD///8AgICAAv///wD///8A////AP///wD///8B////AP///wOfn5t9c3N29GRjn/9oZ8P/c3LU/XVy3P9rbdH/YVt8/9eIrv/5mcz/75bE/t+Mt/67epz/fnN5/4uTkKP///8W////AAAAAAL///8A////AP///wH///8A////A5iYkZJoaH3/bWvB/3h23Px4dt3+dHPX/3Rx1v9sbcv/YFp6/82ApP/rj8D/65DA//KVxv/1mcn83424/5t1iP98gX7A////F////wCAgIAC////AAAAAAL///8AkJCHdWZlf/9zcs7/fHrj/Hh32/92ddn/dXTX/3Zz2P9vcM//YFt7/8t9of/pi73/54u7/+eNvf/vk8P/7ZbE/pxsh/xLVU7/iomJqP///wD///8A////Af///wCfn5Y4ZmV07XZ0zv9+fOX8enjc/3p43P94dtv/dnXZ/3d12f9xcdD/YFp6/8h5nf/mh7r/44e3/+qMvf/ljrv/jV59/3F+V/68wH3/gYNp/4uIlWf///8AAAAAAv///wJ3d22pc3G6/4KA5/t9e9//fHre/3t53f96eNz/eHbZ/3l23P50dNb9Ylx+/ch2m/3jg7b+44W2/+CItv+NXXz/bnpW/8XIgv/Y3Yv8ur57/3Fxbtni4uIa////AHl5XTdoZ4b4f3zb/4F+5P5+fN//fXvf/3x63v96edz/ennf/nt54P9xccb/Xlxy/7Rvj//gf7P/3IKw/olZeP9te1b/xsqD/9DVh//P1IX/09iG/pWYcf9nZ3Nt////AISEb35ycLb/hYPr/YB+4f9/feH/fnzh/3173/99e+L+enjX/2dmmfhlZXiwYGBiknJkaqGVZ37pg11x/3eBXP7GyoP/0NWH/87Uhf/R1ob/2eCK+8HFff9tbXWv////BG9vdb16eMn/hYPq/4F/4/+BfuP/gH7i/4B94v59e93/Y2KO+ISEbX7y8uQT////AP///weAjo5aT1tP5L7Cf//T2Ij+ztOF/9HWhv/S14b/2d+K/s3Sgv+BgnLcg4OSI11deeaBf9r/hYPo/4OB5P+DgeX/gX/i/4OB6v10c77/cXFjmv///wL///8A////AP///wD///8AeHN9apuecP/U2Yf/0teG/tPZhv/V24f/2uCJ/tTahv+KjGj6AABINV9egfuGg+P/hoTp/4WC5v+EgeX/goDk/4WD6/1wbq//WFgAQ////wAzMzMFAAAAAVVVVQP///8AAABjEoiLZ/Df5Y7/2d+J/dfch//W3If/2d+H/9vhif6ZnW3/AAA5P19egvuHheX/iIXq/4eE6P+Gg+f/hILl/4eE7P1xb7D/WFgAQ////wAzMzMFAAAAAVVVVQP///8AAABVEnd1WvDEx37/y9GD/dnfif/g54z/4OeL/97kiv6anWz/AAA1P15eeuaFg93/ioft/4iF6f+Hhen/hoPn/4mG7/14d8L/cXFimv///wP///8A////AP///wD///8AenV3a1dyZf9QY1//bGVW/pORY/+trnD/xch7/tTZhP+Pkmn6AABINXBwdb1/fc7/jYry/4mH6v+Jhur/iIbp/4iG6/6Gg+X/Z2aT+IKEaX7y8uQT////AP///wiXgY9bXndp5HXLnv95z6P+dsif/2Wrjv9RcWv/Z3Rk/nyBYv9xcWbbfHyDI4SEa354d73/kI31/YuI6/+LiOz/iofr/4mG6v+Lh+3+hoPi/21soPhnZ3qwYGBikmNza6JhinXpcsWa/3vdq/6C4a//iOi1/4zpt/+O6Lf/kOu8+3O1l/9jVmKv////BHl5WDdraor4iofn/4+M8v6Miez/i4nt/4uI7P+LiOv/i4jt/ouJ8P9+e83/XGBt/223jv952aj/e9uq/n3aqv+D36//huKy/4rntv+Q7r3/lvTC/nupj/9zYmxt////AP///wJ3d2qpfXvF/5KP9vyNiu7/jYru/4yJ7f+Miu7/i4jr/46L8P6FgN79XmZ5/XTMm/1+3q7+f9ys/4Lgr/+G5LP/iue2/4zot/+V9cL8iNWs/3BxcNni4uIa////AP///wCfn5I4aGd17YeE3v+Ukfj8jovu/46L7/+Oi+//jYru/46M8f+FgNv/XmV3/3jMnf+D4rH/g+Cw/4fks/+K57b/jem4/5TywP6Q57n/bYh5/5KIjWf///8AAAAAAgAAAAL///8AkJCFdWtqhf+IheL/lZL5/JCN8P+PjO//j4zv/5GO8/+Hgt3/X2Z4/33Rof+I5rb/h+Sz/4nmtf+N6rn/lPPB/pHruvxxm4T/h3yDqP///wD///8A////Af///wH///8A////A5aWj5JtbIP/g4DW/5SR9vyVkvj+kY7y/5KP9P+Ig97/X2d4/4DUpf+N6rn/jeq4/5Lwvv+U88D8h9qu/3KVgv+Lgoe/////F////wCAgIAC////AP///wD///8B////AP///wOfn5l9dHR39HVzsf+Fg97/k4/x/ZiW/v+Nh+f/YWl7/4ngrv+W+MX/kuy8/onesP52uZb/dH54/5aOk6P///8W////AAAAAAL///8A////AP///wD///8A////Af///wD///8Ax8fDQJWVjbV5eX79amic/3h2vv93crf/WV9r/3W1kP95vZn/aZ2C/3eCfP+Ui5DNu7i7Xv///wD///8AgICAAv///wD///8A////AP///wD///8A////AP///wEAAAAB////AP///wi6urdDuLitj6WmmcuPj4rvc3Nz/YiEh/Whlp3Wt66zoLq3ulX///8T////AP///wD///8C////AP///wD///8A////AP8B/wD8AH8A+AAfAOAADwDgAAcAwAAHAIAAAwCAAAMAgAABAAB8AQAAfgEAAP4BAAD+AQAAfgEAAHwBAIAAAQCAAAMAgAADAMAABwDgAAcA4AAPAPgAHwD8AH8A/wH/ACgAAAAgAAAAQAAAAAEAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A////AP///wD///8A////AP///wD///8A////AP///wD///8A////KO3t7XXFxcWwqqqq25eXl/RwcHD9mJiY9Kurq9vHx8ew8PDwdf///yj///8A////AP///wD///8A////AP///wD///8A////AP///wD///8AAAAAAP///wD///8A////AP///wD///8A////AP///wD///8A9/f3TLS0tLpvb3H/UVFR/1VVWf9bW27/WVlz/1FRUf96ZG//c2Nr/1lWV/9RUVH/d3V2/7m5ubr5+flM////AP///wD///8A////AP///wD///8A////AP///wAAAAAA////AP///wD///8A////AP///wD///8A/f39HLW1tatbW13/V1de/2FgkP9rabv/cG7S/29u0/9paL7/UVFR/9GIrv/wlcX/75bE/8+Irf+ZcYb/XVhb/2FfYP+8vLyr/v7+HP///wD///8A////AP///wD///8A////AAAAAAD///8A////AP///wD///8A////AOvr60l2dnnpVVVZ/2ZlnP9ycdL/c3LW/3Jx1f9xcNT/cG/T/2ppvv9RUVH/zoSs/+yRwf/uk8L/7pPC/++UxP/pk8D/pXaP/1hVV/9/fn/p7+/vSf///wD///8A////AP///wD///8AAAAAAP///wD///8A////AP///wDg4OBWX19h/11dcv9zccr/d3XZ/3Z02P91c9f/dHLX/3Nx1v9xcNT/a2q//1FRUf/Ngqn/6o6+/+uQwP/tksH/7pPD/++Vxf/xl8b/3I64/3Jiav9lY2T/2dnZVv///wD///8A////AP///wAAAAAA////AP///wD///8A5OTkQ1tbXf9hYIH/eHfY/3l32/94dtr/d3ba/3Z12f90c9f/c3LW/3Nx1f9tbMH/UVFR/8l/pf/mirr/6Iy8/+mOvv/qj7//7JHB/++VxP/xl8b/04qx/1RUUv9ZWVj/6+vrQ////wD///8A////AAAAAAD///8A////APX19Q9mZmjfYF98/3t53P97ed3/enjc/3l32/93dtr/dnXZ/3Z02P91c9f/dXTY/25tw/9RUVH/yHyj/+WIuf/libn/54q7/+mNvf/rj7//7JHB/9WJsf9XVlX/Z2hb/3BxX/9vb23f+fn5D////wD///8AAAAAAP///wD///8AkZCSnFpaZ/97edn/fHre/3t53v97ed3/enjc/3l33P94dtv/dnXZ/3V02P92dNj/cG7D/1FRUf/FeaH/4oW2/+OGt//liLj/54q6/+mOvv/Sha3/V1VV/2doW//FyYL/xcqC/19fWf+fn5+c////AP///wAAAAAA////ANjY2DhRUVL/dXPA/3994f9+fOD/fXvf/3x63v97ed3/enjc/3l32/94dtr/eHba/3d12f9xb8T/UVFR/8J1nf/fgLH/4IOz/+KEtf/liLn/zoCq/1dVVf9naFr/w8eA/83Shf/P1Yb/q652/1JSUf/h4eE4////AAAAAAD///8Afn6ApWNigv+BfuL/gH3h/3584P99e9//fHre/3x63v97ed3/eXjc/3l32/95d9v/d3ba/3Fwxf9RUVH/wXSc/9x9r//fgbL/4YO0/8t9pv9XVVT/Z2ha/8LHgf/N0oX/ztOF/8/Uhv/S14b/c3Vh/42Ni6X///8AAAAAAOfn5xJRUVH/eXbG/4F/4/+AfuL/f33h/3584P99e+D/fHrf/3x63v97ed3/e3ne/3p43f9ycMH/Y2OR/1FRUf+NY3j/w3Sd/9x9rv/IeaL/V1VU/2doWv/DyIH/zNGF/87Thf/Q1Yb/0daG/9TZh/+ytnj/UlJS/+/v7xIAAAAAq6urYF1cav+DgeX/goDk/4F/4/+AfuL/f33h/3994f9+fOD/fXvf/3x63v95d9X/YF99/1FRUb9sbG+AZmVmgG1qa4BRUVG/fWBv/1dVVP9naFr/xMiB/83Shv/O04b/z9WG/9HWhv/S14b/1NqH/9TZhv9fYFn/t7e3YAAAAAB6enyeaGeV/4SC5v+DgeX/goDk/4F+4/+BfuP/f33h/3584P9+fOD/fHrb/1xcbv9sbG6g39/fQP///wD///8A////ANra2kBbWlugVVVT/8LHgf/O04X/ztOF/8/Uhf/Q1Yb/0teG/9PZh//V24f/2N6I/4aIZv+IiIeeAAAAAFpaXMx0crX/hYLm/4SC5v+DgeX/goDk/4KA5P+Bf+P/gH7i/3994f9oZ5v/X19hv/Hx8SD///8A////AP///wD///8A////AOfn5yBcXFq/m55w/9DVhv/R1ob/0teG/9LXhv/T2Yb/1dqH/9jdiP/Z34j/o6Zy/2VlY8wAAAAAUVFR7Hx7zP+GhOj/hYLm/4SB5f+Egub/g4Hl/4KA5P+Bf+P/gX/j/1paZP+xsbFA////AP///wD///8A////AP///wD///8A////AKampkBqa17/0teG/9LXhv/T2Yf/1dqH/9bciP/Y3oj/2N6H/9rgiP+5vXr/UlJS7AAAAABRUVH7gX7W/4eE6P+GhOj/hYPn/4WC5v+EgeX/g4Hl/4OB5f9/fNv/UVFR/9zc3AD///8A////AP///wD///8A////AP///wD///8A0NDQAFRUU//R14b/1dqH/9Xbh//W3If/192H/9jeh//a4Ij/3OKI/8PJfv9RUVH7AAAAAFFRUfuCgNj/iIbq/4eE6P+HhOj/hoPn/4WC5v+Egub/hILm/4B+3f9RUVH/3NzcAP///wD///8A////AP///wD///8A////AP///wDLy8sAUlJR/87ThP/W3Ij/192H/9nfiP/Z34j/2uCI/9vhiP/d44j/xMh+/1FRUfsAAAAAUVFR7H99zv+Jh+v/iIXq/4iF6v+Hhen/hoTo/4aD5/+Fgub/hYLm/1taZP+xsbFA////AP///wD///8A////AP///wD///8A////AJWVlUBSUlH/UlJR/25vXf+Ul2z/ur97/9jeh//c44j/3uWJ/97lif+8wXr/UVFR7AAAAABaWlzMd3W4/4qH6/+Kh+v/iYbq/4iF6f+HhOj/iIXp/4aE6P+GhOj/a2qe/19fYr/x8fEg////AP///wD///8A////AP///wDp6ekgW1xcv2SWe/9nm4D/Wm9k/1daVv9RUVH/VlZT/3p8Yf+ipnH/ys+A/6ircv9lZmPMAAAAAHp6fJ5rapf/i4js/4uJ7f+KiOz/iYbq/4mG6v+Ihur/iIbq/4eE6P+FguP/Xl5w/2xsb6Df399A////AP///wD///8A2traQGVnZqBccGX/d9Sk/3vZqf+B367/g+Gw/33No/9qk33/W2lh/1ZXVP9RUVH/U1NR/3d3dZ4AAAAAq6urYF5ea/+Ni+//jInt/4uI7P+LiOz/iofr/4mG6v+Jhur/iIXp/4iF6f+EguD/ZGSB/1FRUb9ra2+AZWdmgGpubIBRUVG/Xn1u/3TOn/9516f/ftyr/4Herf+G47L/iOa1/4vot/+Q7bz/k+++/4PHo/9SVFP/oaGhYAAAAADn5+cSUVFR/4F/z/+Niu7/jInt/4yJ7f+LiOz/i4nt/4qI7P+LiOz/iofr/4mG6v+Jhur/fnzN/2ZljP9RUVH/YpB5/2++lf921KT/edem/3vZqf+A3q3/hOGw/4bjs/+J5rX/juu6/5Dsu/+U8cD/g8Wi/1JTUv/t7e0SAAAAAP///wB+foClZ2aG/46L8P+Oi/D/jYru/4yJ7f+Mie3/i4js/4uI7P+LiOz/iofr/4qH6/+KiOz/eXjA/1FRUf9zyJz/eden/3zaqf982qn/ftys/4Hfrv+G47L/iOW0/4rntv+N6rn/ku++/5XxwP9le2//iYqKpf///wAAAAAA////ANjY2DhRUVL/gH7L/4+M8P+Oi+//jovv/42K7v+Niu7/jIru/4yK7v+Mie3/jInt/4uI7P96eMD/UVFR/3bLoP972an/ftyr/4Herv+D4K//heOy/4jmtf+L6Lf/juu5/5DtvP+V8cD/gcGf/1NUU//l5eU4////AAAAAAD///8A////AJGRkpxdXGn/jYrp/5CN8f+PjPD/j4zw/46L7/+Oi+//jYru/42K7v+Niu7/jYru/3x5wv9RUVH/ec6i/4Herv+B367/hOGw/4bjsv+I5bT/i+i3/43quf+Q7bz/k+++/4/jtv9bY17/nZ6enP///wD///8AAAAAAP///wD///8A9vb2D2Zmad9mZYL/kI3w/5CN8f+QjfH/j4zw/4+M8P+PjPD/j4zw/46L7/+Oi+//fHrD/1FRUf9/0qb/hOGw/4Xjsv+F47L/iOa1/4vot/+O67r/kO28/5Lvvf+S7b3/ZHht/3N1dN/6+voP////AP///wAAAAAA////AP///wD///8A5OTkQ1xcXv9oZ4j/kI3v/5GN8f+RjfH/kI3x/5CN8f+QjfH/kI3x/5CN8f99e8P/UVFR/3/UqP+I5bT/iOW0/4vot/+L6Lf/juu5/4/su/+S773/kem6/2Z9cf9gYmH/6+vrQ////wD///8A////AAAAAAD///8A////AP///wD///8A4ODgVl9fYf9iYnf/iojh/5KP8/+Sj/P/kY7y/5GO8v+RjvL/kY7y/358xf9RUVH/hdmt/4vot/+O67n/juu5/5Dtu/+Q7bv/ku++/4rXrv9hcGj/aGpp/+np6Vb///8A////AP///wD///8AAAAAAP///wD///8A////AP///wD///8A7OzsSXd3eelWVlr/dXOr/5CO7v+TkPT/k5D0/5KP8/+Sj/P/f33F/1FRUf+I267/kO28/5Dsu/+Q7Lv/ku+9/5Douv9zoon/U1VU/4GDgunx8fFJ////AP///wD///8A////AP///wAAAAAA////AP///wD///8A////AP///wD///8A/f39HLW1tatbW13/WVlg/3Bunv+Gg9T/k4/y/5SR9f+Bfsf/UVFR/4vfsv+S773/kOu7/4TKpf9ul4H/VlpY/19hYP/AwMCr////HP///wD///8A////AP///wD///8A////AAAAAAD///8A////AP///wD///8A////AP///wD///8A////APf390y0tLS6cG9y/1FRUf9WVlr/Y2J2/2Fgdv9RUVH/ZYBx/2N1a/9VV1b/UVFR/3R2df+/v7+6+vr6TP///wD///8A////AP///wD///8A////AP///wD///8AAAAAAP///wD///8A////AP///wD///8A////AP///wD///8A////AP///wD///8o7e3tdcXFxbCqqqrbl5eX9HBwcP2YmJj0q6ur28fHx7Dv7+91////KP///wD///8A////AP///wD///8A////AP///wD///8A////AP///wAAAAAA//Af//+AA//+AAD//AAAf/gAAD/wAAAf4AAAD8AAAAfAAAAHgAAAA4AAAAOAAAADAAfAAQAP4AEAH/ABAB/wAQAf8AEAH/ABAA/gAQAHwAGAAAADgAAAA4AAAAPAAAAHwAAAB+AAAA/wAAAf+AAAP/wAAH/+AAD//4AD///wH/8="/>
  <script id="notfound">window.onload=function(){document.body.innerHTML=""}</script>
  <script language="javascript" type="text/javascript">
{//-----------------------------------------------------------------------------
// 
// PURPOSE
// 
// Krona is a flexible tool for exploring the relative proportions of
// hierarchical data, such as metagenomic classifications, using a
// radial, space-filling display. It is implemented using HTML5 and
// JavaScript, allowing charts to be explored locally or served over the
// Internet, requiring only a current version of any major web
// browser. Krona charts can be created using an Excel template or from
// common bioinformatic formats using the provided conversion scripts.
// 
// 
// COPYRIGHT LICENSE
// 
// Copyright (c) 2011, Battelle National Biodefense Institute (BNBI);
// all rights reserved. Authored by: Brian Ondov, Nicholas Bergman, and
// Adam Phillippy
// 
// This Software was prepared for the Department of Homeland Security
// (DHS) by the Battelle National Biodefense Institute, LLC (BNBI) as
// part of contract HSHQDC-07-C-00020 to manage and operate the National
// Biodefense Analysis and Countermeasures Center (NBACC), a Federally
// Funded Research and Development Center.
// 
// Redistribution and use in source and binary forms, with or without
// modification, are permitted provided that the following conditions are
// met:
// 
// * Redistributions of source code must retain the above copyright
//   notice, this list of conditions and the following disclaimer.
// 
// * Redistributions in binary form must reproduce the above copyright
//   notice, this list of conditions and the following disclaimer in the
//   documentation and/or other materials provided with the distribution.
// 
// * Neither the name of the Battelle National Biodefense Institute nor
//   the names of its contributors may be used to endorse or promote
//   products derived from this software without specific prior written
//   permission.
// 
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
// HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
// 
// 
// TRADEMARK LICENSE
// 
// KRONA(TM) is a trademark of the Department of Homeland Security, and use
// of the trademark is subject to the following conditions:
// 
// * Distribution of the unchanged, official code/software using the
//   KRONA(TM) mark is hereby permitted by the Department of Homeland
//   Security, provided that the software is distributed without charge
//   and modification.
// 
// * Distribution of altered source code/software using the KRONA(TM) mark
//   is not permitted unless written permission has been granted by the
//   Department of Homeland Security.
// 
// 
// FOR MORE INFORMATION VISIT
// 
// https://github.com/marbl/Krona/wiki/
// 
//-----------------------------------------------------------------------------
}


var canvas;
var context;
var svg; // for snapshot mode
var collapse = true;
var collapseCheckBox;
var collapseLast;
var compress;
var compressCheckBox;
var maxAbsoluteDepthText;
var maxAbsoluteDepthButtonDecrease;
var maxAbsoluteDepthButtonIncrease;
var fontSize = 11;
var fontSizeText;
var fontSizeButtonDecrease;
var fontSizeButtonIncrease;
var fontSizeLast;
var radiusButtonDecrease;
var radiusButtonIncrease;
var shorten;
var shortenCheckBox;
var maxAbsoluteDepth;
var backButton;
var upButton;
var forwardButton;
var snapshotButton;
var snapshotMode = false;
var details;
var detailsName;
var search;
var searchResults;
var nSearchResults;
var useHueCheckBox;
var useHueDiv;
var datasetDropDown;
var datasetButtonLast;
var datasetButtonPrev;
var datasetButtonNext;
var keyControl;
var showKeys = true;
var linkButton;
var linkText;
var frame;

// Node references. Note that the meanings of 'selected' and 'focused' are
// swapped in the docs.
//
var head; // the root of the entire tree
var selectedNode = 0; // the root of the current view
var focusNode = 0; // a node chosen for more info (single-click)
var highlightedNode = 0; // mouse hover node
var highlightingHidden = false;
var nodes = new Array();
var currentNodeID = 0; // to iterate while loading

var nodeHistory = new Array();
var nodeHistoryPosition = 0;

var dataEnabled = false; // true when supplemental files are present

// store non-Krona GET variables so they can be passed on to links
//
var getVariables = new Array();

// selectedNodeLast is separate from the history, since we need to check
// properties of the last node viewed when browsing through the history
//
var selectedNodeLast = 0;
var zoomOut = false;

// temporary zoom-in while holding the mouse button on a wedge
//
var quickLook = false; // true when in quick look state
var mouseDown = false;
var mouseDownTime; // to detect mouse button hold
var quickLookHoldLength = 200;

var imageWidth;
var imageHeight;
var centerX;
var centerY;
var gRadius;
var updateViewNeeded = false;

// Determines the angle that the pie chart starts at.  90 degrees makes the
// center label consistent with the children.
//
var rotationOffset = Math.PI / 2;

var buffer;
var bufferFactor = .1;

// The maps are the small pie charts showing the current slice being viewed.
//
var mapBuffer = 10;
var mapRadius = 0;
var maxMapRadius = 25;
var mapWidth = 150;
var maxLabelOverhang = Math.PI * 4.18;

// Keys are the labeled boxes for slices in the highest level that are too thin
// to label.
//
var maxKeySizeFactor = 2; // will be multiplied by font size
var keySize;
var keys;
var keyBuffer = 10;
var currentKey;
var keyMinTextLeft;
var keyMinAngle;

var minRingWidthFactor = 5; // will be multiplied by font size
var maxPossibleDepth; // the theoretical max that can be displayed
var maxDisplayDepth; // the actual depth that will be displayed
var headerHeight = 0;//document.getElementById('options').clientHeight;
var historySpacingFactor = 1.6; // will be multiplied by font size
var historyAlphaDelta = .25;

// appearance
//
var lineOpacity = 0.3;
var saturation = 0.5;
var lightnessBase = 0.6;
var lightnessMax = .8;
var thinLineWidth = .3;
var highlightLineWidth = 1.5;
var labelBoxBuffer = 6;
var labelBoxRounding = 15;
var labelWidthFudge = 1.05; // The width of unshortened labels are set slightly
							// longer than the name width so the animation
							// finishes faster.
var fontNormal;
var fontBold;
var fontFamily = 'sans-serif';
//var fontFaceBold = 'bold Arial';
var nodeRadius;
var angleFactor;
var tickLength;
var compressedRadii;

// colors
//
var highlightFill = 'rgba(255, 255, 255, .3)';
var colorUnclassified = 'rgb(220,220,220)';

// label staggering
//
var labelOffsets; // will store the current offset at each depth
//
// This will store pointers to the last node that had a label in each offset (or "track") of a
// each depth.  These will be used to shorten neighboring labels that would overlap.
// The [nLabelNodes] index will store the last node with a radial label.
// labelFirstNodes is the same, but to check for going all the way around and
// overlapping the first labels.
//
var labelLastNodes;
var labelFirstNodes;
//
var nLabelOffsets = 3; // the number of offsets to use

var mouseX = -1;
var mouseY = -1;
var mouseXRel = -1;
var mouseYRel = -1;

// tweening
//
var progress = 0; // for tweening; goes from 0 to 1.
var progressLast = 0;
var tweenFactor = 0; // progress converted by a curve for a smoother effect.
var tweenLength = 850; // in ms
var tweenCurvature = 13;
//
// tweenMax is used to scale the sigmoid function so its range is [0,1] for the
// domain [0,1]
//
var tweenMax = 1 / (1 + Math.exp(-tweenCurvature / 2));
//
var tweenStartTime;

// for framerate debug
//
var tweenFrames = 0;
var fpsDisplay = document.getElementById('frameRate');

// Arrays to translate xml attribute names into displayable attribute names
//
var attributes = new Array();
//
var magnitudeIndex; // the index of attribute arrays used for magnitude
var membersAssignedIndex;
var membersSummaryIndex;

// For defining gradients
//
var hueDisplayName;
var hueStopPositions;
var hueStopHues;
var hueStopText;

// multiple datasets
//
var currentDataset = 0;
var lastDataset = 0;
var datasets = 1;
var datasetNames;
var datasetSelectSize = 30;
var datasetAlpha = new Tween(0, 0);
var datasetWidths = new Array();
var datasetChanged;
var datasetSelectWidth = 50;

window.onload = load;

var image;
var hiddenPattern;
var loadingImage;
var logoImage;

function backingScale()
{
	if ('devicePixelRatio' in window)
	{
		if (window.devicePixelRatio > 1)
		{
			return window.devicePixelRatio;
		}
	}
	
	return 1;
}

function resize()
{
	imageWidth = window.innerWidth;
	imageHeight = window.innerHeight;
	
	if ( ! snapshotMode )
	{
		context.canvas.width = imageWidth * backingScale();
		context.canvas.height = imageHeight * backingScale();
		context.canvas.style.width = imageWidth + "px"
		context.canvas.style.height = imageHeight + "px"
		context.scale(backingScale(), backingScale());
	}
	
	if ( datasetDropDown )
	{
		var ratio = 
			(datasetDropDown.offsetTop + datasetDropDown.clientHeight) * 2 /
			imageHeight;
		
		if ( ratio > 1 )
		{
			ratio = 1;
		}
		
		ratio = Math.sqrt(ratio);
		
		datasetSelectWidth = 
			(datasetDropDown.offsetLeft + datasetDropDown.clientWidth) * ratio;
	}
	var leftMargin = datasets > 1 ? datasetSelectWidth + 30 : 0;
	var minDimension = imageWidth - mapWidth - leftMargin > imageHeight ?
		imageHeight :
		imageWidth - mapWidth - leftMargin;
	
	maxMapRadius = minDimension * .03;
	buffer = minDimension * bufferFactor;
	margin = minDimension * .015;
	centerX = (imageWidth - mapWidth - leftMargin) / 2 + leftMargin;
	centerY = imageHeight / 2;
	gRadius = minDimension / 2 - buffer;
	//context.font = '11px sans-serif';
}

function handleResize()
{
	updateViewNeeded = true;
}

function Attribute()
{
}

function Tween(start, end)
{
	this.start = start;
	this.end = end;
	this.current = this.start;
	
	this.current = function()
	{
		if ( progress == 1 || this.start == this.end )
		{
			return this.end;
		}
		else
		{
			return this.start + tweenFactor * (this.end - this.start);
		}
	};
	
	this.setTarget = function(target)
	{
		this.start = this.current();
		this.end = target;
	}
}

function Node()
{
	this.id = currentNodeID;
	currentNodeID++;
	nodes[this.id] = this;
	
	this.angleStart = new Tween(Math.PI, 0);
	this.angleEnd = new Tween(Math.PI, 0);
	this.radiusInner = new Tween(1, 1);
	this.labelRadius = new Tween(1, 1);
	this.labelWidth = new Tween(0, 0);
	this.scale = new Tween(1, 1); // TEMP
	this.radiusOuter = new Tween(1, 1);
	
	this.r = new Tween(255, 255);
	this.g = new Tween(255, 255);
	this.b = new Tween(255, 255);
	
	this.alphaLabel = new Tween(0, 1);
	this.alphaLine = new Tween(0, 1);
	this.alphaArc = new Tween(0, 0);
	this.alphaWedge = new Tween(0, 1);
	this.alphaOther = new Tween(0, 1);
	this.alphaPattern = new Tween(0, 0);
	this.children = Array();
	this.parent = 0;
	
	this.attributes = new Array(attributes.length);
	
	this.addChild = function(child)
	{
		this.children.push(child);
	};
	
	this.addLabelNode = function(depth, labelOffset)
	{
		if ( labelHeadNodes[depth][labelOffset] == 0 )
		{
			// this will become the head node for this list
			
			labelHeadNodes[depth][labelOffset] = this;
			this.labelPrev = this;
		}
		
		var head = labelHeadNodes[depth][labelOffset];
		
		this.labelNext = head;
		this.labelPrev = head.labelPrev;
		head.labelPrev.labelNext = this;
		head.labelPrev = this;
	}
	
	this.canDisplayDepth = function()
	{
		// whether this node is at a depth that can be displayed, according
		// to the max absolute depth
		
		return this.depth <= maxAbsoluteDepth;
	}
	
	this.canDisplayHistory = function()
	{
		var radiusInner;
		
		if ( compress )
		{
			radiusInner = compressedRadii[0];
		}
		else
		{
			radiusInner = nodeRadius;
		}
		
		return (
			-this.labelRadius.end * gRadius +
			historySpacingFactor * fontSize / 2 <
			radiusInner * gRadius
			);
	}
	
	this.canDisplayLabelCurrent = function()
	{
		return (
			(this.angleEnd.current() - this.angleStart.current()) *
			(this.radiusInner.current() * gRadius + gRadius) >=
			minWidth());
	}
	
	this.checkHighlight = function()
	{
		if ( this.children.length == 0 && this == focusNode )
		{
			//return false;
		}
		
		if ( this.hide )
		{
			return false;
		}
		
		if ( this.radiusInner.end == 1 )
		{
			// compressed to the outside; don't check
			
			return false;
		}
		
		var highlighted = false;
		
		var angleStartCurrent = this.angleStart.current() + rotationOffset;
		var angleEndCurrent = this.angleEnd.current() + rotationOffset;
		var radiusInner = this.radiusInner.current() * gRadius;
		
		for ( var i = 0; i < this.children.length; i++ )
		{
			highlighted = this.children[i].checkHighlight();
			
			if ( highlighted )
			{
				return true;
			}
		}
		
		if ( this.radial )
		{
			var angleText = (angleStartCurrent + angleEndCurrent) / 2;
			var radiusText = (gRadius + radiusInner) / 2;
			
			context.rotate(angleText);
			context.beginPath();
			context.moveTo(radiusText, -fontSize);
			context.lineTo(radiusText, fontSize);
			context.lineTo(radiusText + centerX, fontSize);
			context.lineTo(radiusText + centerX, -fontSize);
			context.closePath();
			context.rotate(-angleText);
			
			if ( context.isPointInPath(mouseXRel, mouseYRel) )
			{
				var label = String(this.getPercentage()) + '%' + '   ' + this.name;
				
				if ( this.searchResultChildren() )
			    {
					label += searchResultString(this.searchResultChildren());
				}
				
				if
				(
					Math.sqrt((mouseXRel) * (mouseXRel) + (mouseYRel) * (mouseYRel)) / backingScale() <
					radiusText + measureText(label)
				)
				{
					highlighted = true;
				}
			}
		}
		else
		{
		    for ( var i = 0; i < this.hiddenLabels.length; i++ )
		    {
		        var hiddenLabel = this.hiddenLabels[i];
		        
				context.rotate(hiddenLabel.angle);
				context.beginPath();
				context.moveTo(gRadius, -fontSize);
				context.lineTo(gRadius, fontSize);
				context.lineTo(gRadius + centerX, fontSize);
				context.lineTo(gRadius + centerX, -fontSize);
				context.closePath();
				context.rotate(-hiddenLabel.angle);
				
				if ( context.isPointInPath(mouseXRel, mouseYRel) )
				{
					var label = String(hiddenLabel.value) + ' more';
					
					if ( hiddenLabel.search )
				    {
						label += searchResultString(hiddenLabel.search);
					}
					
					if
					(
						Math.sqrt((mouseXRel) * (mouseXRel) + (mouseYRel) * (mouseYRel)) / backingScale() <
						gRadius + fontSize + measureText(label)
					)
					{
						highlighted = true;
						break;
					}
				}
			}
		}
		
		if ( ! highlighted && this != selectedNode && ! this.getCollapse() )
		{
			context.beginPath();
			context.arc(0, 0, radiusInner, angleStartCurrent, angleEndCurrent, false);
			context.arc(0, 0, gRadius, angleEndCurrent, angleStartCurrent, true);
			context.closePath();
			
			if ( context.isPointInPath(mouseXRel, mouseYRel) )
			{
				highlighted = true;
			}
			
			if
			(
				! highlighted &&
				(angleEndCurrent - angleStartCurrent) *
				(radiusInner + gRadius) <
				minWidth() &&
				this.getDepth() == selectedNode.getDepth() + 1
			)
			{
				if ( showKeys && this.checkHighlightKey() )
				{
					highlighted = true;
				}
			}
		}
		
		if ( highlighted )
		{
			if ( this != highlightedNode )
			{
			//	document.body.style.cursor='pointer';
			}
			
			highlightedNode = this;
		}
		
		return highlighted;
	}
	
	this.checkHighlightCenter = function()
	{
		if ( ! this.canDisplayHistory() )
		{
			return;
		}
		
		var cx = centerX;
		var cy = centerY - this.labelRadius.end * gRadius;
		//var dim = context.measureText(this.name);
		
		var width = this.nameWidth;
		
		if ( this.searchResultChildren() )
		{
			var results = searchResultString(this.searchResultChildren());
			var dim = context.measureText(results);
			width += dim.width;
		}
		
		if
		(
			mouseX > cx - width / 2 &&
			mouseX < cx + width / 2 &&
			mouseY > cy - historySpacingFactor * fontSize / 2 &&
			mouseY < cy + historySpacingFactor * fontSize / 2
		)
		{
			highlightedNode = this;
			return;
		}
		
		if ( this.getParent() )
		{
			this.getParent().checkHighlightCenter();
		}
	}
	
	this.checkHighlightKey = function()
	{
		var offset = keyOffset();
		
		var xMin = imageWidth - keySize - margin - this.keyNameWidth - keyBuffer;
		var xMax = imageWidth - margin;
		var yMin = offset;
		var yMax = offset + keySize;
		
		currentKey++;
		
		return (
			mouseX > xMin &&
			mouseX < xMax &&
			mouseY > yMin &&
			mouseY < yMax);
	}
	
	this.checkHighlightMap = function()
	{
		if ( this.parent )
		{
			this.parent.checkHighlightMap();
		}
		
		if ( this.getCollapse() || this == focusNode )
		{
			return;
		}
		
		var box = this.getMapPosition();
		
		if
		(
			mouseX > box.x - mapRadius &&
			mouseX < box.x + mapRadius &&
			mouseY > box.y - mapRadius &&
			mouseY < box.y + mapRadius
		)
		{
			highlightedNode = this;
		}
	}
	
/*	this.collapse = function()
	{
		for (var i = 0; i < this.children.length; i++ )
		{
			this.children[i] = this.children[i].collapse();
		}
		
		if
		(
			this.children.length == 1 &&
			this.children[0].magnitude == this.magnitude
		)
		{
			this.children[0].parent = this.parent;
			this.children[0].getDepth() = this.parent.getDepth() + 1;
			return this.children[0];
		}
		else
		{
			return this;
		}
	}
*/	
	this.draw = function(labelMode, selected, searchHighlighted)
	{
		var depth = this.getDepth() - selectedNode.getDepth() + 1;
//		var hidden = false;
		
		if ( selectedNode == this )
		{
			selected = true;
		}
		
		var angleStartCurrent = this.angleStart.current() + rotationOffset;
		var angleEndCurrent = this.angleEnd.current() + rotationOffset;
		var radiusInner = this.radiusInner.current() * gRadius;
		var canDisplayLabelCurrent = this.canDisplayLabelCurrent();
		var hiddenSearchResults = false;
		
/*		if ( ! this.hide )
		{
			for ( var i = 0; i < this.children.length; i++ )
			{
				if ( this.children[i].hide && this.children[i].searchResults )
				{
					hiddenSearchResults = true;
				}
			}
		}
*/		
		var drawChildren =
			( ! this.hide || ! this.hidePrev && progress < 1 ) &&
			( ! this.hideAlone || ! this.hideAlonePrev && progress < 1 );
		
//		if ( this.alphaWedge.current() > 0 || this.alphaLabel.current() > 0 )
		{
			var lastChildAngleEnd;
			
			if ( this.hasChildren() )//canDisplayChildren )
			{
				lastChildAngleEnd =
					this.children[this.children.length - 1].angleEnd.current()
					+ rotationOffset;
			}
			
			if ( labelMode )
			{
				var drawRadial =
				!(
					this.parent &&
					this.parent != selectedNode &&
					angleEndCurrent == this.parent.angleEnd.current() + rotationOffset
				);
				
				if ( angleStartCurrent != angleEndCurrent )
				{
					this.drawLines(angleStartCurrent, angleEndCurrent, radiusInner, drawRadial, selected);
				}
				
				var alphaOtherCurrent = this.alphaOther.current();
				var childRadiusInner;
				
				if ( this == selectedNode || alphaOtherCurrent )
				{
					childRadiusInner =
						this.children[this.children.length - 1].radiusInner.current() * gRadius;
				}
				
				if ( this == selectedNode )
				{
					this.drawReferenceRings(childRadiusInner);
				}
				
				if
				(
					selected &&
					! searchHighlighted &&
					this != selectedNode &&
					(
						this.isSearchResult ||
						this.hideAlone && this.searchResultChildren() ||
						false
//						this.hide &&
//						this.containsSearchResult
					)
				)
				{
					context.globalAlpha = this.alphaWedge.current();
					
					drawWedge
					(
						angleStartCurrent,
						angleEndCurrent,
						radiusInner,
						gRadius,
						highlightFill,
						0,
						true
					);
					
					if
					(
						this.keyed &&
						! showKeys &&
						this.searchResults &&
						! searchHighlighted &&
						this != highlightedNode &&
						this != focusNode
					)
					{
						var angle = (angleEndCurrent + angleStartCurrent) / 2;
						this.drawLabel(angle, true, false, true, true);
					}
					
					//this.drawHighlight(false);
					searchHighlighted = true;
				}
				
				if
				(
					this == selectedNode ||
//					true
					//(canDisplayLabelCurrent) &&
					this != highlightedNode &&
					this != focusNode
				)
				{
					if ( this.radial != this.radialPrev && this.alphaLabel.end == 1 )
					{
						context.globalAlpha = tweenFactor;
					}
					else
					{
						context.globalAlpha = this.alphaLabel.current();
					}
					
					this.drawLabel
					(
						(angleStartCurrent + angleEndCurrent) / 2,
						this.hideAlone && this.searchResultChildren() ||
						(this.isSearchResult || hiddenSearchResults) && selected,
						this == selectedNode && ! this.radial,
						selected,
						this.radial
					);
					
					if ( this.radial != this.radialPrev && this.alphaLabel.start == 1 && progress < 1 )
					{
						context.globalAlpha = 1 - tweenFactor;
						
						this.drawLabel
						(
							(angleStartCurrent + angleEndCurrent) / 2,
							(this.isSearchResult || hiddenSearchResults) && selected,
							this == selectedNodeLast && ! this.radialPrev,
							selected,
							this.radialPrev
						);
					}
				}
				
				if
				(
					alphaOtherCurrent &&
					lastChildAngleEnd != null
				)
				{
					if
					(
						(angleEndCurrent - lastChildAngleEnd) *
						(childRadiusInner + gRadius) >=
						minWidth()
					)
					{
						//context.font = fontNormal;
						context.globalAlpha = this.alphaOther.current();
						
						drawTextPolar
						(
							this.getUnclassifiedText(),
							this.getUnclassifiedPercentage(),
							(lastChildAngleEnd + angleEndCurrent) / 2,
							(childRadiusInner + gRadius) / 2,
							true,
							false,
							false,
							0,
							0
						);
					}
				}
				
				if ( this == selectedNode && this.keyUnclassified && showKeys )
				{
					this.drawKey
					(
						(lastChildAngleEnd + angleEndCurrent) / 2,
						false,
						false
					);
				}
			}
			else
			{
				var alphaWedgeCurrent = this.alphaWedge.current();
				
				if ( alphaWedgeCurrent || this.alphaOther.current() )
				{
					var currentR = this.r.current();
					var currentG = this.g.current();
					var currentB = this.b.current();
						
					var fill = rgbText(currentR, currentG, currentB);
					
					var radiusOuter;
					var lastChildAngle;
					var truncateWedge =
					(
						this.hasChildren() &&
						! this.keyed &&
						(compress || depth < maxDisplayDepth) &&
						drawChildren
					);
					
					if ( truncateWedge )
					{
						radiusOuter = this.children[0].radiusInner.current() * gRadius;
					}
					else
					{
						radiusOuter = gRadius;
					}
					/*
					if ( this.hasChildren() )
					{
						radiusOuter = this.children[0].getUncollapsed().radiusInner.current() * gRadius + 1;
					}
					else
					{ // TEMP
						radiusOuter = radiusInner + nodeRadius * gRadius;
						
						if ( radiusOuter > gRadius )
						{
							radiusOuter = gRadius;
						}
					}
					*/
					context.globalAlpha = alphaWedgeCurrent;
					
					if ( radiusInner != radiusOuter )
					{
						drawWedge
						(
							angleStartCurrent,
							angleEndCurrent,
							radiusInner,
							radiusOuter,//this.radiusOuter.current() * gRadius,
							//'rgba(0, 200, 0, .1)',
							fill,
							this.alphaPattern.current()
						);
						
						if ( truncateWedge )
						{
							// fill in the extra space if the sum of our childrens'
							// magnitudes is less than ours
							
							if ( lastChildAngleEnd < angleEndCurrent )//&& false) // TEMP
							{
								if ( radiusOuter > 1 )
								{
									// overlap slightly to hide the seam
									
	//								radiusOuter -= 1;
								}
								
								if ( alphaWedgeCurrent < 1 )
								{
									context.globalAlpha = this.alphaOther.current();
									drawWedge
									(
										lastChildAngleEnd,
										angleEndCurrent,
										radiusOuter,
										gRadius,
										colorUnclassified,
										0
									);
									context.globalAlpha = alphaWedgeCurrent;
								}
								
								drawWedge
								(
									lastChildAngleEnd,
									angleEndCurrent,
									radiusOuter,
									gRadius,//this.radiusOuter.current() * gRadius,
									//'rgba(200, 0, 0, .1)',
									fill,
									this.alphaPattern.current()
								);
							}
						}
						
						if ( radiusOuter < gRadius )
						{
							// patch up the seam
							//
							context.beginPath();
							context.arc(0, 0, radiusOuter, angleStartCurrent/*lastChildAngleEnd*/, angleEndCurrent, false);
							context.strokeStyle = fill;
							context.lineWidth = 1;
							context.stroke();
						}
					}
					
					if ( this.keyed && selected && showKeys )//&& progress == 1 )
					{
						this.drawKey
						(
							(angleStartCurrent + angleEndCurrent) / 2,
							(
								this == highlightedNode ||
								this == focusNode ||
								this.searchResults
							),
							this == highlightedNode || this == focusNode
						);
					}
				}
			}
		}
		
		this.hiddenLabels = Array();
		
		if ( drawChildren )
		{
			// draw children
			//
			for ( var i = 0; i < this.children.length; i++ )
			{
				if ( this.drawHiddenChildren(i, selected, labelMode, searchHighlighted) )
				{
					i = this.children[i].hiddenEnd;
				}
				else
				{
					this.children[i].draw(labelMode, selected, searchHighlighted);
				}
			}
		}
	};
	
	this.drawHiddenChildren = function
	(
		firstHiddenChild,
		selected,
		labelMode,
		searchHighlighted
	)
	{
		var firstChild = this.children[firstHiddenChild];
		
		if ( firstChild.hiddenEnd == null || firstChild.radiusInner.current() == 1 )
		{
			return false;
		}
		
		for ( var i = firstHiddenChild; i < firstChild.hiddenEnd; i++ )
		{
			if ( ! this.children[i].hide || ! this.children[i].hidePrev && progress < 1 )
			{
				return false;
			}
		}
		
		var angleStart = firstChild.angleStart.current() + rotationOffset;
		var lastChild = this.children[firstChild.hiddenEnd];
		var angleEnd = lastChild.angleEnd.current() + rotationOffset;
		var radiusInner = gRadius * firstChild.radiusInner.current();
		var hiddenChildren = firstChild.hiddenEnd - firstHiddenChild + 1;
		
		if ( labelMode )
		{
			var hiddenSearchResults = 0;
			
			for ( var i = firstHiddenChild; i <= firstChild.hiddenEnd; i++ )
			{
				hiddenSearchResults += this.children[i].searchResults;
				
				if ( this.children[i].magnitude == 0 )
				{
					hiddenChildren--;
				}
			}
			
			if
			(
				selected &&
				(angleEnd - angleStart) * 
				(gRadius + gRadius) >=
				minWidth() ||
				this == highlightedNode &&
				hiddenChildren ||
				hiddenSearchResults
			)
			{
				context.globalAlpha = this.alphaWedge.current();
				
				this.drawHiddenLabel
				(
					angleStart,
					angleEnd,
					hiddenChildren,
					hiddenSearchResults
				);
			}
		}
		
		var drawWedges = true;
		
		for ( var i = firstHiddenChild; i <= firstChild.hiddenEnd; i++ )
		{
			// all hidden children must be completely hidden to draw together
			
			if ( this.children[i].alphaPattern.current() != this.children[i].alphaWedge.current() )
			{
				drawWedges = false;
				break;
			}
		}
		
		if ( labelMode )
		{
			if ( drawWedges )
			{
				var drawRadial = (angleEnd < this.angleEnd.current() + rotationOffset);
				this.drawLines(angleStart, angleEnd, radiusInner, drawRadial);
			}
			
			if ( hiddenSearchResults && ! searchHighlighted )
			{
				drawWedge
				(
					angleStart,
					angleEnd,
					radiusInner,
					gRadius,//this.radiusOuter.current() * gRadius,
					highlightFill,
					0,
					true
				);
			}
		}
		else if ( drawWedges )
		{
			context.globalAlpha = this.alphaWedge.current();
			
			var fill = rgbText
			(
				firstChild.r.current(),
				firstChild.g.current(),
				firstChild.b.current()
			);
			
			drawWedge
			(
				angleStart,
				angleEnd,
				radiusInner,
				gRadius,//this.radiusOuter.current() * gRadius,
				fill,
				context.globalAlpha,
				false
			);
		}
		
		return drawWedges;
	}
	
	this.drawHiddenLabel = function(angleStart, angleEnd, value, hiddenSearchResults)
	{
		var textAngle = (angleStart + angleEnd) / 2;
		var labelRadius = gRadius + fontSize;//(radiusInner + radius) / 2;
		
		var hiddenLabel = Array();
		
		hiddenLabel.value = value;
		hiddenLabel.angle = textAngle;
		hiddenLabel.search = hiddenSearchResults;
		
		this.hiddenLabels.push(hiddenLabel);
		
		drawTick(gRadius - fontSize * .75, fontSize * 1.5, textAngle);
		drawTextPolar
		(
			value.toString() + ' more',
			0, // inner text
			textAngle,
			labelRadius,
			true, // radial
			hiddenSearchResults, // bubble
			this == highlightedNode || this == focusNode, // bold
			false,
			hiddenSearchResults
		);
	}
	
	this.drawHighlight = function(bold)
	{
		var angleStartCurrent = this.angleStart.current() + rotationOffset;
		var angleEndCurrent = this.angleEnd.current() + rotationOffset;
		var radiusInner = this.radiusInner.current() * gRadius;
		
		//this.setHighlightStyle();
		
		if ( this == focusNode && this == highlightedNode && this.hasChildren() )
		{
//			context.fillStyle = "rgba(255, 255, 255, .3)";
			arrow
			(
				angleStartCurrent,
				angleEndCurrent,
				radiusInner
			);
		}
		else
		{
			drawWedge
			(
				angleStartCurrent,
				angleEndCurrent,
				radiusInner,
				gRadius,
				highlightFill,
				0,
				true
			);
		}
		
		// check if hidden children should be highlighted
		//
		for ( var i = 0; i < this.children.length; i++ )
		{
			if
			(
				this.children[i].getDepth() - selectedNode.getDepth() + 1 <=
				maxDisplayDepth &&
				this.children[i].hiddenEnd != null
			)
			{
				var firstChild = this.children[i];
				var lastChild = this.children[firstChild.hiddenEnd];
				var hiddenAngleStart = firstChild.angleStart.current() + rotationOffset;
				var hiddenAngleEnd = lastChild.angleEnd.current() + rotationOffset;
				var hiddenRadiusInner = gRadius * firstChild.radiusInner.current();
				
				drawWedge
				(
					hiddenAngleStart,
					hiddenAngleEnd,
					hiddenRadiusInner,
					gRadius,
					'rgba(255, 255, 255, .3)',
					0,
					true
				);
				
				if ( false && ! this.searchResults )
				{
					this.drawHiddenLabel
					(
						hiddenAngleStart,
						hiddenAngleEnd,
						firstChild.hiddenEnd - i + 1
					);
				}
				
				i = firstChild.hiddenEnd;
			}
		}
		
//			context.strokeStyle = 'black';
		context.fillStyle = 'black';
		
		var highlight = ! ( progress < 1 && zoomOut && this == selectedNodeLast );
		
		var angle = (angleEndCurrent + angleStartCurrent) / 2;
		
		if ( ! (this.keyed && showKeys) )
		{
			this.drawLabel(angle, true, bold, true, this.radial);
		}
	}
	
	this.drawHighlightCenter = function()
	{
		if ( ! this.canDisplayHistory() )
		{
			return;
		}
		
		context.lineWidth = highlightLineWidth;
		context.strokeStyle = 'black';
		context.fillStyle = "rgba(255, 255, 255, .6)";
		
		context.fillStyle = 'black';
		this.drawLabel(3 * Math.PI / 2, true, true, false);
		context.font = fontNormal;
	}
	
	this.drawKey = function(angle, highlight, bold)
	{
		var offset = keyOffset();
		var color;
		var colorText = this.magnitude == 0 ? 'gray' : 'black';
		var patternAlpha = this.alphaPattern.end;
		var boxLeft = imageWidth - keySize - margin;
		var textY = offset + keySize / 2;
		
		var label;
		var keyNameWidth;
		
		if ( this == selectedNode )
		{
			color = colorUnclassified;
			label =
				this.getUnclassifiedText() +
				'   ' +
				this.getUnclassifiedPercentage();
			keyNameWidth = measureText(label, false);
		}
		else
		{
			label = this.keyLabel;
			color = rgbText(this.r.end, this.g.end, this.b.end);
			
			if ( highlight )
			{
				if ( this.searchResultChildren() )
				{
					label = label + searchResultString(this.searchResultChildren());
				}
				
				keyNameWidth = measureText(label, bold);
			}
			else
			{
				keyNameWidth = this.keyNameWidth;
			}
		}
		
		var textLeft = boxLeft - keyBuffer - keyNameWidth - fontSize / 2;
		var labelLeft = textLeft;
		
		if ( labelLeft > keyMinTextLeft - fontSize / 2 )
		{
			keyMinTextLeft -= fontSize / 2;
			
			if ( keyMinTextLeft < centerX - gRadius + fontSize / 2 )
			{
				keyMinTextLeft = centerX - gRadius + fontSize / 2;
			}
			
			labelLeft = keyMinTextLeft;
		}
		
		var lineX = new Array();
		var lineY = new Array();
		
		var bendRadius;
		var keyAngle = Math.atan((textY - centerY) / (labelLeft - centerX));
		var arcAngle;
		
		if ( keyAngle < 0 )
		{
			keyAngle += Math.PI;
		}
		
		if ( keyMinAngle == 0 || angle < keyMinAngle )
		{
			keyMinAngle = angle;
		}
		
		if ( angle > Math.PI && keyMinAngle > Math.PI )
		{
			// allow lines to come underneath the chart
			
			angle -= Math.PI * 2;
		}
		
		lineX.push(Math.cos(angle) * gRadius);
		lineY.push(Math.sin(angle) * gRadius);
		
		if ( angle < keyAngle && textY > centerY + Math.sin(angle) * (gRadius + buffer * (currentKey - 1) / (keys + 1) / 2 + buffer / 2) )
		{
			bendRadius = gRadius + buffer - buffer * currentKey / (keys + 1) / 2;
		}
		else
		{
			bendRadius = gRadius + buffer * currentKey / (keys + 1) / 2 + buffer / 2;
		}
		
		var outside =
			Math.sqrt
			(
				Math.pow(labelLeft - centerX, 2) +
				Math.pow(textY - centerY, 2)
			) > bendRadius;
		
		if ( ! outside )
		{
			arcAngle = Math.asin((textY - centerY) / bendRadius);
			
			keyMinTextLeft = min(keyMinTextLeft, centerX + bendRadius * Math.cos(arcAngle) - fontSize / 2);
			
			if ( labelLeft < textLeft && textLeft > centerX + bendRadius * Math.cos(arcAngle) )
			{
				lineX.push(textLeft - centerX);
				lineY.push(textY - centerY);
			}
		}
		else
		{
			keyMinTextLeft = min(keyMinTextLeft, labelLeft - fontSize / 2);
			
			if ( angle < keyAngle )
			{
				// flip everything over y = x
				//
				arcAngle = Math.PI / 2 - keyLineAngle
				(
					Math.PI / 2 - angle,
					Math.PI / 2 - keyAngle,
					bendRadius,
					textY - centerY,
					labelLeft - centerX,
					lineY,
					lineX
				);
				
			}
			else
			{
				arcAngle = keyLineAngle
				(
					angle,
					keyAngle,
					bendRadius,
					labelLeft - centerX,
					textY - centerY,
					lineX,
					lineY
				);
			}
		}
		
		if ( labelLeft > centerX + bendRadius * Math.cos(arcAngle) ||
		textY > centerY + bendRadius * Math.sin(arcAngle) + .01)
//		if ( outside ||  )
		{
			lineX.push(labelLeft - centerX);
			lineY.push(textY - centerY);
			
			if ( textLeft != labelLeft )
			{
				lineX.push(textLeft - centerX);
				lineY.push(textY - centerY);
			}
		}
		
		context.globalAlpha = this.alphaWedge.current();
		
		if ( snapshotMode )
		{
			var labelSVG;
			
			if ( this == selectedNode )
			{
				labelSVG =
					this.getUnclassifiedText() +
					spacer() +
					this.getUnclassifiedPercentage();
			}
			else
			{
				labelSVG = this.name + spacer() + this.getPercentage() + '%';
			}
			
			svg +=
				'<rect fill="' + color + '" ' +
				'x="' + boxLeft + '" y="' + offset +
				'" width="' + keySize + '" height="' + keySize + '"/>';
			
			if ( patternAlpha )
			{
				svg +=
					'<rect fill="url(#hiddenPattern)" style="stroke:none" ' +
					'x="' + boxLeft + '" y="' + offset +
					'" width="' + keySize + '" height="' + keySize + '"/>';
			}
			
			svg +=
				'<path class="line' +
				(highlight ? ' highlight' : '') +
				'" d="M ' + (lineX[0] + centerX) + ',' +
				(lineY[0] + centerY);
			
			if ( angle != arcAngle )
			{
				svg +=
					' L ' + (centerX + bendRadius * Math.cos(angle)) + ',' +
					(centerY + bendRadius * Math.sin(angle)) +
					' A ' + bendRadius + ',' + bendRadius + ' 0 ' +
					'0,' + (angle > arcAngle ? '0' : '1') + ' ' +
					(centerX + bendRadius * Math.cos(arcAngle)) + ',' +
					(centerY + bendRadius * Math.sin(arcAngle));
			}
			
			for ( var i = 1; i < lineX.length; i++ )
			{
				svg +=
					' L ' + (centerX + lineX[i]) + ',' +
					(centerY + lineY[i]);
			}
			
			svg += '"/>';
			
			if ( highlight )
			{
				if ( this.searchResultChildren() )
				{
					labelSVG = labelSVG + searchResultString(this.searchResultChildren());
				}
				
				drawBubbleSVG
				(
					boxLeft - keyBuffer - keyNameWidth - fontSize / 2,
					textY - fontSize,
					keyNameWidth + fontSize,
					fontSize * 2,
					fontSize,
					0
				);
				
				if ( this.isSearchResult )
				{
					drawSearchHighlights
					(
						label,
						boxLeft - keyBuffer - keyNameWidth,
						textY,
						0
					)
				}
			}
			
			svg += svgText(labelSVG, boxLeft - keyBuffer, textY, 'end', bold, colorText);
		}
		else
		{
			context.fillStyle = color;
			context.translate(-centerX, -centerY);
			context.strokeStyle = 'black';
				context.globalAlpha = 1;//this.alphaWedge.current();
			
			context.fillRect(boxLeft, offset, keySize, keySize);
			
			if ( patternAlpha )
			{
				context.globalAlpha = patternAlpha;
				context.fillStyle = hiddenPattern;
				
				// make clipping box for Firefox performance
				context.beginPath();
				context.moveTo(boxLeft, offset);
				context.lineTo(boxLeft + keySize, offset);
				context.lineTo(boxLeft + keySize, offset + keySize);
				context.lineTo(boxLeft, offset + keySize);
				context.closePath();
				context.save();
				context.clip();
				
				context.fillRect(boxLeft, offset, keySize, keySize);
				context.fillRect(boxLeft, offset, keySize, keySize);
				
				context.restore(); // remove clipping region
			}
			
			if ( highlight )
			{
				this.setHighlightStyle();
				context.fillRect(boxLeft, offset, keySize, keySize);
			}
			else
			{
				context.lineWidth = thinLineWidth;
			}
			
			context.strokeRect(boxLeft, offset, keySize, keySize);
			
			if ( lineX.length )
			{
				context.beginPath();
				context.moveTo(lineX[0] + centerX, lineY[0] + centerY);
				
				context.arc(centerX, centerY, bendRadius, angle, arcAngle, angle > arcAngle);
				
				for ( var i = 1; i < lineX.length; i++ )
				{
					context.lineTo(lineX[i] + centerX, lineY[i] + centerY);
				}
				
				context.globalAlpha = this == selectedNode ?
					this.children[0].alphaWedge.current() :
					this.alphaWedge.current();
				context.lineWidth = highlight ? highlightLineWidth : thinLineWidth;
				context.stroke();
				context.globalAlpha = 1;
			}
			
			if ( highlight )
			{
				drawBubbleCanvas
				(
					boxLeft - keyBuffer - keyNameWidth - fontSize / 2,
					textY - fontSize,
					keyNameWidth + fontSize,
					fontSize * 2,
					fontSize,
					0
				);
				
				if ( this.isSearchResult )
				{
					drawSearchHighlights
					(
						label,
						boxLeft - keyBuffer - keyNameWidth,
						textY,
						0
					)
				}
			}
			
			drawText(label, boxLeft - keyBuffer, offset + keySize / 2, 0, 'end', bold, colorText);
			
			context.translate(centerX, centerY);
		}
		
		currentKey++;
	}
	
	this.drawLabel = function(angle, bubble, bold, selected, radial)
	{
		if ( context.globalAlpha == 0 )
		{
			return;
		}
		
		var innerText;
		var label;
		var radius;
		
		if ( radial )
		{
			radius = (this.radiusInner.current() + 1) * gRadius / 2;
		}
		else
		{
			radius = this.labelRadius.current() * gRadius;
		}
		
		if ( radial && (selected || bubble ) )
		{
			var percentage = this.getPercentage();
			innerText = percentage + '%';
		}
		
		if
		(
			! radial &&
			this != selectedNode &&
			! bubble &&
			( !zoomOut || this != selectedNodeLast)
		)
		{
			label = this.shortenLabel();
		}
		else
		{
			label = this.name;
		}
		
		var flipped = drawTextPolar
		(
			label,
			innerText,
			angle,
			radius,
			radial,
			bubble,
			bold,
//			this.isSearchResult && this.shouldAddSearchResultsString() && (!selected || this == selectedNode || highlight),
			this.isSearchResult && (!selected || this == selectedNode || bubble),
			(this.hideAlone || !selected || this == selectedNode ) ? this.searchResultChildren() : 0
		);
		
		var depth = this.getDepth() - selectedNode.getDepth() + 1;
		
		if
		(
			! radial &&
			! bubble &&
			this != selectedNode &&
			this.angleEnd.end != this.angleStart.end &&
			nLabelOffsets[depth - 2] > 2 &&
			this.labelWidth.current() > (this.angleEnd.end - this.angleStart.end) * Math.abs(radius) &&
			! ( zoomOut && this == selectedNodeLast ) &&
			this.labelRadius.end > 0
		)
		{
			// name extends beyond wedge; draw tick mark towards the central
			// radius for easier identification
			
			var radiusCenter = compress ?
				(compressedRadii[depth - 1] + compressedRadii[depth - 2]) / 2 :
				(depth - .5) * nodeRadius;
			
			if ( this.labelRadius.end > radiusCenter )
			{
				if ( flipped )
				{
					drawTick(radius - tickLength * 1.4 , tickLength, angle);
				}
				else
				{
					drawTick(radius - tickLength * 1.7, tickLength, angle);
				}
			}
			else
			{
				if ( flipped )
				{
					drawTick(radius + tickLength * .7, tickLength, angle);
				}
				else
				{
					drawTick(radius + tickLength * .4, tickLength, angle);
				}
			}
		}
	}
	
	this.drawLines = function(angleStart, angleEnd, radiusInner, drawRadial, selected)
	{
		if ( snapshotMode )
		{
			if ( this != selectedNode)
			{
				if ( angleEnd == angleStart + Math.PI * 2 )
				{
					// fudge to prevent overlap, which causes arc ambiguity
					//
					angleEnd -= .1 / gRadius;
				}
				
				var longArc = angleEnd - angleStart > Math.PI ? 1 : 0;
				
				var x1 = centerX + radiusInner * Math.cos(angleStart);
				var y1 = centerY + radiusInner * Math.sin(angleStart);
				
				var x2 = centerX + gRadius * Math.cos(angleStart);
				var y2 = centerY + gRadius * Math.sin(angleStart);
				
				var x3 = centerX + gRadius * Math.cos(angleEnd);
				var y3 = centerY + gRadius * Math.sin(angleEnd);
				
				var x4 = centerX + radiusInner * Math.cos(angleEnd);
				var y4 = centerY + radiusInner * Math.sin(angleEnd);
				
				if ( this.alphaArc.end )
				{
					var dArray =
					[
						" M ", x4, ",", y4,
						" A ", radiusInner, ",", radiusInner, " 0 ", longArc,
							" 0 ", x1, ",", y1
					];
					
					svg += '<path class="line" d="' + dArray.join('') + '"/>';
				}
				
				if ( drawRadial && this.alphaLine.end )
				{
					svg += '<line x1="' + x3 + '" y1="' + y3 + '" x2="' + x4 + '" y2="' + y4 + '"/>';
				}
			}
		}
		else
		{
			context.lineWidth = thinLineWidth;
			context.strokeStyle = 'black';
			context.beginPath();
			context.arc(0, 0, radiusInner, angleStart, angleEnd, false);
			context.globalAlpha = this.alphaArc.current();
			context.stroke();
			
			if ( drawRadial )
			{
				var x1 = radiusInner * Math.cos(angleEnd);
				var y1 = radiusInner * Math.sin(angleEnd);
				var x2 = gRadius * Math.cos(angleEnd);
				var y2 = gRadius * Math.sin(angleEnd);
				
				context.beginPath();
				context.moveTo(x1, y1);
				context.lineTo(x2, y2);
				
//				if ( this.getCollapse() )//( selected && this != selectedNode )
				{
					context.globalAlpha = this.alphaLine.current();
				}
				
				context.stroke();
			}
		}
	}
	
	this.drawMap = function(child)
	{
		if ( this.parent )
		{
			this.parent.drawMap(child);
		}
		
		if ( this.getCollapse() && this != child || this == focusNode )
		{
			return;
		}
		
		var angleStart =
			(child.baseMagnitude - this.baseMagnitude) / this.magnitude * Math.PI * 2 +
			rotationOffset;
		var angleEnd =
			(child.baseMagnitude - this.baseMagnitude + child.magnitude) /
			this.magnitude * Math.PI * 2 +
			rotationOffset;
		
		var box = this.getMapPosition();
		
		context.save();
		context.fillStyle = 'black';
		context.textAlign = 'end';
		context.textBaseline = 'middle';
		
		var textX = box.x - mapRadius - mapBuffer;
		var percentage = getPercentage(child.magnitude / this.magnitude);
		
		var highlight = this == selectedNode || this == highlightedNode;
		
		if ( highlight )
		{
			context.font = fontBold;
		}
		else
		{
			context.font = fontNormal;
		}
		
		context.fillText(percentage + '% of', textX, box.y - mapRadius / 3);
		context.fillText(this.name, textX, box.y + mapRadius / 3);
		
		if ( highlight )
		{
			context.font = fontNormal;
		}
		
		if ( this == highlightedNode && this != selectedNode )
		{
			context.fillStyle = 'rgb(245, 245, 245)';
//			context.fillStyle = 'rgb(200, 200, 200)';
		}
		else
		{
			context.fillStyle = 'rgb(255, 255, 255)';
		}
		
		context.beginPath();
		context.arc(box.x, box.y, mapRadius, 0, Math.PI * 2, true);
		context.closePath();
		context.fill();
		
		if ( this == selectedNode )
		{
			context.lineWidth = 1;
			context.fillStyle = 'rgb(100, 100, 100)';
		}
		else
		{
			if ( this == highlightedNode )
			{
				context.lineWidth = .2;
				context.fillStyle = 'rgb(190, 190, 190)';
			}
			else
			{
				context.lineWidth = .2;
				context.fillStyle = 'rgb(200, 200, 200)';
			}
		}
		
		var maxDepth = this.getMaxDepth();
		
		if ( ! compress && maxDepth > maxPossibleDepth + this.getDepth() - 1 )
		{
			maxDepth = maxPossibleDepth + this.getDepth() - 1;
		}
		
		if ( this.getDepth() < selectedNode.getDepth() )
		{
			if ( child.getDepth() - 1 >= maxDepth )
			{
				maxDepth = child.getDepth();
			}
		}
		
		var radiusInner;
		
		if ( compress )
		{
			radiusInner = 0;
//				Math.atan(child.getDepth() - this.getDepth()) /
//				Math.PI * 2 * .9;
		}
		else
		{
			radiusInner =
				(child.getDepth() - this.getDepth()) /
				(maxDepth - this.getDepth() + 1);
		}
		
		context.stroke();
		context.beginPath();
		
		if ( radiusInner == 0 )
		{
			context.moveTo(box.x, box.y);
		}
		else
		{
			context.arc(box.x, box.y, mapRadius * radiusInner, angleEnd, angleStart, true);
		}
		
		context.arc(box.x, box.y, mapRadius, angleStart, angleEnd, false);
		context.closePath();
		context.fill();
		
		if ( this == highlightedNode && this != selectedNode )
		{
			context.lineWidth = 1;
			context.stroke();
		}
		
		context.restore();
	}
	
	this.drawReferenceRings = function(childRadiusInner)
	{
		if ( snapshotMode )
		{
			svg +=
				'<circle cx="' + centerX + '" cy="' + centerY +
				'" r="' + childRadiusInner + '"/>';
			svg +=
				'<circle cx="' + centerX + '" cy="' + centerY +
				'" r="' + gRadius + '"/>';
		}
		else
		{
			context.globalAlpha = 1 - this.alphaLine.current();//this.getUncollapsed().alphaLine.current();
			context.beginPath();
			context.arc(0, 0, childRadiusInner, 0, Math.PI * 2, false);
			context.stroke();
			context.beginPath();
			context.arc(0, 0, gRadius, 0, Math.PI * 2, false);
			context.stroke();
		}
	}
	
	this.getCollapse = function()
	{
		return (
			collapse &&
			this.collapse &&
			this.depth != maxAbsoluteDepth
			);
	}
	
	this.getDepth = function()
	{
		if ( collapse )
		{
			return this.depthCollapsed;
		}
		else
		{
			return this.depth;
		}
	}
	
	this.getMagnitude = function()
	{
		return this.attributes[magnitudeIndex][currentDataset];
	}
	
	this.getMapPosition = function()
	{
		return {
			x : (details.offsetLeft + details.clientWidth - mapRadius),
			y : ((focusNode.getDepth() - this.getDepth()) *
				(mapBuffer + mapRadius * 2) - mapRadius) +
				details.clientHeight + details.offsetTop
		};
	}
	
	this.getMaxDepth = function(limit)
	{
		var max;
		
		if ( collapse )
		{
			return this.maxDepthCollapsed;
		}
		else
		{
			if ( this.maxDepth > maxAbsoluteDepth )
			{
				return maxAbsoluteDepth;
			}
			else
			{
				return this.maxDepth;
			}
		}
	}
	
	this.getData = function(index, summary)
	{
		var files = new Array();
		
		if
		(
			this.attributes[index] != null &&
			this.attributes[index][currentDataset] != null &&
			this.attributes[index][currentDataset] != ''
		)
		{
			files.push
			(
				document.location +
				'.files/' +
				this.attributes[index][currentDataset]
			);
		}
		
		if ( summary )
		{
			for ( var i = 0; i < this.children.length; i++ )
			{
				files = files.concat(this.children[i].getData(index, true));
			}
		}
		
		return files;
	}
	
	this.getList = function(index, summary)
	{
		var list;
		
		if
		(
			this.attributes[index] != null &&
			this.attributes[index][currentDataset] != null
		)
		{
			list = this.attributes[index][currentDataset];
		}
		else
		{
			list = new Array();
		}
		
		if ( summary )
		{
			for ( var i = 0; i < this.children.length; i++ )
			{
				list = list.concat(this.children[i].getList(index, true));
			}
		}
		
		return list;
	}
	
	this.getParent = function()
	{
		// returns parent, accounting for collapsing or 0 if doesn't exist
		
		var parent = this.parent;
		
		while ( parent != 0 && parent.getCollapse() )
		{
			parent = parent.parent;
		}
		
		return parent;
	}
	
	this.getPercentage = function()
	{
		return getPercentage(this.magnitude / selectedNode.magnitude);
	}
	
	this.getUnclassifiedPercentage = function()
	{
		var lastChild = this.children[this.children.length - 1];
		
		return getPercentage
		(
			(
				this.baseMagnitude +
				this.magnitude -
				lastChild.magnitude -
				lastChild.baseMagnitude
			) / this.magnitude
		) + '%';
	}
	
	this.getUnclassifiedText = function()
	{
		return '[unassigned '+ this.name + ']';
	}
	
	this.getUncollapsed = function()
	{
		// recurse through collapsed children until uncollapsed node is found
		
		if ( this.getCollapse() )
		{
			return this.children[0].getUncollapsed();
		}
		else
		{
			return this;
		}
	}
	
	this.hasChildren = function()
	{
		return this.children.length && this.depth < maxAbsoluteDepth && this.magnitude;
	}
	
	this.hasParent = function(parent)
	{
		if ( this.parent )
		{
			if ( this.parent == parent )
			{
				return true;
			}
			else
			{
				return this.parent.hasParent(parent);
			}
		}
		else
		{
			return false;
		}
	}
	
	this.maxVisibleDepth = function(maxDepth)
	{
		var childInnerRadius;
		var depth = this.getDepth() - selectedNode.getDepth() + 1;
		var currentMaxDepth = depth;
		
		if ( this.hasChildren() && depth < maxDepth)
		{
			var lastChild = this.children[this.children.length - 1];
			
			if ( this.name == 'Pseudomonadaceae' )
			{
				var x = 3;
			}
			
			if
			(
				lastChild.baseMagnitude + lastChild.magnitude <
				this.baseMagnitude + this.magnitude
			)
			{
				currentMaxDepth++;
			}
			
			if ( compress )
			{
				childInnerRadius = compressedRadii[depth - 1];
			}
			else
			{
				childInnerRadius = (depth) / maxDepth;
			}
			
			for ( var i = 0; i < this.children.length; i++ )
			{
				if
				(//true ||
					this.children[i].magnitude *
					angleFactor *
					(childInnerRadius + 1) *
					gRadius >=
					minWidth()
				)
				{
					var childMaxDepth = this.children[i].maxVisibleDepth(maxDepth);
					
					if ( childMaxDepth > currentMaxDepth )
					{
						currentMaxDepth = childMaxDepth;
					}
				}
			}
		}
		
		return currentMaxDepth;
	}
	
	this.resetLabelWidth = function()
	{
		var nameWidthOld = this.nameWidth;
		
		if ( true || ! this.radial )//&& fontSize != fontSizeLast )
		{
			var dim = context.measureText(this.name);
			this.nameWidth = dim.width;
		}
		
		if ( fontSize != fontSizeLast && this.labelWidth.end == nameWidthOld * labelWidthFudge )
		{
			// font size changed; adjust start of tween to match
			
			this.labelWidth.start = this.nameWidth * labelWidthFudge;
		}
		else
		{
			this.labelWidth.start = this.labelWidth.current();
		}
		
		this.labelWidth.end = this.nameWidth * labelWidthFudge;
	}
	
	this.restrictLabelWidth = function(width)
	{
		if ( width < this.labelWidth.end )
		{
			this.labelWidth.end = width;
		}
	}
	
	this.search = function()
	{
		this.isSearchResult = false;
		this.searchResults = 0;
		
		if
		(
			! this.getCollapse() &&
			search.value != '' &&
			this.name.toLowerCase().indexOf(search.value.toLowerCase()) != -1
		)
		{
			this.isSearchResult = true;
			this.searchResults = 1;
			nSearchResults++;
		}
		
		for ( var i = 0; i < this.children.length; i++ )
		{
			this.searchResults += this.children[i].search();
		}
		
		return this.searchResults;
	}
	
	this.searchResultChildren = function()
	{
		if ( this.isSearchResult )
		{
			return this.searchResults - 1;
		}
		else
		{
			return this.searchResults;
		}
	}
	
	this.setDepth = function(depth, depthCollapsed)
	{
		this.depth = depth;
		this.depthCollapsed = depthCollapsed;
		
		if
		(
			this.children.length == 1 &&
//			this.magnitude > 0 &&
			this.children[0].magnitude == this.magnitude &&
			( head.children.length > 1 || this.children[0].children.length )
		)
		{
			this.collapse = true;
		}
		else
		{
			this.collapse = false;
			depthCollapsed++;
		}
		
		for ( var i = 0; i < this.children.length; i++ )
		{
			this.children[i].setDepth(depth + 1, depthCollapsed);
		}
	}
	
	this.setHighlightStyle = function()
	{
		context.lineWidth = highlightLineWidth;
		
		if ( this.hasChildren() || this != focusNode || this != highlightedNode )
		{
			context.strokeStyle = 'black';
			context.fillStyle = "rgba(255, 255, 255, .3)";
		}
		else
		{
			context.strokeStyle = 'rgb(90,90,90)';
			context.fillStyle = "rgba(155, 155, 155, .3)";
		}
	}
	
	this.setLabelWidth = function(node)
	{
		if ( ! shorten || this.radial )
		{
			return; // don't need to set width
		}
		
		if ( node.hide )
		{
			alert('wtf');
			return;
		}
		
		var angle = (this.angleStart.end + this.angleEnd.end) / 2;
		var a; // angle difference
		
		if ( node == selectedNode )
		{
			a = Math.abs(angle - node.angleOther);
		}
		else
		{
			a = Math.abs(angle - (node.angleStart.end + node.angleEnd.end) / 2);
		}
		
		if ( a == 0 )
		{
			return;
		}
		
		if ( a > Math.PI )
		{
			a = 2 * Math.PI - a;
		}
		
		if ( node.radial || node == selectedNode )
		{
			var nodeLabelRadius;
			
			if ( node == selectedNode )
			{
				// radial 'other' label
				
				nodeLabelRadius = (node.children[0].radiusInner.end + 1) / 2;
			}
			else
			{
				nodeLabelRadius = (node.radiusInner.end + 1) / 2;
			}
			
			if ( a < Math.PI / 2 )
			{
				var r = this.labelRadius.end * gRadius + .5 * fontSize
				var hypotenuse = r / Math.cos(a);
				var opposite = r * Math.tan(a);
				var fontRadius = .8 * fontSize;
				
				if
				(
					nodeLabelRadius * gRadius < hypotenuse &&
					this.labelWidth.end / 2 + fontRadius > opposite
				)
				{
					this.labelWidth.end = 2 * (opposite - fontRadius);
				}
			}
		}
		else if
		(
			this.labelRadius.end == node.labelRadius.end &&
			a < Math.PI / 4
		)
		{
			// same radius with small angle; use circumferential approximation
			
			var dist = a * this.labelRadius.end * gRadius - fontSize * (1 - a * 4 / Math.PI) * 1.3;
			
			if ( this.labelWidth.end < dist )
			{
				node.restrictLabelWidth((dist - this.labelWidth.end / 2) * 2);
			}
			else if ( node.labelWidth.end < dist )
			{
				this.restrictLabelWidth((dist - node.labelWidth.end / 2) * 2);
			}
			else
			{
				// both labels reach halfway point; restrict both
				
				this.labelWidth.end = dist;
				node.labelWidth.end = dist
			}
		}
		else
		{
			var r1 = this.labelRadius.end * gRadius;
			var r2 = node.labelRadius.end * gRadius;
			
			// first adjust the radii to account for the height of the font by shifting them
			// toward each other
			//
			var fontFudge = .35 * fontSize;
			//
			if ( this.labelRadius.end < node.labelRadius.end )
			{
				r1 += fontFudge;
				r2 -= fontFudge;
			}
			else if ( this.labelRadius.end > node.labelRadius.end )
			{
				r1 -= fontFudge;
				r2 += fontFudge;
			}
			else
			{
				r1 -= fontFudge;
				r2 -= fontFudge;
			}
			
			var r1s = r1 * r1;
			var r2s = r2 * r2;
			
			// distance between the centers of the two labels
			//
			var dist = Math.sqrt(r1s + r2s - 2 * r1 * r2 * Math.cos(a));
			
			// angle at our label center between our radius and the line to the other label center
			//
			var b = Math.acos((r1s + dist * dist - r2s) / (2 * r1 * dist));
			
			// distance from our label center to the intersection of the two tangents
			//
			var l1 = Math.sin(a + b - Math.PI / 2) * dist / Math.sin(Math.PI - a);
			
			// distance from other label center the the intersection of the two tangents
			//
			var l2 = Math.sin(Math.PI / 2 - b) * dist / Math.sin(Math.PI - a);
			
			l1 = Math.abs(l1) - .4 * fontSize;
			l2 = Math.abs(l2) - .4 * fontSize;
/*			
			// amount to shorten the distances because of the height of the font
			//
			var l3 = 0;
			var fontRadius = fontSize * .55;
			//
			if ( l1 < 0 || l2 < 0 )
			{
				var l4 = fontRadius / Math.tan(a);
			l1 = Math.abs(l1);
			l2 = Math.abs(l2);
			
				l1 -= l4;
				l2 -= l4;
			}
			else
			{
				var c = Math.PI - a;
				
				l3 = fontRadius * Math.tan(c / 2);
			}
*/			
			if ( this.labelWidth.end / 2 > l1 && node.labelWidth.end / 2 > l2 )
			{
				// shorten the farthest one from the intersection
				
				if ( l1 > l2 )
				{
					this.restrictLabelWidth(2 * (l1));// - l3 - fontRadius));
				}
				else
				{
					node.restrictLabelWidth(2 * (l2));// - l3 - fontRadius));
				}
			}/*
			else if ( this.labelWidth.end / 2 > l1 + l3 && node.labelWidth.end / 2 > l2 - l3 )
			{
				node.restrictLabelWidth(2 * (l2 - l3));
			}
			else if ( this.labelWidth.end / 2 > l1 - l3 && node.labelWidth.end / 2 > l2 + l3 )
			{
				this.restrictLabelWidth(2 * (l1 - l3));
			}*/
		}
	}
	
	this.setMagnitudes = function(baseMagnitude)
	{
		this.magnitude = this.getMagnitude();
		this.baseMagnitude = baseMagnitude;
		
		for ( var i = 0; i < this.children.length; i++ )
		{
			this.children[i].setMagnitudes(baseMagnitude);
			baseMagnitude += this.children[i].magnitude;
		}
		
		this.maxChildMagnitude = baseMagnitude;
	}
	
	this.setMaxDepths = function()
	{
		this.maxDepth = this.depth;
		this.maxDepthCollapsed = this.depthCollapsed;
		
		for ( i in this.children )
		{
			var child = this.children[i];
			
			child.setMaxDepths();
			
			if ( child.maxDepth > this.maxDepth )
			{
				this.maxDepth = child.maxDepth;
			}
			
			if
			(
				child.maxDepthCollapsed > this.maxDepthCollapsed &&
				(child.depth <= maxAbsoluteDepth || maxAbsoluteDepth == 0)
			)
			{
				this.maxDepthCollapsed = child.maxDepthCollapsed;
			}
		}
	}
	
	this.setTargetLabelRadius = function()
	{
		var depth = this.getDepth() - selectedNode.getDepth() + 1;
		var index = depth - 2;
		var labelOffset = labelOffsets[index];
		
		if ( this.radial )
		{
			//this.labelRadius.setTarget((this.radiusInner.end + 1) / 2);
			var max =
				depth == maxDisplayDepth ?
				1 :
				compressedRadii[index + 1];
			
			this.labelRadius.setTarget((compressedRadii[index] + max) / 2);
		}
		else
		{
			var radiusCenter;
			var width;
			
			if ( compress )
			{
				if ( nLabelOffsets[index] > 1 )
				{
					this.labelRadius.setTarget
					(
						lerp
						(
							labelOffset + .75,
							0,
							nLabelOffsets[index] + .5,
							compressedRadii[index],
							compressedRadii[index + 1]
						)
					);
				}
				else
				{
					this.labelRadius.setTarget((compressedRadii[index] + compressedRadii[index + 1]) / 2);
				}
			}
			else
			{
				radiusCenter =
					nodeRadius * (depth - 1) +
					nodeRadius / 2;
				width = nodeRadius;
				
				this.labelRadius.setTarget
				(
					radiusCenter + width * ((labelOffset + 1) / (nLabelOffsets[index] + 1) - .5)
				);
			}
		}
		
		if ( ! this.hide && ! this.keyed && nLabelOffsets[index] )
		{
			// check last and first labels in each track for overlap
			
			for ( var i = 0; i < maxDisplayDepth - 1; i++ )
			{
				for ( var j = 0; j <= nLabelOffsets[i]; j++ )
				{
					var last = labelLastNodes[i][j];
					var first = labelFirstNodes[i][j];
					
					if ( last )
					{
						if ( j == nLabelOffsets[i] )
						{
							// last is radial
							this.setLabelWidth(last);
						}
						else
						{
							last.setLabelWidth(this);
						}
					}
					
					if ( first )
					{
						if ( j == nLabelOffsets[i] )
						{
							this.setLabelWidth(first);
						}
						else
						{
							first.setLabelWidth(this);
						}
					}
				}
			}
			
			if ( selectedNode.canDisplayLabelOther )
			{
				this.setLabelWidth(selectedNode); // in case there is an 'other' label
			}
			
			if ( this.radial )
			{
				// use the last 'track' of this depth for radial
				
				labelLastNodes[index][nLabelOffsets[index]] = this;
				
				if ( labelFirstNodes[index][nLabelOffsets[index]] == 0 )
				{
					labelFirstNodes[index][nLabelOffsets[index]] = this;
				}
			}
			else
			{
				labelLastNodes[index][labelOffset] = this;
				
				// update offset
				
				labelOffsets[index] += 1;
				
				if ( labelOffsets[index] > nLabelOffsets[index] )
				{
					labelOffsets[index] -= nLabelOffsets[index];
					
					if ( !(nLabelOffsets[index] & 1) )
					{
						labelOffsets[index]--;
					}
				}
				else if ( labelOffsets[index] == nLabelOffsets[index] )
				{
					labelOffsets[index] -= nLabelOffsets[index];
					
					if ( false && !(nLabelOffsets[index] & 1) )
					{
						labelOffsets[index]++;
					}
				}
				
				if ( labelFirstNodes[index][labelOffset] == 0 )
				{
					labelFirstNodes[index][labelOffset] = this;
				}
			}
		}
		else if ( this.hide )
		{
			this.labelWidth.end = 0;
		}
	}
	
	this.setTargets = function()
	{
		if ( this == selectedNode )
		{
			this.setTargetsSelected
			(
				0,
				1,
				lightnessBase,
				false,
				false
			);
			return;
		}
		
		var depthRelative = this.getDepth() - selectedNode.getDepth();
		
		var parentOfSelected = selectedNode.hasParent(this);
/*		(
//			! this.getCollapse() &&
			this.baseMagnitude <= selectedNode.baseMagnitude &&
			this.baseMagnitude + this.magnitude >=
			selectedNode.baseMagnitude + selectedNode.magnitude
		);
*/		
		if ( parentOfSelected )
		{
			this.resetLabelWidth();
		}
		else
		{
			//context.font = fontNormal;
			var dim = context.measureText(this.name);
			this.nameWidth = dim.width;
			//this.labelWidth.setTarget(this.labelWidth.end);
			this.labelWidth.setTarget(0);
		}
		
		// set angles
		//
		if ( this.baseMagnitude <= selectedNode.baseMagnitude )
		{
			this.angleStart.setTarget(0);
		}
		else
		{
			this.angleStart.setTarget(Math.PI * 2);
		}
		//
		if
		(
			parentOfSelected ||
			this.baseMagnitude + this.magnitude >=
			selectedNode.baseMagnitude + selectedNode.magnitude
		)
		{
			this.angleEnd.setTarget(Math.PI * 2);
		}
		else
		{
			this.angleEnd.setTarget(0);
		}
		
		// children
		//
		for ( var i = 0; i < this.children.length; i++ )
		{
			this.children[i].setTargets();
		}
		
		if ( this.getDepth() <= selectedNode.getDepth() )
		{
			// collapse in
			
			this.radiusInner.setTarget(0);
			
			if ( parentOfSelected )
			{
				this.labelRadius.setTarget
				(
					(depthRelative) *
					historySpacingFactor * fontSize / gRadius
				);
				//this.scale.setTarget(1 - (selectedNode.getDepth() - this.getDepth()) / 18); // TEMP
			}
			else
			{
				this.labelRadius.setTarget(0);
				//this.scale.setTarget(1); // TEMP
			}
		}
		else if ( depthRelative + 1 > maxDisplayDepth )
		{
			// collapse out
			
			this.radiusInner.setTarget(1);
			this.labelRadius.setTarget(1);
			//this.scale.setTarget(1); // TEMP
		}
		else
		{
			// don't collapse
			
			if ( compress )
			{
				this.radiusInner.setTarget(compressedRadii[depthRelative - 1]);
			}
			else
			{
				this.radiusInner.setTarget(nodeRadius * (depthRelative));
			}
			
			//this.scale.setTarget(1); // TEMP
			
			if ( this == selectedNode )
			{
				this.labelRadius.setTarget(0);
			}
			else
			{
				if ( compress )
				{
					this.labelRadius.setTarget
					(
						(compressedRadii[depthRelative - 1] + compressedRadii[depthRelative]) / 2
					);
				}
				else
				{
					this.labelRadius.setTarget(nodeRadius * (depthRelative) + nodeRadius / 2);
				}
			}
		}
		
//		this.r.start = this.r.end;
//		this.g.start = this.g.end;
//		this.b.start = this.b.end;
		
		this.r.setTarget(255);
		this.g.setTarget(255);
		this.b.setTarget(255);

		this.alphaLine.setTarget(0);
		this.alphaArc.setTarget(0);
		this.alphaWedge.setTarget(0);
		this.alphaPattern.setTarget(0);
		this.alphaOther.setTarget(0);
		
		if ( parentOfSelected && ! this.getCollapse() )
		{
			var alpha =
			(
				1 -
				(selectedNode.getDepth() - this.getDepth()) /
				(Math.floor((compress ? compressedRadii[0] : nodeRadius) * gRadius / (historySpacingFactor * fontSize) - .5) + 1)
			);
			
			if ( alpha < 0 )
			{
				alpha = 0;
			}
			
			this.alphaLabel.setTarget(alpha);
			this.radial = false;
		}
		else
		{
			this.alphaLabel.setTarget(0);
		}
		
		this.hideAlonePrev = this.hideAlone;
		this.hidePrev = this.hide;
		
		if ( parentOfSelected )
		{
			this.hideAlone = false;
			this.hide = false;
		}
		
		if ( this.getParent() == selectedNode.getParent() )
		{
			this.hiddenEnd = null;
		}
		
		this.radialPrev = this.radial;
	}
	
	this.setTargetsSelected = function(hueMin, hueMax, lightness, hide, nextSiblingHidden)
	{
		var collapse = this.getCollapse();
		var depth = this.getDepth() - selectedNode.getDepth() + 1;
		var canDisplayChildLabels = false;
		var lastChild;
		
		if ( this.hasChildren() )//&& ! hide )
		{
			lastChild = this.children[this.children.length - 1];
			this.hideAlone = true;
		}
		else
		{
			this.hideAlone = false;
		}
		
		// set child wedges
		//
		for ( var i = 0; i < this.children.length; i++ )
		{
			this.children[i].setTargetWedge();
			
			if
			(
				! this.children[i].hide &&
				( collapse || depth < maxDisplayDepth ) &&
				this.depth < maxAbsoluteDepth
			)
			{
				canDisplayChildLabels = true;
				this.hideAlone = false;
			}
		}
		
		if ( this == selectedNode || lastChild && lastChild.angleEnd.end < this.angleEnd.end - .01)
		{
			this.hideAlone = false;
		}
		
		if ( this.hideAlonePrev == undefined )
		{
			this.hideAlonePrev = this.hideAlone;
		}
		
		if ( this == selectedNode )
		{
			var otherArc = 
				angleFactor *
				(
					this.baseMagnitude + this.magnitude -
					lastChild.baseMagnitude - lastChild.magnitude
				);
			this.canDisplayLabelOther =
				otherArc *
				(this.children[0].radiusInner.end + 1) * gRadius >=
				minWidth();
			
			this.keyUnclassified = false;
			
			if ( this.canDisplayLabelOther )
			{
				this.angleOther = Math.PI * 2 - otherArc / 2;
			}
			else if ( otherArc > 0.0000000001 )
			{
				this.keyUnclassified = true;
				keys++;
			}
			
			this.angleStart.setTarget(0);
			this.angleEnd.setTarget(Math.PI * 2);
			this.radiusInner.setTarget(0);
			this.hidePrev = this.hide;
			this.hide = false;
			this.hideAlonePrev = this.hideAlone;
			this.hideAlone = false;
			this.keyed = false;
		}
		
		if ( hueMax - hueMin > 1 / 12 )
		{
			hueMax = hueMin + 1 / 12;
		}
		
		// set lightness
		//
		if ( ! ( hide || this.hideAlone ) )
		{
			if ( useHue() )
			{
				lightness = (lightnessBase + lightnessMax) / 2;
			}
			else
			{
				lightness = lightnessBase + (depth - 1) * lightnessFactor;
				
				if ( lightness > lightnessMax )
				{
					lightness = lightnessMax;
				}
			}
		}
		
		if ( hide )
		{
			this.hide = true;
		}
		
		if ( this.hidePrev == undefined )
		{
			this.hidePrev = this.hide;
		}
		
		var hiddenStart = -1;
		var hiddenHueNumer = 0;
		var hiddenHueDenom = 0;
		var i = 0;
		
		if ( ! this.hide )
		{
			this.hiddenEnd = null;
		}
		
		while ( true )
		{
			if ( ! this.hideAlone && ! hide && ( i == this.children.length || ! this.children[i].hide ) )
			{
				// reached a non-hidden child or the end; set targets for
				// previous group of hidden children (if any) using their
				// average hue
				
				if ( hiddenStart != -1 )
				{
					var hiddenHue = hiddenHueDenom ? hiddenHueNumer / hiddenHueDenom : hueMin;
					
					for ( var j = hiddenStart; j < i; j++ )
					{
						this.children[j].setTargetsSelected
						(
							hiddenHue,
							null,
							lightness,
							false,
							j < i - 1
						);
						
						this.children[j].hiddenEnd = null;
					}
					
					this.children[hiddenStart].hiddenEnd = i - 1;
				}
			}
			
			if ( i == this.children.length )
			{
				break;
			}
			
			var child = this.children[i];
			var childHueMin;
			var childHueMax;
			
			if ( this.magnitude > 0 && ! this.hide && ! this.hideAlone )
			{
				if ( useHue() )
				{
					childHueMin = child.hues[currentDataset];
				}
				else if ( this == selectedNode )
				{
					var min = 0.0;
					var max = 1.0;
					
					if ( this.children.length > 6 )
					{
						childHueMin = lerp((1 - Math.pow(1 - i / this.children.length, 1.4)) * .95, 0, 1, min, max);
						childHueMax = lerp((1 - Math.pow(1 - (i + .55) / this.children.length, 1.4)) * .95, 0, 1, min, max);
					}
					else
					{
						childHueMin = lerp(i / this.children.length, 0, 1, min, max);
						childHueMax = lerp((i + .55) / this.children.length, 0, 1, min, max);
					}
				}
				else
				{
					childHueMin = lerp
					(
						child.baseMagnitude,
						this.baseMagnitude, 
						this.baseMagnitude + this.magnitude,
						hueMin,
						hueMax
					);
					childHueMax = lerp
					(
						child.baseMagnitude + child.magnitude * .99,
						this.baseMagnitude,
						this.baseMagnitude + this.magnitude,
						hueMin,
						hueMax
					);
				}
			}
			else
			{
				childHueMin = hueMin;
				childHueMax = hueMax;
			}
			
			if ( ! this.hideAlone && ! hide && ! this.hide && child.hide )
			{
				if ( hiddenStart == -1 )
				{
					hiddenStart = i;
				}
				
				if ( useHue() )
				{
					hiddenHueNumer += childHueMin * child.magnitude;
					hiddenHueDenom += child.magnitude;
				}
				else
				{
					hiddenHueNumer += childHueMin;
					hiddenHueDenom++;
				}
			}
			else
			{
				hiddenStart = -1;
				
				this.children[i].setTargetsSelected
				(
					childHueMin,
					childHueMax,
					lightness,
					hide || this.keyed || this.hideAlone || this.hide && ! collapse,
					false
				);
			}
			
			i++;
		}
		
	 	if ( this.hue && this.magnitude )
	 	{
		 	this.hue.setTarget(this.hues[currentDataset]);
			
			if ( this.attributes[magnitudeIndex][lastDataset] == 0 )
			{
				this.hue.start = this.hue.end;
			}
		}
	 	
		this.radialPrev = this.radial;
		
		if ( this == selectedNode )
		{
			this.resetLabelWidth();
			this.labelWidth.setTarget(this.nameWidth * labelWidthFudge);
			this.alphaWedge.setTarget(0);
			this.alphaLabel.setTarget(1);
			this.alphaOther.setTarget(1);
			this.alphaArc.setTarget(0);
			this.alphaLine.setTarget(0);
			this.alphaPattern.setTarget(0);
			this.r.setTarget(255);
			this.g.setTarget(255);
			this.b.setTarget(255);
			this.radial = false;
			this.labelRadius.setTarget(0);
		}
		else
		{
			var rgb = hslToRgb
			(
				hueMin,
				saturation,
				lightness
			);
			
			this.r.setTarget(rgb.r);
			this.g.setTarget(rgb.g);
			this.b.setTarget(rgb.b);
			this.alphaOther.setTarget(0);
			
			this.alphaWedge.setTarget(1);
			
			if ( this.hide || this.hideAlone )
			{
				this.alphaPattern.setTarget(1);
			}
			else
			{
				this.alphaPattern.setTarget(0);
			}
			
			// set radial
			//
			if ( ! ( hide || this.hide ) )//&& ! this.keyed )
			{
				if ( this.hideAlone )
				{
					this.radial = true;
				}
				else if ( false && canDisplayChildLabels )
				{
					this.radial = false;
				}
				else
				{
					this.radial = true;
					
					if ( this.hasChildren() && depth < maxDisplayDepth )
					{
						var lastChild = this.children[this.children.length - 1];
						
						if
						(
							lastChild.angleEnd.end == this.angleEnd.end ||
							(
								(this.angleStart.end + this.angleEnd.end) / 2 -
								lastChild.angleEnd.end
							) * (this.radiusInner.end + 1) * gRadius * 2 <
							minWidth()
						)
						{
							this.radial = false;
						}
					}
				}
			}
			
			// set alphaLabel
			//
			if
			(
				collapse ||
				hide ||
				this.hide ||
				this.keyed ||
				depth > maxDisplayDepth ||
				! this.canDisplayDepth()
			)
			{
				this.alphaLabel.setTarget(0);
			}
			else
			{
				if
				(
					(this.radial || nLabelOffsets[depth - 2])
				)
				{
					this.alphaLabel.setTarget(1);
				}
				else
				{
					this.alphaLabel.setTarget(0);
					
					if ( this.radialPrev )
					{
						this.alphaLabel.start = 0;
					}
				}
			}
			
			// set alphaArc
			//
			if
			(
				collapse ||
				hide ||
				depth > maxDisplayDepth ||
				! this.canDisplayDepth()
			)
			{
				this.alphaArc.setTarget(0);
			}
			else
			{
				this.alphaArc.setTarget(1);
			}
			
			// set alphaLine
			//
			if
			(
				hide ||
				this.hide && nextSiblingHidden ||
				depth > maxDisplayDepth ||
				! this.canDisplayDepth()
			)
			{
				this.alphaLine.setTarget(0);
			}
			else
			{
				this.alphaLine.setTarget(1);
			}
			
			//if (  ! this.radial )
			{
				this.resetLabelWidth();
			}
			
			// set labelRadius target
			//
			if ( collapse )
			{
				this.labelRadius.setTarget(this.radiusInner.end);
			}
			else
			{
				if ( depth > maxDisplayDepth || ! this.canDisplayDepth() )
				{
					this.labelRadius.setTarget(1);
				}
				else
				{
					this.setTargetLabelRadius();
				}
			}
		}
	}
	
	this.setTargetWedge = function()
	{
		var depth = this.getDepth() - selectedNode.getDepth() + 1;
		
		// set angles
		//
		var baseMagnitudeRelative = this.baseMagnitude - selectedNode.baseMagnitude;
		//
		this.angleStart.setTarget(baseMagnitudeRelative * angleFactor);
		this.angleEnd.setTarget((baseMagnitudeRelative + this.magnitude) * angleFactor);
		
		// set radiusInner
		//
		if ( depth > maxDisplayDepth || ! this.canDisplayDepth() )
		{
			this.radiusInner.setTarget(1);
		}
		else
		{
			if ( compress )
			{
				this.radiusInner.setTarget(compressedRadii[depth - 2]);
			}
			else
			{
				this.radiusInner.setTarget(nodeRadius * (depth - 1));
			}
		}
		
		if ( this.hide != undefined )
		{
			this.hidePrev = this.hide;
		}
		
		if ( this.hideAlone != undefined )
		{
			this.hideAlonePrev = this.hideAlone;
		}
		
		// set hide
		//
		if
		(
			(this.angleEnd.end - this.angleStart.end) *
			(this.radiusInner.end * gRadius + gRadius) <
			minWidth()
		)
		{
			if ( depth == 2 && ! this.getCollapse() && this.depth <= maxAbsoluteDepth )
			{
				this.keyed = true;
				keys++;
				this.hide = false;
				
				var percentage = this.getPercentage();
				this.keyLabel = this.name + '   ' + percentage + '%';
				var dim = context.measureText(this.keyLabel);
				this.keyNameWidth = dim.width;
			}
			else
			{
				this.keyed = false;
				this.hide = depth > 2;
			}
		}
		else
		{
			this.keyed = false;
			this.hide = false;
		}
	}
	
	this.shortenLabel = function()
	{
		var label = this.name;
		
		var labelWidth = this.nameWidth;
		var maxWidth = this.labelWidth.current();
		var minEndLength = 0;
		
		if ( labelWidth > maxWidth && label.length > minEndLength * 2 )
		{
			var endLength =
				Math.floor((label.length - 1) * maxWidth / labelWidth / 2);
			
			if ( endLength < minEndLength )
			{
				endLength = minEndLength;
			}
			
			return (
				label.substring(0, endLength) +
				'...' +
				label.substring(label.length - endLength));
		}
		else
		{
			return label;
		}
	}
	
/*	this.shouldAddSearchResultsString = function()
	{
		if ( this.isSearchResult )
		{
			return this.searchResults > 1;
		}
		else
		{
			return this.searchResults > 0;
		}
	}
*/	
	this.sort = function()
	{
		this.children.sort(function(a, b){return b.getMagnitude() - a.getMagnitude()});
		
		for (var i = 0; i < this.children.length; i++)
		{
			this.children[i].sort();
		}
	}
}

var options;

function addOptionElement(position, innerHTML, title)
{
	var div = document.createElement("div");
//	div.style.position = 'absolute';
//	div.style.top = position + 'px';
	div.innerHTML = innerHTML;
//	div.style.display = 'block';
	div.style.padding = '2px';
	
	if ( title )
	{
		div.title = title;
	}
	
	options.appendChild(div);
	var height = 0;//div.clientHeight;
	return position + height;
}

function addOptionElements(hueName, hueDefault)
{
	options = document.createElement('div');
	options.style.position = 'absolute';
	options.style.top = '0px';
	options.addEventListener('mousedown', function(e) {mouseClick(e)}, false);
//	options.onmouseup = function(e) {mouseUp(e)}
	document.body.appendChild(options);
	
	document.body.style.font = '11px sans-serif';
	var position = 5;
	
	details = document.createElement('div');
	details.style.position = 'absolute';
	details.style.top = '1%';
	details.style.right = '2%';
	details.style.textAlign = 'right';
	document.body.insertBefore(details, canvas);
//		<div id="details" style="position:absolute;top:1%;right:2%;text-align:right;">

	details.innerHTML = '\
<span id="detailsName" style="font-weight:bold"></span>&nbsp;\
<input type="button" id="detailsExpand" onclick="expand(focusNode);"\
value="&harr;" title="Expand this wedge to become the new focus of the chart"/><br/>\
<div id="detailsInfo" style="float:right"></div>';

	keyControl = document.createElement('input');
	keyControl.type = 'button';
	keyControl.value = showKeys ? 'x' : '…';
	keyControl.style.position = '';
	keyControl.style.position = 'fixed';
	keyControl.style.visibility = 'hidden';
	
	document.body.insertBefore(keyControl, canvas);
	
	var logoElement = document.getElementById('logo');
	
	if ( logoElement )
	{
		logoImage = logoElement.src;
	}
	else
	{
		logoImage = 'http://marbl.github.io/Krona/img/logo-med.png';
	}
	
//	document.getElementById('options').style.fontSize = '9pt';
	position = addOptionElement
	(
		position,
'<a style="margin:2px" target="_blank" href="https://github.com/marbl/Krona/wiki"><img style="vertical-align:middle;width:108px;height:30px;" src="' + logoImage + '"/></a><input type="button" id="back" value="&larr;" title="Go back (Shortcut: &larr;)"/>\
<input type="button" id="forward" value="&rarr;" title="Go forward (Shortcut: &rarr;)"/> \
&nbsp;Search: <input type="text" id="search"/>\
<input id="searchClear" type="button" value="x" onclick="clearSearch()"/> \
<span id="searchResults"></span>'
	);
	
	if ( datasets > 1 )
	{
		var size = datasets < datasetSelectSize ? datasets : datasetSelectSize;
		
		var select =
			'<table style="border-collapse:collapse;padding:0px"><tr><td style="padding:0px">' +
			'<select id="datasets" style="min-width:100px" size="' + size + '" onchange="onDatasetChange()">';
		
		for ( var i = 0; i < datasetNames.length; i++ )
		{
			select += '<option>' + datasetNames[i] + '</option>';
		}
		
		select +=
			'</select></td><td style="vertical-align:top;padding:1px;">' +
			'<input style="display:block" title="Previous dataset (Shortcut: &uarr;)" id="prevDataset" type="button" value="&uarr;" onclick="prevDataset()" disabled="true"/>' +
			'<input title="Next dataset (Shortcut: &darr;)" id="nextDataset" type="button" value="&darr;" onclick="nextDataset()"/><br/></td>' +
			'<td style="padding-top:1px;vertical-align:top"><input title="Switch to the last dataset that was viewed (Shortcut: TAB)" id="lastDataset" type="button" style="font:11px Times new roman" value="last" onclick="selectLastDataset()"/></td></tr></table>';
		
		position = addOptionElement(position + 5, select);
		
		datasetDropDown = document.getElementById('datasets');
		datasetButtonLast = document.getElementById('lastDataset');
		datasetButtonPrev = document.getElementById('prevDataset');
		datasetButtonNext = document.getElementById('nextDataset');
		
		position += datasetDropDown.clientHeight;
	}
	
	position = addOptionElement
	(
		position + 5,
'<input type="button" id="maxAbsoluteDepthDecrease" value="-"/>\
<span id="maxAbsoluteDepth"></span>\
&nbsp;<input type="button" id="maxAbsoluteDepthIncrease" value="+"/> Max depth',
'Maximum depth to display, counted from the top level \
and including collapsed wedges.'
	);
	
	position = addOptionElement
	(
		position,
'<input type="button" id="fontSizeDecrease" value="-"/>\
<span id="fontSize"></span>\
&nbsp;<input type="button" id="fontSizeIncrease" value="+"/> Font size'
	);
	
	position = addOptionElement
	(
		position,
'<input type="button" id="radiusDecrease" value="-"/>\
<input type="button" id="radiusIncrease" value="+"/> Chart size'
	);
	
	if ( hueName )
	{
		hueDisplayName = attributes[attributeIndex(hueName)].displayName;
		
		position = addOptionElement
		(
			position + 5,
			'<input type="checkbox" id="useHue" style="float:left" ' +
			'/><div>Color by<br/>' + hueDisplayName +
			'</div>'
		);
		
		useHueCheckBox = document.getElementById('useHue');
		useHueCheckBox.checked = hueDefault;
		useHueCheckBox.onclick = handleResize;
		useHueCheckBox.onmousedown = suppressEvent;
	}
	/*
	position = addOptionElement
	(
		position + 5,
		'&nbsp;<input type="checkbox" id="shorten" checked="checked" />Shorten labels</div>',
		'Prevent labels from overlapping by shortening them'
	);
	
	position = addOptionElement
	(
		position,
		'&nbsp;<input type="checkbox" id="compress" checked="checked" />Compress',
		'Compress wedges if needed to show the entire depth'
	);
	*/
	position = addOptionElement
	(
		position,
		'<input type="checkbox" id="collapse" checked="checked" />Collapse',
		'Collapse wedges that are redundant (entirely composed of another wedge)'
	);
	
	position = addOptionElement
	(
		position + 5,
		'<input type="button" id="snapshot" value="Snapshot"/>',
'Render the current view as SVG (Scalable Vector Graphics), a publication-\
quality format that can be printed and saved (see Help for browser compatibility)'
	);
	
	position = addOptionElement
	(
		position + 5,
'<input type="button" id="linkButton" value="Link"/>\
<input type="text" size="30" id="linkText"/>',
'Show a link to this view that can be copied for bookmarking or sharing'
	);
	
	position = addOptionElement
	(
		position + 5,
'<input type="button" id="help" value="?"\
onclick="window.open(\'https://github.com/marbl/Krona/wiki/Browsing%20Krona%20charts\', \'help\')"/>',
'Help'
	);
}

function arrow(angleStart, angleEnd, radiusInner)
{
	if ( context.globalAlpha == 0 )
	{
		return;
	}
	
	var angleCenter = (angleStart + angleEnd) / 2;
	var radiusArrowInner = radiusInner - gRadius / 10;//nodeRadius * gRadius;
	var radiusArrowOuter = gRadius * 1.1;//(1 + nodeRadius);
	var radiusArrowCenter = (radiusArrowInner + radiusArrowOuter) / 2;
	var pointLength = (radiusArrowOuter - radiusArrowInner) / 5;
	
	context.fillStyle = highlightFill;
	context.lineWidth = highlightLineWidth;
	
	// First, mask out the first half of the arrow.  This will prevent the tips
	// from superimposing if the arrow goes most of the way around the circle.
	// Masking is done by setting the clipping region to the inverse of the
	// half-arrow, which is defined by cutting the half-arrow out of a large
	// rectangle
	//
	context.beginPath();
	context.arc(0, 0, radiusInner, angleCenter, angleEnd, false);
	context.lineTo
	(
		radiusArrowInner * Math.cos(angleEnd),
		radiusArrowInner * Math.sin(angleEnd)
	);
	context.lineTo
	(
		radiusArrowCenter * Math.cos(angleEnd) - pointLength * Math.sin(angleEnd),
		radiusArrowCenter * Math.sin(angleEnd) + pointLength * Math.cos(angleEnd)
	);
	context.lineTo
	(
		radiusArrowOuter * Math.cos(angleEnd),
		radiusArrowOuter * Math.sin(angleEnd)
	);
	context.arc(0, 0, gRadius, angleEnd, angleCenter, true);
	context.closePath();
	context.moveTo(-imageWidth, -imageHeight);
	context.lineTo(imageWidth, -imageHeight);
	context.lineTo(imageWidth, imageHeight);
	context.lineTo(-imageWidth, imageHeight);
	context.closePath();
	context.save();
	context.clip();
	
	// Next, draw the other half-arrow with the first half masked out
	//
	context.beginPath();
	context.arc(0, 0, radiusInner, angleCenter, angleStart, true);
	context.lineTo
	(
		radiusArrowInner * Math.cos(angleStart),
		radiusArrowInner * Math.sin(angleStart)
	);
	context.lineTo
	(
		radiusArrowCenter * Math.cos(angleStart) + pointLength * Math.sin(angleStart),
		radiusArrowCenter * Math.sin(angleStart) - pointLength * Math.cos(angleStart)
	);
	context.lineTo
	(
		radiusArrowOuter * Math.cos(angleStart),
		radiusArrowOuter * Math.sin(angleStart)
	);
	context.arc(0, 0, gRadius, angleStart, angleCenter, false);
	context.fill();
	context.stroke();
	
	// Finally, remove the clipping region and draw the first half-arrow.  This
	// half is extended slightly to fill the seam.
	//
	context.restore();
	context.beginPath();
	context.arc(0, 0, radiusInner, angleCenter - 2 / (2 * Math.PI * radiusInner), angleEnd, false);
	context.lineTo
	(
		radiusArrowInner * Math.cos(angleEnd),
		radiusArrowInner * Math.sin(angleEnd)
	);
	context.lineTo
	(
		radiusArrowCenter * Math.cos(angleEnd) - pointLength * Math.sin(angleEnd),
		radiusArrowCenter * Math.sin(angleEnd) + pointLength * Math.cos(angleEnd)
	);
	context.lineTo
	(
		radiusArrowOuter * Math.cos(angleEnd),
		radiusArrowOuter * Math.sin(angleEnd)
	);
	context.arc(0, 0, gRadius, angleEnd, angleCenter - 2 / (2 * Math.PI * gRadius), true);
	context.fill();
	context.stroke();
}

function attributeIndex(aname)
{
	for ( var i = 0 ; i < attributes.length; i++ )
	{
		if ( aname == attributes[i].name )
		{
			return i;
		}
	}
	
	return null;
}

function checkHighlight()
{
	var lastHighlightedNode = highlightedNode;
	var lastHighlightingHidden = highlightingHidden;
	
	highlightedNode = selectedNode;
	resetKeyOffset();
	
	if ( progress == 1 )
	{
		selectedNode.checkHighlight();
		if ( selectedNode.getParent() )
		{
			selectedNode.getParent().checkHighlightCenter();
		}
		
		focusNode.checkHighlightMap();
	}
	
	if ( highlightedNode != selectedNode )
	{
		if ( highlightedNode == focusNode )
		{
//			canvas.style.display='none';
//			window.resizeBy(1,0);
//			canvas.style.cursor='ew-resize';
//			window.resizeBy(-1,0);
//			canvas.style.display='inline';
		}
		else
		{
//			canvas.style.cursor='pointer';
		}
	}
	else
	{
//		canvas.style.cursor='auto';
	}
	
	if
	(
		(
			true ||
			highlightedNode != lastHighlightedNode ||
			highlightingHidden != highlightingHiddenLast
		) &&
		progress == 1
	)
	{
		draw(); // TODO: handle in update()
	}
}

function checkSelectedCollapse()
{
	var newNode = selectedNode;
	
	while ( newNode.getCollapse() )
	{
		newNode = newNode.children[0];
	}
	
	if ( newNode.children.length == 0 )
	{
		newNode = newNode.getParent();
	}
	
	if ( newNode != selectedNode )
	{
		selectNode(newNode);
	}
}

function clearSearch()
{
	if ( search.value != '' )
	{
		search.value = '';
		onSearchChange();
	}
}

function createSVG()
{
	svgNS = "http://www.w3.org/2000/svg";
	var SVG = {};
	SVG.xlinkns = "http://www.w3.org/1999/xlink";
	
	var newSVG = document.createElementNS(svgNS, "svg:svg");
	
	newSVG.setAttribute("id", "canvas");
	// How big is the canvas in pixels
	newSVG.setAttribute("width", '100%');
	newSVG.setAttribute("height", '100%');
	// Set the coordinates used by drawings in the canvas
//	newSVG.setAttribute("viewBox", "0 0 " + imageWidth + " " + imageHeight);
	// Define the XLink namespace that SVG uses
	newSVG.setAttributeNS
	(
		"http://www.w3.org/2000/xmlns/",
		"xmlns:xlink",
		SVG.xlinkns
	);
	
	return newSVG;
}

function degrees(radians)
{
	return radians * 180 / Math.PI;
}

function draw()
{
	tweenFrames++;
	//resize();
//	context.fillRect(0, 0, imageWidth, imageHeight);
	context.clearRect(0, 0, imageWidth, imageHeight);
	
	context.font = fontNormal;
	context.textBaseline = 'middle';
	
	//context.strokeStyle = 'rgba(0, 0, 0, 0.3)';
	context.translate(centerX, centerY);
	
	resetKeyOffset();
	
	head.draw(false, false); // draw pie slices
	head.draw(true, false); // draw labels
	
	var pathRoot = selectedNode;
	
	if ( focusNode != 0 && focusNode != selectedNode )
	{
		context.globalAlpha = 1;
		focusNode.drawHighlight(true);
		pathRoot = focusNode;
	}
	
	if
	(
		highlightedNode &&
		highlightedNode.getDepth() >= selectedNode.getDepth() &&
		highlightedNode != focusNode
	)
	{
		if
		(
			progress == 1 &&
			highlightedNode != selectedNode &&
			(
				highlightedNode != focusNode ||
				focusNode.children.length > 0
			)
		)
		{
			context.globalAlpha = 1;
			highlightedNode.drawHighlight(true);
		}
		
		//pathRoot = highlightedNode;
	}
	else if
	(
		progress == 1 &&
		highlightedNode.getDepth() < selectedNode.getDepth()
	)
	{
		context.globalAlpha = 1;
		highlightedNode.drawHighlightCenter();
	}
	
	if ( quickLook && false) // TEMP
	{
		context.globalAlpha = 1 - progress / 2;
		selectedNode.drawHighlight(true);
	}
	else if ( progress < 1 )//&& zoomOut() )
	{
		if ( !zoomOut)//() )
		{
			context.globalAlpha = selectedNode.alphaLine.current();
			selectedNode.drawHighlight(true);
		}
		else if ( selectedNodeLast )
		{
			context.globalAlpha = 1 - 4 * Math.pow(progress - .5, 2);
			selectedNodeLast.drawHighlight(false);
		}
	}
	
	drawDatasetName();
	
	//drawHistory();
	
	context.translate(-centerX, -centerY);
	context.globalAlpha = 1;
	
	mapRadius =
		(imageHeight / 2 - details.clientHeight - details.offsetTop) /
		(pathRoot.getDepth() - 1) * 3 / 4 / 2;
	
	if ( mapRadius > maxMapRadius )
	{
		mapRadius = maxMapRadius;
	}
	
	mapBuffer = mapRadius / 2;
	
	//context.font = fontNormal;
	pathRoot.drawMap(pathRoot);
	
	if ( hueDisplayName && useHue() )
	{
		drawLegend();
	}
}

function drawBubble(angle, radius, width, radial, flip)
{
	var height = fontSize * 2;
	var x;
	var y;
	
	width = width + fontSize;
	
	if ( radial )
	{
		y = -fontSize;
		
		if ( flip )
		{
			x = radius - width + fontSize / 2;
		}
		else
		{
			x = radius - fontSize / 2;
		}
	}
	else
	{
		x = -width / 2;
		y = -radius - fontSize;
	}
	
	if ( snapshotMode )
	{
		drawBubbleSVG(x + centerX, y + centerY, width, height, fontSize, angle);
	}
	else
	{
		drawBubbleCanvas(x, y, width, height, fontSize, angle);
	}
}

function drawBubbleCanvas(x, y, width, height, radius, rotation)
{
	context.strokeStyle = 'black';
	context.lineWidth = highlightLineWidth;
	context.fillStyle = 'rgba(255, 255, 255, .75)';
	context.rotate(rotation);
	roundedRectangle(x, y, width, fontSize * 2, fontSize);
	context.fill();
	context.stroke();
	context.rotate(-rotation);
}

function drawBubbleSVG(x, y, width, height, radius, rotation)
{
	svg +=
		'<rect x="' + x + '" y="' + y +
		'" width="' + width +
		'" height="' + height +
		'" rx="' + radius +
		'" ry="' + radius +
		'" fill="rgba(255, 255, 255, .75)' +
		'" class="highlight" ' +
		'transform="rotate(' +
		degrees(rotation) + ',' + centerX + ',' + centerY +
		')"/>';
}

function drawDatasetName()
{
	var alpha = datasetAlpha.current();
	
	if ( alpha > 0 )
	{
		var radius = gRadius * compressedRadii[0] / -2;
		
		if ( alpha > 1 )
		{
			alpha = 1;
		}
		
		context.globalAlpha = alpha;
		
		drawBubble(0, -radius, datasetWidths[currentDataset], false, false);
		drawText(datasetNames[currentDataset], 0, radius, 0, 'center', true);
	}
}

function drawHistory()
{
	var alpha = 1;
	context.textAlign = 'center';
	
	for ( var i = 0; i < nodeHistoryPosition && alpha > 0; i++ )
	{
		
		context.globalAlpha = alpha - historyAlphaDelta * tweenFactor;
		context.fillText
		(
			nodeHistory[nodeHistoryPosition - i - 1].name,
			0,
			(i + tweenFactor) * historySpacingFactor * fontSize - 1
		);
		
		if ( alpha > 0 )
		{
			alpha -= historyAlphaDelta;
		}
	}
	
	context.globalAlpha = 1;
}

function drawLegend()
{
	var left = imageWidth * .01;
	var width = imageHeight * .0265;
	var height = imageHeight * .15;
	var top = imageHeight - fontSize * 3.5 - height;
	var textLeft = left + width + fontSize / 2;
	
	context.fillStyle = 'black';
	context.textAlign = 'start';
	context.font = fontNormal;
//	context.fillText(valueStartText, textLeft, top + height);
//	context.fillText(valueEndText, textLeft, top);
	context.fillText(hueDisplayName, left, imageHeight - fontSize * 1.5);
	
	var gradient = context.createLinearGradient(0, top + height, 0, top);
	
	for ( var i = 0; i < hueStopPositions.length; i++ )
	{
		gradient.addColorStop(hueStopPositions[i], hueStopHsl[i]);
		
		var textY = top + (1 - hueStopPositions[i]) * height;
		
		if
		(
			i == 0 ||
			i == hueStopPositions.length - 1 ||
			textY > top + fontSize && textY < top + height - fontSize
		)
		{
			context.fillText(hueStopText[i], textLeft, textY);
		}
	}
	
	context.fillStyle = gradient;
	context.fillRect(left, top, width, height);
	context.lineWidth = thinLineWidth;
	context.strokeRect(left, top, width, height);
}

function drawLegendSVG()
{
	var left = imageWidth * .01;
	var width = imageHeight * .0265;
	var height = imageHeight * .15;
	var top = imageHeight - fontSize * 3.5 - height;
	var textLeft = left + width + fontSize / 2;

	var text = '';
	
	text += svgText(hueDisplayName, left, imageHeight - fontSize * 1.5);
	
	var svgtest = '<linearGradient id="gradient" x1="0%" y1="100%" x2="0%" y2="0%">';
	
	for ( var i = 0; i < hueStopPositions.length; i++ )
	{
		svgtest +=
			'<stop offset="' + round(hueStopPositions[i] * 100) +
			'%" style="stop-color:' + hueStopHsl[i] + '"/>';
		
		var textY = top + (1 - hueStopPositions[i]) * height;
		
		if
		(
			i == 0 ||
			i == hueStopPositions.length - 1 ||
			textY > top + fontSize && textY < top + height - fontSize
		)
		{
			text += svgText(hueStopText[i], textLeft, textY);
		}
	}
	
	svgtest += '</linearGradient>';
	//alert(svgtest);
	svg += svgtest;
	svg +=
		'<rect style="fill:url(#gradient)" x="' + left + '" y="' + top +
		'" width="' + width + '" height="' + height + '"/>';
	
	svg += text;
}

function drawSearchHighlights(label, bubbleX, bubbleY, rotation, center)
{
	var index = -1;
	var labelLength = label.length;
	
	bubbleX -= fontSize / 4;
	
	do
	{
		index = label.toLowerCase().indexOf(search.value.toLowerCase(), index + 1);
		
		if ( index != -1 && index < labelLength )
		{
			var dim = context.measureText(label.substr(0, index));
			var x = bubbleX + dim.width;
			
			dim = context.measureText(label.substr(index, search.value.length));
			
			var y = bubbleY - fontSize * 3 / 4;
			var width = dim.width + fontSize / 2;
			var height = fontSize * 3 / 2;
			var radius = fontSize / 2;
			
			if ( snapshotMode )
			{
				if ( center )
				{
					x += centerX;
					y += centerY;
				}
				
				svg +=
					'<rect x="' + x + '" y="' + y +
					'" width="' + width +
					'" height="' + height +
					'" rx="' + radius +
					'" ry="' + radius +
					'" class="searchHighlight' +
					'" transform="rotate(' +
					degrees(rotation) + ',' + centerX + ',' + centerY +
					')"/>';
			}
			else
			{
				context.fillStyle = 'rgb(255, 255, 100)';
				context.rotate(rotation);
				roundedRectangle(x, y, width, height, radius);
				context.fill();
				context.rotate(-rotation);
			}
		}
	}
	while ( index != -1 && index < labelLength );
}

function drawText(text, x, y, angle, anchor, bold, color)
{
	if ( color == undefined )
	{
		color = 'black';
	}
	
	if ( snapshotMode )
	{
		svg +=
			'<text x="' + (centerX + x) + '" y="' + (centerY + y) +
			'" text-anchor="' + anchor + '" style="font-color:' + color + ';font-weight:' + (bold ? 'bold' : 'normal') +
			'" transform="rotate(' + degrees(angle) + ',' + centerX + ',' + centerY + ')">' +
			text + '</text>';
	}
	else
	{
		context.fillStyle = color;
		context.textAlign = anchor;
		context.font = bold ? fontBold : fontNormal;
		context.rotate(angle);
		context.fillText(text, x, y);
		context.rotate(-angle);
	}
}

function drawTextPolar
(
	text,
	innerText,
	angle,
	radius,
	radial,
	bubble,
	bold, 
	searchResult,
	searchResults
)
{
	var anchor;
	var textX;
	var textY;
	var spacer;
	var totalText = text;
	var flip;
	
	if ( snapshotMode )
	{
		spacer = '&#160;&#160;&#160;';
	}
	else
	{
		spacer = '   ';
	}
	
	if ( radial )
	{
		flip = angle < 3 * Math.PI / 2;
		
		if ( flip )
		{
			angle -= Math.PI;
			radius = -radius;
			anchor = 'end';
			
			if ( innerText )
			{
				totalText = text + spacer + innerText;
			}
		}
		else
		{
			anchor = 'start';
			
			if ( innerText )
			{
				totalText = innerText + spacer + text;
			}
		}
		
		textX = radius;
		textY = 0;
	}
	else
	{
		flip = angle < Math.PI || angle > 2 * Math.PI;
		var label;
		
		anchor = snapshotMode ? 'middle' : 'center';
		
		if ( flip )
		{
			angle -= Math.PI;
			radius = -radius;
		}
		
		angle += Math.PI / 2;
		textX = 0;
		textY = -radius;
	}
	
	if ( bubble )
	{
		var textActual = totalText;
		
		if ( innerText && snapshotMode )
		{
			if ( flip )
			{
				textActual = text + '   ' + innerText;
			}
			else
			{
				textActual = innerText + '   ' + text;
			}
		}
		
		if ( searchResults )
		{
			textActual = textActual + searchResultString(searchResults);
		}
		
		var textWidth = measureText(textActual, bold);
		
		var x = textX;
		
		if ( anchor == 'end' )
		{
			x -= textWidth;
		}
		else if ( anchor != 'start' )
		{
			// centered
			x -= textWidth / 2;
		}
		
		drawBubble(angle, radius, textWidth, radial, flip);
		
		if ( searchResult )
		{
			drawSearchHighlights
			(
				textActual,
				x,
				textY,
				angle,
				true
			)
		}
	}
	
	if ( searchResults )
	{
		totalText = totalText + searchResultString(searchResults);
	}
	
	drawText(totalText, textX, textY, angle, anchor, bold);
	
	return flip;
}

function drawTick(start, length, angle)
{
	if ( snapshotMode )
	{
		svg +=
			'<line x1="' + (centerX + start) +
			'" y1="' + centerY +
			'" x2="' + (centerX + start + length) +
			'" y2="' + centerY +
			'" class="tick" transform="rotate(' +
			degrees(angle) + ',' + centerX + ',' + centerY +
			')"/>';
	}
	else
	{
		context.rotate(angle);
		context.beginPath();
		context.moveTo(start, 0);
		context.lineTo(start + length, 0);
		context.lineWidth = thinLineWidth * 2;
		context.stroke();
		context.rotate(-angle);
	}
}

function drawWedge
(
	angleStart,
	angleEnd,
	radiusInner,
	radiusOuter,
	color,
	patternAlpha,
	highlight
)
{
	if ( context.globalAlpha == 0 )
	{
		return;
	}
	
	if ( snapshotMode )
	{
		if ( angleEnd == angleStart + Math.PI * 2 )
		{
			// fudge to prevent overlap, which causes arc ambiguity
			//
			angleEnd -= .1 / gRadius;
		}
		
		var longArc = angleEnd - angleStart > Math.PI ? 1 : 0;
		
		var x1 = centerX + radiusInner * Math.cos(angleStart);
		var y1 = centerY + radiusInner * Math.sin(angleStart);
		
		var x2 = centerX + gRadius * Math.cos(angleStart);
		var y2 = centerY + gRadius * Math.sin(angleStart);
		
		var x3 = centerX + gRadius * Math.cos(angleEnd);
		var y3 = centerY + gRadius * Math.sin(angleEnd);
		
		var x4 = centerX + radiusInner * Math.cos(angleEnd);
		var y4 = centerY + radiusInner * Math.sin(angleEnd);
		
		var dArray =
		[
			" M ", x1, ",", y1,
			" L ", x2, ",", y2,
			" A ", gRadius, ",", gRadius, " 0 ", longArc, ",1 ", x3, ",", y3,
			" L ", x4, ",", y4,
			" A ", radiusInner, ",", radiusInner, " 0 ", longArc, " 0 ", x1, ",", y1,
			" Z "
		];
		
		svg +=
			'<path class="'+ (highlight ? 'highlight' : 'wedge') + '" fill="' + color +
			'" d="' + dArray.join('') + '"/>';
		
		if ( patternAlpha > 0 )
		{
			svg +=
				'<path class="wedge" fill="url(#hiddenPattern)" d="' +
				dArray.join('') + '"/>';
		}
	}
	else
	{
		// fudge to prevent seams during animation
		//
		angleEnd += 1 / gRadius;
		
		context.fillStyle = color;
		context.beginPath();
		context.arc(0, 0, radiusInner, angleStart, angleEnd, false);
		context.arc(0, 0, radiusOuter, angleEnd, angleStart, true);
		context.closePath();
		context.fill();
		
		if ( patternAlpha > 0 )
		{
			context.save();
			context.clip();
			context.globalAlpha = patternAlpha;
			context.fillStyle = hiddenPattern;
			context.fill();
			context.restore();
		}
		
		if ( highlight )
		{
			context.lineWidth = highlight ? highlightLineWidth : thinLineWidth;
			context.strokeStyle = 'black';
			context.stroke();
		}
	}
}

function expand(node)
{
	selectNode(node);
	updateView();
}

function focusLost()
{
	mouseX = -1;
	mouseY = -1;
	checkHighlight();
	document.body.style.cursor = 'auto';
}

function fontSizeDecrease()
{
	if ( fontSize > 1 )
	{
		fontSize--;
		updateViewNeeded = true;
	}
}

function fontSizeIncrease()
{
	fontSize++;
	updateViewNeeded = true;
}

function getGetString(name, value, bool)
{
	return name + '=' + (bool ? value ? 'true' : 'false' : value);
}

function hideLink()
{
	hide(linkText);
	show(linkButton);
}

function show(object)
{
	object.style.display = 'inline';
}

function hide(object)
{
	object.style.display = 'none';
}

function showLink()
{
	var urlHalves = String(document.location).split('?');
	var newGetVariables = new Array();
	
	newGetVariables.push
	(
		getGetString('dataset', currentDataset, false),
		getGetString('node', selectedNode.id, false),
		getGetString('collapse', collapse, true),
		getGetString('color', useHue(), true),
		getGetString('depth', maxAbsoluteDepth - 1, false),
		getGetString('font', fontSize, false),
		getGetString('key', showKeys, true)
	);
	
	hide(linkButton);
	show(linkText);
	linkText.value = urlHalves[0] + '?' + getVariables.concat(newGetVariables).join('&');
	//linkText.disabled = false;
	linkText.focus();
	linkText.select();
	//linkText.disabled = true;
//	document.location = urlHalves[0] + '?' + getVariables.join('&');
}

function getFirstChild(element)
{
	element = element.firstChild;
	
	if ( element && element.nodeType != 1 )
	{
		element = getNextSibling(element);
	}
	
	return element;
}

function getNextSibling(element)
{
	do
	{
		element = element.nextSibling;
	}
	while ( element && element.nodeType != 1 );
	
	return element;
}

function getPercentage(fraction)
{
	return round(fraction * 100);
}

function hslText(hue)
{
	if ( 1 || snapshotMode )
	{
		// Safari doesn't seem to allow hsl() in SVG
		
		var rgb = hslToRgb(hue, saturation, (lightnessBase + lightnessMax) / 2);
		
		return rgbText(rgb.r, rgb.g, rgb.b);
	}
	else
	{
		var hslArray =
		[
			'hsl(',
			Math.floor(hue * 360),
			',',
			Math.floor(saturation * 100),
			'%,',
			Math.floor((lightnessBase + lightnessMax) * 50),
			'%)'
		];
		
		return hslArray.join('');
	}
}

function hslToRgb(h, s, l)
{
	var m1, m2;
	var r, g, b;
	
	if (s == 0)
	{
		r = g = b = Math.floor((l * 255));
	}
	else
	{
		if (l <= 0.5)
		{
			m2 = l * (s + 1);
		}
		else
		{
			m2 = l + s - l * s;
		}
		
		m1 = l * 2 - m2;
		
		r = Math.floor(hueToRgb(m1, m2, h + 1 / 3));
		g = Math.floor(hueToRgb(m1, m2, h));
		b = Math.floor(hueToRgb(m1, m2, h - 1/3));
	}
	
	return {r: r, g: g, b: b};
}

function hueToRgb(m1, m2, hue)
{
	var v;
	
	while (hue < 0)
	{
		hue += 1;
	}
	
	while (hue > 1)
	{
		hue -= 1;
	}
	
	if (6 * hue < 1)
		v = m1 + (m2 - m1) * hue * 6;
	else if (2 * hue < 1)
		v = m2;
	else if (3 * hue < 2)
		v = m1 + (m2 - m1) * (2/3 - hue) * 6;
	else
		v = m1;

	return 255 * v;
}

function interpolateHue(hueStart, hueEnd, valueStart, valueEnd)
{
	// since the gradient will be RGB based, we need to add stops to hit all the
	// colors in the hue spectrum
	
	hueStopPositions = new Array();
	hueStopHsl = new Array();
	hueStopText = new Array();
	
	hueStopPositions.push(0);
	hueStopHsl.push(hslText(hueStart));
	hueStopText.push(round(valueStart));
	
	for
	(
		var i = (hueStart > hueEnd ? 5 / 6 : 1 / 6);
		(hueStart > hueEnd ? i > 0 : i < 1);
		i += (hueStart > hueEnd ? -1 : 1) / 6
	)
	{
		if
		(
			hueStart > hueEnd ?
				i > hueEnd && i < hueStart :
				i > hueStart && i < hueEnd
		)
		{
			hueStopPositions.push(lerp(i, hueStart, hueEnd, 0, 1));
			hueStopHsl.push(hslText(i));
			hueStopText.push(round(lerp
			(
				i,
				hueStart,
				hueEnd,
				valueStart,
				valueEnd
			)));
		}
	}
	
	hueStopPositions.push(1);
	hueStopHsl.push(hslText(hueEnd));
	hueStopText.push(round(valueEnd));
}

function keyLineAngle(angle, keyAngle, bendRadius, keyX, keyY, pointsX, pointsY)
{
	if ( angle < Math.PI / 2 && keyY < bendRadius * Math.sin(angle) 
	|| angle > Math.PI / 2 && keyY < bendRadius)
	{
		return Math.asin(keyY / bendRadius);
	}
	else
	{
		// find the angle of the normal to a tangent line that goes to
		// the label
		
		var textDist = Math.sqrt
		(
			Math.pow(keyX, 2) +
			Math.pow(keyY, 2)
		);
		
		var tanAngle = Math.acos(bendRadius / textDist) + keyAngle;
		
		if ( angle < tanAngle || angle < Math.PI / 2 )//|| labelLeft < centerX )
		{
			// angle doesn't reach far enough for tangent; collapse and
			// connect directly to label
			
			if ( keyY / Math.tan(angle) > 0 )
			{
				pointsX.push(keyY / Math.tan(angle));
				pointsY.push(keyY);
			}
			else
			{
				pointsX.push(bendRadius * Math.cos(angle));
				pointsY.push(bendRadius * Math.sin(angle));
			}
			
			return angle;
		}
		else
		{
			return tanAngle;
		}
	}
}

function keyOffset()
{
	return imageHeight - (keys - currentKey + 1) * (keySize + keyBuffer) + keyBuffer - margin;
}

function lerp(value, fromStart, fromEnd, toStart, toEnd)
{
	return (value - fromStart) *
		(toEnd - toStart) /
		(fromEnd - fromStart) +
		toStart;
}

function createCanvas()
{
	canvas = document.createElement('canvas');
	document.body.appendChild(canvas);
	context = canvas.getContext('2d');
}

function load()
{
	document.body.style.overflow = "hidden";
	document.body.style.margin = 0;
	
	createCanvas();
	
	if ( context == undefined )
	{
		document.body.innerHTML = '\
<br/>This browser does not support HTML5 (see \
<a href="https://github.com/marbl/Krona/wiki/Browser%20support">Browser support</a>).\
	';
		return;
	}

	if ( typeof context.fillText != 'function' )
	{
		document.body.innerHTML = '\
<br/>This browser does not support HTML5 canvas text (see \
<a href="https://github.com/marbl/Krona/wiki/Browser%20support">Browser support</a>).\
	';
		return;
	}
	
	resize();
	
	var kronaElement = document.getElementsByTagName('krona')[0];
	
	var magnitudeName;
	var hueName;
	var hueDefault;
	var hueStart;
	var hueEnd;
	var valueStart;
	var valueEnd;
	
	if ( kronaElement.getAttribute('collapse') != undefined )
	{
		collapse = kronaElement.getAttribute('collapse') == 'true';
	}
	
	if ( kronaElement.getAttribute('key') != undefined )
	{
		showKeys = kronaElement.getAttribute('key') == 'true';
	}
	
	for
	(
		var element = getFirstChild(kronaElement);
		element;
		element = getNextSibling(element)
	)
	{
		switch ( element.tagName.toLowerCase() )
		{
			case 'attributes':
				magnitudeName = element.getAttribute('magnitude');
				//
				for
				(
					var attributeElement = getFirstChild(element);
					attributeElement;
					attributeElement = getNextSibling(attributeElement)
				)
				{
					var tag = attributeElement.tagName.toLowerCase();
					
					if ( tag == 'attribute' )
					{
						var attribute = new Attribute();
						attribute.name = attributeElement.firstChild.nodeValue.toLowerCase();
						attribute.displayName = attributeElement.getAttribute('display');
						
						if ( attributeElement.getAttribute('hrefBase') )
						{
							attribute.hrefBase = attributeElement.getAttribute('hrefBase');
						}
						
						if ( attributeElement.getAttribute('target') )
						{
							attribute.target = attributeElement.getAttribute('target');
						}
						
						if ( attribute.name == magnitudeName )
						{
							magnitudeIndex = attributes.length;
						}
						
						if ( attributeElement.getAttribute('listAll') )
						{
							attribute.listAll = attributeElement.getAttribute('listAll').toLowerCase();
						}
						else if ( attributeElement.getAttribute('listNode') )
						{
							attribute.listNode = attributeElement.getAttribute('listNode').toLowerCase();
						}
						else if ( attributeElement.getAttribute('dataAll') )
						{
							attribute.dataAll = attributeElement.getAttribute('dataAll').toLowerCase();
						}
						else if ( attributeElement.getAttribute('dataNode') )
						{
							attribute.dataNode = attributeElement.getAttribute('dataNode').toLowerCase();
						}
						
						if ( attributeElement.getAttribute('postUrl') )
						{
							attribute.postUrl = attributeElement.getAttribute('postUrl');
						}
						
						if ( attributeElement.getAttribute('postVar') )
						{
							attribute.postVar = attributeElement.getAttribute('postVar');
						}
						
						if ( attributeElement.getAttribute('mono') )
						{
							attribute.mono = true;
						}
						
						attributes.push(attribute);
					}
					else if ( tag == 'list' )
					{
						var attribute = new Attribute();
						
						attribute.name = attributeElement.firstChild.nodeValue;
						attribute.list = true;
						attributes.push(attribute);
					}
					else if ( tag == 'data' )
					{
						var attribute = new Attribute();
						
						attribute.name = attributeElement.firstChild.nodeValue;
						attribute.data = true;
						attributes.push(attribute);
						
						var enableScript = document.createElement('script');
						var date = new Date();
						enableScript.src =
							attributeElement.getAttribute('enable') + '?' +
							date.getTime();
						document.body.appendChild(enableScript);
					}
				}
				break;
			
			case 'color':
				hueName = element.getAttribute('attribute');
				hueStart = Number(element.getAttribute('hueStart')) / 360;
				hueEnd = Number(element.getAttribute('hueEnd')) / 360;
				valueStart = Number(element.getAttribute('valueStart'));
				valueEnd = Number(element.getAttribute('valueEnd'));
				//
				interpolateHue(hueStart, hueEnd, valueStart, valueEnd);
				//
				if ( element.getAttribute('default') == 'true' )
				{
					hueDefault = true;
				}
				break;
			
			case 'datasets':
				datasetNames = new Array();
				//
				for ( j = getFirstChild(element); j; j = getNextSibling(j) )
				{
					datasetNames.push(j.firstChild.nodeValue);
				}
				datasets = datasetNames.length;
				break;
			
			case 'node':
				head = loadTreeDOM
				(
					element,
					magnitudeName,
					hueName,
					hueStart,
					hueEnd,
					valueStart,
					valueEnd
				);
				break;
		}
	}
	
	// get GET options
	//
	var urlHalves = String(document.location).split('?');
	var datasetDefault = 0;
	var maxDepthDefault;
	var nodeDefault = 0;
	//
	if ( urlHalves[1] )
	{
		var vars = urlHalves[1].split('&');
		
		for ( i = 0; i < vars.length; i++ )
		{
			var pair = vars[i].split('=');
			
			switch ( pair[0] )
			{
				case 'collapse':
					collapse = pair[1] == 'true';
					break;
				
				case 'color':
					hueDefault = pair[1] == 'true';
					break;
				
				case 'dataset':
					datasetDefault = Number(pair[1]);
					break;
					
				case 'depth':
					maxDepthDefault = Number(pair[1]) + 1;
					break;
				
				case 'key':
					showKeys = pair[1] == 'true';
					break;
				
				case 'font':
					fontSize = Number(pair[1]);
					break;
				
				case 'node':
					nodeDefault = Number(pair[1]);
					break;
				
				default:
					getVariables.push(pair[0] + '=' + pair[1]);
					break;
			}
		}
	}
	
	addOptionElements(hueName, hueDefault);
	setCallBacks();
	
	head.sort();
	maxAbsoluteDepth = 0;
	selectDataset(datasetDefault);
	
	if ( maxDepthDefault && maxDepthDefault < head.maxDepth )
	{
		maxAbsoluteDepth = maxDepthDefault;
	}
	else
	{
		maxAbsoluteDepth = head.maxDepth;
	}
	
	selectNode(nodes[nodeDefault]);
	
	setInterval(update, 20);
	
	window.onresize = handleResize;
	updateMaxAbsoluteDepth();
	updateViewNeeded = true;
}

function loadTreeDOM
(
	domNode,
	magnitudeName,
	hueName,
	hueStart,
	hueEnd,
	valueStart,
	valueEnd
)
{
	var newNode = new Node();
	
	newNode.name = domNode.getAttribute('name');
	
	if ( domNode.getAttribute('href') )
	{
		newNode.href = domNode.getAttribute('href');
	}
	
	if ( hueName )
	{
		newNode.hues = new Array();
	}
	
	for ( var i = getFirstChild(domNode); i; i = getNextSibling(i) )
	{
		switch ( i.tagName.toLowerCase() )
		{
		case 'node': 
			var newChild = loadTreeDOM
			(
				i,
				magnitudeName,
				hueName,
				hueStart,
				hueEnd,
				valueStart,
				valueEnd
			);
			newChild.parent = newNode;
			newNode.children.push(newChild);
			break;
			
		default:
			var attributeName = i.tagName.toLowerCase();
			var index = attributeIndex(attributeName);
			//
			newNode.attributes[index] = new Array();
			//
			for ( var j = getFirstChild(i); j; j = getNextSibling(j) )
			{
				if ( attributes[index] == undefined )
				{
					var x = 5;
				}
				if ( attributes[index].list )
				{
					newNode.attributes[index].push(new Array());
					
					for ( var k = getFirstChild(j); k; k = getNextSibling(k) )
					{
						newNode.attributes[index][newNode.attributes[index].length - 1].push(k.firstChild.nodeValue);
					}
				}
				else
				{
					var value = j.firstChild ? j.firstChild.nodeValue : '';
					
					if ( j.getAttribute('href') )
					{
						var target;
						
						if ( attributes[index].target )
						{
							target = ' target="' + attributes[index].target + '"';
						}
						
						value = '<a href="' + attributes[index].hrefBase + j.getAttribute('href') + '"' + target + '>' + value + '</a>';
					}
					
					newNode.attributes[index].push(value);
				}
			}
			//
			if ( attributeName == magnitudeName || attributeName == hueName )
			{
				for ( j = 0; j < datasets; j++ )
				{
					var value = newNode.attributes[index][j] == undefined ? 0 : Number(newNode.attributes[index][j]);
					
					newNode.attributes[index][j] = value;
					
					if ( attributeName == hueName )
					{
						var hue = lerp
						(
							value,
							valueStart,
							valueEnd,
							hueStart,
							hueEnd
						);
						
						if ( hue < hueStart == hueStart < hueEnd )
						{
							hue = hueStart;
						}
						else if ( hue > hueEnd == hueStart < hueEnd )
						{
							hue = hueEnd;
						}
						
						newNode.hues[j] = hue;
					}
				}
				
				if ( attributeName == hueName )
				{
					newNode.hue = new Tween(newNode.hues[0], newNode.hues[0]);
				}
			}
			break;
		}
	}
	
	return newNode;
}

function maxAbsoluteDepthDecrease()
{
	if ( maxAbsoluteDepth > 2 )
	{
		maxAbsoluteDepth--;
		head.setMaxDepths();
		handleResize();
	}
}

function maxAbsoluteDepthIncrease()
{
	if ( maxAbsoluteDepth < head.maxDepth )
	{
		maxAbsoluteDepth++;
		head.setMaxDepths();
		handleResize();
	}
}

function measureText(text, bold)
{
	context.font = bold ? fontBold : fontNormal;
	var dim = context.measureText(text);
	return dim.width;
}

function min(a, b)
{
	return a < b ? a : b;
}

function minWidth()
{
	// Min wedge width (at center) for displaying a node (or for displaying a
	// label if it's at the highest level being viewed, multiplied by 2 to make
	// further calculations simpler
	
	return (fontSize * 2.3);
}

function mouseMove(e)
{
	mouseX = e.pageX;
	mouseY = e.pageY - headerHeight;
	mouseXRel = (mouseX - centerX) * backingScale()
	mouseYRel = (mouseY - centerY) * backingScale()
	
	if ( head && ! quickLook )
	{
		checkHighlight();
	}
}

function mouseClick(e)
{
	if ( highlightedNode == focusNode && focusNode != selectedNode || selectedNode.hasParent(highlightedNode) )
	{
		if ( highlightedNode.hasChildren() )
		{
			expand(highlightedNode);
		}
	}
	else if ( progress == 1 )//( highlightedNode != selectedNode )
	{
		setFocus(highlightedNode);
//		document.body.style.cursor='ew-resize';
		draw();
		checkHighlight();
		var date = new Date();
		mouseDownTime = date.getTime();
		mouseDown = true;
	}
}

function mouseUp(e)
{
	if ( quickLook )
	{
		navigateBack();
		quickLook = false;
	}
	
	mouseDown = false;
}

function navigateBack()
{
	if ( nodeHistoryPosition > 0 )
	{
		nodeHistory[nodeHistoryPosition] = selectedNode;
		nodeHistoryPosition--;
		
		if ( nodeHistory[nodeHistoryPosition].collapse )
		{
			collapseCheckBox.checked = collapse = false;
		}
		
		setSelectedNode(nodeHistory[nodeHistoryPosition]);
		updateDatasetButtons();
		updateView();
	}
}

function navigateUp()
{
	if ( selectedNode.getParent() )
	{
		selectNode(selectedNode.getParent());
		updateView();
	}
}

function navigateForward()
{
	if ( nodeHistoryPosition < nodeHistory.length - 1 )
	{
		nodeHistoryPosition++;
		var newNode = nodeHistory[nodeHistoryPosition];
		
		if ( newNode.collapse )
		{
			collapseCheckBox.checked = collapse = false;
		}
		
		if ( nodeHistoryPosition == nodeHistory.length - 1 )
		{
			// this will ensure the forward button is disabled
			
			nodeHistory.length = nodeHistoryPosition;
		}
		
		setSelectedNode(newNode);
		updateDatasetButtons();
		updateView();
	}
}

function nextDataset()
{
	var newDataset = currentDataset;
	
	do
	{
		if ( newDataset == datasets - 1 )
		{
			newDataset = 0;
		}
		else
		{
			newDataset++;
		}
	}
	while ( datasetDropDown.options[newDataset].disabled )
	
	selectDataset(newDataset);
}

function onDatasetChange()
{
	selectDataset(datasetDropDown.selectedIndex);
}

function onKeyDown(event)
{
	if
	(
		event.keyCode == 37 &&
		document.activeElement.id != 'search' &&
		document.activeElement.id != 'linkText'
	)
	{
		navigateBack();
		event.preventDefault();
	}
	else if
	(
		event.keyCode == 39 &&
		document.activeElement.id != 'search' &&
		document.activeElement.id != 'linkText'
	)
	{
		navigateForward();
		event.preventDefault();
	}
	else if ( event.keyCode == 38 && datasets > 1 )
	{
		prevDataset();
		
		//if ( document.activeElement.id == 'datasets' )
		{
			event.preventDefault();
		}
	}
	else if ( event.keyCode == 40 && datasets > 1 )
	{
		nextDataset();
		
		//if ( document.activeElement.id == 'datasets' )
		{
			event.preventDefault();
		}
	}
	else if ( event.keyCode == 9 && datasets > 1 )
	{
		selectLastDataset();
		event.preventDefault();
	}
	else if ( event.keyCode == 83 )
	{
		progress += .2;
	}
	else if ( event.keyCode == 66 )
	{
		progress -= .2;
	}
	else if ( event.keyCode == 70 )
	{
		progress = 1;
	}
}

function onKeyPress(event)
{
	if ( event.keyCode == 38 && datasets > 1 )
	{
//		prevDataset();
		
		//if ( document.activeElement.id == 'datasets' )
		{
			event.preventDefault();
		}
	}
	else if ( event.keyCode == 40 && datasets > 1 )
	{
//		nextDataset();
		
		//if ( document.activeElement.id == 'datasets' )
		{
			event.preventDefault();
		}
	}
}

function onKeyUp(event)
{
	if ( event.keyCode == 27 && document.activeElement.id == 'search' )
	{
		search.value = '';
		onSearchChange();
	}
	else if ( event.keyCode == 38 && datasets > 1 )
	{
//		prevDataset();
		
		//if ( document.activeElement.id == 'datasets' )
		{
			event.preventDefault();
		}
	}
	else if ( event.keyCode == 40 && datasets > 1 )
	{
//		nextDataset();
		
		//if ( document.activeElement.id == 'datasets' )
		{
			event.preventDefault();
		}
	}
}

function onSearchChange()
{
	nSearchResults = 0;
	head.search();
	
	if ( search.value == '' )
	{
		searchResults.innerHTML = '';
	}
	else
	{
		searchResults.innerHTML = nSearchResults + ' results';
	}
	
	setFocus(selectedNode);
	draw();
}

function post(url, variable, value, postWindow)
{
	var form = document.createElement('form');
	var input = document.createElement('input');
	var inputDataset = document.createElement('input');
	
	form.appendChild(input);
	form.appendChild(inputDataset);
	
	form.method = "POST";
	form.action = url;
	
	if ( postWindow == undefined )
	{
		form.target = '_blank';
		postWindow = window;
	}
	
	input.type = 'hidden';
	input.name = variable;
	input.value = value;
	
	inputDataset.type = 'hidden';
	inputDataset.name = 'dataset';
	inputDataset.value = currentDataset;
	
	postWindow.document.body.appendChild(form);
	form.submit();
}

function prevDataset()
{
	var newDataset = currentDataset;
	
	do
	{
		if ( newDataset == 0 )
		{
			newDataset = datasets - 1;
		}
		else
		{
			newDataset--;
		}
	}
	while ( datasetDropDown.options[newDataset].disabled );
	
	selectDataset(newDataset);
}

function radiusDecrease()
{
	if ( bufferFactor < .309 )
	{
		bufferFactor += .03;
		updateViewNeeded = true;
	}
}

function radiusIncrease()
{
	if ( bufferFactor > .041 )
	{
		bufferFactor -= .03;
		updateViewNeeded = true;
	}
}

function resetKeyOffset()
{
	currentKey = 1;
	keyMinTextLeft = centerX + gRadius + buffer - buffer / (keys + 1) / 2 + fontSize / 2;
	keyMinAngle = 0;
}

function rgbText(r, g, b)
{
	var rgbArray =
	[
		"rgb(",
		Math.floor(r),
		",",
		Math.floor(g),
		",",
		Math.floor(b),
		")"
	];
	
	return rgbArray.join('');
}

function round(number)
{
	if ( number >= 1 || number <= -1 )
	{
		return number.toFixed(0);
	}
	else
	{
		return number.toPrecision(1);
	}
}

function roundedRectangle(x, y, width, height, radius)
{
	if ( radius * 2 > width )
	{
		radius = width / 2;
	}
	
	if ( radius * 2 > height )
	{
		radius = height / 2;
	}
	
	context.beginPath();
	context.arc(x + radius, y + radius, radius, Math.PI, Math.PI * 3 / 2, false);
	context.lineTo(x + width - radius, y);
	context.arc(x + width - radius, y + radius, radius, Math.PI * 3 / 2, Math.PI * 2, false);
	context.lineTo(x + width, y + height - radius);
	context.arc(x + width - radius, y + height - radius, radius, 0, Math.PI / 2, false);
	context.lineTo(x + radius, y + height);
	context.arc(x + radius, y + height - radius, radius, Math.PI / 2, Math.PI, false);
	context.lineTo(x, y + radius);
}

function passClick(e)
{
	mouseClick(e);
}

function searchResultString(results)
{
	var searchResults = this.searchResults;
	
	if ( this.isSearchResult )
	{
		// don't count ourselves
		searchResults--;
	}
	
	return ' - ' + results + (results > 1 ? ' results' : ' result');
}

function setCallBacks()
{
	canvas.onselectstart = function(){return false;} // prevent unwanted highlighting
	options.onselectstart = function(){return false;} // prevent unwanted highlighting
	document.onmousemove = mouseMove;
	window.onblur = focusLost;
	window.onmouseout = focusLost;
	document.onkeyup = onKeyUp;
	document.onkeydown = onKeyDown;
	canvas.onmousedown = mouseClick;
	document.onmouseup = mouseUp;
	keyControl.onclick = toggleKeys;
	collapseCheckBox = document.getElementById('collapse');
	collapseCheckBox.checked = collapse;
	collapseCheckBox.onclick = handleResize;
	collapseCheckBox.onmousedown = suppressEvent;
	maxAbsoluteDepthText = document.getElementById('maxAbsoluteDepth');
	maxAbsoluteDepthButtonDecrease = document.getElementById('maxAbsoluteDepthDecrease');
	maxAbsoluteDepthButtonIncrease = document.getElementById('maxAbsoluteDepthIncrease');
	maxAbsoluteDepthButtonDecrease.onclick = maxAbsoluteDepthDecrease;
	maxAbsoluteDepthButtonIncrease.onclick = maxAbsoluteDepthIncrease;
	maxAbsoluteDepthButtonDecrease.onmousedown = suppressEvent;
	maxAbsoluteDepthButtonIncrease.onmousedown = suppressEvent;
	fontSizeText = document.getElementById('fontSize');
	fontSizeButtonDecrease = document.getElementById('fontSizeDecrease');
	fontSizeButtonIncrease = document.getElementById('fontSizeIncrease');
	fontSizeButtonDecrease.onclick = fontSizeDecrease;
	fontSizeButtonIncrease.onclick = fontSizeIncrease;
	fontSizeButtonDecrease.onmousedown = suppressEvent;
	fontSizeButtonIncrease.onmousedown = suppressEvent;
	radiusButtonDecrease = document.getElementById('radiusDecrease');
	radiusButtonIncrease = document.getElementById('radiusIncrease');
	radiusButtonDecrease.onclick = radiusDecrease;
	radiusButtonIncrease.onclick = radiusIncrease;
	radiusButtonDecrease.onmousedown = suppressEvent;
	radiusButtonIncrease.onmousedown = suppressEvent;
	maxAbsoluteDepth = 0;
	backButton = document.getElementById('back');
	backButton.onclick = navigateBack;
	backButton.onmousedown = suppressEvent;
	forwardButton = document.getElementById('forward');
	forwardButton.onclick = navigateForward;
	forwardButton.onmousedown = suppressEvent;
	snapshotButton = document.getElementById('snapshot');
	snapshotButton.onclick = snapshot;
	snapshotButton.onmousedown = suppressEvent;
	detailsName = document.getElementById('detailsName');
	detailsExpand = document.getElementById('detailsExpand');
	detailsInfo = document.getElementById('detailsInfo');
	search = document.getElementById('search');
	search.onkeyup = onSearchChange;
	search.onmousedown = suppressEvent;
	searchResults = document.getElementById('searchResults');
	useHueDiv = document.getElementById('useHueDiv');
	linkButton = document.getElementById('linkButton');
	linkButton.onclick = showLink;
	linkButton.onmousedown = suppressEvent;
	linkText = document.getElementById('linkText');
	linkText.onblur = hideLink;
	linkText.onmousedown = suppressEvent;
	hide(linkText);
	var helpButton = document.getElementById('help');
	helpButton.onmousedown = suppressEvent;
	var searchClear = document.getElementById('searchClear');
	searchClear.onmousedown = suppressEvent;
	if ( datasets > 1 )
	{
		datasetDropDown.onmousedown = suppressEvent;
		var prevDatasetButton = document.getElementById('prevDataset');
		prevDatasetButton.onmousedown = suppressEvent;
		var nextDatasetButton = document.getElementById('nextDataset');
		nextDatasetButton.onmousedown = suppressEvent;
		var lastDatasetButton = document.getElementById('lastDataset');
		lastDatasetButton.onmousedown = suppressEvent;
	}
	
	image = document.getElementById('hiddenImage');
	
	if ( image.complete )
	{
		hiddenPattern = context.createPattern(image, 'repeat');
	}
	else
	{
		image.onload = function()
		{
			hiddenPattern = context.createPattern(image, 'repeat');
		}
	}
	
	var loadingImageElement = document.getElementById('loadingImage');
	
	if ( loadingImageElement )
	{
		loadingImage = loadingImageElement.src;
	}
}

function selectDataset(newDataset)
{
	lastDataset = currentDataset;
	currentDataset = newDataset
	if ( datasets > 1 )
	{
		datasetDropDown.selectedIndex = currentDataset;
		updateDatasetButtons();
		datasetAlpha.start = 1.5;
		datasetChanged = true;
	}
	head.setMagnitudes(0);
	head.setDepth(1, 1);
	head.setMaxDepths();
	handleResize();
}

function selectLastDataset()
{
	selectDataset(lastDataset);
	handleResize();
}

function selectNode(newNode)
{
	if ( selectedNode != newNode )
	{
		// truncate history at current location to create a new branch
		//
		nodeHistory.length = nodeHistoryPosition;
		
		if ( selectedNode != 0 )
		{
			nodeHistory.push(selectedNode);
			nodeHistoryPosition++;
		}
		
		setSelectedNode(newNode);
		//updateView();
	}
	
	updateDatasetButtons();
}

function setFocus(node)
{
	if ( node == focusNode )
	{
//		return;
	}
	
	focusNode = node;
	
	if ( node.href )
	{
		detailsName.innerHTML =
			'<a target="_blank" href="' + node.href + '">' + node.name + '</a>';
	}
	else
	{
		detailsName.innerHTML = node.name;
	}
	
	var table = '<table>';
	
	table += '<tr><td></td></tr>';
	
	for ( var i = 0; i < node.attributes.length; i++ )
	{
		if ( attributes[i].displayName && node.attributes[i] != undefined )
		{
			var index = node.attributes[i].length == 1 && attributes[i].mono ? 0 : currentDataset;
			
			if ( typeof node.attributes[i][currentDataset] == 'number' || node.attributes[i][index] != undefined && node.attributes[i][currentDataset] != '' )
			{
				var value = node.attributes[i][index];
				
				if ( attributes[i].listNode != undefined )
				{
					value =
						'<a href="" onclick="showList(' +
						attributeIndex(attributes[i].listNode) + ',' + i +
						',false);return false;" title="Show list">' +
						value + '</a>';
				}
				else if ( attributes[i].listAll != undefined )
				{
					value =
						'<a href="" onclick="showList(' +
						attributeIndex(attributes[i].listAll) + ',' + i +
						',true);return false;" title="Show list">' +
						value + '</a>';
				}
				else if ( attributes[i].dataNode != undefined && dataEnabled )
				{
					value =
						'<a href="" onclick="showData(' +
						attributeIndex(attributes[i].dataNode) + ',' + i +
						',false);return false;" title="Show data">' +
						value + '</a>';
				}
				else if ( attributes[i].dataAll != undefined && dataEnabled )
				{
					value =
						'<a href="" onclick="showData(' +
						attributeIndex(attributes[i].dataAll) + ',' + i +
						',true);return false;" title="Show data">' +
						value + '</a>';
				}
				
				table +=
					'<tr><td><strong>' + attributes[i].displayName + ':</strong></td><td>' +
					value + '</td></tr>';
			}
		}
	}
	
	table += '</table>';
	detailsInfo.innerHTML = table;
	
	detailsExpand.disabled = !focusNode.hasChildren() || focusNode == selectedNode;
}

function setSelectedNode(newNode)
{
	if ( selectedNode && selectedNode.hasParent(newNode) )
	{
		zoomOut = true;
	}
	else
	{
		zoomOut = false;
	}
	
	selectedNodeLast = selectedNode;
	selectedNode = newNode;
	
	//if ( focusNode != selectedNode )
	{
		setFocus(selectedNode);
	}
}

function waitForData(dataWindow, target, title, time, postUrl, postVar)
{
	if ( nodeData.length == target )
	{
		if ( postUrl != undefined )
		{
			for ( var i = 0; i < nodeData.length; i++ )
			{
				nodeData[i] = nodeData[i].replace(/\n/g, ',');
			}
			
			var postString = nodeData.join('');
			postString = postString.slice(0, -1);
			
			dataWindow.document.body.removeChild(dataWindow.document.getElementById('loading'));
			document.body.removeChild(document.getElementById('data'));
			
			post(postUrl, postVar, postString, dataWindow);
		}
		else
		{
			//dataWindow.document.body.removeChild(dataWindow.document.getElementById('loading'));
			//document.body.removeChild(document.getElementById('data'));
			
			dataWindow.document.open();
			dataWindow.document.write('<pre>' + nodeData.join('') + '</pre>');
			dataWindow.document.close();
		}
		
		dataWindow.document.title = title; // replace after document.write()
	}
	else
	{
		var date = new Date();
		
		if ( date.getTime() - time > 10000 )
		{
			dataWindow.document.body.removeChild(dataWindow.document.getElementById('loading'));
			document.body.removeChild(document.getElementById('data'));
			dataWindow.document.body.innerHTML =
				'Timed out loading supplemental files for:<br/>' + document.location;
		}
		else
		{
			setTimeout(function() {waitForData(dataWindow, target, title, time, postUrl, postVar);}, 100);
		}
	}
}

function data(newData)
{
	nodeData.push(newData);
}

function enableData()
{
	dataEnabled = true;
}

function showData(indexData, indexAttribute, summary)
{
	var dataWindow = window.open('', '_blank');
	var title = 'Krona - ' + attributes[indexAttribute].displayName + ' - ' + focusNode.name;
	dataWindow.document.title = title;
	
	nodeData = new Array();
	
	if ( dataWindow && dataWindow.document && dataWindow.document.body != null )
	{
		//var loadImage = document.createElement('img');
		//loadImage.src = "file://localhost/Users/ondovb/Krona/KronaTools/img/loading.gif";
		//loadImage.id = "loading";
		//loadImage.alt = "Loading...";
		//dataWindow.document.body.appendChild(loadImage);
		dataWindow.document.body.innerHTML =
			'<img id="loading" src="' + loadingImage + '" alt="Loading..."></img>';
	}
	
	var scripts = document.createElement('div');
	scripts.id = 'data';
	document.body.appendChild(scripts);
	
	var files = focusNode.getData(indexData, summary);
	
	var date = new Date();
	var time = date.getTime();
	
	for ( var i = 0; i < files.length; i++ )
	{
		var script = document.createElement('script');
		script.src = files[i] + '?' + time;
		scripts.appendChild(script);
	}
	
	waitForData(dataWindow, files.length, title, time, attributes[indexAttribute].postUrl, attributes[indexAttribute].postVar);
	
	return false;
}

function showList(indexList, indexAttribute, summary)
{
	var list = focusNode.getList(indexList, summary);
	
	if ( attributes[indexAttribute].postUrl != undefined )
	{
		post(attributes[indexAttribute].postUrl, attributes[indexAttribute].postVar, list.join(','));
	}
	else
	{
		var dataWindow = window.open('', '_blank');
		
		if ( true || navigator.appName == 'Microsoft Internet Explorer' ) // :(
		{
			dataWindow.document.open();
			dataWindow.document.write('<pre>' + list.join('\n') + '</pre>');
			dataWindow.document.close();
		}
		else
		{
			var pre = document.createElement('pre');
			dataWindow.document.body.appendChild(pre);
			pre.innerHTML = list;
		}
		
		dataWindow.document.title = 'Krona - ' + attributes[indexAttribute].displayName + ' - ' + focusNode.name;
	}
}

function snapshot()
{
	svg = svgHeader();
	
	resetKeyOffset();
	
	snapshotMode = true;
	
	selectedNode.draw(false, true);
	selectedNode.draw(true, true);
	
	if ( focusNode != 0 && focusNode != selectedNode )
	{
		context.globalAlpha = 1;
		focusNode.drawHighlight(true);
	}
	
	if ( hueDisplayName && useHue() )
	{
		drawLegendSVG();
	}
	
	snapshotMode = false;
	
	svg += svgFooter();
	
	snapshotWindow = window.open
	(
		'data:image/svg+xml;charset=utf-8,' + encodeURIComponent(svg),
		'_blank'
	);
/*	var data = window.open('data:text/plain;charset=utf-8,hello', '_blank');
	var data = window.open('', '_blank');
	data.document.open('text/plain');
	data.document.write('hello');
	data.document.close();
	var button = document.createElement('input');
	button.type = 'button';
	button.value = 'save';
	button.onclick = save;
	data.document.body.appendChild(button);
//	snapshotWindow.document.write(svg);
//	snapshotWindow.document.close();
*/	
}

function save()
{
	alert(document.body.innerHTML);
}

function spacer()
{
	if ( snapshotMode )
	{
		return '&#160;&#160;&#160;';
	}
	else
	{
		return '   ';
	}
}

function suppressEvent(e)
{
	e.cancelBubble = true;
	if (e.stopPropagation) e.stopPropagation();
}

function svgFooter()
{
	return '</svg>';
}

function svgHeader()
{
	var patternWidth = fontSize * .6;//radius / 50;
	
	return '\
<?xml version="1.0" standalone="no"?>\
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" \
	"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">\
<svg width="' + imageWidth + '" height="' + imageHeight + '" version="1.1"\
	xmlns="http://www.w3.org/2000/svg">\
<title>Krona (snapshot) - ' +
(datasets > 1 ? datasetNames[currentDataset] + ' - ' : '') + selectedNode.name +
'</title>\
<defs>\
	<style type="text/css">\
	text {font-size: ' + fontSize + 'px; font-family: ' + fontFamily + '; dominant-baseline:central}\
	path {stroke-width:' + thinLineWidth * fontSize / 12 + ';}\
	path.wedge {stroke:none}\
	path.line {fill:none;stroke:black;}\
	line {stroke:black;stroke-width:' + thinLineWidth * fontSize / 12 + ';}\
	line.tick {stroke-width:' + thinLineWidth * fontSize / 6 + ';}\
	line.pattern {stroke-width:' + thinLineWidth * fontSize / 18 + ';}\
	circle {fill:none;stroke:black;stroke-width:' + thinLineWidth * fontSize / 12 + ';}\
	rect {stroke:black;stroke-width:' + thinLineWidth * fontSize / 12 + ';}\
	.highlight {stroke:black;stroke-width:'+ highlightLineWidth * fontSize / 12 + ';}\
	.searchHighlight {fill:rgb(255, 255, 100);stroke:none;}\
	</style>\
<pattern id="hiddenPattern" patternUnits="userSpaceOnUse" \
x="0" y="0" width="' + patternWidth + '" height="' + patternWidth + '">\
<line class="pattern" x1="0" y1="0" x2="' + patternWidth / 2 + '" y2="' + patternWidth / 2 + '"/>\
<line class="pattern" x1="' + patternWidth / 2 + '" y1="' + patternWidth +
'" x2="' + patternWidth + '" y2="' + patternWidth / 2 + '"/>\
</pattern>\
</defs>\
';
}

function svgText(text, x, y, anchor, bold, color)
{
	if ( typeof(anchor) == 'undefined' )
	{
		anchor = 'start';
	}
	
	if ( color == undefined )
	{
		color = 'black';
	}
	
	return '<text x="' + x + '" y="' + y +
		'" style="font-color:' + color + ';font-weight:' + (bold ? 'bold' : 'normal') +
		'" text-anchor="' + anchor + '">' + text + '</text>';
}

function toggleKeys()
{
	if ( showKeys )
	{
		keyControl.value = '…';
		showKeys = false;
	}
	else
	{
		keyControl.value = 'x';
		showKeys = true;
	}
	
	updateKeyControl();
	
	if ( progress == 1 )
	{
		draw();
	}
}

function update()
{
	if ( ! head )
	{
		return;
	}
	
	if ( mouseDown && focusNode != selectedNode )
	{
		var date = new Date();
		
		if ( date.getTime() - mouseDownTime > quickLookHoldLength )
		{
			if ( focusNode.hasChildren() )
			{
				expand(focusNode);
				quickLook = true;
			}
		}
	}
	
	if ( updateViewNeeded )
	{
		resize();
		mouseX = -1;
		mouseY = -1;
		
		collapse = collapseCheckBox.checked;
		compress = true;//compressCheckBox.checked;
		shorten = true;//shortenCheckBox.checked;
		
		checkSelectedCollapse();
		updateMaxAbsoluteDepth();
		
		if ( focusNode.getCollapse() || focusNode.depth > maxAbsoluteDepth )
		{
			setFocus(selectedNode);
		}
		else
		{
			setFocus(focusNode);
		}
		
		updateView();
		
		updateViewNeeded = false;
	}
	
	var date = new Date();
	progress = (date.getTime() - tweenStartTime) / tweenLength;
//	progress += .01;
	
	if ( progress >= 1 )
	{
		progress = 1;
	}
	
	if ( progress != progressLast )
	{
		tweenFactor =// progress;
			(1 / (1 + Math.exp(-tweenCurvature * (progress - .5))) - .5) /
			(tweenMax - .5) / 2 + .5;
		
		if ( progress == 1 )
		{
			snapshotButton.disabled = false;
			zoomOut = false;
			
			//updateKeyControl();
			
			if ( ! quickLook )
			{
				//checkHighlight();
			}
			
			
			if ( fpsDisplay )
			{
				fpsDisplay.innerHTML = 'fps: ' + Math.round(tweenFrames * 1000 / tweenLength);
			}
		}
		
		draw();
	}
	
	progressLast = progress;
}

function updateDatasetButtons()
{
	if ( datasets == 1 )
	{
		return;
	}
	
	var node = selectedNode ? selectedNode : head;
	
	datasetButtonLast.disabled =
		node.attributes[magnitudeIndex][lastDataset] == 0;
	
	datasetButtonPrev.disabled = true;
	datasetButtonNext.disabled = true;
	
	for ( var i = 0; i < datasets; i++ )
	{
		var disable = node.attributes[magnitudeIndex][i] == 0;
		
		datasetDropDown.options[i].disabled = disable;
		
		if ( ! disable )
		{
			if ( i != currentDataset )
			{
				datasetButtonPrev.disabled = false;
				datasetButtonNext.disabled = false;
			}
		}
	}
}

function updateDatasetWidths()
{
	if ( datasets > 1 )
	{
		for ( var i = 0; i < datasets; i++ )
		{
			context.font = fontBold;
			var dim = context.measureText(datasetNames[i]);
			datasetWidths[i] = dim.width;
		}
	}
}

function updateKeyControl()
{
	if ( keys == 0 )//|| progress != 1 )
	{
		keyControl.style.visibility = 'hidden';
	}
	else
	{
		keyControl.style.visibility = 'visible';
		keyControl.style.right = margin + 'px';
		
		if ( showKeys )
		{
			keyControl.style.top =
				imageHeight -
				(
					keys * (keySize + keyBuffer) -
					keyBuffer +
					margin +
					keyControl.clientHeight * 1.5
				) + 'px';
		}
		else
		{
			keyControl.style.top =
				(imageHeight - margin - keyControl.clientHeight) + 'px';
		}
	}
}

function updateView()
{
	if ( selectedNode.depth > maxAbsoluteDepth - 1 )
	{
		maxAbsoluteDepth = selectedNode.depth + 1;
	}
	
	highlightedNode = selectedNode;
	
	angleFactor = 2 * Math.PI / (selectedNode.magnitude);
	
	maxPossibleDepth = Math.floor(gRadius / (fontSize * minRingWidthFactor));
	
	if ( maxPossibleDepth < 4 )
	{
		maxPossibleDepth = 4;
	}
	
	var minRadiusInner = fontSize * 8 / gRadius;
	var minRadiusFirst = fontSize * 6 / gRadius;
	var minRadiusOuter = fontSize * 5 / gRadius;
	
	if ( .25 < minRadiusInner )
	{
		minRadiusInner = .25;
	}
	
	if ( .15 < minRadiusFirst )
	{
		minRadiusFirst = .15;
	}
	
	if ( .15 < minRadiusOuter )
	{
		minRadiusOuter = .15;
	}
	
	// visibility of nodes depends on the depth they are displayed at,
	// so we need to set the max depth assuming they can all be displayed
	// and iterate it down based on the deepest child node we can display
	//
	var maxDepth;
	var newMaxDepth = selectedNode.getMaxDepth() - selectedNode.getDepth() + 1;
	//
	do
	{
		maxDepth = newMaxDepth;
		
		if ( ! compress && maxDepth > maxPossibleDepth )
		{
			maxDepth = maxPossibleDepth;
		}
		
		if ( compress )
		{
			compressedRadii = new Array(maxDepth);
			
			compressedRadii[0] = minRadiusInner;
			
			var offset = 0;
			
			while
			(
				lerp
				(
					Math.atan(offset + 2),
					Math.atan(offset + 1),
					Math.atan(maxDepth + offset - 1),
					minRadiusInner,
					1 - minRadiusOuter
				) - minRadiusInner > minRadiusFirst &&
				offset < 10
			)
			{
				offset++;
			}
			
			offset--;
			
			for ( var i = 1; i < maxDepth; i++ )
			{
				compressedRadii[i] = lerp
				(
					Math.atan(i + offset),
					Math.atan(offset),
					Math.atan(maxDepth + offset - 1),
					minRadiusInner,
					1 - minRadiusOuter
				)
			}
		}
		else
		{
			nodeRadius = 1 / maxDepth;
		}
		
		newMaxDepth = selectedNode.maxVisibleDepth(maxDepth);
		
		if ( compress )
		{
			if ( newMaxDepth <= maxPossibleDepth )
			{
//				compress
			}
		}
		else
		{
			if ( newMaxDepth > maxPossibleDepth )
			{
				newMaxDepth = maxPossibleDepth;
			}
		}
	}
	while ( newMaxDepth < maxDepth );
	
	maxDisplayDepth = maxDepth;
	
	lightnessFactor = (lightnessMax - lightnessBase) / (maxDepth > 8 ? 8 : maxDepth);
	keys = 0;
	
	nLabelOffsets = new Array(maxDisplayDepth - 1);
	labelOffsets = new Array(maxDisplayDepth - 1);
	labelLastNodes = new Array(maxDisplayDepth - 1);
	labelFirstNodes = new Array(maxDisplayDepth - 1);
	
	for ( var i = 0; i < maxDisplayDepth - 1; i++ )
	{
		if ( compress )
		{
			if ( i == maxDisplayDepth - 1 )
			{
				nLabelOffsets[i] = 0;
			}
			else
			{
				var width =
					(compressedRadii[i + 1] - compressedRadii[i]) *
					gRadius;
				
				nLabelOffsets[i] = Math.floor(width / fontSize / 1.2);
				
				if ( nLabelOffsets[i] > 2 )
				{
					nLabelOffsets[i] = min
					(
						Math.floor(width / fontSize / 1.75),
						5
					);
				}
			}
		}
		else
		{
			nLabelOffsets[i] = Math.max
			(
				Math.floor(Math.sqrt((nodeRadius * gRadius / fontSize)) * 1.5),
				3
			);
		}
		
		labelOffsets[i] = Math.floor((nLabelOffsets[i] - 1) / 2);
		labelLastNodes[i] = new Array(nLabelOffsets[i] + 1);
		labelFirstNodes[i] = new Array(nLabelOffsets[i] + 1);
		
		for ( var j = 0; j <= nLabelOffsets[i]; j++ )
		{
			// these arrays will allow nodes with neighboring labels to link to
			// each other to determine max label length
			
			labelLastNodes[i][j] = 0;
			labelFirstNodes[i][j] = 0;
		}
	}
	
	fontSizeText.innerHTML = fontSize;
	fontNormal = fontSize + 'px ' + fontFamily;
	context.font = fontNormal;
	fontBold = 'bold ' + fontSize + 'px ' + fontFamily;
	tickLength = fontSize * .7;
	
	head.setTargets(0);
	
	keySize = ((imageHeight - margin * 3) * 1 / 2) / keys * 3 / 4;
	
	if ( keySize > fontSize * maxKeySizeFactor )
	{
		keySize = fontSize * maxKeySizeFactor;
	}
	
	keyBuffer = keySize / 3;
	
	fontSizeLast = fontSize;
	
	if ( datasetChanged )
	{
		datasetChanged = false;
	}
	else
	{
		datasetAlpha.start = 0;
	}
	
	var date = new Date();
	tweenStartTime = date.getTime();
	progress = 0;
	tweenFrames = 0;
	
	updateKeyControl();
	updateDatasetWidths();
	
	document.title = 'Krona - ' + selectedNode.name;
	updateNavigationButtons();
	snapshotButton.disabled = true;
	
	maxAbsoluteDepthText.innerHTML = maxAbsoluteDepth - 1;
	
	maxAbsoluteDepthButtonDecrease.disabled = (maxAbsoluteDepth == 2);
	maxAbsoluteDepthButtonIncrease.disabled = (maxAbsoluteDepth == head.maxDepth);
	
	if ( collapse != collapseLast && search.value != '' )
	{
		onSearchChange();
		collapseLast = collapse;
	}
}

function updateMaxAbsoluteDepth()
{
	while ( selectedNode.depth > maxAbsoluteDepth - 1 )
	{
		selectedNode = selectedNode.getParent();
	}
}

function updateNavigationButtons()
{
	backButton.disabled = (nodeHistoryPosition == 0);
//	upButton.disabled = (selectedNode.getParent() == 0);
	forwardButton.disabled = (nodeHistoryPosition == nodeHistory.length);
}

function useHue()
{
	return useHueCheckBox && useHueCheckBox.checked;
}
/*
function zoomOut()
{
	return (
		selectedNodeLast != 0 &&
		selectedNodeLast.getDepth() < selectedNode.getDepth());
}
*/
  </script>
 </head>
 <body>
  <img id="hiddenImage" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAgAAAAICAYAAADED76LAAAAAXNSR0IArs4c6QAAAAZiS0dEAP8A/wD/oL2nkwAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9oLCBQhNQwWVnsAAAAidEVYdENvbW1lbnQAQ3JlYXRlZCB3aXRoIEdJTVAgb24gYSBNYWOHqHdDAAABE0lEQVQYGQEIAff+AwAAABkAAAAAAAAA+gAAAAAAAAAAAAAAAAAAAAAAAAAMAwAAAAAAAAANAAAAAAAAAPoAAAAAAAAADAAAAAYAAAD0AwAAAPoAAAAAAAAAAAAAAPoAAAAMAAAADQAAAPoAAAD6AAAAAAAAAAAAAAAAAAAAAAwAAAAZAAAADAAAAAAAAAAAAAAAAAAAAAAAAAAADAAAABkAAAAMAAAAAAAAAAAAAAAAAAAAAAAAAAAMAAAAGQAAAAwAAAAAAAAADAAAAAwAAAAABAAAAAAAAAAAAAAA8wAAAPQAAAAAAAAAAAAAAA0AAAAAAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwAAAAZRssKC5OpXwYAAAAASUVORK5CYII=" style="display:none"/>
  <img id="loadingImage" src="data:image/gif;base64,R0lGODlhEAAQAPIAAP///wAAAMLCwkJCQgAAAGJiYoKCgpKSkiH/C05FVFNDQVBFMi4wAwEAAAAh/hpDcmVhdGVkIHdpdGggYWpheGxvYWQuaW5mbwAh+QQJCgAAACwAAAAAEAAQAAADMwi63P4wyklrE2MIOggZnAdOmGYJRbExwroUmcG2LmDEwnHQLVsYOd2mBzkYDAdKa+dIAAAh+QQJCgAAACwAAAAAEAAQAAADNAi63P5OjCEgG4QMu7DmikRxQlFUYDEZIGBMRVsaqHwctXXf7WEYB4Ag1xjihkMZsiUkKhIAIfkECQoAAAAsAAAAABAAEAAAAzYIujIjK8pByJDMlFYvBoVjHA70GU7xSUJhmKtwHPAKzLO9HMaoKwJZ7Rf8AYPDDzKpZBqfvwQAIfkECQoAAAAsAAAAABAAEAAAAzMIumIlK8oyhpHsnFZfhYumCYUhDAQxRIdhHBGqRoKw0R8DYlJd8z0fMDgsGo/IpHI5TAAAIfkECQoAAAAsAAAAABAAEAAAAzIIunInK0rnZBTwGPNMgQwmdsNgXGJUlIWEuR5oWUIpz8pAEAMe6TwfwyYsGo/IpFKSAAAh+QQJCgAAACwAAAAAEAAQAAADMwi6IMKQORfjdOe82p4wGccc4CEuQradylesojEMBgsUc2G7sDX3lQGBMLAJibufbSlKAAAh+QQJCgAAACwAAAAAEAAQAAADMgi63P7wCRHZnFVdmgHu2nFwlWCI3WGc3TSWhUFGxTAUkGCbtgENBMJAEJsxgMLWzpEAACH5BAkKAAAALAAAAAAQABAAAAMyCLrc/jDKSatlQtScKdceCAjDII7HcQ4EMTCpyrCuUBjCYRgHVtqlAiB1YhiCnlsRkAAAOwAAAAAAAAAAAA==" style="display:none"/>
  <img id="logo" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAANgAAAA8CAYAAAAE9XR5AAAC0WlDQ1BJQ0MgUHJvZmlsZQAAOI2NlM9LFGEYx7+zjRgoQWBme4ihQ0ioTBZlROWuv9i0bVl/lBLE7Oy7u5Ozs9PM7JoiEV46ZtE9Kg8e+gM8eOiUl8LALALpblFEgpeS7Xlnxt0R7ccLM/N5nx/f53nf4X2BGlkxTT0kAXnDsZJ9Uen66JhU+xEhHEEdwqhTVNuMJBIDoMFjsWtsvofAvyute/v/OurStpoHhP1A6Eea2Sqw7xfZC1lqBBC5XsOEYzrE9zhbnv0x55TH8659KNlFvEh8QDUtHv+auEPNKWmgRiRuyQZiUgHO60XV7+cgPfXMGB6k73Hq6S6ze3wWZtJKdz9xG/HnNOvu4ZrE8xmtN0bcTM9axuod9lg4oTmxIY9DI4YeH/C5yUjFr/qaoulEk9v6dmmwZ9t+S7mcIA4TJ8cL/TymkXI7p3JD1zwW9KlcV9znd1Yxyeseo5g5U3f/F/UWeoVR6GDQYNDbgIQk+hBFK0xYKCBDHo0iNLIyN8YitjG+Z6SORIAl8q9TzrqbcxtFyuZZI4jGMdNSUZDkD/JXeVV+Ks/JX2bDxeaqZ8a6qanLD76TLq+8ret7/Z48fZXqRsirI0vWfGVNdqDTQHcZYzZcVeI12P34ZmCVLFCpFSlXadytVHJ9Nr0jgWp/2j2KXZpebKrWWhUXbqzUL03v2KvCrlWxyqp2zqtxwXwmHhVPijGxQzwHSbwkdooXxW6anRcHKhnDpKJhwlWyoVCWgUnymjv+mRcL76y5o6GPGczSVImf/4RVyGg6CxzRf7j/c/B7xaOxIvDCBg6frto2ku4dIjQuV23OFeDCN7oP3lZtzXQeDj0BFs6oRavkSwvCG4pmdxw+6SqYk5aWzTlSuyyflSJ0JTEpZqhtLZKi65LrsiWL2cwqsXQb7Mypdk+lnnal5lO5vEHnr/YRsPWwXP75rFzeek49rAEv9d/AvP1FSOihagAAAAZiS0dEAP8A/wD/oL2nkwAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB+AEEhYRJKJxkPUAACAASURBVHgBAGaAmX8B/////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAvUj3EAAAIABJREFUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////AAAAAAD+/v4AAAAAAP///wD///8AAAAAAP///wAAAAAAAQEBAAEBAQAAAAAAAQEBAAICAgAAAAAAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wAAAAAA/v7+AP7+/gD9/f0A/v7+AP39/QD+/v4A/f39AP39/QD9/f0AAQIBAAMCAwAEBAQAAwMDAAMDAwACAgIAAwMDAAEBAQABAQEAAgICAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////AP7+/gD///8A/v7+APv7+wD7+/sA+/v7AN7e3gDd3d0A5eXlAPLy8gDn5+cA////ABUVFQAQEBAAGRkZACMjIwAfHx8ABQUFAAYGBgAFBQUABAQEAAQEBAACAgIAAgICAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAf////8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////AP39/QD8/PwA+vr6APb29gDFxcUAxMPDAN3a3ADz9/UACQQEABwKDAAQBQQA0e7tAAgVBwAVHg0A9e76AO/o9wD59voACwwKACQjJQA1NTUAODg4AAsLCwAJCQkABgYGAAQEBAACAgIAAgICAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH/////AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAZ2dnAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJmZmQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANra2gCRkZEA/Pz8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA8PDwB1dXUAFRUVAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/v7+AP39/QD5+fkA6+vrALKysgC/u7sA9vf3ADgUFQA1ERIAMBEQABUHCAACAAEAAAAAAF3JxgAZPhMAQ2gmAAABAAAAAAAA8uj5AOTW8ADey+wA383uAAkICQA7PDwAS0tLABMTEwAKCgoABgYGAAUFBQADAwMAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAODg4ACUlJQA/Pz8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAsLCwBwcHAAFRUVAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/f39APv7+wD39/cAr6+vAK+urgAI/v4AWhscAEYXGQAIAwMAAAAAAAAA/wAA//8A/gAAAAAAAAAAAP4A/wD/AP/+/wAAAP8AAAAAAA0UBQApPhUAKkEXAPjz/ADXwekAwa6+AMTEwABISEwASEhIAA4ODgAKCgoABgYGAAICAgACAgIAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAObm5gCRkZEA+fn5AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAcHBwBvb28AGhoaAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/Pz8APr6+gDg4OAAj4yMAOTj4wBbHh8ATxscAAUBAgAAAP8A///+AAAAAAABAQEAAAAAAAD//wD///8A/wAAAP79/gD9/f0AAQAAAAAAAAD/AAAAAAAAAAMBAQAHCwMAMUwbAC5GGgDNseMAsK2tAM/PzgBhYWIAICAgAAwMDAAJCQkAAAAAAAICAgABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAOvr6wCQkJAA9vb2AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAUFBQBtbW0AHx8fAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/Pz8APj4+ADGxsYAeHR0AAXv8ACZMTQAVBwdAAUBAQD///4A////AAD/AAAA/wAA/v7+AP7+/gD///8AAAAAAAAAAAD///8A+/r6AP39/QD9/f0A/v3+AP79/gD9/P0A/fz9AP/+/wACBAAAOFYfAF2PNwD1AO0AiYiGAMnJyQDw8PAA9PT0APv7+wD+/v4A////AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAO/v7wCPj48A8fHxAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMDAwBoaGgAJSUlAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/Pz8APf39wC5ubkAiYiIAD4SEgCLLC4AFQcHAP///wD+/v8A//7/AP/+/wAAAAAA/wD/AP///wD///8A/v/+AP7//gAAAAAA//7/AP7+/wD9/f0A/f39AP39/QD9/f0A/v7+AP7+/gD9/P0A/fz9AP39/gASGgkAV4cwAB42DgCWlpUAv7+/APDw8AD29vYA+/v7AP7+/gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABpHbeyAAAgAElEQVQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPPz8wCQkJAA7e3tAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQBkZGQALCwsAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/Pz8APj4+AC/v78Al5WWAFsdHQBYHiAA/wAAAP8AAAD//v4AAAAAAAAAAQAAAAAA/f79AAAAAAAAAQEAAAAAAAIAAAAAAAAAAAAAAP8A/wD9/P0AAAICAAAAAAD/AAAAAAAAAAAAAAAAAAAAAAAAAAIDAgD9/f0AAAAAAAQEAQBIbCgAChAGAKSkowAEBAQAOjo6AAICAgACAgIAAAAAAAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPf39wCRkZEA6OjoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABfX18AMjIyAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/f39APv7+wDR0dEAl5SUAFMaGwBXHh8A/v//AAEBAAAAAAAA/v8AAAAAAAAAAf8AAP//AAAAAAAAAAAAAP//AAD/AAD9AP8AAAAAAAD/AAD//v8A/f79AAAA/wAAAAAA/gAAAAQCAwD+/P0AAgQDAP39/QAAAAAA/gD/AAAAAAAAAAMAAwMBADxbIgD9+f8ApKWiAA0NDQAwMDAAAQEBAAoKCgACAgIAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPn5+QCTk5MA4+PjAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABXV1cAOTk5AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA/v7+APz8/ADr6+sAkY2NADUREgBeICEA////AP7/AAAA//8AAP8AAAABAAAAAAAAAP4AAAAAAAD/AP8AAAAAAAD//wAAAAAAAAEBAAAAAAAAAAAA/wD/AP79/AACAAQAAAAAAAEAAAD+AAEAAAD/AP/8AAAAAAAAAAD/AAMDBAD+/f4AAQAAAAAAAwABAgEAQmMkAPDo9wCjo6IAGRkZAAQEBAAEBAQAAQEBAAMDAwABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPv7+wCYmJgA3d3dAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABPT08AQkJCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA/v7+AMHBwQDBwcEA6urqAAcHBwAfHx8ANzc3ADY2NgADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////AP7+/gD5+fkAmpeXAAT6+gBuJykA/v/+AAIBAQAAAP8AAAAAAAABAAD///8A////AAEAAQAAAAAA///+AAAAAAAAAQEAAAAAAAD+/gAAAAAA/wD/AP/+/wD9/f4A/f35AAMCAwD/AAEAAwT/AAIAAQD+AAEAAAEAAAH+AQAAAPwAAQMDAAD9/QD/AAAAA/8CAP0DAgBLcioAnm3JAKqqqgBiYmIABgYGAAMDAwABAQEAAwMDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAOLi4gDIyMgA2traAOrq6gD7+/sADQ0NABISEgAqKioANDQ0ABoaGgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA+Pj4ANLS0gDV1dUA5eXlAO/v7wD4+PgA/f39AAUFBQALCwsAExMTAB8fHwAtLS0AKCgoAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP39/QCdnZ0A1tbWAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABGRkYASkpKAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABnZ2cAAAAAAAAAAAAAAAAAAAAAAAAAAACZmZkA/f39AJqamgDQ0NAAAAAAAAAAAAD29vYA19fXANfX1wDAwMAAeXl5AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP7+/gD7+/sAzc3NAMbJyQCFKy0ABwICAAAAAAD9/v4AAAAAAAD//wAAAAAAAAEBAAAAAAD///8AAAAAAP8AAAAAAAAAAP//AAAAAAAAAAAA/v//AAAAAAD///8A/v3+AAAAAAAAAP4AAAAAAAH8/wD+AAAAAgQAAAMCAgD9/PwAAAIDAAIAAQAAAwIAAAAAAAABAQD//v8ABAkCAEdqKQCwhc0A1zk5AAgICAAGBgYAAwMDAAICAgABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGdnZwAAAAAAAAAAAAAAAAAAAAAAAAAAAJmZmQAAAAAA8PDwAKioqADPz88AAAAAAAAAAAAAAAAAAAAAAPHx8QDx8fEA39/fALW1tQA4ODgAVFRUAA0NDQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwcHBALCwsAD29vYAAAAAAAAAAAAAAAAAAAAAAAAAAAD6+voA+vr6AO/v7wDc3NwAvb29ABwcHABfX18AHh4eAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wCjo6MAz8/PAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA+Pj4AU1NTAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKampgDQ0NAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAgIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wD+/v4A+Pj4AKejowA7FBUALA4OAP///wD///8AAAAAAAAAAAD///8A////AP7+/gD///8A/wD/AP3//gAAAAAAAAAAAP7//wD+/v4A/v/+AAAA/wAAAAAA////AP39/QD9/f4A/v3+APz8/gD9/v0A/f7+AP79/AD9/v0AAAAAAP3+/QD9/P0A/f3+AP79/QD+/f0AAAAAAAD/AAAjNBMAHzAUALa3tQDy8vIA9/f3APz8/AD+/v4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA4ODgAIWFhQDPz88AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAyMjIAI2NjQDx8fEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAp6enAKampgD29vYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADk5OQAiYmJANbW1gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACpqakAx8fHAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA2NjYAWlpaAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMrKygDExMQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACcnJwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD+/v4A+/v7AM/PzwDb398AWBsbAP///wAA//8AAP//AP///wD+/v4A/wAAAP8AAAAAAAAAAAAAAP7//wAAAAAA/v7+AP38/QD//f4A//7/AAD/AAAA/wAAAAAAAP7//wD9/f4A/f38AP39/QD//v0A/v3/AAEAAAD+/f4A/vz9AP3+/gAAAQEAAQAAAAAAAAAAAAEA+/39AP38/QD//f4A/P39AD1dIwDd3N4A2dnZAPX19QD7+/sA/v7+AP///wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA5OTkAJCQkADy8vIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADn5+cApKSkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwcHBAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/ACbm5sA9PT0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACwsLAAv7+/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAvLy8AYGBgAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACnp6cAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAoKCgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/v7+APv7+wC1tbUANBISAB0ICgAA/wAA/wAAAP7//wD///8A/v//AP7//wD+/v4A/v7+AP3+/QD//f4A//3+AP/+/wAAAAAAAAAAAAAAAAD//v8A9fz8AP3//wD79/0A+vX6AP79/gD9/f4AAAABAP39/gD7/P4A/wD/AP8AAQD9/P0A/f39AP3+/QD9/P0A/f38AAIAAAAAAQAAAAEBAAIBAQAXIg0AHSoTAMLEwgDz8/MA9/j3APz8/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJaWlgD39/cAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANLS0gDf398AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA9PT0AK2trQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA9vb2AMnJyQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAC5ubkAt7e3AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHBwcAAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA9vb2AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHBwcAHx8fAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP39/QDv7+8A1tPTAG8hIwD+//4AAAAAAAAAAAD+//8AAAAAAAAAAAAA//8A/wAAAAEAAAAA/gAAAAAAAP8A/wAAAAAAAAAAAP8AAADD7u4AuOzqANvz8wAAAAAA8NH1AOPQ9AAUHgoAIT0LAB8xDgAAAAAAAwD/AP79AwAAAP0AAwICAAIAAAAAAAEAAQAAAAAAAQAAAAAAAAAAAAAA/wD+//8AAgEAAC9IGgC7l9gA7u7uAA0NDQAJCAkABQUFAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADt7e0AAAAAAAAAAAAaGhoATExMACIiIgAICAgA3d3dAKysrADn5+cAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAy8vLACEhIQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMDAwAD5+fkAAAAAAAAAAAAAAAAAAAAAAAAAAABCQkIAQkJCABEREQD39/cA2dnZAKioqADz8/MAAAAAAAAAAAAAAAAAAAAAAAAAAADa2toAQ0NDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAArq6uAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALi4uAA1NTUAGRkZAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJiYmAAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP7+/gD9/f0AzMzMAGfP6fwAACAASURBVP///wA0DxAAAAAAAAAAAAD+//8AAP//AAABAQAA//8A/v7+AAAAAAD/AAAAAAAAAP8AAAD+//8AAgAAAPP9/ACM3d8A3PbzACQnJwBNJCQAERERAAAAAADv7+8A3NzcAM2o2QDkzPUANlwVAAMMBwAAAAAA/Pz+AAAE/wAA/AAAAgQAAAAA/wD//AAAAAD/AP4AAAABAAAAAgABAOXV7wDcyesA+ff7AMbGxgATExMADw8PAAgICAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAtLS0AaGhoAAQEBAAAAAAAAAAAACwsLAAfHx8Ag4ODAPHx8QAAAAAAAAAAAAAAAAAAAAAAAAAAAN/f3wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADLy8sAAAAAAAAAAAAAAAAAAAAAAAAAAAA4ODgAV1dXAAAAAAAAAAAADQ0NADQ0NAABAQEAmpqaAAAAAAAAAAAAAAAAAAAAAAAAAAAA6enpAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD9/f0ApKSkAHR0dAABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQBHR0cAOzs7ABISEgD7+/sA8fHxAAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wD///8A/Pz8ANLS0gApDQ4ABAIBAP7//wAA//8AAAAAAP8AAAD//f4AAAAAAAAAAAAAAAAA/wD/AAAAAAD+/wAAAAAAAPf+/QCC3dsA9QQFAFlcXAAaGhoACwsLAAYGBgABAQEA/v7+ADg4OADk5OQAqKmnAMSa6QA/bBoAAAYDAAMDAwAAAAEAAAAAAAAAAAAAAQEAAQIBAPPt+gDXveoA3s/vAPLr/AD++vsAAgkJAAMKCQDn5+cA9/f3APr6+gD9/f0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMDAwAZWVlAAQEBAAAAAAAAAAAAAAAAAAAAAAADQ0NAEJCQgCvr68AAAAAAAAAAAAAAAAAAAAAAAAAAADu7u4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAc3NzAOfn5wDm5uYA5+fnAObm5gDm5uYARkZGAAAAAAAAAAAAAAAAAAAAAAAAAAAAKCgoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPLy8gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANzc3ADDw8MAPj4+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABYWFgAQEBAAAAAAAAAAAAABAQEAAkJCQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////APz8/ADj398AOAsLAP7//wAA//8A/wAAAAEAAAAA/v8AAAD/AAAAAAD///8AAAAAAP7//wAAAAAAAAAAAAAAAACj5OQA8P8AAFZVVQAbHBsAEhESAA4ODgAGBgYAAQEBAAAAAAD7+/sAExMTABMTEwCRkY4A1bbuADBQFgD+/wAAAgD/AP37AADawewA4M7xAO3m+wD59fkAAgMDAA0oJgAXQS4ADS0rAAgfHQDknqMA8NbaAPr6+gD7+/sA////AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/AAcHBwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlJSUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA9fX1AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA4ODgAODg4AJycnAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAoKCgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD5+fkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAArq6uAC8vLwAPDw8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAATExMAPj4+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP39/QD8/fwA6+vrABscHQAAAAAA/wD/AP/+AAD/AAAA/wD/AP7//wAAAAAAAAAAAAD//wAAAAAA////AAIBAADs+voAvu3sAEdISAAZGRkAFhUWABISEgAKCgoABQUFAAEBAQD///8A/v7+AAsLCwAFBQUASkpNAK2tqwAOGQYA6NX3AOvi+QD59PwA//7+AAkdHQARQy4ACy4rAAslJAACBwYAAAAAAAAAAAAAAAAACiAgAO7GygD8/PwA/f39AP39/QAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACQkJAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAERERAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA6enpANLS0gDKysoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA/Pz8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANDQ0AMzMzAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAPf39wCnp6cAaGhoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMzMzAAICAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP7+/gD///8A9PP0APX5+QAVCAgAAP//AAD+AAAAAP8AAAAAAAD//wAAAAAAAP//AAAAAAD///8AAAAAAP///gD9AAAAxu/vAA4MDAArKysAERERABAQEAANDQ0ABwcHAAMDAwAAAAAAAAAAAP7+/gADAwMA////ADs7OwCssK8AzKnrAAQQDgANMDYAFywqAAopJgAFDQ0AAAABAAEDAwAAAQAAAAAAAAAAAAAAAAAAAAAAAAYTEQD49/cA////AP7+/gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAASEhIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAkJCQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD9/f0A5eXlANzc3ADe3t4A29vbAPDw8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKCgoAcHBwABkZGQDc3NwA5OTkAAAAAAAAAAAAAAAAAAAAAAAAAAAAz8/PANHR0QAvLy8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAgIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////APX19QAAAAAACQABAP7+/gAAAAAA////AP4AAAAAAAAAAf/+AP7/AAAAAAAA/wD+AAEAAgAAAAAA////AN/39wAUFxcACgoKAAwMDAAKCgoABgYGAAMDAwABAQEAAAAAAAAAAAADAwMA////AAcHBwALCwsAHRscAO8TEQAdYl4AAQQDAAAAAAAAAQAAAP7+AP/+/gAAAP8AAAEBAAEAAAD/AgIAAAAAAAECAQADCgoA/v39APn5+QD///8A/v7+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFBQUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////ANXV1QDJyckA09PTAPf39wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGBgYAbm5uABwcHAAAAAAAXV1dAIiIiAD///8AAAAAAAAAAAAAAAAAAAAAAAAAAACmpqYAJycnAAgICAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAASEhIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD4+PgAAAAAAAcDAwAAAAAA/f//AAD//wAAAAAAAAAAAAAAAAAAAAAA///+AP8A/wD///0AAAAAAP///wDz//4ACQkJAAYGBgAGBgYABQUFAAEBAQABAQEAAAAAAAAAAAAAAAAAAAAAAAEBAQAEBAQACAgIABQSEgDw+/oA/v79AAD6/QACAgMAAAAAAP//AAABAAAAAPz+AP8AAAAA/wAAAQQAAAAEAwD//v8AAgcIAAAAAAD5+fkA////AP///wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAcHBwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwMDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA6enpAJ2dnQDh4eEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQAa2trACEhIQAAAAAAAAAAACAgIABCQkIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA8PDwAK+vrwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABwcHAAAAAAD5/fwA/wD/AAAAAAAAAAAA////AP///wD9/v4A/v/9AP///wAA/wAA//8AAP//AAAAAAAADAEBAPn5+QABAQEAAAAAAAAAAAABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAACAgIAAQEBAAICAgD2+PgAAAgHAAD+/QAA/P0A/v39AP7+/QAA/P0A//z9AP/+/QD//v0A//z9AP/8/QAA+/0AAf38AP34+AAAAAAABgYGAP7+/gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA4uLiAIWFhQDh4eEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAQEAZmZmACgoKAAAAAAAAAAAAAAAAAAAAAAATk5OABISEgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAwMAA39/fAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAkJCQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAsLCwAAAAAA9vz9AP7//wAA//8A////AP///gD///4A/wD/AAD/AAD/AAAA/wAAAP7//wD+//4A/v//AB8HCADr6OgA+/z7APz8/AD9/f0A////AP///wAAAAAAAAAAAAAAAAAAAAAA/v7+AP7+/gD+/v4A4ePjAAkdHAAA/P4AAP79AAH//QAA/v0A/v79AP7+/QD+/P0AAQAAAAD+/QAA/v0AAP79AP8AAAD99fUAAQICAAcHBwACAgIAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgICAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA+fn5AJqamgD5+fkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQEBAAnJycAUFBQAH5+fgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAYmJiAC8vLwAAAAAAAAAAAAAAAAAAAAAAAAAAAAgICABbW1sAk5OTAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKKiogB8fHwAAwMDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGBgYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQAKCgoACwcHAOj5+QAgCQkAAQD/AP8AAAAAAAAAAAAAAP//AAAA/wAA/v//AAAAAAAAAP8A//8AAAEBAAA1Dg0AqunoANTT1AAVFBUA+fn5AAUFBQABAQEAAgICAAEBAQAAAAAA/f39AP39/QD4+PgAv7+/APj19QAbKSgA//z8AAAAAAD//wAAAQEBAAEEAAAAAAEAAAABAP/8AAAAAAAAAAAAAP/+AAABBAIA++zrAAoLCwACAgIAAgICAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMzMzADr6+sAAAAAAAAAAAAAAAAAAAAAgelArwAAIABJREFUAAAAAAAAAAAAMjIyADs7OwAnJycABQUFAAAAAAD///8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFWSqm1AgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANjY2AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAsLCwANTU1AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAObm5gC5ubkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgICAAEBAQACAgIAFBQUAOX49wD///8A/f7+AP7//QD+//8A/v//AAAAAAAAAP8AAAAAAAAA/wD///8A///+AP7//wASBAUAPhARALy5uADo6OgA8vLyAPf39wD6+voA/Pz8APv7+wD7+/sA+/v7APb29gDc3NwApaioABA3NAABBwYA//79AP/+/gAA/f0A//z8AP78/QD//PwA//78AP7+/QAAAAAA//z9AP/+/QAA/P4A9eDhABISEgAFBQUABAQEAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMrKygAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABLS0sAZ2dnACwsLAAFBQUAAAAAAAAAAAD7+/sAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFZWVgAKCgoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAtbW1AOnp6QAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgICAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAgIABAQEABoeHgDY9vUAAAAAAP8A/wAA/wAAAQAAAAAAAAAA//8A/v//AP///wD+//8AAAAAAAAAAAD9/v4A/v//AFcWFgAN/v4Am5eXANzc3ADv7+8A8vPyAPPz8wD19fUA9fX1APHy8QDU1NQAfn5/APf29gAPOjYAAf7/AAD8/gD//PwA//39AP/+/gAA/P4A//8AAP/7/QAA/v0A/v79AAAAAAABAAAAAAIAAPzf4gAWFBQABgYGAAMDAwADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADo6OgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQ0NDAAAAAAAAAAAAAAAAAAAAAAAAAAAA8/PzAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAODg4AVFRUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/AClpaUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQEBAAQEBAArKysA2fPzAPr9/AABAAEAAAEAAP///wD+/v8A/v8AAP//AAD///4A//7/AP7//wD9//8AAP//AAD//wAGAQEAdBocAA739gCLiYkAmpqaAMTDxADb29sA2traAL+/vwCPjo8Afn5/ABEFHQACAgQA5KKoAPz19AD+/fwA//39AP/9/QAA/P0AAP79AAD9/QAA/f0AAPz+AAH+/gD+/v0A//z9APvw8gDy3+AAHx8fAAYGBgAFBQUAAgICAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA8/PzAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/AAAAAAAAAAAAAAAAAAAAAAAAAAAANPT0wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA/f39AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADs7OwDJyckA2dnZAAAAAAAAAAAAAAAAAAAAAAAAAAAA2dnZAMXFxQA7OzsAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQEBAAEBAQADAwMAMjIyAKKfnwDO8/IANQ4PAAD//wAAAAAAAAD/AAAAAAAAAP8A/wAAAAAA/wAA/wEAAAAAAAAAAAAAAAAA////AAoBAQB1Gx0AyfPzAL7V1ACTyckAAQEBAP///wACAQIAFQsfADwaYQAgDzAA8vvqALLeiQDinqQAHmNdAAIHBwABAQIA/v/9AAEA/wABAwMAAAMDAAEA/wD//PwAAAIDAP4EAADxzM0ABwkJACQkJAAHBwcABgYGAAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAcHBwD29vYAAAAAAAAAAAAAAAAAAAAAAAAAAADLy8sAREREAAAAAAAAAAAAAAAAAOvr6wC0tLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/AADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAW1tbAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACsrKwA8vLyAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAQEABAQEAA8PDwAnKioAwe/vAAD/AAAAAAAA/v8AAP8AAAD+/v4A/v//AP7//wD+//8A/v/+AAD+/wD+/v4A/v7+AP///wD///8AAAAAADYMDAB2GRsAkCMkACwJCQANBhQAVSWJAEYecwAhEDIAAgIDABEJGwBSJXsA/O7xAOKdogD99fYA/v37AAD+/gAB/v8A//39AAD+/QD//P4AAAAAAAECAAAAAAAA8cjMACEdHgAMDAwACAgIAAUFBQABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAaGhoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAr6+vAMDAwAD09PQA9PT0AMTExACIiIgA8/PzAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD29vYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABcXFwBMTEwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA9vb2AKqqqgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgICAAMDAwAHBwcAREVEANDx8QDj+PgA/v//AP///gD//v8AAAAAAAD//wAA//8AAAAAAAAAAAD/AP8AAAEAAP8AAAD/AP8A/gAAAP3//gD+//8AAP//AAgBAQAGAgIAAgEDAAUDCAABAQIAAQEBAAMCAwAEAgEAEgoaAFUofQD97vIA4qGlAP319QAAAAAAAAAAAP/9/QD//P0A//38AP/8/QD++/0A9tvcAPbe3wA2NjYACwsLAAkJCQAEBAQAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALy8vAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/ACnp6cAc3NzAHNzcwCjo6MA9PT0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA7+/vAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAARkZGABsbGwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADLy8sA1NTUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQACAgIABgYGAC8uLwAgHR0AsOvqAP//AAD///8A////AP//AAD+AAAA////AP///wD///8A////AP/+/wD+/v4A/v7/AAD//wAB//8AAQD/AP///wD+/v4AAAAAAAABAAADAwMAAgIBAAECAgABAgAA/wAAAAEAAQAUCx0AVyuAAP7u8gDioacA/fX2AP79/AACAQIAAQABAAABAAAAAAAAAQEBAOy3uwAfISEAHx8fAAwMDAAHBwcABAQEAAICAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD4+PgANDQ0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQEBAAcHBwAAAAAAAAAAAAAAAAAAAAAAAAAAAOHh4QAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMDAwBdXV0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKampgD5+fkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwMDAAUFBQAKCgoAUVRUAMrx8ADX9fQAAAEBAAABAAD+AP8AAP//AAEBAQAAAAAA////AP8A/wAAAAAAAAAAAP///wD///8A/v//AP7//wD+/v4A////AAD/AAABAAEABAMDAAMCAwAEAgEAAgIDAAICAwABAwIA/gAAABIKHABbLoQA/u7yAOOhpQD99vYA/v37AP79/AD//P0A//39APLP0QD1290AOzk5AA4ODgALCwsABwcHAAEBAQABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAICAgAZWVlAAICAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABgYGAGxsbAAyMjIAAAAAAAAAAAAAAAAAAAAAAAAAAADMzMwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAISEhAEJCQgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADt7e0AsrKyAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQACAgIADExMQAzMDAAnunoAPj9/QAAAAAAAAAAAAAAAAD///8A////AP///wD+/v8A/v//AAH//wD///8A////AP///wD///8AAAEAAAABAAAAAP8AAAAAAAECAQABAgAAAQIDAAIBAQACAQEAAQECAAMCAQACAgEAEgsaAFsvhQD97/IA46ClAP75+gACAQIAAAAAAP349gDqrrEALi0tACMjIwAPDw8ACQkJAAUFBQADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACcnJwBxcXEADw8PAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKCgoAHp6egAsLCwAIyMjAAAAAAAAAAAAAAAAAAAAAAAAAAAA3NzcAOPj4wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABPT08AERERAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAL29vQDh4eEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAd8jdrAAAgAElEQVQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAQEAAQEBAAcHBwALCwsAW15eAP0GBgCe5+cA/v7+AP/+/gD+/v4A////AP8AAAD+AP8AAAAAAAEA/wD+AAAAAAAAAAAAAAD///8A/v7/AP/+/wD//v8A/wAAAAEBAQACAgMABQQFAAMCAgABAwMAAQMDAAMCAQACAQIAAgEBAAIBAgAWDR4AYDKHAP3v8gDinaMA/Pf3AP/+/wDop60ACQUGAENEQwAREREADQ0NAAgICAADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJiYmAISEhABZWVkAKysrAA8PDwADAwMACQkJABkZGQA+Pj4AcnJyAHFxcQAZGRkAAAAAACcnJwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACtra0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH/////AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wD9/f0A+vr6APHx8QDLzMsAlZGSAEQODwBSFhYA/v7+AP///wD+AP8AAP8AAP///wAAAP8A//8AAP///gD//wAAAAEAAP7//wAA/wAAAAD/AKDr6wDfAA8AWC2IAAD/AAAAAAAAAwIBAAICAgAAAQAAAgIBAAECAwADAgIAAQEAAAICAwDv9+UAncp1AAMSDwAJGhgA+NjbAFNRUQAsLCwAGhoaABEREQAKCgoABAQEAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAQEAAwMDAAUFBQALCwsALy4vAFhbWwC47+4As+rqAAAAAAAAAAAAAAABAAEA/wAAAAAA////AP///wD//wAA/v//AP7//wAA//8AAAD/AAAAAAD/AAAAAAAAAAQCAgADAwEABQUDAAIDAgACAwEAAgEBAAEBAwADAQIAAgEAAAQCAwADAgIAFQwfAGQ5jAD77O4A8tLVAEE/PwAiIiIAEBAQAAwMDAAHBwcAAgICAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQACAgIACAgIAA0NDQA/Pz8A+/v7AKqpqQDw/f0AVxUVAAIA/wAAAAAAAAAAAP//AAD///4AAAABAAIBAQD+//4A////AP///wAAAAAAAAAAAAEAAAABAQEAAwMCAAAAAAAA/wAAAgECAAIDAwAAAAAAAQIAAAECAwD/AgIAAgECAP0C/ADH3rAAAwECADMzMQAsLCwAExMTABEREQAKCgoABQUFAAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwMDAAAAAAAKCgoA////AENDQwALCwsApaKiAOPm5QBiFRQAEgMFAP8A/wAAAAAAAAEBAP7+/wAAAP8A//8AAAAAAAAAAAAAAAAAAAD//wAAAgEAAwQEAAEA/wAAAAAAAgMCAAAAAAACAwIAAAAAAAMBAQABAgIAAgECAPD56ACu0IkA8vvqAEtLSQAnJycAFRUVABQUFAAMDAwABQUFAAICAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAgIABQUFAAsLCwABAQEAOTk5ABYWFgCinp4AguPkAEoREQBBDg4AAv8AAP4A/wAAAP8A////AAAAAAD///8A/v//AAD//wD/AAAAAQABAAMDAgAA/wAAAAAAAAICAgAAAAAAAQEBAAECAgD///8A/v/7AMTerwDG3qwAGBkXAFlKWQAbHBsAEhESABEREQAPDw8ACAgIAAQEBAABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQEBAAICAgAFBQUACgoKAAICAgAEBAQAdHh4ALi4uAC6v78AwPT0AEMMDAA3CwwABgD/AP///wAAAAAA////AAAAAAD+AAAAAAAAAAABAAACAgIAAQEBAAAAAAACAAAA/wICAAIAAAD5/fYAz+S6AMnhsAABA/4AOz46ADo5OgAREREAERERAA4PDgAKCQoACAgIAAICAgABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAQEAAgICAAUFBQACAgIAAgICAAUFBQBPT08ANzg3AL28vQC4zMwAsO7uAC0KCgAoBwcAJggIAA8BAgD+//4AAAAAAAD/AAAAAAAABAMDAAAAAAAAAAAA7/boAOLw1QDb68oA3uvPAAcHCAAtMCsANzc3ABUVFQAQEBAADg4OAAwMDAAJCQkABQUFAAMDAwABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAB/////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////APz8/AD7+/sA9/f3APX19QD09PQA7+/vAM7OzgDS0dEA5uTkAPb5+QAHAQEAFwcIAAsA/gDl+vsA+AEHABwRKQD2+e8A7vfoAPr89wALCgwAGRoZACYmJQArKysADg4OAA0NDQANDQ0ACwwLAAoJCgAHBwcAAwMDAAICAgABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAf////8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/v7+AP39/QD5+fkA+fn5APf39wD19fUA9/f3APb29gDi4uIA5OTkAOvr6wD29vYA7e3tAAAAAAAQEBAACAgIABYWFgAaGhoAGhoaAAgICAAICAgACgoKAAsLCwAJCgkACAcIAAYGBgAFBQUAAwMDAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJItZQAAAArMSURBVAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH/////AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/f39APz8/AD9/f0A+fn5APr6+gD6+voA+fn5APr6+gD6+voA+fn5APj5+AD9/P0AAgICAAYGBgAHBwcABgYGAAYGBgAICAgABgYGAAcHBwAGBgYABQUFAAMDAwACAgIAAgICAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAB/////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wAAAAAA/v7+AP7+/gD8/PwA/v7+APz8/AD8/PwA/P38AP38/QD7+/sA/Pz8AAMDAwAFBQUAAgICAAUFBQAEBAQAAwMDAAQEBAADAwMAAgICAAEBAQACAgIAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAf////8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/v7+AAAAAAD///8A////AP///wD9/f0A/v7+AAAAAAAAAAAAAQEBAAEBAQADAwMAAQEBAAEBAQABAQEAAgICAAAAAAABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACanuzMvN2MUAAAAAElFTkSuQmCC" style="display:none"/>
  <noscript>Javascript must be enabled to view this page.</noscript>
  <div style="display:none">
  <krona collapse="true" key="true">
   <attributes magnitude="magnitude">
    <list>members</list>
    <attribute display="Abundance">magnitude</attribute>
    <attribute display="Unassigned">magnitudeUnassigned</attribute>
    <attribute display="Avg. log e-value">score</attribute>
    <attribute display="Rank" mono="true">rank</attribute>
   </attributes>
   <datasets>
    <dataset>AMD_functional_UBA</dataset>
    <dataset>AMD_functional_5way</dataset>
   </datasets>
   <color attribute="score" hueStart="120" hueEnd="0" valueStart="-191" valueEnd="0" default="false" ></color>
<node name="all">
 <magnitude><val>11038</val><val>46416</val></magnitude>
 <node name="Protein Metabolism">
  <rank><val>level 1</val></rank>
  <score><val>-45.6645</val><val>-69.0709</val></score>
  <magnitude><val>883</val><val>3792</val></magnitude>
  <node name="Protein folding">
   <rank><val>level 2</val></rank>
   <score><val>-75.2218</val><val>-83.0986</val></score>
   <magnitude><val>47</val><val>317</val></magnitude>
   <node name="GroEL GroES">
    <rank><val>level 3</val></rank>
    <magnitude><val>12</val><val>87</val></magnitude>
    <score><val>-72.3333</val><val>-91.2759</val></score>
    <node name="Heat shock protein 60 family co-chaperone GroES">
     <magnitude><val>1</val></magnitude>
     <score><val>-21</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Heat shock protein 60 family chaperone GroEL">
     <score><val>-77</val><val>-91.2759</val></score>
     <magnitude><val>11</val><val>87</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Peptidyl-prolyl cis-trans isomerase">
    <magnitude><val>2</val></magnitude>
    <score><val>-78</val></score>
    <rank><val>level 3</val></rank>
    <node name="Peptidyl-prolyl cis-trans isomerase ppiD (EC 5.2.1.8)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-78</val></score>
    </node>
   </node>
   <node name="Thermosome, archaeal">
    <magnitude><val>9</val><val>87</val></magnitude>
    <score><val>-86.7143</val><val>-91.2759</val></score>
    <rank><val>level 3</val></rank>
    <node name="Thermosome subunit">
     <score><val>-86.7143</val><val>-91.2759</val></score>
     <magnitude><val>9</val><val>87</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Protein chaperones">
    <score><val>-72.1248</val><val>-73.1485</val></score>
    <magnitude><val>24</val><val>143</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="DnaJ-class molecular chaperone CbpA">
     <magnitude><val>3</val><val>5</val></magnitude>
     <score><val>-15.6667</val><val>-1</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Heat shock protein GrpE">
     <score><val>0</val><val>-4.5</val></score>
     <magnitude><val></val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="ClpB protein">
     <rank><val>function</val></rank>
     <score><val>-43.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Chaperone protein DnaK">
     <rank><val>function</val></rank>
     <score><val>-103.214</val><val>-86.6364</val></score>
     <magnitude><val>14</val><val>90</val></magnitude>
    </node>
    <node name="Chaperone protein DnaJ">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>46</val></magnitude>
     <score><val>-30.4</val><val>-57.5862</val></score>
    </node>
   </node>
  </node>
  <node name="Selenoproteins">
   <rank><val>level 2</val></rank>
   <magnitude><val>25</val><val>190</val></magnitude>
   <score><val>-40.76</val><val>-64.9994</val></score>
   <node name="Glycine reductase, sarcosine reductase and betaine reductase">
    <score><val>-39.3636</val><val>-64.9994</val></score>
    <magnitude><val>22</val><val>190</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Thioredoxin reductase (EC 1.8.1.9)">
     <score><val>-39.3636</val><val>-67.8525</val></score>
     <magnitude><val>11</val><val>91</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Thioredoxin">
     <rank><val>function</val></rank>
     <score><val>-39.3636</val><val>-62.3768</val></score>
     <magnitude><val>11</val><val>99</val></magnitude>
    </node>
   </node>
   <node name="Selenocysteine metabolism">
    <rank><val>level 3</val></rank>
    <score><val>-51</val></score>
    <magnitude><val>3</val></magnitude>
    <node name="Selenide,water dikinase (EC 2.7.9.3)">
     <rank><val>function</val></rank>
     <score><val>-34</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="selenocysteine-containing">
     <magnitude><val>2</val></magnitude>
     <score><val>-59.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Protein processing and modification">
   <rank><val>level 2</val></rank>
   <magnitude><val>122</val><val>450</val></magnitude>
   <score><val>-49.1232</val><val>-72.0001</val></score>
   <node name="G3E family of P-loop GTPases (metallocenter biosynthesis)">
    <rank><val>level 3</val></rank>
    <magnitude><val>32</val><val>92</val></magnitude>
    <score><val>-68.7813</val><val>-68.3186</val></score>
    <node name="putative periplasmic protein kinase ArgK and related GTPases of G3E family">
     <magnitude><val>7</val><val>14</val></magnitude>
     <score><val>-35.7143</val><val>-36.75</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Methylmalonyl-CoA mutase (EC 5.4.99.2)">
     <score><val>-102</val><val>-92.2258</val></score>
     <magnitude><val>16</val><val>46</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="[NiFe] hydrogenase metallocenter assembly protein HypE">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-87</val></score>
    </node>
    <node name="Cob(I)alamin adenosyltransferase PduO (EC 2.5.1.17)">
     <rank><val>function</val></rank>
     <score><val>-25</val><val>-43</val></score>
     <magnitude><val>4</val><val>21</val></magnitude>
    </node>
    <node name="Urease accessory protein UreG">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-9</val></score>
    </node>
    <node name="[NiFe] hydrogenase metallocenter assembly protein HypF">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-44.5</val></score>
    </node>
    <node name="B12 binding domain of Methylmalonyl-CoA mutase (EC 5.4.99.2)">
     <score><val>-34</val><val>-56.8571</val></score>
     <magnitude><val>1</val><val>11</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Peptide methionine sulfoxide reductase">
    <rank><val>level 3</val></rank>
    <score><val>-25.6111</val><val>-56.15</val></score>
    <magnitude><val>9</val><val>33</val></magnitude>
    <node name="Peptide methionine sulfoxide reductase MsrA (EC 1.8.4.11)">
     <score><val>-25.5</val><val>-56.15</val></score>
     <magnitude><val>7</val><val>33</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Peptide methionine sulfoxide reductase MsrB (EC 1.8.4.12)">
     <score><val>-26</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Signal peptidase">
    <rank><val>level 3</val></rank>
    <magnitude><val>6</val><val>8</val></magnitude>
    <score><val>-11.5</val><val>-95.6667</val></score>
    <node name="Signal peptidase I (EC 3.4.21.89)">
     <score><val>-11.5</val><val>-95.6667</val></score>
     <magnitude><val>6</val><val>8</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Ribosomal protein S12p Asp methylthiotransferase">
    <score><val>-25.2262</val><val>-40.2083</val></score>
    <magnitude><val>12</val><val>32</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Ribosomal protein S12p Asp88 (E. coli) methylthiotransferase">
     <score><val>-20</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S12p (S23e)">
     <score><val>-57</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA-i(6)A37 methylthiotransferase">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>32</val></magnitude>
     <score><val>-22.8571</val><val>-40.2083</val></score>
    </node>
   </node>
   <node name="N-linked Glycosylation in Bacteria">
    <rank><val>level 3</val></rank>
    <magnitude><val>18</val><val>60</val></magnitude>
    <score><val>-34.8235</val><val>-54.0256</val></score>
    <node name="UDP-glucose 4-epimerase (EC 5.1.3.2)">
     <score><val>-34.8235</val><val>-54.0256</val></score>
     <magnitude><val>18</val><val>60</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Lipoprotein Biosynthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-12</val></score>
    <node name="Prolipoprotein diacylglyceryl transferase (EC 2.4.99.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Protein Acetylation and Deacetylation in Bacteria">
    <score><val>-60.8437</val><val>-78.6799</val></score>
    <magnitude><val>32</val><val>156</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Acetate permease ActP (cation/acetate symporter)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-25</val></score>
    </node>
    <node name="Acetyl-coenzyme A synthetase (EC 6.2.1.1)">
     <magnitude><val>16</val><val>73</val></magnitude>
     <score><val>-37.5</val><val>-67.283</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Protein acetyltransferase">
     <score><val>-12</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Propionate--CoA ligase (EC 6.2.1.17)">
     <rank><val>function</val></rank>
     <magnitude><val>14</val><val>83</val></magnitude>
     <score><val>-93.5714</val><val>-88.7037</val></score>
    </node>
   </node>
   <node name="Inteins">
    <magnitude><val>12</val><val>69</val></magnitude>
    <score><val>-50.3333</val><val>-97.0172</val></score>
    <rank><val>level 3</val></rank>
    <node name="intein-containing">
     <magnitude><val>12</val><val>69</val></magnitude>
     <score><val>-50.3333</val><val>-97.0172</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Protein biosynthesis">
   <rank><val>level 2</val></rank>
   <score><val>-42.6445</val><val>-67.0031</val></score>
   <magnitude><val>616</val><val>2582</val></magnitude>
   <node name="tRNA aminoacylation, Pro">
    <score><val>-70.7143</val><val>-68.3182</val></score>
    <magnitude><val>7</val><val>34</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Prolyl-tRNA synthetase (EC 6.1.1.15)">
     <score><val>-70.7143</val><val>-68.3182</val></score>
     <magnitude><val>7</val><val>34</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Ribosome LSU bacterial">
    <score><val>-29</val></score>
    <magnitude><val>4</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="LSU ribosomal protein L5p (L11e)">
     <rank><val>function</val></rank>
     <score><val>-49</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="LSU ribosomal protein L7/L12 (P1/P2)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-25</val></score>
    </node>
    <node name="LSU ribosomal protein L25p">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-3</val></score>
    </node>
    <node name="LSU ribosomal protein L6p (L9e)">
     <score><val>-39</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA aminoacylation, Leu">
    <rank><val>level 3</val></rank>
    <score><val>-41.375</val><val>-87.6098</val></score>
    <magnitude><val>16</val><val>57</val></magnitude>
    <node name="Leucyl-tRNA synthetase (EC 6.1.1.4)">
     <score><val>-41.375</val><val>-87.6098</val></score>
     <magnitude><val>16</val><val>57</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA aminoacylation, Arg">
    <magnitude><val>12</val><val>26</val></magnitude>
    <score><val>-20.6364</val><val>-65.381</val></score>
    <rank><val>level 3</val></rank>
    <node name="Arginyl-tRNA synthetase (EC 6.1.1.19)">
     <magnitude><val>12</val><val>26</val></magnitude>
     <score><val>-20.6364</val><val>-65.381</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA aminoacylation, Lys">
    <rank><val>level 3</val></rank>
    <magnitude><val>8</val><val>43</val></magnitude>
    <score><val>-36.25</val><val>-85.9667</val></score>
    <node name="Lysyl-tRNA synthetase (class I) (EC 6.1.1.6)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>43</val></magnitude>
     <score><val>-33.4286</val><val>-85.9667</val></score>
    </node>
    <node name="Lysyl-tRNA synthetase (class II) (EC 6.1.1.6)">
     <magnitude><val>1</val></magnitude>
     <score><val>-56</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA aminoacylation, Tyr">
    <rank><val>level 3</val></rank>
    <score><val>-42</val><val>-61.9565</val></score>
    <magnitude><val>8</val><val>33</val></magnitude>
    <node name="Tyrosyl-tRNA synthetase (EC 6.1.1.1)">
     <score><val>-42</val><val>-61.9565</val></score>
     <magnitude><val>8</val><val>33</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA aminoacylation, Cys">
    <magnitude><val>4</val><val>22</val></magnitude>
    <score><val>-52.75</val><val>-99.9613</val></score>
    <rank><val>level 3</val></rank>
    <node name="Cysteinyl-tRNA synthetase paralog 2">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-41</val></score>
    </node>
    <node name="Cysteinyl-tRNA synthetase (EC 6.1.1.16)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>21</val></magnitude>
     <score><val>-52.75</val><val>-102.769</val></score>
    </node>
   </node>
   <node name="tRNA aminoacylation, Ser">
    <score><val>-63.3636</val><val>-81.4286</val></score>
    <magnitude><val>12</val><val>17</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Seryl-tRNA synthetase (EC 6.1.1.11)">
     <rank><val>function</val></rank>
     <score><val>-63.3636</val><val>-81.4286</val></score>
     <magnitude><val>12</val><val>17</val></magnitude>
    </node>
   </node>
   <node name="tRNA aminoacylation, Glu and Gln">
    <rank><val>level 3</val></rank>
    <magnitude><val>25</val><val>160</val></magnitude>
    <score><val>-42.32</val><val>-76.1863</val></score>
    <node name="Glutamyl-tRNA(Gln) amidotransferase transferase subunit (EC 6.3.5.7)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>45</val></magnitude>
     <score><val>0</val><val>-73.2424</val></score>
    </node>
    <node name="Glutamyl-tRNA(Gln) amidotransferase subunit C (EC 6.3.5.7)">
     <magnitude><val>1</val><val>2</val></magnitude>
     <score><val>-3</val><val>-3</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glutamyl-tRNA(Gln) amidotransferase asparaginase subunit (EC 6.3.5.7)">
     <score><val>-15.75</val><val>-70.8824</val></score>
     <magnitude><val>4</val><val>23</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glutamyl-tRNA(Gln) amidotransferase subunit A (EC 6.3.5.7)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-49</val></score>
    </node>
    <node name="Glutamyl-tRNA(Gln) synthetase (EC 6.1.1.24)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>45</val></magnitude>
     <score><val>-49</val><val>-80.64</val></score>
    </node>
    <node name="Glutamyl-tRNA(Gln) amidotransferase subunit B (EC 6.3.5.7)">
     <rank><val>function</val></rank>
     <score><val>-61</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Glutamyl-tRNA synthetase (EC 6.1.1.17)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>45</val></magnitude>
     <score><val>-49</val><val>-80.64</val></score>
    </node>
   </node>
   <node name="Ribosome SSU bacterial">
    <rank><val>level 3</val></rank>
    <score><val>-28.8</val><val>-69</val></score>
    <magnitude><val>5</val><val>1</val></magnitude>
    <node name="SSU ribosomal protein S12p (S23e)">
     <magnitude><val>1</val></magnitude>
     <score><val>-57</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S7p (S5e)">
     <magnitude><val>1</val></magnitude>
     <score><val>-17</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S8p (S15Ae)">
     <rank><val>function</val></rank>
     <score><val>-17.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="SSU ribosomal protein S3p (S3e)">
     <score><val>-35</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S1p">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-69</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA aminoacylation, Gly">
    <magnitude><val>13</val><val>48</val></magnitude>
    <score><val>-22.6154</val><val>-59.7586</val></score>
    <rank><val>level 3</val></rank>
    <node name="Glycyl-tRNA synthetase (EC 6.1.1.14)">
     <rank><val>function</val></rank>
     <magnitude><val>11</val><val>48</val></magnitude>
     <score><val>-24.9091</val><val>-59.7586</val></score>
    </node>
    <node name="Glycyl-tRNA synthetase alpha chain (EC 6.1.1.14)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-10</val></score>
    </node>
    <node name="Glycyl-tRNA synthetase beta chain (EC 6.1.1.14)">
     <magnitude><val>1</val></magnitude>
     <score><val>-10</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Translation initiation factors eukaryotic and archaeal">
    <rank><val>level 3</val></rank>
    <magnitude><val>48</val><val>235</val></magnitude>
    <score><val>-44.4414</val><val>-57.8004</val></score>
    <node name="Translation initiation factor 1A">
     <score><val>-38</val><val>-11.3333</val></score>
     <magnitude><val>1</val><val>28</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Eukaryotic translation initiation factor 2 gamma subunit">
     <rank><val>function</val></rank>
     <score><val>-60.4</val><val>-75</val></score>
     <magnitude><val>16</val><val>40</val></magnitude>
    </node>
    <node name="Eukaryotic translation initiation factor 6">
     <magnitude><val>5</val><val>23</val></magnitude>
     <score><val>-36</val><val>-64.3333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Translation initiation factor SUI1">
     <magnitude><val></val><val>5</val></magnitude>
     <score><val>0</val><val>-44.6667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Eukaryotic translation initiation factor 2 beta subunit">
     <magnitude><val>1</val><val>27</val></magnitude>
     <score><val>-19</val><val>-38.25</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Eukaryotic translation initiation factor 5A">
     <magnitude><val>4</val><val>15</val></magnitude>
     <score><val>-39</val><val>-46.8571</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Translation initiation factor SUI1-related protein">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>5</val></magnitude>
     <score><val>0</val><val>-44.6667</val></score>
    </node>
    <node name="Translation initiation factor 2">
     <magnitude><val>15</val><val>46</val></magnitude>
     <score><val>-35.7857</val><val>-92.871</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Eukaryotic translation initiation factor 2 alpha subunit">
     <magnitude><val>2</val><val>31</val></magnitude>
     <score><val>-40.5</val><val>-52.5455</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Eukaryotic translation initiation factor 5">
     <magnitude><val>4</val><val>15</val></magnitude>
     <score><val>-39</val><val>-46.8571</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA aminoacylation, Phe">
    <magnitude><val>16</val><val>72</val></magnitude>
    <score><val>-28.5909</val><val>-72.6951</val></score>
    <rank><val>level 3</val></rank>
    <node name="Phenylalanyl-tRNA synthetase beta chain (EC 6.1.1.20)">
     <score><val>-8</val><val>-73.25</val></score>
     <magnitude><val>4</val><val>25</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phenylalanyl-tRNA synthetase alpha chain (EC 6.1.1.20)">
     <rank><val>function</val></rank>
     <score><val>-35.4545</val><val>-72.4</val></score>
     <magnitude><val>12</val><val>47</val></magnitude>
    </node>
   </node>
   <node name="Translation elongation factors bacterial">
    <rank><val>level 3</val></rank>
    <score><val>-30</val></score>
    <magnitude><val>8</val></magnitude>
    <node name="Translation elongation factor G">
     <magnitude><val>5</val></magnitude>
     <score><val>-36.2</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Translation elongation factor G-related protein">
     <magnitude><val>3</val></magnitude>
     <score><val>-19.6667</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Ribosome SSU eukaryotic and archaeal">
    <magnitude><val>68</val><val>302</val></magnitude>
    <score><val>-35.5485</val><val>-37.806</val></score>
    <rank><val>level 3</val></rank>
    <node name="SSU ribosomal protein S11e (S17p)">
     <magnitude><val></val><val>4</val></magnitude>
     <score><val>0</val><val>-52</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S16e (S9p)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>7</val></magnitude>
     <score><val>-31.5</val><val>-36</val></score>
    </node>
    <node name="SSU ribosomal protein S8e">
     <score><val>-37</val><val>-23</val></score>
     <magnitude><val>3</val><val>13</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S20e (S10p)">
     <magnitude><val>1</val><val>12</val></magnitude>
     <score><val>-45</val><val>-31.8889</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S18e (S13p)">
     <magnitude><val>6</val><val>18</val></magnitude>
     <score><val>-41.4</val><val>-38.2308</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S19e">
     <rank><val>function</val></rank>
     <score><val>-24.8</val><val>-16.5455</val></score>
     <magnitude><val>5</val><val>39</val></magnitude>
    </node>
    <node name="SSU ribosomal protein S13e (S15p)">
     <magnitude><val>1</val><val>14</val></magnitude>
     <score><val>-30</val><val>-29.7273</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S3e (S3p)">
     <magnitude><val>3</val><val>2</val></magnitude>
     <score><val>-60.6667</val><val>-97</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S24e">
     <rank><val>function</val></rank>
     <score><val>-16</val><val>-15.9091</val></score>
     <magnitude><val>3</val><val>20</val></magnitude>
    </node>
    <node name="SSU ribosomal protein S5e (S7p)">
     <magnitude><val>2</val><val>21</val></magnitude>
     <score><val>-24</val><val>-63.5714</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S14e (S11p)">
     <score><val>-32.75</val><val>-42.9091</val></score>
     <magnitude><val>4</val><val>15</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S23e (S12p)">
     <score><val>-46.2</val><val>-52.6</val></score>
     <magnitude><val>8</val><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S2e (S5p)">
     <magnitude><val>2</val><val>14</val></magnitude>
     <score><val>-39</val><val>-59.8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S27e">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>10</val></magnitude>
     <score><val>0</val><val>-23.875</val></score>
    </node>
    <node name="SSU ribosomal protein S3Ae">
     <rank><val>function</val></rank>
     <score><val>-8.25</val><val>-31.3333</val></score>
     <magnitude><val>4</val><val>27</val></magnitude>
    </node>
    <node name="SSU ribosomal protein S6e">
     <score><val>-38</val><val>-16</val></score>
     <magnitude><val>5</val><val>8</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S15Ae (S8p)">
     <rank><val>function</val></rank>
     <score><val>-51.5</val><val>-39.8462</val></score>
     <magnitude><val>2</val><val>15</val></magnitude>
    </node>
    <node name="SSU ribosomal protein S15e (S19p)">
     <score><val>-29.6667</val><val>-71.75</val></score>
     <magnitude><val>3</val><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S28e">
     <rank><val>function</val></rank>
     <score><val>-25</val><val>-17</val></score>
     <magnitude><val>1</val><val>4</val></magnitude>
    </node>
    <node name="SSU ribosomal protein S17e">
     <score><val>0</val><val>-22</val></score>
     <magnitude><val></val><val>10</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S9e (S4p)">
     <rank><val>function</val></rank>
     <score><val>-26.75</val><val>-70.4286</val></score>
     <magnitude><val>4</val><val>15</val></magnitude>
    </node>
    <node name="SSU ribosomal protein S4e">
     <score><val>-25</val><val>-20</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein S27Ae">
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-4</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="SSU ribosomal protein SAe (S2p)">
     <score><val>-51.8</val><val>-66.3571</val></score>
     <magnitude><val>6</val><val>21</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Ribosome LSU eukaryotic and archaeal">
    <score><val>-26.8214</val><val>-45.9368</val></score>
    <magnitude><val>56</val><val>308</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="LSU ribosomal protein L27Ae (L15p)">
     <score><val>0</val><val>-43.4</val></score>
     <magnitude><val></val><val>6</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L31e">
     <score><val>-13</val><val>-19.4</val></score>
     <magnitude><val>2</val><val>7</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L3e (L3p)">
     <score><val>-34.2222</val><val>-62.381</val></score>
     <magnitude><val>9</val><val>33</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L5e (L18p)">
     <magnitude><val></val><val>9</val></magnitude>
     <score><val>0</val><val>-52.8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L23e (L14p)">
     <magnitude><val></val><val>6</val></magnitude>
     <score><val>0</val><val>-33.6</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L19e">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-20</val></score>
     <magnitude><val></val><val>4</val></magnitude>
    </node>
    <node name="LSU ribosomal protein L18e">
     <magnitude><val>2</val><val>9</val></magnitude>
     <score><val>-21.5</val><val>-46.7143</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L12a (P1/P2)">
     <score><val>-12</val><val>-28.6</val></score>
     <magnitude><val>3</val><val>24</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L37Ae">
     <magnitude><val></val><val>4</val></magnitude>
     <score><val>0</val><val>-4.75</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L12e (L11p)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>9</val></magnitude>
     <score><val>0</val><val>-38.5</val></score>
    </node>
    <node name="LSU ribosomal protein L10Ae (L1p)">
     <score><val>-11</val><val>-61.5</val></score>
     <magnitude><val>1</val><val>13</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L23Ae (L23p)">
     <magnitude><val>1</val><val>2</val></magnitude>
     <score><val>-21</val><val>-14.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L17e (L22p)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>8</val></magnitude>
     <score><val>-19</val><val>-68.6667</val></score>
    </node>
    <node name="LSU ribosomal protein L32e">
     <score><val>0</val><val>-26</val></score>
     <magnitude><val></val><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L7Ae">
     <score><val>-34.8333</val><val>-38.4</val></score>
     <magnitude><val>6</val><val>20</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L9e (L6p)">
     <score><val>-39</val><val>-61.5556</val></score>
     <magnitude><val>2</val><val>16</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L10e (L16p)">
     <score><val>-36.25</val><val>-46.3</val></score>
     <magnitude><val>8</val><val>15</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L44e">
     <magnitude><val>1</val><val>11</val></magnitude>
     <score><val>-6</val><val>-8.66667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L7e (L30p)">
     <magnitude><val>4</val><val>5</val></magnitude>
     <score><val>-29.5</val><val>-65</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L8e (L2p)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>14</val></magnitude>
     <score><val>-37</val><val>-82</val></score>
    </node>
    <node name="LSU ribosomal protein L13Ae (L13p)">
     <magnitude><val>3</val><val>8</val></magnitude>
     <score><val>-16</val><val>-29.5714</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L26e (L24p)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>11</val></magnitude>
     <score><val>0</val><val>-53.7143</val></score>
    </node>
    <node name="LSU ribosomal protein L1e (L4p)">
     <magnitude><val>2</val><val>19</val></magnitude>
     <score><val>-23</val><val>-55.5625</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L11e (L5p)">
     <score><val>-37</val><val>-47.5</val></score>
     <magnitude><val>1</val><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L24e">
     <score><val>0</val><val>-34</val></score>
     <magnitude><val></val><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein P0 (L10p)">
     <magnitude><val>1</val><val>21</val></magnitude>
     <score><val>-38</val><val>-56.8125</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L21e">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>5</val></magnitude>
     <score><val>-17</val><val>-21.6</val></score>
    </node>
    <node name="LSU ribosomal protein L15e">
     <magnitude><val>5</val><val>18</val></magnitude>
     <score><val>-16</val><val>-26.0769</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA aminoacylation, Asp and Asn">
    <score><val>-41.7681</val><val>-51.962</val></score>
    <magnitude><val>23</val><val>67</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Asparaginyl-tRNA synthetase (EC 6.1.1.22)">
     <rank><val>function</val></rank>
     <score><val>-37.6667</val><val>-43.2</val></score>
     <magnitude><val>13</val><val>20</val></magnitude>
    </node>
    <node name="Aspartyl-tRNA synthetase (EC 6.1.1.12)">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>45</val></magnitude>
     <score><val>-52</val><val>-58.0323</val></score>
    </node>
    <node name="Aspartyl-tRNA(Asn) amidotransferase subunit A (EC 6.3.5.6)">
     <score><val>-49</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Aspartyl-tRNA(Asn) synthetase (EC 6.1.1.23)">
     <score><val>-46</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Aspartyl-tRNA(Asn) amidotransferase subunit C (EC 6.3.5.6)">
     <score><val>-3</val><val>-3</val></score>
     <magnitude><val>1</val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Aspartyl-tRNA(Asn) amidotransferase subunit B (EC 6.3.5.6)">
     <rank><val>function</val></rank>
     <score><val>-61</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Translation elongation factor G family">
    <rank><val>level 3</val></rank>
    <score><val>-73.8561</val><val>-110.173</val></score>
    <magnitude><val>22</val><val>82</val></magnitude>
    <node name="Translation elongation factor 2">
     <rank><val>function</val></rank>
     <magnitude><val>14</val><val>82</val></magnitude>
     <score><val>-98.9167</val><val>-110.173</val></score>
    </node>
    <node name="Translation elongation factor G-related protein">
     <magnitude><val>3</val></magnitude>
     <score><val>-19.6667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Translation elongation factor G">
     <rank><val>function</val></rank>
     <score><val>-36.2</val></score>
     <magnitude><val>5</val></magnitude>
    </node>
   </node>
   <node name="tRNA aminoacylation, Trp">
    <rank><val>level 3</val></rank>
    <score><val>-37.2857</val><val>-88.5</val></score>
    <magnitude><val>7</val><val>47</val></magnitude>
    <node name="Tryptophanyl-tRNA synthetase (EC 6.1.1.2)">
     <score><val>-37.2857</val><val>-88.5</val></score>
     <magnitude><val>7</val><val>47</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA aminoacylation, Thr">
    <magnitude><val>11</val><val>55</val></magnitude>
    <score><val>-48.4</val><val>-70.7568</val></score>
    <rank><val>level 3</val></rank>
    <node name="Threonyl-tRNA synthetase (EC 6.1.1.3)">
     <rank><val>function</val></rank>
     <magnitude><val>11</val><val>55</val></magnitude>
     <score><val>-48.4</val><val>-70.7568</val></score>
    </node>
   </node>
   <node name="tRNA aminoacylation, Met">
    <magnitude><val>13</val><val>27</val></magnitude>
    <score><val>-27.2596</val><val>-75.9062</val></score>
    <rank><val>level 3</val></rank>
    <node name="Methionyl-tRNA synthetase (EC 6.1.1.10)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>11</val></magnitude>
     <score><val>-27.375</val><val>-138.8</val></score>
    </node>
    <node name="Structure-specific tRNA-binding protein trbp111">
     <rank><val>function</val></rank>
     <score><val>-27</val><val>-32.6667</val></score>
     <magnitude><val>2</val><val>8</val></magnitude>
    </node>
    <node name="Structure-specific tRNA-binding protein">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>8</val></magnitude>
     <score><val>-27</val><val>-32.6667</val></score>
    </node>
   </node>
   <node name="Universal GTPases">
    <rank><val>level 3</val></rank>
    <score><val>-57.661</val><val>-87.1044</val></score>
    <magnitude><val>82</val><val>295</val></magnitude>
    <node name="Translation initiation factor 2">
     <magnitude><val>15</val><val>46</val></magnitude>
     <score><val>-35.7857</val><val>-92.871</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="COG0536: GTP-binding protein Obg">
     <magnitude><val>2</val></magnitude>
     <score><val>-15.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP-binding protein TypA/BipA">
     <rank><val>function</val></rank>
     <score><val>-55</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="GTP-binding and nucleic acid-binding protein YchF">
     <rank><val>function</val></rank>
     <score><val>-17.75</val><val>-52.7143</val></score>
     <magnitude><val>8</val><val>37</val></magnitude>
    </node>
    <node name="Translation elongation factor G">
     <rank><val>function</val></rank>
     <magnitude><val>5</val></magnitude>
     <score><val>-36.2</val></score>
    </node>
    <node name="Signal recognition particle receptor protein FtsY (=alpha subunit) (TC 3.A.5.1.1)">
     <rank><val>function</val></rank>
     <magnitude><val>13</val><val>31</val></magnitude>
     <score><val>-51.25</val><val>-62.16</val></score>
    </node>
    <node name="Translation elongation factor 1 alpha subunit">
     <magnitude><val>10</val><val>51</val></magnitude>
     <score><val>-91.3333</val><val>-89.0606</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Translation elongation factor 2">
     <score><val>-98.9167</val><val>-110.173</val></score>
     <magnitude><val>14</val><val>82</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP-binding protein Era">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
    </node>
    <node name="50S ribosomal subunit maturation GTPase RbgA (B. subtilis YlqF)">
     <score><val>-11</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Signal recognition particle, subunit Ffh SRP54 (TC 3.A.5.1.1)">
     <score><val>-66.25</val><val>-82.7097</val></score>
     <magnitude><val>12</val><val>48</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Translation initiation factors bacterial">
    <magnitude><val>17</val><val>79</val></magnitude>
    <score><val>-34.2227</val><val>-60.9207</val></score>
    <rank><val>level 3</val></rank>
    <node name="Translation initiation factor 1">
     <score><val>-38</val><val>-11.3333</val></score>
     <magnitude><val>1</val><val>28</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Methionyl-tRNA formyltransferase (EC 2.1.2.9)">
     <score><val>-7</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Translation initiation factor 2">
     <score><val>-35.7857</val><val>-92.871</val></score>
     <magnitude><val>15</val><val>46</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Translation initiation factor SUI1-related protein">
     <magnitude><val></val><val>5</val></magnitude>
     <score><val>0</val><val>-44.6667</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA aminoacylation, Val">
    <magnitude><val>15</val><val>58</val></magnitude>
    <score><val>-31.5714</val><val>-63.35</val></score>
    <rank><val>level 3</val></rank>
    <node name="Valyl-tRNA synthetase (EC 6.1.1.9)">
     <rank><val>function</val></rank>
     <score><val>-31.5714</val><val>-63.35</val></score>
     <magnitude><val>15</val><val>58</val></magnitude>
    </node>
   </node>
   <node name="tRNA aminoacylation, Ala">
    <score><val>-37.4353</val><val>-70.1437</val></score>
    <magnitude><val>22</val><val>109</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Alanyl-tRNA synthetase (EC 6.1.1.7)">
     <rank><val>function</val></rank>
     <magnitude><val>12</val><val>59</val></magnitude>
     <score><val>-48.9091</val><val>-90.0233</val></score>
    </node>
    <node name="Alanyl-tRNA synthetase domain protein">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>50</val></magnitude>
     <score><val>-23.6667</val><val>-46.6857</val></score>
    </node>
   </node>
   <node name="Translation termination factors bacterial">
    <score><val>-49.3067</val><val>-61.7041</val></score>
    <magnitude><val>15</val><val>32</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Methionine aminopeptidase (EC 3.4.11.18)">
     <rank><val>function</val></rank>
     <score><val>-56.8</val><val>-63.5333</val></score>
     <magnitude><val>12</val><val>31</val></magnitude>
    </node>
    <node name="Peptide chain release factor 1">
     <score><val>-36</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Peptide chain release factor 2">
     <score><val>-11</val><val>-5</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Nucleolar protein complex">
    <rank><val>level 3</val></rank>
    <score><val>-21.5714</val><val>-42.9091</val></score>
    <magnitude><val>7</val><val>15</val></magnitude>
    <node name="RNA helicase, putative">
     <score><val>-9</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="NADH dehydrogenase subunit 5">
     <rank><val>function</val></rank>
     <score><val>-2</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="SSU ribosomal protein S14e (S11p)">
     <magnitude><val>4</val><val>15</val></magnitude>
     <score><val>-32.75</val><val>-42.9091</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Ribosome biogenesis bacterial">
    <score><val>-14.725</val><val>-60.5637</val></score>
    <magnitude><val>10</val><val>78</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Dimethyladenosine transferase (EC 2.1.1.-)">
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-2</val><val>-54</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribosomal-protein-S18p-alanine acetyltransferase (EC 2.3.1.-)">
     <magnitude><val></val><val>9</val></magnitude>
     <score><val>0</val><val>-52.375</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribosomal protein S6 glutaminyl transferase">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>22</val></magnitude>
     <score><val>0</val><val>-56.1875</val></score>
    </node>
    <node name="tRNA (Guanine37-N1) -methyltransferase (EC 2.1.1.31)">
     <rank><val>function</val></rank>
     <score><val>-19.25</val><val>-79.9412</val></score>
     <magnitude><val>5</val><val>26</val></magnitude>
    </node>
    <node name="Ribosomal protein L11 methyltransferase (EC 2.1.1.-)">
     <rank><val>function</val></rank>
     <score><val>-14</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Ribosomal large subunit pseudouridine synthase D (EC 4.2.1.70)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-5</val></score>
    </node>
    <node name="Ribosomal-protein-L7p-serine acetyltransferase">
     <score><val>-19</val><val>-40</val></score>
     <magnitude><val>1</val><val>14</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU m5C1962 methyltransferase RlmI">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-11</val></score>
    </node>
   </node>
   <node name="rRNAs">
    <rank><val>level 3</val></rank>
    <score><val>-16.5</val></score>
    <magnitude><val>2</val></magnitude>
    <node name="LSU rRNA">
     <rank><val>function</val></rank>
     <score><val>-13</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="SSU rRNA">
     <score><val>-20</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Programmed frameshift">
    <rank><val>level 3</val></rank>
    <score><val>-11</val><val>-5</val></score>
    <magnitude><val>4</val><val>2</val></magnitude>
    <node name="programmed frameshift-containing">
     <rank><val>function</val></rank>
     <score><val>-11</val><val>-5</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
    <node name="Peptide chain release factor 2">
     <rank><val>function</val></rank>
     <score><val>-11</val><val>-5</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
   </node>
   <node name="tRNA aminoacylation, Ile">
    <rank><val>level 3</val></rank>
    <magnitude><val>8</val><val>94</val></magnitude>
    <score><val>-56.75</val><val>-78.9091</val></score>
    <node name="Isoleucyl-tRNA synthetase (EC 6.1.1.5)">
     <score><val>-56.75</val><val>-78.9091</val></score>
     <magnitude><val>8</val><val>94</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA aminoacylation, His">
    <magnitude><val>6</val><val>6</val></magnitude>
    <score><val>-19.8333</val><val>-32.2</val></score>
    <rank><val>level 3</val></rank>
    <node name="Histidyl-tRNA synthetase (EC 6.1.1.21)">
     <score><val>-19.8333</val><val>-32.2</val></score>
     <magnitude><val>6</val><val>6</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Translation elongation factors eukaryotic and archaeal">
    <rank><val>level 3</val></rank>
    <score><val>-68.496</val><val>-88.4439</val></score>
    <magnitude><val>42</val><val>178</val></magnitude>
    <node name="Translation elongation factor 1 beta subunit">
     <magnitude><val>4</val><val>3</val></magnitude>
     <score><val>-9.33333</val><val>-25</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Translation elongation factor 1 alpha-related protein">
     <rank><val>function</val></rank>
     <score><val>-17.6667</val><val>-44.4</val></score>
     <magnitude><val>8</val><val>29</val></magnitude>
    </node>
    <node name="Translation elongation factor 2">
     <score><val>-98.9167</val><val>-110.173</val></score>
     <magnitude><val>14</val><val>82</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Translation elongation factor 1 alpha subunit">
     <score><val>-91.3333</val><val>-89.0606</val></score>
     <magnitude><val>10</val><val>51</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Eukaryotic peptide chain release factor subunit 1">
     <magnitude><val>6</val><val>13</val></magnitude>
     <score><val>-66.6667</val><val>-61.8571</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Protein degradation">
   <rank><val>level 2</val></rank>
   <score><val>-48.0177</val><val>-70.4455</val></score>
   <magnitude><val>73</val><val>253</val></magnitude>
   <node name="Metallocarboxypeptidases (EC 3.4.17.-)">
    <magnitude><val>7</val><val>31</val></magnitude>
    <score><val>-26.4286</val><val>-84.2258</val></score>
    <rank><val>level 3</val></rank>
    <node name="D-alanyl-D-alanine carboxypeptidase (EC 3.4.16.4)">
     <score><val>-1.5</val><val>-7</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Muramoyltetrapeptide carboxypeptidase (EC 3.4.17.13)">
     <score><val>-36.4</val><val>-86.8</val></score>
     <magnitude><val>5</val><val>30</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Proteolysis in bacteria, ATP-dependent">
    <rank><val>level 3</val></rank>
    <magnitude><val>10</val></magnitude>
    <score><val>-62.2</val></score>
    <node name="ATP-dependent Clp protease, ATP-binding subunit ClpC">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-88</val></score>
    </node>
    <node name="ATP-dependent Clp protease ATP-binding subunit ClpA">
     <score><val>-163</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="ATP-dependent protease La (EC 3.4.21.53)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-51</val></score>
    </node>
    <node name="ATP-dependent protease domain protein (EC 3.4.21.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-3</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="ATP-dependent Clp protease adaptor protein ClpS">
     <score><val>-4</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="ATP-dependent Clp protease ATP-binding subunit ClpX">
     <magnitude><val>3</val></magnitude>
     <score><val>-75.3333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="ClpB protein">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-43.5</val></score>
    </node>
   </node>
   <node name="Putative TldE-TldD proteolytic complex">
    <score><val>-32.5</val><val>-50.3385</val></score>
    <magnitude><val>10</val><val>13</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="TldD protein, part of proposed TldE/TldD proteolytic complex (PMID 12029038)">
     <magnitude><val>5</val><val>5</val></magnitude>
     <score><val>-46.2</val><val>-75.2</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="TldE/PmbA family protein, Actinobacterial subgroup">
     <score><val>-38</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="TldE/PmbA protein, part of proposed TldE/TldD proteolytic complex (PMID 12029038)">
     <rank><val>function</val></rank>
     <score><val>-14</val><val>-34.8</val></score>
     <magnitude><val>4</val><val>8</val></magnitude>
    </node>
   </node>
   <node name="Omega peptidases (EC 3.4.19.-)">
    <rank><val>level 3</val></rank>
    <magnitude><val>3</val><val>16</val></magnitude>
    <score><val>-36.6667</val><val>-27.6354</val></score>
    <node name="Pyrrolidone-carboxylate peptidase (EC 3.4.19.3)">
     <score><val>0</val><val>-26.3333</val></score>
     <magnitude><val></val><val>11</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Acylamino-acid-releasing enzyme (EC 3.4.19.1)">
     <rank><val>function</val></rank>
     <score><val>-36.6667</val><val>-30.5</val></score>
     <magnitude><val>3</val><val>5</val></magnitude>
    </node>
   </node>
   <node name="Proteasome bacterial">
    <score><val>-60.2541</val><val>-84.7862</val></score>
    <magnitude><val>30</val><val>126</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="LonB like ATP-ase no protease domain">
     <rank><val>function</val></rank>
     <score><val>-45.3529</val><val>-77.8108</val></score>
     <magnitude><val>17</val><val>51</val></magnitude>
    </node>
    <node name="proteasome-activating AAA-ATPase (PAN)">
     <score><val>-103</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="ATP-dependent protease LonB Type II">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>75</val></magnitude>
     <score><val>-78.625</val><val>-89.5294</val></score>
    </node>
    <node name="ATP-dependent Clp protease ATP-binding subunit ClpX">
     <magnitude><val>3</val></magnitude>
     <score><val>-75.3333</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Proteasome archaeal">
    <magnitude><val>10</val><val>67</val></magnitude>
    <score><val>-42.0666</val><val>-51.2252</val></score>
    <rank><val>level 3</val></rank>
    <node name="Proteasome subunit beta (EC 3.4.25.1), archaeal">
     <score><val>-54.3333</val><val>-49.7826</val></score>
     <magnitude><val>5</val><val>41</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Proteasome subunit alpha (EC 3.4.25.1), archaeal">
     <score><val>-8</val><val>-57.0833</val></score>
     <magnitude><val>2</val><val>24</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Bacterial proteasome-activating AAA-ATPase (PAN)">
     <magnitude><val>1</val></magnitude>
     <score><val>-103</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Proteasome-activating AAA-ATPase (PAN), archaeal">
     <score><val>-15</val><val>-10.5</val></score>
     <magnitude><val>2</val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Aminopeptidases (EC 3.4.11.-)">
    <score><val>-11.6667</val></score>
    <magnitude><val>3</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Cytosol aminopeptidase PepA (EC 3.4.11.1)">
     <score><val>-9.5</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Xaa-Pro aminopeptidase (EC 3.4.11.9)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-16</val></score>
    </node>
   </node>
  </node>
 </node>
 <node name="Miscellaneous">
  <rank><val>level 1</val></rank>
  <score><val>-34.0047</val><val>-64.1574</val></score>
  <magnitude><val>885</val><val>3418</val></magnitude>
  <node name="Broadly distributed proteins not in subsystems">
   <rank><val>level 2</val></rank>
   <magnitude><val>2</val></magnitude>
   <score><val>-13</val></score>
   <node name="FIG001943: hypothetical protein YajQ">
    <rank><val>function</val></rank>
    <score><val>-13</val></score>
    <magnitude><val>2</val></magnitude>
   </node>
  </node>
  <node name="Archease2">
   <score><val>0</val><val>-10.1429</val></score>
   <magnitude><val></val><val>10</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Archease">
    <score><val>0</val><val>-10.1429</val></score>
    <magnitude><val></val><val>10</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="YbbK">
   <rank><val>level 2</val></rank>
   <score><val>-45.4546</val><val>-71.9996</val></score>
   <magnitude><val>11</val><val>69</val></magnitude>
   <node name="Putative stomatin/prohibitin-family membrane protease subunit aq_911">
    <rank><val>function</val></rank>
    <magnitude><val>7</val><val>39</val></magnitude>
    <score><val>-57.4286</val><val>-67.9524</val></score>
   </node>
   <node name="Putative membrane-bound ClpP-class protease associated with aq_911">
    <score><val>-24.5</val><val>-77.2609</val></score>
    <magnitude><val>4</val><val>30</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Muconate lactonizing enzyme family">
   <rank><val>level 2</val></rank>
   <score><val>-33.0909</val><val>-79.4</val></score>
   <magnitude><val>13</val><val>28</val></magnitude>
   <node name="Muconate cycloisomerase (EC 5.5.1.1)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-56</val></score>
   </node>
   <node name="O-succinylbenzoate-CoA synthase (EC 4.2.1.-)">
    <rank><val>function</val></rank>
    <magnitude><val>12</val><val>28</val></magnitude>
    <score><val>-31.1818</val><val>-79.4</val></score>
   </node>
  </node>
  <node name="Plant-Prokaryote DOE project">
   <rank><val>level 2</val></rank>
   <score><val>-33.5717</val><val>-63.1997</val></score>
   <magnitude><val>811</val><val>3067</val></magnitude>
   <node name="PROSC">
    <rank><val>level 3</val></rank>
    <magnitude><val>98</val><val>344</val></magnitude>
    <score><val>-37.3424</val><val>-69.6879</val></score>
    <node name="Serine hydroxymethyltransferase (EC 2.1.2.1)">
     <rank><val>function</val></rank>
     <magnitude><val>11</val><val>67</val></magnitude>
     <score><val>-76</val><val>-79.3077</val></score>
    </node>
    <node name="Proline dehydrogenase (EC 1.5.99.8) (Proline oxidase)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>1</val></magnitude>
     <score><val>-15.5</val><val>-25</val></score>
    </node>
    <node name="Glucosamine-1-phosphate N-acetyltransferase (EC 2.3.1.157)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-8</val></score>
    </node>
    <node name="Ornithine cyclodeaminase (EC 4.3.1.12)">
     <rank><val>function</val></rank>
     <score><val>-8.85714</val><val>-64.6905</val></score>
     <magnitude><val>7</val><val>60</val></magnitude>
    </node>
    <node name="LL-diaminopimelate aminotransferase, predicted alternative">
     <magnitude><val>1</val></magnitude>
     <score><val>-22</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetylornithine aminotransferase (EC 2.6.1.11)">
     <magnitude><val>8</val><val>34</val></magnitude>
     <score><val>-29.8333</val><val>-75</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="N-acetylglucosamine-1-phosphate uridyltransferase (EC 2.7.7.23)">
     <score><val>-8</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Delta-1-pyrroline-5-carboxylate dehydrogenase (EC 1.5.1.12)">
     <magnitude><val>21</val><val>14</val></magnitude>
     <score><val>-34.8824</val><val>-40.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glucosamine--fructose-6-phosphate aminotransferase [isomerizing] (EC 2.6.1.16)">
     <magnitude><val>6</val><val>28</val></magnitude>
     <score><val>-35.8</val><val>-77.25</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Pyrroline-5-carboxylate reductase (EC 1.5.1.2)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>14</val></magnitude>
     <score><val>0</val><val>-60.5455</val></score>
    </node>
    <node name="Aspartate-semialdehyde dehydrogenase (EC 1.2.1.11)">
     <rank><val>function</val></rank>
     <score><val>-23.3333</val><val>-64.8462</val></score>
     <magnitude><val>3</val><val>14</val></magnitude>
    </node>
    <node name="Aspartate aminotransferase (EC 2.6.1.1)">
     <rank><val>function</val></rank>
     <magnitude><val>21</val><val>33</val></magnitude>
     <score><val>-48.4737</val><val>-49.2692</val></score>
    </node>
    <node name="Glutamate-1-semialdehyde aminotransferase (EC 5.4.3.8)">
     <score><val>-48.75</val><val>-91.2963</val></score>
     <magnitude><val>5</val><val>42</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Aspartokinase (EC 2.7.2.4)">
     <rank><val>function</val></rank>
     <score><val>-23</val><val>-58.0769</val></score>
     <magnitude><val>2</val><val>14</val></magnitude>
    </node>
    <node name="Homoserine dehydrogenase (EC 1.1.1.3)">
     <score><val>-14.8571</val><val>-62.7692</val></score>
     <magnitude><val>8</val><val>23</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Homoserine O-acetyltransferase (EC 2.3.1.31)">
     <magnitude><val>1</val></magnitude>
     <score><val>-10</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="At2g23840">
    <score><val>-49.4</val><val>-102.769</val></score>
    <magnitude><val>5</val><val>21</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Cysteinyl-tRNA synthetase (EC 6.1.1.16)">
     <rank><val>function</val></rank>
     <score><val>-52.75</val><val>-102.769</val></score>
     <magnitude><val>4</val><val>21</val></magnitude>
    </node>
    <node name="Peptide chain release factor 1">
     <magnitude><val>1</val></magnitude>
     <score><val>-36</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="At4g10620 At3g57180 At3g47450">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val><val>14</val></magnitude>
    <score><val>-4</val><val>-73.2857</val></score>
    <node name="Shikimate 5-dehydrogenase I alpha (EC 1.1.1.25)">
     <score><val>-4</val><val>-73.2857</val></score>
     <magnitude><val>1</val><val>14</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="COG3146 experimental">
    <score><val>-30.9583</val><val>-63.5157</val></score>
    <magnitude><val>24</val><val>102</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Maleylacetoacetate isomerase (EC 5.2.1.2)">
     <rank><val>function</val></rank>
     <score><val>-27</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 2">
     <magnitude><val>2</val><val>5</val></magnitude>
     <score><val>-8</val><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Fumarylacetoacetase (EC 3.7.1.2)">
     <score><val>-15</val><val>-104.667</val></score>
     <magnitude><val>2</val><val>7</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>16</val></magnitude>
     <score><val>-64</val><val>-47.7778</val></score>
    </node>
    <node name="Dihydroneopterin aldolase (EC 4.1.2.25)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-13</val></score>
    </node>
    <node name="2-amino-4-hydroxy-6-hydroxymethyldihydropteridine pyrophosphokinase (EC 2.7.6.3)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-3</val></score>
    </node>
    <node name="4-hydroxyphenylpyruvate dioxygenase (EC 1.13.11.27)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>20</val></magnitude>
     <score><val>-74</val><val>-85.2308</val></score>
    </node>
    <node name="Homogentisate 1,2-dioxygenase (EC 1.13.11.5)">
     <score><val>-49.25</val><val>-81.5556</val></score>
     <magnitude><val>4</val><val>25</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Queuosine biosynthesis QueD, PTPS-I">
     <magnitude><val>2</val><val>16</val></magnitude>
     <score><val>-9</val><val>-34.9</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="4-hydroxybenzoate polyprenyltransferase (EC 2.5.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>13</val></magnitude>
     <score><val>-33.4444</val><val>-49.5833</val></score>
    </node>
   </node>
   <node name="At5g37530">
    <score><val>-31.9792</val><val>-51.2374</val></score>
    <magnitude><val>36</val><val>107</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Aspartyl-tRNA synthetase (EC 6.1.1.12)">
     <rank><val>function</val></rank>
     <score><val>-52</val><val>-58.0323</val></score>
     <magnitude><val>6</val><val>45</val></magnitude>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7), SufS subfamily">
     <magnitude><val>9</val><val>18</val></magnitude>
     <score><val>-31</val><val>-37.6667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7)">
     <rank><val>function</val></rank>
     <magnitude><val>15</val><val>38</val></magnitude>
     <score><val>-29.4167</val><val>-52.625</val></score>
    </node>
    <node name="Histidyl-tRNA synthetase (EC 6.1.1.21)">
     <rank><val>function</val></rank>
     <score><val>-19.8333</val><val>-32.2</val></score>
     <magnitude><val>6</val><val>6</val></magnitude>
    </node>
   </node>
   <node name="At5g63420">
    <rank><val>level 3</val></rank>
    <score><val>-45.625</val><val>-79.7014</val></score>
    <magnitude><val>16</val><val>54</val></magnitude>
    <node name="Zn-dependent hydrolase, RNA-metabolising">
     <magnitude><val>3</val></magnitude>
     <score><val>-49.3333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribosomal-protein-S18p-alanine acetyltransferase (EC 2.3.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>9</val></magnitude>
     <score><val>0</val><val>-52.375</val></score>
    </node>
    <node name="Zn-dependent hydrolase, RNA-metabolising, CPSF 100 kDa analog">
     <score><val>-49.3333</val></score>
     <magnitude><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="YgjD/Kae1/Qri7 family, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>45</val></magnitude>
     <score><val>-43.4</val><val>-85.1667</val></score>
    </node>
   </node>
   <node name="COG0523">
    <rank><val>level 3</val></rank>
    <score><val>-29.6688</val><val>-78.9541</val></score>
    <magnitude><val>36</val><val>128</val></magnitude>
    <node name="Phosphoribosyl-ATP pyrophosphatase (EC 3.6.1.31)">
     <rank><val>function</val></rank>
     <score><val>-34</val><val>-57.1111</val></score>
     <magnitude><val>1</val><val>11</val></magnitude>
    </node>
    <node name="Phosphoribosyl-AMP cyclohydrolase (EC 3.5.4.19)">
     <score><val>-34</val><val>-57.1111</val></score>
     <magnitude><val>1</val><val>11</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Ubiquinone/menaquinone biosynthesis methyltransferase UbiE (EC 2.1.1.-)">
     <magnitude><val>8</val><val>3</val></magnitude>
     <score><val>-22.625</val><val>-55</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 2">
     <score><val>-8</val><val>-7</val></score>
     <magnitude><val>2</val><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydroorotase (EC 3.5.2.3)">
     <score><val>-37.5385</val><val>-72.9545</val></score>
     <magnitude><val>15</val><val>31</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="CobN component of cobalt chelatase involved in B12 biosynthesis">
     <rank><val>function</val></rank>
     <score><val>-50</val><val>-112.333</val></score>
     <magnitude><val>3</val><val>50</val></magnitude>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <rank><val>function</val></rank>
     <score><val>-64</val><val>-47.7778</val></score>
     <magnitude><val>1</val><val>16</val></magnitude>
    </node>
    <node name="D-alanyl-D-alanine carboxypeptidase (EC 3.4.16.4)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>1</val></magnitude>
     <score><val>-1.5</val><val>-7</val></score>
    </node>
    <node name="50S ribosomal subunit maturation GTPase RbgA (B. subtilis YlqF)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-11</val></score>
    </node>
    <node name="Zinc ABC transporter, ATP-binding protein ZnuC">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-6</val></score>
    </node>
    <node name="Creatinine amidohydrolase (EC 3.5.2.10)">
     <rank><val>function</val></rank>
     <score><val>-6</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Experimental-yggC">
    <score><val>-127</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Fructose-1,6-bisphosphatase, GlpX type (EC 3.1.3.11)">
     <magnitude><val>1</val></magnitude>
     <score><val>-127</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="At1g69340 At2g40600">
    <magnitude><val>2</val><val>22</val></magnitude>
    <score><val>-28</val><val>-36.7273</val></score>
    <rank><val>level 3</val></rank>
    <node name="Cardiolipin synthetase (EC 2.7.8.-)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-49</val></score>
     <magnitude><val></val><val>12</val></magnitude>
    </node>
    <node name="DNA/RNA-binding protein Alba">
     <magnitude><val>2</val><val>10</val></magnitude>
     <score><val>-28</val><val>-22</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="COG1836">
    <magnitude><val>37</val><val>143</val></magnitude>
    <score><val>-23.5179</val><val>-57.1678</val></score>
    <rank><val>level 3</val></rank>
    <node name="4-hydroxybenzoate polyprenyltransferase (EC 2.5.1.-)">
     <rank><val>function</val></rank>
     <score><val>-33.4444</val><val>-49.5833</val></score>
     <magnitude><val>9</val><val>13</val></magnitude>
    </node>
    <node name="Digeranylgeranylglycerophospholipid reductase">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>25</val></magnitude>
     <score><val>-32</val><val>-85.6667</val></score>
    </node>
    <node name="COG1836">
     <magnitude><val>7</val><val>18</val></magnitude>
     <score><val>-14.8333</val><val>-44.375</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Geranylgeranyl hydrogenase BchP">
     <rank><val>function</val></rank>
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Isopentenyl phosphate kinase">
     <score><val>-6.5</val><val>-49.5833</val></score>
     <magnitude><val>3</val><val>24</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="(S)-2,3-di-O-geranylgeranylglyceryl phosphate synthase">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>21</val></magnitude>
     <score><val>-22.4</val><val>-55.5714</val></score>
    </node>
    <node name="Undecaprenyl pyrophosphate synthetase (EC 2.5.1.31)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>28</val></magnitude>
     <score><val>-29.7143</val><val>-58.65</val></score>
    </node>
    <node name="phytol kinase">
     <magnitude><val>1</val><val>14</val></magnitude>
     <score><val>-9</val><val>-42.2</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="At3g50560">
    <score><val>-33.6667</val><val>-88.8649</val></score>
    <magnitude><val>6</val><val>63</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Gamma-glutamyltranspeptidase (EC 2.3.2.2)">
     <score><val>-33.6667</val><val>-88.8649</val></score>
     <magnitude><val>6</val><val>63</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="COG1565">
    <rank><val>level 3</val></rank>
    <score><val>-15</val><val>-36.6985</val></score>
    <magnitude><val>9</val><val>37</val></magnitude>
    <node name="Prolipoprotein diacylglyceryl transferase (EC 2.4.99.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Queuosine biosynthesis QueD, PTPS-I">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>16</val></magnitude>
     <score><val>-9</val><val>-34.9</val></score>
    </node>
    <node name="2-amino-4-hydroxy-6-hydroxymethyldihydropteridine pyrophosphokinase (EC 2.7.6.3)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-3</val></score>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <rank><val>function</val></rank>
     <score><val>-64</val><val>-47.7778</val></score>
     <magnitude><val>1</val><val>16</val></magnitude>
    </node>
    <node name="COG1565: Uncharacterized conserved protein">
     <score><val>-9</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydroneopterin aldolase (EC 4.1.2.25)">
     <score><val>-13</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 2">
     <magnitude><val>2</val><val>5</val></magnitude>
     <score><val>-8</val><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="COG2363">
    <rank><val>level 3</val></rank>
    <score><val>-32.6204</val><val>-49.7733</val></score>
    <magnitude><val>45</val><val>177</val></magnitude>
    <node name="Thiamine biosynthesis protein ThiI">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>49</val></magnitude>
     <score><val>-26.6667</val><val>-63.875</val></score>
    </node>
    <node name="Thiazole biosynthesis protein ThiG">
     <magnitude><val>3</val></magnitude>
     <score><val>-53</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Thiaminase II (EC 3.5.99.2)">
     <score><val>-21.2</val><val>-35.75</val></score>
     <magnitude><val>5</val><val>37</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphomethylpyrimidine kinase (EC 2.7.4.7)">
     <score><val>-6</val><val>-91.3333</val></score>
     <magnitude><val>1</val><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Thiamin-phosphate pyrophosphorylase (EC 2.5.1.3)">
     <magnitude><val>1</val><val>1</val></magnitude>
     <score><val>-12</val><val>-87</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="ThiJ/PfpI family protein">
     <rank><val>function</val></rank>
     <score><val>-10.5</val><val>-39.1667</val></score>
     <magnitude><val>2</val><val>29</val></magnitude>
    </node>
    <node name="Thiamin biosynthesis protein ThiC">
     <score><val>-128.5</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7), SufS subfamily">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>18</val></magnitude>
     <score><val>-31</val><val>-37.6667</val></score>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7)">
     <score><val>-29.4167</val><val>-52.625</val></score>
     <magnitude><val>15</val><val>38</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Iojap">
    <score><val>-27.9841</val><val>-41.1494</val></score>
    <magnitude><val>21</val><val>94</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Inorganic pyrophosphatase (EC 3.6.1.1)">
     <rank><val>function</val></rank>
     <score><val>-43.6667</val><val>-35.2308</val></score>
     <magnitude><val>7</val><val>17</val></magnitude>
    </node>
    <node name="COG0536: GTP-binding protein Obg">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-15.5</val></score>
    </node>
    <node name="N-acetylglucosamine-1-phosphate uridyltransferase (EC 2.7.7.23)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-8</val></score>
    </node>
    <node name="Phosphatidate cytidylyltransferase (EC 2.7.7.41)">
     <score><val>0</val><val>-61</val></score>
     <magnitude><val></val><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA delta(2)-isopentenylpyrophosphate transferase (EC 2.5.1.8)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-20</val></score>
    </node>
    <node name="Adenylate cyclase (EC 4.6.1.1)">
     <rank><val>function</val></rank>
     <score><val>-28</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Nicotinamidase (EC 3.5.1.19)">
     <magnitude><val>2</val><val>8</val></magnitude>
     <score><val>-13.5</val><val>-42.2857</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Nicotinamide-nucleotide adenylyltransferase, NadM family (EC 2.7.7.1)">
     <score><val>-30</val><val>-58.5333</val></score>
     <magnitude><val>2</val><val>25</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Orotate phosphoribosyltransferase (EC 2.4.2.10)">
     <magnitude><val>4</val><val>32</val></magnitude>
     <score><val>-20</val><val>-37.75</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="V-type ATP synthase subunit K (EC 3.6.3.14)">
     <score><val>-28</val><val>-8.5</val></score>
     <magnitude><val>1</val><val>9</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="COG0451">
    <rank><val>level 3</val></rank>
    <magnitude><val>26</val><val>128</val></magnitude>
    <score><val>-15.8846</val><val>-51.4633</val></score>
    <node name="Ribulose-5-phosphate 4-epimerase and related epimerases and aldolases">
     <score><val>0</val><val>-75.9091</val></score>
     <magnitude><val></val><val>22</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycosyltransferase">
     <magnitude><val>26</val><val>106</val></magnitude>
     <score><val>-15.8846</val><val>-46.3896</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Experimental - Histidine Degradation">
    <rank><val>level 3</val></rank>
    <magnitude><val>48</val><val>174</val></magnitude>
    <score><val>-52.1458</val><val>-72.8596</val></score>
    <node name="Glutamate formiminotransferase (EC 2.1.2.5)">
     <score><val>-46</val><val>-37</val></score>
     <magnitude><val>3</val><val>7</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Imidazolonepropionase (EC 3.5.2.7)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>3</val></magnitude>
     <score><val>-22</val><val>-20.3333</val></score>
    </node>
    <node name="Methenyltetrahydrofolate cyclohydrolase (EC 3.5.4.9)">
     <magnitude><val>5</val><val>17</val></magnitude>
     <score><val>-14.2</val><val>-53.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Serine hydroxymethyltransferase (EC 2.1.2.1)">
     <magnitude><val>11</val><val>67</val></magnitude>
     <score><val>-76</val><val>-79.3077</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Histidine ammonia-lyase (EC 4.3.1.3)">
     <score><val>-43.5556</val><val>-89.6071</val></score>
     <magnitude><val>9</val><val>41</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Urocanate hydratase (EC 4.2.1.49)">
     <magnitude><val>14</val><val>18</val></magnitude>
     <score><val>-69.8571</val><val>-75.0769</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Formiminotetrahydrofolate cyclodeaminase (EC 4.3.1.4)">
     <magnitude><val>4</val><val>21</val></magnitude>
     <score><val>-11</val><val>-52.8182</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="COG3380 COG2907">
    <rank><val>level 3</val></rank>
    <magnitude><val>7</val><val>3</val></magnitude>
    <score><val>-30.7857</val><val>-14.3333</val></score>
    <node name="Deoxyribodipyrimidine photolyase (EC 4.1.99.3)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>3</val></magnitude>
     <score><val>-41.5</val><val>-14.3333</val></score>
    </node>
    <node name="Phytoene synthase (EC 2.5.1.32)">
     <magnitude><val>2</val></magnitude>
     <score><val>-24</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="S-adenosyl-L-methionine dependent methyltransferase, similar to cyclopropane-fatty-acyl-phospholipid synthase">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-24</val></score>
    </node>
    <node name="COG2907: Amine oxidase, flavin-containing">
     <rank><val>function</val></rank>
     <score><val>-19</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Scaffold proteins for [4Fe-4S] cluster assembly (MRP family)">
    <magnitude><val>23</val><val>35</val></magnitude>
    <score><val>-27.4873</val><val>-89.5588</val></score>
    <rank><val>level 3</val></rank>
    <node name="Scaffold protein for [4Fe-4S] cluster assembly ApbC, MRP-like">
     <rank><val>function</val></rank>
     <score><val>-1</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="PaaD-like protein (DUF59) involved in Fe-S cluster assembly">
     <magnitude><val>11</val><val>7</val></magnitude>
     <score><val>-30.1667</val><val>-17.8333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Methionyl-tRNA synthetase (EC 6.1.1.10)">
     <score><val>-27.375</val><val>-138.8</val></score>
     <magnitude><val>9</val><val>11</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Iron-sulfur cluster-binding protein">
     <magnitude><val>2</val><val>17</val></magnitude>
     <score><val>-26.5</val><val>-87.2308</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Niacin-Choline transport and metabolism">
    <magnitude><val>5</val><val>8</val></magnitude>
    <score><val>-36.4</val><val>-42.2857</val></score>
    <rank><val>level 3</val></rank>
    <node name="Sarcosine dehydrogenase (EC 1.5.99.1)">
     <magnitude><val>2</val></magnitude>
     <score><val>-62</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="BAT1-like">
     <score><val>-31</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Nicotinamidase (EC 3.5.1.19)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>8</val></magnitude>
     <score><val>-13.5</val><val>-42.2857</val></score>
    </node>
   </node>
   <node name="DMT transporter">
    <rank><val>level 3</val></rank>
    <score><val>-19.9481</val><val>-66.6672</val></score>
    <magnitude><val>21</val><val>65</val></magnitude>
    <node name="Permease of the drug/metabolite transporter (DMT) superfamily">
     <rank><val>function</val></rank>
     <score><val>-21.9091</val><val>-50.1053</val></score>
     <magnitude><val>12</val><val>27</val></magnitude>
    </node>
    <node name="Dihydroneopterin aldolase (EC 4.1.2.25)">
     <magnitude><val>1</val></magnitude>
     <score><val>-13</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydropteroate synthase (EC 2.5.1.15)">
     <magnitude><val>7</val><val>38</val></magnitude>
     <score><val>-20</val><val>-78.4348</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2-amino-4-hydroxy-6-hydroxymethyldihydropteridine pyrophosphokinase (EC 2.7.6.3)">
     <rank><val>function</val></rank>
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="COG2509">
    <magnitude><val>9</val><val>28</val></magnitude>
    <score><val>-37.0889</val><val>-77.25</val></score>
    <rank><val>level 3</val></rank>
    <node name="Hypothetical protein YbbP, contains nucleotide-binding domain of DisA bacterial checkpoint controller">
     <rank><val>function</val></rank>
     <score><val>-39.6667</val></score>
     <magnitude><val>3</val></magnitude>
    </node>
    <node name="Glucosamine--fructose-6-phosphate aminotransferase [isomerizing] (EC 2.6.1.16)">
     <rank><val>function</val></rank>
     <score><val>-35.8</val><val>-77.25</val></score>
     <magnitude><val>6</val><val>28</val></magnitude>
    </node>
   </node>
   <node name="Competence or DNA damage-inducible protein CinA and related protein families">
    <rank><val>level 3</val></rank>
    <magnitude><val>17</val><val>121</val></magnitude>
    <score><val>-25.0252</val><val>-54.0927</val></score>
    <node name="CDP-diacylglycerol--glycerol-3-phosphate 3-phosphatidyltransferase (EC 2.7.8.5)">
     <score><val>-18.4286</val><val>-35.6429</val></score>
     <magnitude><val>8</val><val>19</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Competence/damage-inducible protein CinA family">
     <score><val>0</val><val>-57.6667</val></score>
     <magnitude><val></val><val>21</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Competence/damage-inducible protein CinA family, archaeal branch">
     <magnitude><val></val><val>21</val></magnitude>
     <score><val>0</val><val>-57.6667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Lipoate synthase">
     <magnitude><val>8</val><val>39</val></magnitude>
     <score><val>-32.625</val><val>-57.3077</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Competence/damage-inducible protein CinA">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>21</val></magnitude>
     <score><val>-17</val><val>-57.6667</val></score>
    </node>
   </node>
   <node name="Experimental-Ubiquinone BiosynthesisVDC">
    <rank><val>level 3</val></rank>
    <magnitude><val>50</val><val>117</val></magnitude>
    <score><val>-37.1491</val><val>-67.2983</val></score>
    <node name="Ubiquinone biosynthesis monooxygenase UbiB">
     <rank><val>function</val></rank>
     <magnitude><val>12</val><val>30</val></magnitude>
     <score><val>-39.8182</val><val>-80.0476</val></score>
    </node>
    <node name="4-hydroxybenzoate polyprenyltransferase (EC 2.5.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>13</val></magnitude>
     <score><val>-33.4444</val><val>-49.5833</val></score>
    </node>
    <node name="Ubiquinone/menaquinone biosynthesis methyltransferase UbiE (EC 2.1.1.-)">
     <score><val>-22.625</val><val>-55</val></score>
     <magnitude><val>8</val><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="3-polyprenyl-4-hydroxybenzoate carboxy-lyase UbiX (EC 4.1.1.-)">
     <magnitude><val>8</val><val>18</val></magnitude>
     <score><val>-32.875</val><val>-49.2143</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="3-polyprenyl-4-hydroxybenzoate carboxy-lyase (EC 4.1.1.-)">
     <rank><val>function</val></rank>
     <score><val>-48.8182</val><val>-71.2647</val></score>
     <magnitude><val>13</val><val>53</val></magnitude>
    </node>
   </node>
   <node name="Iron-sulfur cluster assembly">
    <rank><val>level 3</val></rank>
    <score><val>-39.284</val><val>-66.9384</val></score>
    <magnitude><val>70</val><val>198</val></magnitude>
    <node name="Iron-sulfur cluster assembly protein SufD">
     <rank><val>function</val></rank>
     <score><val>-36</val><val>-59.35</val></score>
     <magnitude><val>4</val><val>33</val></magnitude>
    </node>
    <node name="PaaD-like protein (DUF59) involved in Fe-S cluster assembly">
     <rank><val>function</val></rank>
     <score><val>-30.1667</val><val>-17.8333</val></score>
     <magnitude><val>11</val><val>7</val></magnitude>
    </node>
    <node name="Iron-sulfur cluster assembly ATPase protein SufC">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>23</val></magnitude>
     <score><val>-53.1111</val><val>-58.2941</val></score>
    </node>
    <node name="Putative iron-sulfur cluster assembly scaffold protein for SUF system, SufE2">
     <rank><val>function</val></rank>
     <score><val>-9</val><val>-22</val></score>
     <magnitude><val>3</val><val>1</val></magnitude>
    </node>
    <node name="Iron-sulfur cluster regulator SufR">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>18</val></magnitude>
     <score><val>-16.25</val><val>-49.6429</val></score>
    </node>
    <node name="Scaffold protein for [4Fe-4S] cluster assembly ApbC, MRP-like">
     <rank><val>function</val></rank>
     <score><val>-1</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Iron-sulfur cluster assembly protein SufB">
     <rank><val>function</val></rank>
     <magnitude><val>12</val><val>60</val></magnitude>
     <score><val>-79.5455</val><val>-103.939</val></score>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7)">
     <rank><val>function</val></rank>
     <score><val>-29.4167</val><val>-52.625</val></score>
     <magnitude><val>15</val><val>38</val></magnitude>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7), SufS subfamily">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>18</val></magnitude>
     <score><val>-31</val><val>-37.6667</val></score>
    </node>
    <node name="Ferredoxin, 2Fe-2S">
     <rank><val>function</val></rank>
     <score><val>-12</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="At2g33980 At1g28960">
    <magnitude><val>1</val><val>2</val></magnitude>
    <score><val>-5</val><val>-16.5</val></score>
    <rank><val>level 3</val></rank>
    <node name="tRNA nucleotidyltransferase (EC 2.7.7.21) (EC 2.7.7.25)">
     <magnitude><val>1</val><val>2</val></magnitude>
     <score><val>-5</val><val>-16.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="At5g04520 AT1G06240">
    <rank><val>level 3</val></rank>
    <magnitude><val>19</val><val>96</val></magnitude>
    <score><val>-37.2368</val><val>-55.8808</val></score>
    <node name="Amidophosphoribosyltransferase (EC 2.4.2.14)">
     <score><val>-45.875</val><val>-77.3571</val></score>
     <magnitude><val>8</val><val>42</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Bacterioferritin">
     <rank><val>function</val></rank>
     <score><val>-31</val><val>-39.5625</val></score>
     <magnitude><val>1</val><val>23</val></magnitude>
    </node>
    <node name="Thiol peroxidase, Bcp-type (EC 1.11.1.15)">
     <rank><val>function</val></rank>
     <score><val>-33.25</val><val>-40.6471</val></score>
     <magnitude><val>6</val><val>28</val></magnitude>
    </node>
    <node name="Peptide methionine sulfoxide reductase MsrB (EC 1.8.4.12)">
     <score><val>-26</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Large exoproteins involved in heme utilization or adhesion">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>3</val></magnitude>
     <score><val>-29</val><val>-22.5</val></score>
    </node>
   </node>
   <node name="COG4319">
    <magnitude><val>10</val><val>54</val></magnitude>
    <score><val>-22</val><val>-69.3512</val></score>
    <rank><val>level 3</val></rank>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <rank><val>function</val></rank>
     <score><val>-64</val><val>-47.7778</val></score>
     <magnitude><val>1</val><val>16</val></magnitude>
    </node>
    <node name="Dihydroneopterin aldolase (EC 4.1.2.25)">
     <magnitude><val>1</val></magnitude>
     <score><val>-13</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydropteroate synthase (EC 2.5.1.15)">
     <magnitude><val>7</val><val>38</val></magnitude>
     <score><val>-20</val><val>-78.4348</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2-amino-4-hydroxy-6-hydroxymethyldihydropteridine pyrophosphokinase (EC 2.7.6.3)">
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="COG2302">
    <magnitude><val>3</val><val>17</val></magnitude>
    <score><val>-12</val><val>-17.3333</val></score>
    <rank><val>level 3</val></rank>
    <node name="Cell division protein FtsZ (EC 3.4.24.-)">
     <score><val>-12</val><val>-17.3333</val></score>
     <magnitude><val>3</val><val>17</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="At1g26220 At1g32070">
    <score><val>-14</val><val>-7</val></score>
    <magnitude><val>3</val><val>3</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Tyrosine recombinase XerC">
     <magnitude><val>1</val><val>3</val></magnitude>
     <score><val>-1</val><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Site-specific recombinase XerD">
     <rank><val>function</val></rank>
     <score><val>-20.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
   </node>
   <node name="At3g21300">
    <score><val>-27.4444</val><val>-3</val></score>
    <magnitude><val>9</val><val>4</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Glutamyl-tRNA(Gln) amidotransferase subunit A (EC 6.3.5.7)">
     <magnitude><val>1</val></magnitude>
     <score><val>-49</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Aspartyl-tRNA(Asn) amidotransferase subunit A (EC 6.3.5.6)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-49</val></score>
    </node>
    <node name="Aspartyl-tRNA(Asn) amidotransferase subunit C (EC 6.3.5.6)">
     <rank><val>function</val></rank>
     <score><val>-3</val><val>-3</val></score>
     <magnitude><val>1</val><val>2</val></magnitude>
    </node>
    <node name="Aspartyl-tRNA(Asn) amidotransferase subunit B (EC 6.3.5.6)">
     <rank><val>function</val></rank>
     <score><val>-61</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Glutamyl-tRNA(Gln) amidotransferase subunit B (EC 6.3.5.7)">
     <rank><val>function</val></rank>
     <score><val>-61</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Glutamyl-tRNA(Gln) amidotransferase subunit C (EC 6.3.5.7)">
     <magnitude><val>1</val><val>2</val></magnitude>
     <score><val>-3</val><val>-3</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Transcription regulator [contains diacylglycerol kinase catalytic domain]">
     <score><val>-7</val></score>
     <magnitude><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="YrdC-YciO-Sua5 protein family">
    <score><val>-51.1842</val><val>-76.5474</val></score>
    <magnitude><val>38</val><val>198</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="COG0009 Sua5 subfamily, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>29</val></magnitude>
     <score><val>-28.5</val><val>-56.4615</val></score>
    </node>
    <node name="YgjD/Kae1/Qri7 family, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
     <magnitude><val>10</val><val>45</val></magnitude>
     <score><val>-43.4</val><val>-85.1667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Serine hydroxymethyltransferase (EC 2.1.2.1)">
     <score><val>-76</val><val>-79.3077</val></score>
     <magnitude><val>11</val><val>67</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="p53-regulating protein kinase (human PRPK/yeast YGR262c)">
     <magnitude><val>8</val><val>45</val></magnitude>
     <score><val>-50.625</val><val>-85.1667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Peptide chain release factor 1">
     <rank><val>function</val></rank>
     <score><val>-36</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Rossmann fold nucleotide-binding protein Smf possibly involved in DNA uptake">
     <magnitude><val>3</val><val>3</val></magnitude>
     <score><val>-30.5</val><val>-23</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribosomal-protein-S18p-alanine acetyltransferase (EC 2.3.1.-)">
     <score><val>0</val><val>-52.375</val></score>
     <magnitude><val></val><val>9</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Conserved gene cluster possibly involved in RNA metabolism">
    <rank><val>level 3</val></rank>
    <score><val>-52.75</val><val>-102.769</val></score>
    <magnitude><val>4</val><val>21</val></magnitude>
    <node name="Cysteinyl-tRNA synthetase (EC 6.1.1.16)">
     <rank><val>function</val></rank>
     <score><val>-52.75</val><val>-102.769</val></score>
     <magnitude><val>4</val><val>21</val></magnitude>
    </node>
   </node>
   <node name="Omega-amidase">
    <rank><val>level 3</val></rank>
    <score><val>-52.1273</val><val>-60.65</val></score>
    <magnitude><val>22</val><val>99</val></magnitude>
    <node name="Methylthioribose-1-phosphate isomerase (EC 5.3.1.23)">
     <rank><val>function</val></rank>
     <score><val>-71.3333</val><val>-57.7941</val></score>
     <magnitude><val>12</val><val>59</val></magnitude>
    </node>
    <node name="2,3-diketo-5-methylthiopentyl-1-phosphate enolase-phosphatase (EC 3.1.3.77)">
     <magnitude><val>2</val><val>2</val></magnitude>
     <score><val>-17.5</val><val>-28.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Methylthioribulose-1-phosphate dehydratase (EC 4.2.1.109)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-3</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="1,2-dihydroxy-3-keto-5-methylthiopentene dioxygenase (EC 1.13.11.54)">
     <magnitude><val></val><val>4</val></magnitude>
     <score><val>0</val><val>-18.75</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="5&#39;-methylthioadenosine phosphorylase (EC 2.4.2.28)">
     <rank><val>function</val></rank>
     <score><val>-36.8</val><val>-77.5</val></score>
     <magnitude><val>6</val><val>31</val></magnitude>
    </node>
    <node name="2,3-diketo-5-methylthiopentyl-1-phosphate enolase">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>2</val></magnitude>
     <score><val>-17.5</val><val>-28.5</val></score>
    </node>
   </node>
   <node name="At4g38090">
    <rank><val>level 3</val></rank>
    <magnitude><val>2</val><val>18</val></magnitude>
    <score><val>-6</val><val>-48.8</val></score>
    <node name="Ferrochelatase, protoheme ferro-lyase (EC 4.99.1.1)">
     <score><val>-6</val><val>-48.8</val></score>
     <magnitude><val>2</val><val>18</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="COG0398">
    <rank><val>level 3</val></rank>
    <score><val>-20.2727</val><val>-70.9167</val></score>
    <magnitude><val>12</val><val>31</val></magnitude>
    <node name="Mercuric ion reductase (EC 1.16.1.1)">
     <rank><val>function</val></rank>
     <score><val>-20.2727</val><val>-70.9167</val></score>
     <magnitude><val>12</val><val>31</val></magnitude>
    </node>
   </node>
   <node name="Experimental-PTPS">
    <score><val>-20.5909</val><val>-48.8161</val></score>
    <magnitude><val>22</val><val>110</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <magnitude><val>1</val><val>16</val></magnitude>
     <score><val>-64</val><val>-47.7778</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydroneopterin aldolase (EC 4.1.2.25)">
     <score><val>-13</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Queuosine Biosynthesis QueE Radical SAM">
     <score><val>-14</val><val>-8</val></score>
     <magnitude><val>1</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 2">
     <rank><val>function</val></rank>
     <score><val>-8</val><val>-7</val></score>
     <magnitude><val>2</val><val>5</val></magnitude>
    </node>
    <node name="Cell division inhibitor">
     <score><val>-27</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Queuosine biosynthesis QueD, PTPS-I">
     <rank><val>function</val></rank>
     <score><val>-9</val><val>-34.9</val></score>
     <magnitude><val>2</val><val>16</val></magnitude>
    </node>
    <node name="2-amino-4-hydroxy-6-hydroxymethyldihydropteridine pyrophosphokinase (EC 2.7.6.3)">
     <rank><val>function</val></rank>
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Dihydropteroate synthase (EC 2.5.1.15)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>38</val></magnitude>
     <score><val>-20</val><val>-78.4348</val></score>
    </node>
    <node name="COG1565: Uncharacterized conserved protein">
     <magnitude><val>1</val></magnitude>
     <score><val>-9</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Queuosine Biosynthesis QueC ATPase">
     <rank><val>function</val></rank>
     <score><val>-29.8</val><val>-30.1</val></score>
     <magnitude><val>5</val><val>34</val></magnitude>
    </node>
   </node>
   <node name="COG0277">
    <score><val>-19.1852</val><val>-70.7368</val></score>
    <magnitude><val>9</val><val>53</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="D-3-phosphoglycerate dehydrogenase (EC 1.1.1.95)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>35</val></magnitude>
     <score><val>-25.2</val><val>-72.6071</val></score>
    </node>
    <node name="Fumarylacetoacetate hydrolase family protein">
     <magnitude><val>4</val><val>18</val></magnitude>
     <score><val>-11.6667</val><val>-67.1</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Biotin biosynthesis Experimental">
    <magnitude><val>4</val></magnitude>
    <score><val>-35</val></score>
    <rank><val>level 3</val></rank>
    <node name="Biotin synthase (EC 2.8.1.6)">
     <score><val>-37.5</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Adenosylmethionine-8-amino-7-oxononanoate aminotransferase (EC 2.6.1.62)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-32.5</val></score>
    </node>
   </node>
   <node name="COG2078">
    <rank><val>level 3</val></rank>
    <magnitude><val>11</val><val>26</val></magnitude>
    <score><val>-15.1454</val><val>-65.0571</val></score>
    <node name="COG1180: Radical SAM, Pyruvate-formate lyase-activating enzyme like">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>8</val></magnitude>
     <score><val>-28.5</val><val>-37.4</val></score>
    </node>
    <node name="COG1355, Predicted dioxygenase">
     <magnitude><val>6</val><val>13</val></magnitude>
     <score><val>-11.6</val><val>-96.7143</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="COG2078: Uncharacterized ACR">
     <rank><val>function</val></rank>
     <score><val>-13.3333</val><val>-27</val></score>
     <magnitude><val>3</val><val>5</val></magnitude>
    </node>
   </node>
   <node name="DOE COG2016">
    <rank><val>level 3</val></rank>
    <score><val>-27.9565</val><val>-52.609</val></score>
    <magnitude><val>23</val><val>134</val></magnitude>
    <node name="Putative snRNP Sm-like protein">
     <magnitude><val>7</val><val>10</val></magnitude>
     <score><val>-19.7143</val><val>-23.1429</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Small nuclear ribonucleoprotein F">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>5</val></magnitude>
     <score><val>0</val><val>-12</val></score>
    </node>
    <node name="Thioredoxin">
     <rank><val>function</val></rank>
     <magnitude><val>11</val><val>99</val></magnitude>
     <score><val>-39.3636</val><val>-62.3768</val></score>
    </node>
    <node name="Translation initiation factor SUI1-related protein">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>5</val></magnitude>
     <score><val>0</val><val>-44.6667</val></score>
    </node>
    <node name="conserved protein with predicted RNA binding PUA domain">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>14</val></magnitude>
     <score><val>-6</val><val>-25.1818</val></score>
    </node>
    <node name="Putative snRNP Sm-like protein, Archaeal">
     <magnitude><val>3</val><val>1</val></magnitude>
     <score><val>-20</val><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="At1g54520">
    <magnitude><val>1</val><val>1</val></magnitude>
    <score><val>-12</val><val>-87</val></score>
    <rank><val>level 3</val></rank>
    <node name="Thiamin-phosphate pyrophosphorylase (EC 2.5.1.3)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>1</val></magnitude>
     <score><val>-12</val><val>-87</val></score>
    </node>
   </node>
   <node name="YggW">
    <rank><val>level 3</val></rank>
    <magnitude><val></val><val>15</val></magnitude>
    <score><val>0</val><val>-57.5091</val></score>
    <node name="A/G-specific adenine glycosylase (EC 3.2.2.-)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-15</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Pyrroline-5-carboxylate reductase (EC 1.5.1.2)">
     <magnitude><val></val><val>14</val></magnitude>
     <score><val>0</val><val>-60.5455</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="COG3533">
    <rank><val>level 3</val></rank>
    <magnitude><val>5</val><val>2</val></magnitude>
    <score><val>-8.8</val><val>-31.5</val></score>
    <node name="Alpha-galactosidase (EC 3.2.1.22)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-8</val></score>
    </node>
    <node name="sugar ABC transporter, permease protein">
     <rank><val>function</val></rank>
     <score><val>-8</val><val>-55</val></score>
     <magnitude><val>1</val><val>1</val></magnitude>
    </node>
    <node name="ABC transport system, sugar-binding protein">
     <magnitude><val>2</val></magnitude>
     <score><val>-9</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="N-Acetyl-D-glucosamine ABC transport system, sugar-binding protein">
     <score><val>-9</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Archease">
   <magnitude><val></val><val>10</val></magnitude>
   <score><val>0</val><val>-10.1429</val></score>
   <rank><val>level 2</val></rank>
   <node name="Archease">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>10</val></magnitude>
    <score><val>0</val><val>-10.1429</val></score>
   </node>
  </node>
  <node name="ZZ gjo need homes">
   <magnitude><val>48</val><val>234</val></magnitude>
   <score><val>-39.8195</val><val>-77.1897</val></score>
   <rank><val>level 2</val></rank>
   <node name="Protein secretion chaperonin CsaA">
    <magnitude><val>1</val></magnitude>
    <score><val>-14</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Quinone oxidoreductase (EC 1.6.5.5)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-7</val></score>
   </node>
   <node name="Uncharacterized ATP-dependent helicase MJ0294">
    <magnitude><val>14</val><val>98</val></magnitude>
    <score><val>-47.9286</val><val>-93.3377</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Peptidyl-tRNA hydrolase, archaeal type (EC 3.1.1.29)">
    <rank><val>function</val></rank>
    <score><val>-16</val><val>-27.25</val></score>
    <magnitude><val>4</val><val>13</val></magnitude>
   </node>
   <node name="Na+/H+ antiporter">
    <magnitude><val>3</val><val>17</val></magnitude>
    <score><val>-108</val><val>-60.6429</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Proposed peptidoglycan lipid II flippase MurJ">
    <magnitude><val>1</val></magnitude>
    <score><val>-29</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Pyrrolidone-carboxylate peptidase (EC 3.4.19.3)">
    <rank><val>function</val></rank>
    <score><val>0</val><val>-26.3333</val></score>
    <magnitude><val></val><val>11</val></magnitude>
   </node>
   <node name="GTP-binding protein, gtp1/obg family">
    <score><val>-36.6667</val><val>-72</val></score>
    <magnitude><val>5</val><val>5</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="GTP-binding protein RBG1">
    <rank><val>function</val></rank>
    <score><val>-39</val><val>-78.4211</val></score>
    <magnitude><val>9</val><val>27</val></magnitude>
   </node>
   <node name="RNase L inhibitor">
    <rank><val>function</val></rank>
    <score><val>-30.625</val><val>-80.25</val></score>
    <magnitude><val>8</val><val>51</val></magnitude>
   </node>
   <node name="AMP/CMP kinase AK6">
    <rank><val>function</val></rank>
    <score><val>-11.5</val><val>-38.8333</val></score>
    <magnitude><val>2</val><val>10</val></magnitude>
   </node>
   <node name="Putative heat shock protein YegD">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>2</val></magnitude>
    <score><val>0</val><val>-141</val></score>
   </node>
  </node>
 </node>
 <node name="Photosynthesis">
  <rank><val>level 1</val></rank>
  <score><val>-19.2857</val><val>-24.5429</val></score>
  <magnitude><val>7</val><val>7</val></magnitude>
  <node name="Proteorhodopsin">
   <score><val>-19.2857</val><val>-24.5429</val></score>
   <magnitude><val>7</val><val>7</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Phytoene dehydrogenase (EC 1.14.99.-)">
    <magnitude><val>1</val><val>1</val></magnitude>
    <score><val>-2</val><val>-23</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Beta-carotene ketolase (EC 1.14.-.-)">
    <score><val>-23</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Phytoene synthase (EC 2.5.1.32)">
    <rank><val>function</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-24</val></score>
   </node>
   <node name="Octaprenyl-diphosphate synthase (EC 2.5.1.-)">
    <magnitude><val>3</val><val>6</val></magnitude>
    <score><val>-20.6667</val><val>-24.8</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
 </node>
 <node name="Cofactors, Vitamins, Prosthetic Groups, Pigments">
  <magnitude><val>946</val><val>4150</val></magnitude>
  <score><val>-32.6497</val><val>-64.3013</val></score>
  <rank><val>level 1</val></rank>
  <node name="Thiamin biosynthesis">
   <rank><val>level 2</val></rank>
   <score><val>-33.5238</val><val>-47.2859</val></score>
   <magnitude><val>21</val><val>77</val></magnitude>
   <node name="Thiamin ABC transporter, ATPase component">
    <rank><val>function</val></rank>
    <score><val>-34</val><val>-33</val></score>
    <magnitude><val>2</val><val>3</val></magnitude>
   </node>
   <node name="Thiamin-phosphate synthase ThiN (EC 2.5.1.3)">
    <magnitude><val>1</val></magnitude>
    <score><val>-6</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Predicted thiamin transporter ThiT">
    <rank><val>function</val></rank>
    <magnitude><val>3</val><val>7</val></magnitude>
    <score><val>-19</val><val>-26.8</val></score>
   </node>
   <node name="Thiaminase II (EC 3.5.99.2)">
    <rank><val>function</val></rank>
    <magnitude><val>5</val><val>37</val></magnitude>
    <score><val>-21.2</val><val>-35.75</val></score>
   </node>
   <node name="Thiazole biosynthesis protein ThiG">
    <rank><val>function</val></rank>
    <magnitude><val>3</val></magnitude>
    <score><val>-53</val></score>
   </node>
   <node name="Thiamine-monophosphate kinase (EC 2.7.4.16)">
    <rank><val>function</val></rank>
    <score><val>-11</val><val>-62</val></score>
    <magnitude><val>3</val><val>24</val></magnitude>
   </node>
   <node name="Phosphomethylpyrimidine kinase (EC 2.7.4.7)">
    <score><val>-6</val><val>-91.3333</val></score>
    <magnitude><val>1</val><val>5</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Thiamin biosynthesis protein ThiC">
    <rank><val>function</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-128.5</val></score>
   </node>
   <node name="Thiamin-phosphate pyrophosphorylase (EC 2.5.1.3)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val><val>1</val></magnitude>
    <score><val>-12</val><val>-87</val></score>
   </node>
  </node>
  <node name="Tetrapyrroles">
   <magnitude><val>117</val><val>876</val></magnitude>
   <score><val>-23.9905</val><val>-67.6294</val></score>
   <rank><val>level 2</val></rank>
   <node name="Coenzyme B12 biosynthesis">
    <magnitude><val>43</val><val>389</val></magnitude>
    <score><val>-25.0965</val><val>-70.7632</val></score>
    <rank><val>level 3</val></rank>
    <node name="Nicotinate-nucleotide--dimethylbenzimidazole phosphoribosyltransferase (EC 2.4.2.21)">
     <rank><val>function</val></rank>
     <score><val>-20.2</val><val>-57.1111</val></score>
     <magnitude><val>7</val><val>41</val></magnitude>
    </node>
    <node name="Adenosylcobinamide amidohydrolase (EC 3.5.1.90)">
     <score><val>0</val><val>-60.6667</val></score>
     <magnitude><val></val><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="5,6-dimethylbenzimidazole synthase, flavin destructase family">
     <rank><val>function</val></rank>
     <score><val>-29</val><val>-39.4</val></score>
     <magnitude><val>1</val><val>8</val></magnitude>
    </node>
    <node name="Cob(I)alamin adenosyltransferase PduO (EC 2.5.1.17)">
     <magnitude><val>4</val><val>21</val></magnitude>
     <score><val>-25</val><val>-43</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cobalamin biosynthesis protein CbiG">
     <magnitude><val></val><val>24</val></magnitude>
     <score><val>0</val><val>-80.0625</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="L-threonine 3-O-phosphate decarboxylase (EC 4.1.1.81)">
     <score><val>-1</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cobalt-precorrin-4 C11-methyltransferase (EC 2.1.1.133)">
     <score><val>0</val><val>-65.9</val></score>
     <magnitude><val></val><val>14</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cobalamin synthase">
     <score><val>-8</val><val>-69.0769</val></score>
     <magnitude><val>1</val><val>18</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="CobN component of cobalt chelatase involved in B12 biosynthesis">
     <rank><val>function</val></rank>
     <score><val>-50</val><val>-112.333</val></score>
     <magnitude><val>3</val><val>50</val></magnitude>
    </node>
    <node name="Cobyric acid synthase">
     <rank><val>function</val></rank>
     <score><val>-20.3333</val><val>-92.48</val></score>
     <magnitude><val>6</val><val>31</val></magnitude>
    </node>
    <node name="Cobalt-precorrin-8x methylmutase (EC 5.4.1.2)">
     <score><val>-7.33333</val></score>
     <magnitude><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cobalt-precorrin-6y C15-methyltransferase [decarboxylating] (EC 2.1.1.-)">
     <rank><val>function</val></rank>
     <score><val>-14</val><val>-88.7778</val></score>
     <magnitude><val>1</val><val>21</val></magnitude>
    </node>
    <node name="Cobalt-precorrin-3b C17-methyltransferase">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-33</val></score>
     <magnitude><val></val><val>24</val></magnitude>
    </node>
    <node name="Cobalt-precorrin-6y C5-methyltransferase (EC 2.1.1.-)">
     <score><val>-14</val><val>-88.7778</val></score>
     <magnitude><val>1</val><val>21</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Predicted cobalt transporter in Thermoplasma">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-17</val></score>
     <magnitude><val></val><val>2</val></magnitude>
    </node>
    <node name="Cobalt-precorrin-2 C20-methyltransferase (EC 2.1.1.130)">
     <magnitude><val></val><val>12</val></magnitude>
     <score><val>0</val><val>-74.6364</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Aerobic cobaltochelatase CobT subunit (EC 6.6.1.2)">
     <score><val>-141</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cobalt-precorrin-6x reductase (EC 1.3.1.54)">
     <score><val>0</val><val>-47.8889</val></score>
     <magnitude><val></val><val>15</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cob(I)alamin adenosyltransferase (EC 2.5.1.17)">
     <rank><val>function</val></rank>
     <score><val>-31.75</val></score>
     <magnitude><val>5</val></magnitude>
    </node>
    <node name="ChlD component of cobalt chelatase involved in B12 biosynthesis">
     <magnitude><val>1</val><val>21</val></magnitude>
     <score><val>-22</val><val>-50.8824</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Adenosylcobinamide-phosphate synthase">
     <magnitude><val>5</val><val>18</val></magnitude>
     <score><val>-23.4</val><val>-61.7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="ChlI component of cobalt chelatase involved in B12 biosynthesis">
     <magnitude><val>1</val><val>21</val></magnitude>
     <score><val>-22</val><val>-50.8824</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cobalt-precorrin-6 synthase, anaerobic">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-91</val></score>
     <magnitude><val></val><val>1</val></magnitude>
    </node>
    <node name="Cobyrinic acid A,C-diamide synthase">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>21</val></magnitude>
     <score><val>-8.5</val><val>-75.5</val></score>
    </node>
   </node>
   <node name="Heme biosynthesis orphans">
    <rank><val>level 3</val></rank>
    <magnitude><val>5</val><val>4</val></magnitude>
    <score><val>-16.6</val><val>-12.6667</val></score>
    <node name="Radical SAM domain heme biosynthesis protein">
     <rank><val>function</val></rank>
     <score><val>-16.6</val><val>-12.6667</val></score>
     <magnitude><val>5</val><val>4</val></magnitude>
    </node>
   </node>
   <node name="Cobalamin synthesis">
    <score><val>-19.8758</val><val>-65.2167</val></score>
    <magnitude><val>31</val><val>227</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Cobyrinic acid A,C-diamide synthase">
     <rank><val>function</val></rank>
     <score><val>-8.5</val><val>-75.5</val></score>
     <magnitude><val>2</val><val>21</val></magnitude>
    </node>
    <node name="Cobalamin biosynthesis protein BluB">
     <score><val>-29</val><val>-39.4</val></score>
     <magnitude><val>1</val><val>8</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Adenosylcobinamide-phosphate synthase">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>18</val></magnitude>
     <score><val>-23.4</val><val>-61.7</val></score>
    </node>
    <node name="Cobalt-precorrin-6 synthase, anaerobic">
     <score><val>0</val><val>-91</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cob(I)alamin adenosyltransferase (EC 2.5.1.17)">
     <rank><val>function</val></rank>
     <score><val>-31.75</val></score>
     <magnitude><val>5</val></magnitude>
    </node>
    <node name="Cobalt-precorrin-8x methylmutase (EC 5.4.1.2)">
     <rank><val>function</val></rank>
     <score><val>-7.33333</val></score>
     <magnitude><val>3</val></magnitude>
    </node>
    <node name="Cobyric acid synthase">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>31</val></magnitude>
     <score><val>-20.3333</val><val>-92.48</val></score>
    </node>
    <node name="Cobalt-precorrin-6x reductase (EC 1.3.1.54)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>15</val></magnitude>
     <score><val>0</val><val>-47.8889</val></score>
    </node>
    <node name="Cobalamin synthase">
     <rank><val>function</val></rank>
     <score><val>-8</val><val>-69.0769</val></score>
     <magnitude><val>1</val><val>18</val></magnitude>
    </node>
    <node name="Cobalt-precorrin-2 C20-methyltransferase (EC 2.1.1.130)">
     <score><val>0</val><val>-74.6364</val></score>
     <magnitude><val></val><val>12</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cobalamin biosynthesis protein CbiG">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>24</val></magnitude>
     <score><val>0</val><val>-80.0625</val></score>
    </node>
    <node name="L-threonine 3-O-phosphate decarboxylase (EC 4.1.1.81)">
     <score><val>-1</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cobalt-precorrin-4 C11-methyltransferase (EC 2.1.1.133)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>14</val></magnitude>
     <score><val>0</val><val>-65.9</val></score>
    </node>
    <node name="Cobalt-precorrin-3b C17-methyltransferase">
     <magnitude><val></val><val>24</val></magnitude>
     <score><val>0</val><val>-33</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Nicotinate-nucleotide--dimethylbenzimidazole phosphoribosyltransferase (EC 2.4.2.21)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>41</val></magnitude>
     <score><val>-20.2</val><val>-57.1111</val></score>
    </node>
   </node>
   <node name="Chlorophyll Biosynthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-3</val></score>
    <node name="Geranylgeranyl hydrogenase BchP">
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Heme and Siroheme Biosynthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>37</val><val>256</val></magnitude>
    <score><val>-27.7185</val><val>-65.8655</val></score>
    <node name="Glutamyl-tRNA reductase (EC 1.2.1.70)">
     <magnitude><val>1</val><val>32</val></magnitude>
     <score><val>-10</val><val>-49.5769</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Uroporphyrinogen-III synthase (EC 4.2.1.75)">
     <magnitude><val></val><val>16</val></magnitude>
     <score><val>0</val><val>-48.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Precorrin-2 oxidase (EC 1.3.1.76)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-34.8571</val></score>
     <magnitude><val></val><val>15</val></magnitude>
    </node>
    <node name="Uroporphyrinogen III decarboxylase (EC 4.1.1.37)">
     <magnitude><val>7</val><val>26</val></magnitude>
     <score><val>-8.83333</val><val>-68.2381</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glutamate-1-semialdehyde aminotransferase (EC 5.4.3.8)">
     <score><val>-48.75</val><val>-91.2963</val></score>
     <magnitude><val>5</val><val>42</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Porphobilinogen deaminase (EC 2.5.1.61)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>18</val></magnitude>
     <score><val>-15.8</val><val>-71.9375</val></score>
    </node>
    <node name="Glutamyl-tRNA synthetase (EC 6.1.1.17)">
     <magnitude><val>9</val><val>45</val></magnitude>
     <score><val>-49</val><val>-80.64</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ferrochelatase, protoheme ferro-lyase (EC 4.99.1.1)">
     <magnitude><val>2</val><val>18</val></magnitude>
     <score><val>-6</val><val>-48.8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Protoporphyrinogen IX oxidase, aerobic, HemY (EC 1.3.3.4)">
     <score><val>-8</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Sirohydrochlorin ferrochelatase (EC 4.99.1.4)">
     <magnitude><val></val><val>15</val></magnitude>
     <score><val>0</val><val>-34.8571</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Porphobilinogen synthase (EC 4.2.1.24)">
     <score><val>-24.2857</val><val>-70.4375</val></score>
     <magnitude><val>7</val><val>29</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Pyridoxine">
   <rank><val>level 2</val></rank>
   <score><val>-34.7407</val><val>-70.8322</val></score>
   <magnitude><val>27</val><val>127</val></magnitude>
   <node name="Pyridoxin (Vitamin B6) Biosynthesis">
    <magnitude><val>27</val><val>127</val></magnitude>
    <score><val>-34.7407</val><val>-70.8322</val></score>
    <rank><val>level 3</val></rank>
    <node name="NAD-dependent glyceraldehyde-3-phosphate dehydrogenase (EC 1.2.1.12)">
     <score><val>-11</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="D-3-phosphoglycerate dehydrogenase (EC 1.1.1.95)">
     <magnitude><val>5</val><val>35</val></magnitude>
     <score><val>-25.2</val><val>-72.6071</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Pyridoxine biosynthesis glutamine amidotransferase, glutaminase subunit (EC 2.4.2.-)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>23</val></magnitude>
     <score><val>-25</val><val>-53.8125</val></score>
    </node>
    <node name="Pyridoxine biosynthesis glutamine amidotransferase, synthase subunit (EC 2.4.2.-)">
     <rank><val>function</val></rank>
     <magnitude><val>14</val><val>42</val></magnitude>
     <score><val>-44.2857</val><val>-80.7083</val></score>
    </node>
    <node name="4-hydroxythreonine-4-phosphate dehydrogenase (EC 1.1.1.262)">
     <magnitude><val>3</val><val>27</val></magnitude>
     <score><val>-27</val><val>-67.6667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="1-deoxy-D-xylulose 5-phosphate synthase (EC 2.2.1.7)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-25</val></score>
    </node>
   </node>
  </node>
  <node name="Quinone cofactors">
   <magnitude><val>146</val><val>387</val></magnitude>
   <score><val>-38.9358</val><val>-69.8102</val></score>
   <rank><val>level 2</val></rank>
   <node name="Coenzyme PQQ synthesis">
    <rank><val>level 3</val></rank>
    <score><val>-54</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="Coenzyme PQQ synthesis protein E">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-54</val></score>
    </node>
   </node>
   <node name="Pyrroloquinoline Quinone biosynthesis">
    <magnitude><val>1</val></magnitude>
    <score><val>-54</val></score>
    <rank><val>level 3</val></rank>
    <node name="Coenzyme PQQ synthesis protein E">
     <score><val>-54</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Menaquinone Biosynthesis via Futalosine -- gjo">
    <score><val>-42.1513</val><val>-67.3294</val></score>
    <magnitude><val>66</val><val>198</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="3-polyprenyl-4-hydroxybenzoate carboxy-lyase (EC 4.1.1.-)">
     <rank><val>function</val></rank>
     <score><val>-48.8182</val><val>-71.2647</val></score>
     <magnitude><val>13</val><val>53</val></magnitude>
    </node>
    <node name="Gene SCO4494, often clustered with other genes in menaquinone via futalosine pathway">
     <rank><val>function</val></rank>
     <score><val>-48.7778</val><val>-84.3871</val></score>
     <magnitude><val>10</val><val>37</val></magnitude>
    </node>
    <node name="3-polyprenyl-4-hydroxybenzoate carboxy-lyase UbiX (EC 4.1.1.-)">
     <rank><val>function</val></rank>
     <score><val>-32.875</val><val>-49.2143</val></score>
     <magnitude><val>8</val><val>18</val></magnitude>
    </node>
    <node name="Menaquinone via futalosine step 3">
     <rank><val>function</val></rank>
     <score><val>-49.2857</val><val>-71.2857</val></score>
     <magnitude><val>16</val><val>47</val></magnitude>
    </node>
    <node name="Menaquinone via futalosine step 4">
     <score><val>-63</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Menaquinone via futalosine step 1">
     <magnitude><val></val><val>27</val></magnitude>
     <score><val>0</val><val>-51.3333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="4-hydroxybenzoate polyprenyltransferase (EC 2.5.1.-)">
     <rank><val>function</val></rank>
     <score><val>-33.4444</val><val>-49.5833</val></score>
     <magnitude><val>9</val><val>13</val></magnitude>
    </node>
    <node name="Ubiquinone/menaquinone biosynthesis methyltransferase UbiE (EC 2.1.1.-)">
     <magnitude><val>8</val><val>3</val></magnitude>
     <score><val>-22.625</val><val>-55</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Menaquinone and Phylloquinone Biosynthesis">
    <magnitude><val>26</val><val>32</val></magnitude>
    <score><val>-30.3531</val><val>-75.0688</val></score>
    <rank><val>level 3</val></rank>
    <node name="O-succinylbenzoate-CoA synthase (EC 4.2.1.-)">
     <score><val>-31.1818</val><val>-79.4</val></score>
     <magnitude><val>12</val><val>28</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="1,4-dihydroxy-2-naphthoate octaprenyltransferase (EC 2.5.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-14</val></score>
    </node>
    <node name="Ubiquinone/menaquinone biosynthesis methyltransferase UbiE (EC 2.1.1.-)">
     <magnitude><val>8</val><val>3</val></magnitude>
     <score><val>-22.625</val><val>-55</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2-succinyl-5-enolpyruvyl-6-hydroxy-3-cyclohexene-1-carboxylic-acid synthase (EC 2.2.1.9)">
     <magnitude><val>1</val></magnitude>
     <score><val>-48</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2-heptaprenyl-1,4-naphthoquinone methyltransferase (EC 2.1.1.-)">
     <magnitude><val>2</val></magnitude>
     <score><val>-8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Naphthoate synthase (EC 4.1.3.36)">
     <score><val>-56.6667</val></score>
     <magnitude><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Tocopherol Biosynthesis">
    <rank><val>level 3</val></rank>
    <score><val>-74</val><val>-85.2308</val></score>
    <magnitude><val>1</val><val>20</val></magnitude>
    <node name="4-hydroxyphenylpyruvate dioxygenase (EC 1.13.11.27)">
     <magnitude><val>1</val><val>20</val></magnitude>
     <score><val>-74</val><val>-85.2308</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Ubiquinone Biosynthesis">
    <score><val>-37.1491</val><val>-67.2983</val></score>
    <magnitude><val>50</val><val>117</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Ubiquinone/menaquinone biosynthesis methyltransferase UbiE (EC 2.1.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>3</val></magnitude>
     <score><val>-22.625</val><val>-55</val></score>
    </node>
    <node name="4-hydroxybenzoate polyprenyltransferase (EC 2.5.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>13</val></magnitude>
     <score><val>-33.4444</val><val>-49.5833</val></score>
    </node>
    <node name="Ubiquinone biosynthesis monooxygenase UbiB">
     <magnitude><val>12</val><val>30</val></magnitude>
     <score><val>-39.8182</val><val>-80.0476</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="3-polyprenyl-4-hydroxybenzoate carboxy-lyase UbiX (EC 4.1.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>18</val></magnitude>
     <score><val>-32.875</val><val>-49.2143</val></score>
    </node>
    <node name="3-polyprenyl-4-hydroxybenzoate carboxy-lyase (EC 4.1.1.-)">
     <score><val>-48.8182</val><val>-71.2647</val></score>
     <magnitude><val>13</val><val>53</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Plastoquinone Biosynthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val><val>20</val></magnitude>
    <score><val>-74</val><val>-85.2308</val></score>
    <node name="4-hydroxyphenylpyruvate dioxygenase (EC 1.13.11.27)">
     <magnitude><val>1</val><val>20</val></magnitude>
     <score><val>-74</val><val>-85.2308</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Coenzyme A">
   <rank><val>level 2</val></rank>
   <magnitude><val>19</val><val>88</val></magnitude>
   <score><val>-23.406</val><val>-58.6401</val></score>
   <node name="Coenzyme A Biosynthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>19</val><val>88</val></magnitude>
    <score><val>-23.406</val><val>-58.6401</val></score>
    <node name="Phosphopantetheine adenylyltransferase, type II eukaryotic (EC 2.7.7.3)">
     <rank><val>function</val></rank>
     <score><val>-10.4</val><val>-55.7391</val></score>
     <magnitude><val>5</val><val>31</val></magnitude>
    </node>
    <node name="Pantothenate kinase (EC 2.7.1.33)">
     <rank><val>function</val></rank>
     <score><val>-19</val><val>-39.5</val></score>
     <magnitude><val>2</val><val>7</val></magnitude>
    </node>
    <node name="Pantothenate kinase type III, CoaX-like (EC 2.7.1.33)">
     <magnitude><val>1</val></magnitude>
     <score><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dephospho-CoA kinase archaeal, predicted (EC 2.7.1.24)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>22</val></magnitude>
     <score><val>-27.7143</val><val>-41.3333</val></score>
    </node>
    <node name="3-methyl-2-oxobutanoate hydroxymethyltransferase (EC 2.1.2.11)">
     <rank><val>function</val></rank>
     <score><val>-15</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Phosphopantothenate synthetase, archaeal">
     <rank><val>function</val></rank>
     <score><val>-14</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Ketol-acid reductoisomerase (EC 1.1.1.86)">
     <score><val>-97</val><val>-80.2353</val></score>
     <magnitude><val>1</val><val>28</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Biotin">
   <rank><val>level 2</val></rank>
   <magnitude><val>55</val><val>163</val></magnitude>
   <score><val>-44.9218</val><val>-77.9165</val></score>
   <node name="Biotin biosynthesis">
    <score><val>-44.9218</val><val>-77.9165</val></score>
    <magnitude><val>55</val><val>163</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Biotin synthase (EC 2.8.1.6)">
     <score><val>-37.5</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Long-chain-fatty-acid--CoA ligase (EC 6.2.1.3)">
     <rank><val>function</val></rank>
     <magnitude><val>24</val><val>51</val></magnitude>
     <score><val>-40.9524</val><val>-80.3</val></score>
    </node>
    <node name="3-ketoacyl-CoA thiolase (EC 2.3.1.16)">
     <score><val>-49.92</val><val>-76.8312</val></score>
     <magnitude><val>27</val><val>112</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Adenosylmethionine-8-amino-7-oxononanoate aminotransferase (EC 2.6.1.62)">
     <score><val>-32.5</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Folate and pterines">
   <magnitude><val>455</val><val>1948</val></magnitude>
   <score><val>-32.9212</val><val>-63.9896</val></score>
   <rank><val>level 2</val></rank>
   <node name="Pterin metabolism">
    <magnitude><val>3</val><val>21</val></magnitude>
    <score><val>-26.6667</val><val>-38.0688</val></score>
    <rank><val>level 3</val></rank>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>16</val></magnitude>
     <score><val>-64</val><val>-47.7778</val></score>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 2">
     <score><val>-8</val><val>-7</val></score>
     <magnitude><val>2</val><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Methanopterin biosynthesis2">
    <score><val>-16.2857</val><val>-36.6985</val></score>
    <magnitude><val>7</val><val>37</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="2-amino-4-hydroxy-6-hydroxymethyldihydropteridine pyrophosphokinase (EC 2.7.6.3)">
     <rank><val>function</val></rank>
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Queuosine biosynthesis QueD, PTPS-I">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>16</val></magnitude>
     <score><val>-9</val><val>-34.9</val></score>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 2">
     <rank><val>function</val></rank>
     <score><val>-8</val><val>-7</val></score>
     <magnitude><val>2</val><val>5</val></magnitude>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>16</val></magnitude>
     <score><val>-64</val><val>-47.7778</val></score>
    </node>
    <node name="Dihydroneopterin aldolase (EC 4.1.2.25)">
     <rank><val>function</val></rank>
     <score><val>-13</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Pterin metabolism 3">
    <score><val>-26.4713</val><val>-73.3639</val></score>
    <magnitude><val>29</val><val>129</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <rank><val>function</val></rank>
     <score><val>-64</val><val>-47.7778</val></score>
     <magnitude><val>1</val><val>16</val></magnitude>
    </node>
    <node name="Dihydroneopterin aldolase (EC 4.1.2.25)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-13</val></score>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 2">
     <magnitude><val>2</val><val>5</val></magnitude>
     <score><val>-8</val><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Fumarylacetoacetase (EC 3.7.1.2)">
     <rank><val>function</val></rank>
     <score><val>-15</val><val>-104.667</val></score>
     <magnitude><val>2</val><val>7</val></magnitude>
    </node>
    <node name="Maleylacetoacetate isomerase (EC 5.2.1.2)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-27</val></score>
    </node>
    <node name="COG1565: Uncharacterized conserved protein">
     <score><val>-9</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Homogentisate 1,2-dioxygenase (EC 1.13.11.5)">
     <rank><val>function</val></rank>
     <score><val>-49.25</val><val>-81.5556</val></score>
     <magnitude><val>4</val><val>25</val></magnitude>
    </node>
    <node name="Fumarylacetoacetate hydrolase family protein">
     <rank><val>function</val></rank>
     <score><val>-11.6667</val><val>-67.1</val></score>
     <magnitude><val>4</val><val>18</val></magnitude>
    </node>
    <node name="Acetoacetyl-CoA synthetase (EC 6.2.1.16)">
     <score><val>-37</val></score>
     <magnitude><val>4</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydropteroate synthase (EC 2.5.1.15)">
     <magnitude><val>7</val><val>38</val></magnitude>
     <score><val>-20</val><val>-78.4348</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2-amino-4-hydroxy-6-hydroxymethyldihydropteridine pyrophosphokinase (EC 2.7.6.3)">
     <rank><val>function</val></rank>
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="4-hydroxyphenylpyruvate dioxygenase (EC 1.13.11.27)">
     <score><val>-74</val><val>-85.2308</val></score>
     <magnitude><val>1</val><val>20</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Folate Biosynthesis">
    <rank><val>level 3</val></rank>
    <score><val>-23.2369</val><val>-63.4297</val></score>
    <magnitude><val>41</val><val>145</val></magnitude>
    <node name="Folylpolyglutamate synthase (EC 6.3.2.17)">
     <magnitude><val>8</val><val>30</val></magnitude>
     <score><val>-22.8571</val><val>-65.56</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydroneopterin aldolase (EC 4.1.2.25)">
     <rank><val>function</val></rank>
     <score><val>-13</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <score><val>-64</val><val>-47.7778</val></score>
     <magnitude><val>1</val><val>16</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 2">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>5</val></magnitude>
     <score><val>-8</val><val>-7</val></score>
    </node>
    <node name="Dihydrofolate synthase (EC 6.3.2.12)">
     <score><val>-22.8571</val><val>-65.56</val></score>
     <magnitude><val>8</val><val>30</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="2-amino-4-hydroxy-6-hydroxymethyldihydropteridine pyrophosphokinase (EC 2.7.6.3)">
     <rank><val>function</val></rank>
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Dihydropteroate synthase (EC 2.5.1.15)">
     <score><val>-20</val><val>-78.4348</val></score>
     <magnitude><val>7</val><val>38</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Thymidylate synthase thyX (EC 2.1.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>13</val><val>26</val></magnitude>
     <score><val>-27</val><val>-57.0667</val></score>
    </node>
   </node>
   <node name="YgfZ-Iron">
    <rank><val>level 3</val></rank>
    <magnitude><val>25</val><val>71</val></magnitude>
    <score><val>-30.2831</val><val>-56.9048</val></score>
    <node name="Transcriptional regulator, FUR family">
     <rank><val>function</val></rank>
     <score><val>-5</val><val>-5</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
    <node name="Dihydroorotase (EC 3.5.2.3)">
     <rank><val>function</val></rank>
     <score><val>-37.5385</val><val>-72.9545</val></score>
     <magnitude><val>15</val><val>31</val></magnitude>
    </node>
    <node name="Heme O synthase, protoheme IX farnesyltransferase (EC 2.5.1.-) COX10-CtaB">
     <magnitude><val>7</val><val>39</val></magnitude>
     <score><val>-21</val><val>-45.4783</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cytochrome c heme lyase subunit CcmF">
     <score><val>-37</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Pterin carbinolamine dehydratase">
    <score><val>-31.2222</val><val>-81.1996</val></score>
    <magnitude><val>12</val><val>70</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Maleylacetoacetate isomerase (EC 5.2.1.2)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-27</val></score>
    </node>
    <node name="4-hydroxyphenylpyruvate dioxygenase (EC 1.13.11.27)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>20</val></magnitude>
     <score><val>-74</val><val>-85.2308</val></score>
    </node>
    <node name="Homogentisate 1,2-dioxygenase (EC 1.13.11.5)">
     <rank><val>function</val></rank>
     <score><val>-49.25</val><val>-81.5556</val></score>
     <magnitude><val>4</val><val>25</val></magnitude>
    </node>
    <node name="Fumarylacetoacetate hydrolase family protein">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>18</val></magnitude>
     <score><val>-11.6667</val><val>-67.1</val></score>
    </node>
    <node name="Fumarylacetoacetase (EC 3.7.1.2)">
     <magnitude><val>2</val><val>7</val></magnitude>
     <score><val>-15</val><val>-104.667</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Molybdenum cofactor biosynthesis">
    <magnitude><val>22</val><val>158</val></magnitude>
    <score><val>-19.2045</val><val>-66.6354</val></score>
    <rank><val>level 3</val></rank>
    <node name="Periplasmic molybdate-binding domain">
     <rank><val>function</val></rank>
     <score><val>-16.75</val><val>-87.0833</val></score>
     <magnitude><val>6</val><val>34</val></magnitude>
    </node>
    <node name="Molybdenum cofactor biosynthesis protein MoaE">
     <magnitude><val></val><val>21</val></magnitude>
     <score><val>0</val><val>-82</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP cyclohydrolase II (EC 3.5.4.25)">
     <score><val>-17.3333</val><val>-29.4286</val></score>
     <magnitude><val>6</val><val>19</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <score><val>-64</val><val>-47.7778</val></score>
     <magnitude><val>1</val><val>16</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Molybdopterin biosynthesis protein MoeB">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Molybdenum cofactor biosynthesis protein MoaB">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>4</val></magnitude>
     <score><val>0</val><val>-63</val></score>
    </node>
    <node name="Molybdenum transport system permease protein ModB (TC 3.A.1.8.1)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-20</val></score>
     <magnitude><val></val><val>2</val></magnitude>
    </node>
    <node name="Molybdenum cofactor biosynthesis protein MoaA">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>1</val></magnitude>
     <score><val>-24.5</val><val>-88</val></score>
    </node>
    <node name="Molybdopterin-guanine dinucleotide biosynthesis protein MobA">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>5</val></magnitude>
     <score><val>0</val><val>-17.2</val></score>
    </node>
    <node name="Molybdopterin biosynthesis protein MoeA">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>43</val></magnitude>
     <score><val>-15</val><val>-80.7667</val></score>
    </node>
    <node name="Molybdenum cofactor biosynthesis protein MoaC">
     <magnitude><val></val><val>12</val></magnitude>
     <score><val>0</val><val>-48</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="YgfZ">
    <magnitude><val>207</val><val>818</val></magnitude>
    <score><val>-31.0138</val><val>-62.0047</val></score>
    <rank><val>level 3</val></rank>
    <node name="2-amino-4-hydroxy-6-hydroxymethyldihydropteridine pyrophosphokinase (EC 2.7.6.3)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-3</val></score>
    </node>
    <node name="Chorismate synthase (EC 4.2.3.5)">
     <magnitude><val>5</val><val>42</val></magnitude>
     <score><val>-28.75</val><val>-65.6552</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Molybdenum cofactor biosynthesis protein MoaA">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>1</val></magnitude>
     <score><val>-24.5</val><val>-88</val></score>
    </node>
    <node name="Transcriptional regulator, FUR family">
     <score><val>-5</val><val>-5</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydroorotase (EC 3.5.2.3)">
     <rank><val>function</val></rank>
     <score><val>-37.5385</val><val>-72.9545</val></score>
     <magnitude><val>15</val><val>31</val></magnitude>
    </node>
    <node name="GTP cyclohydrolase II (EC 3.5.4.25)">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>19</val></magnitude>
     <score><val>-17.3333</val><val>-29.4286</val></score>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7), SufS subfamily">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>18</val></magnitude>
     <score><val>-31</val><val>-37.6667</val></score>
    </node>
    <node name="Aconitate hydratase (EC 4.2.1.3)">
     <score><val>-76</val><val>-103.438</val></score>
     <magnitude><val>2</val><val>64</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="5,10-methylenetetrahydrofolate reductase (EC 1.5.1.20)">
     <magnitude><val>3</val><val>6</val></magnitude>
     <score><val>-9</val><val>-7.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Succinate dehydrogenase iron-sulfur protein (EC 1.3.99.1)">
     <magnitude><val>3</val><val>18</val></magnitude>
     <score><val>-30.6667</val><val>-94.9231</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydroorotate dehydrogenase, catalytic subunit (EC 1.3.3.1)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>18</val></magnitude>
     <score><val>-47.5</val><val>-68.7857</val></score>
    </node>
    <node name="Phosphoribosylformylglycinamidine cyclo-ligase (EC 6.3.3.1)">
     <rank><val>function</val></rank>
     <score><val>-67.8571</val><val>-79.9259</val></score>
     <magnitude><val>12</val><val>48</val></magnitude>
    </node>
    <node name="NADH-ubiquinone oxidoreductase chain F (EC 1.6.5.3)">
     <rank><val>function</val></rank>
     <score><val>-43</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Coenzyme PQQ synthesis protein E">
     <magnitude><val>1</val></magnitude>
     <score><val>-54</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Iron-sulfur cluster assembly ATPase protein SufC">
     <rank><val>function</val></rank>
     <score><val>-53.1111</val><val>-58.2941</val></score>
     <magnitude><val>9</val><val>23</val></magnitude>
    </node>
    <node name="Queuosine Biosynthesis QueC ATPase">
     <magnitude><val>5</val><val>34</val></magnitude>
     <score><val>-29.8</val><val>-30.1</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoglycerate kinase (EC 2.7.2.3)">
     <magnitude><val>7</val><val>13</val></magnitude>
     <score><val>-25.7143</val><val>-63.5714</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribose-phosphate pyrophosphokinase (EC 2.7.6.1)">
     <score><val>-12.6667</val><val>-57.5263</val></score>
     <magnitude><val>7</val><val>33</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Amidophosphoribosyltransferase (EC 2.4.2.14)">
     <score><val>-45.875</val><val>-77.3571</val></score>
     <magnitude><val>8</val><val>42</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="L-aspartate oxidase (EC 1.4.3.16)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
    </node>
    <node name="tRNA-i(6)A37 methylthiotransferase">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>32</val></magnitude>
     <score><val>-22.8571</val><val>-40.2083</val></score>
    </node>
    <node name="Lipoate synthase">
     <score><val>-32.625</val><val>-57.3077</val></score>
     <magnitude><val>8</val><val>39</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7)">
     <score><val>-29.4167</val><val>-52.625</val></score>
     <magnitude><val>15</val><val>38</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Twin-arginine translocation protein TatA">
     <score><val>-13</val><val>-29.75</val></score>
     <magnitude><val>1</val><val>13</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cystathionine gamma-synthase (EC 2.5.1.48)">
     <score><val>-17.1818</val><val>-62.12</val></score>
     <magnitude><val>13</val><val>42</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Adenosylcobinamide-phosphate synthase">
     <magnitude><val>5</val><val>18</val></magnitude>
     <score><val>-23.4</val><val>-61.7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Aminomethyltransferase (glycine cleavage system T protein) (EC 2.1.2.10)">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>38</val></magnitude>
     <score><val>-21.125</val><val>-58.8571</val></score>
    </node>
    <node name="Ferredoxin">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>61</val></magnitude>
     <score><val>-16.7778</val><val>-61.9091</val></score>
    </node>
    <node name="Dihydroorotate dehydrogenase (EC 1.3.3.1)">
     <score><val>-44</val><val>-51.5714</val></score>
     <magnitude><val>1</val><val>7</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydropteroate synthase (EC 2.5.1.15)">
     <rank><val>function</val></rank>
     <score><val>-20</val><val>-78.4348</val></score>
     <magnitude><val>7</val><val>38</val></magnitude>
    </node>
    <node name="Phosphoribosyl-AMP cyclohydrolase (EC 3.5.4.19)">
     <magnitude><val>1</val><val>11</val></magnitude>
     <score><val>-34</val><val>-57.1111</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Aspartate carbamoyltransferase (EC 2.1.3.2)">
     <score><val>-42.1</val><val>-52.3333</val></score>
     <magnitude><val>10</val><val>24</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydroorotate dehydrogenase electron transfer subunit (EC 1.3.3.1)">
     <score><val>-19.5</val><val>-36.4167</val></score>
     <magnitude><val>8</val><val>14</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Orotate phosphoribosyltransferase (EC 2.4.2.10)">
     <score><val>-20</val><val>-37.75</val></score>
     <magnitude><val>4</val><val>32</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Biotin synthase (EC 2.8.1.6)">
     <magnitude><val>2</val></magnitude>
     <score><val>-37.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Pterin biosynthesis">
    <score><val>-64</val><val>-47.7778</val></score>
    <magnitude><val>1</val><val>16</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <magnitude><val>1</val><val>16</val></magnitude>
     <score><val>-64</val><val>-47.7778</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="5-FCL-like protein">
    <score><val>-46.5433</val><val>-66.452</val></score>
    <magnitude><val>108</val><val>483</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Isocitrate dehydrogenase [NADP] (EC 1.1.1.42)">
     <score><val>-79.7778</val><val>-78.8929</val></score>
     <magnitude><val>12</val><val>46</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Thiamin-phosphate synthase ThiN (EC 2.5.1.3)">
     <magnitude><val>1</val></magnitude>
     <score><val>-6</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Serine hydroxymethyltransferase (EC 2.1.2.1)">
     <rank><val>function</val></rank>
     <magnitude><val>11</val><val>67</val></magnitude>
     <score><val>-76</val><val>-79.3077</val></score>
    </node>
    <node name="Phosphomethylpyrimidine kinase (EC 2.7.4.7)">
     <magnitude><val>1</val><val>5</val></magnitude>
     <score><val>-6</val><val>-91.3333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Butyryl-CoA dehydrogenase (EC 1.3.99.2)">
     <rank><val>function</val></rank>
     <magnitude><val>21</val><val>25</val></magnitude>
     <score><val>-54.6</val><val>-56.4444</val></score>
    </node>
    <node name="Glutamate formiminotransferase (EC 2.1.2.5)">
     <magnitude><val>3</val><val>7</val></magnitude>
     <score><val>-46</val><val>-37</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Alcohol dehydrogenase (EC 1.1.1.1)">
     <rank><val>function</val></rank>
     <score><val>-21.5</val><val>-68.2</val></score>
     <magnitude><val>12</val><val>34</val></magnitude>
    </node>
    <node name="Formate--tetrahydrofolate ligase (EC 6.3.4.3)">
     <rank><val>function</val></rank>
     <score><val>-67.1429</val><val>-86.5909</val></score>
     <magnitude><val>8</val><val>64</val></magnitude>
    </node>
    <node name="Thiamine-monophosphate kinase (EC 2.7.4.16)">
     <score><val>-11</val><val>-62</val></score>
     <magnitude><val>3</val><val>24</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylaminoimidazolecarboxamide formyltransferase (EC 2.1.2.3)">
     <rank><val>function</val></rank>
     <score><val>-17.6</val><val>-65.4545</val></score>
     <magnitude><val>6</val><val>32</val></magnitude>
    </node>
    <node name="5,10-methylenetetrahydrofolate reductase (EC 1.5.1.20)">
     <magnitude><val>3</val><val>6</val></magnitude>
     <score><val>-9</val><val>-7.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Thiaminase II (EC 3.5.99.2)">
     <score><val>-21.2</val><val>-35.75</val></score>
     <magnitude><val>5</val><val>37</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Succinate dehydrogenase iron-sulfur protein (EC 1.3.99.1)">
     <rank><val>function</val></rank>
     <score><val>-30.6667</val><val>-94.9231</val></score>
     <magnitude><val>3</val><val>18</val></magnitude>
    </node>
    <node name="Methylenetetrahydrofolate dehydrogenase (NADP+) (EC 1.5.1.5) / Methenyltetrahydrofolate cyclohydrolase (EC 3.5.4.9)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>17</val></magnitude>
     <score><val>-14.2</val><val>-53.5</val></score>
    </node>
    <node name="H4MPt-dependent SHMT">
     <rank><val>function</val></rank>
     <score><val>-85.8571</val><val>-71.2121</val></score>
     <magnitude><val>7</val><val>43</val></magnitude>
    </node>
    <node name="Thiamin-phosphate pyrophosphorylase (EC 2.5.1.3)">
     <magnitude><val>1</val><val>1</val></magnitude>
     <score><val>-12</val><val>-87</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylglycinamide formyltransferase (EC 2.1.2.2)">
     <magnitude><val>2</val><val>21</val></magnitude>
     <score><val>-25</val><val>-38.3571</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Formiminotetrahydrofolate cyclodeaminase (EC 4.3.1.4)">
     <rank><val>function</val></rank>
     <score><val>-11</val><val>-52.8182</val></score>
     <magnitude><val>4</val><val>21</val></magnitude>
    </node>
    <node name="Dihydrolipoamide dehydrogenase (EC 1.8.1.4)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>10</val></magnitude>
     <score><val>0</val><val>-44.7</val></score>
    </node>
    <node name="MTHFR (MetF)-like">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>5</val></magnitude>
     <score><val>0</val><val>-17.4</val></score>
    </node>
   </node>
  </node>
  <node name="NAD and NADP">
   <rank><val>level 2</val></rank>
   <magnitude><val>56</val><val>170</val></magnitude>
   <score><val>-27.7839</val><val>-50.825</val></score>
   <node name="NAD consumption">
    <magnitude><val>2</val></magnitude>
    <score><val>-43</val></score>
    <rank><val>level 3</val></rank>
    <node name="DNA ligase (EC 6.5.1.2)">
     <rank><val>function</val></rank>
     <score><val>-43</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
   </node>
   <node name="NAD regulation">
    <magnitude><val>21</val><val>70</val></magnitude>
    <score><val>-26.6762</val><val>-53.659</val></score>
    <rank><val>level 3</val></rank>
    <node name="L-aspartate oxidase (EC 1.4.3.16)">
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Nicotinamide-nucleotide adenylyltransferase, NadM family (EC 2.7.7.1)">
     <rank><val>function</val></rank>
     <score><val>-30</val><val>-58.5333</val></score>
     <magnitude><val>2</val><val>25</val></magnitude>
    </node>
    <node name="NAD synthetase (EC 6.3.1.5)">
     <rank><val>function</val></rank>
     <score><val>-28</val><val>-60.8571</val></score>
     <magnitude><val>5</val><val>8</val></magnitude>
    </node>
    <node name="Nicotinamidase (EC 3.5.1.19)">
     <rank><val>function</val></rank>
     <score><val>-13.5</val><val>-42.2857</val></score>
     <magnitude><val>2</val><val>8</val></magnitude>
    </node>
    <node name="Nicotinate phosphoribosyltransferase (EC 2.4.2.11)">
     <magnitude><val>11</val><val>29</val></magnitude>
     <score><val>-29.2</val><val>-50.6087</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="NAD and NADP cofactor biosynthesis global">
    <score><val>-27.5667</val><val>-48.8413</val></score>
    <magnitude><val>33</val><val>100</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Glutamine amidotransferase chain of NAD synthetase">
     <magnitude><val>1</val></magnitude>
     <score><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Nicotinamidase/isochorismatase family protein">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-14.5</val></score>
    </node>
    <node name="ADP-ribose pyrophosphatase (EC 3.6.1.13)">
     <score><val>-17</val><val>-42.6667</val></score>
     <magnitude><val>1</val><val>15</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="NAD synthetase (EC 6.3.1.5)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>8</val></magnitude>
     <score><val>-28</val><val>-60.8571</val></score>
    </node>
    <node name="L-Aspartate dehydrogenase (EC 1.4.1.21)">
     <rank><val>function</val></rank>
     <score><val>-15</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="L-aspartate oxidase (EC 1.4.3.16)">
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Amidases related to nicotinamidase">
     <score><val>0</val><val>-8</val></score>
     <magnitude><val></val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Nicotinate phosphoribosyltransferase (EC 2.4.2.11)">
     <score><val>-29.2</val><val>-50.6087</val></score>
     <magnitude><val>11</val><val>29</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Nicotinamide-nucleotide adenylyltransferase, NadM family (EC 2.7.7.1)">
     <rank><val>function</val></rank>
     <score><val>-30</val><val>-58.5333</val></score>
     <magnitude><val>2</val><val>25</val></magnitude>
    </node>
    <node name="Nicotinamidase (EC 3.5.1.19)">
     <rank><val>function</val></rank>
     <score><val>-13.5</val><val>-42.2857</val></score>
     <magnitude><val>2</val><val>8</val></magnitude>
    </node>
    <node name="NAD kinase (EC 2.7.1.23)">
     <magnitude><val>9</val><val>11</val></magnitude>
     <score><val>-34.5</val><val>-40.2727</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Coenzyme M">
   <rank><val>level 2</val></rank>
   <magnitude><val>3</val><val>36</val></magnitude>
   <score><val>-9</val><val>-58.1875</val></score>
   <node name="coenzyme M biosynthesis">
    <score><val>-9</val><val>-58.1875</val></score>
    <magnitude><val>3</val><val>36</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Phosphosulfolactate synthase (EC 4.4.1.19)">
     <rank><val>function</val></rank>
     <score><val>-11</val><val>-86.1</val></score>
     <magnitude><val>1</val><val>15</val></magnitude>
    </node>
    <node name="2-phosphosulfolactate phosphatase (EC 3.1.3.71)">
     <magnitude><val>2</val><val>21</val></magnitude>
     <score><val>-8</val><val>-38.25</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Riboflavin, FMN, FAD">
   <magnitude><val>34</val><val>190</val></magnitude>
   <score><val>-28.2605</val><val>-50.8953</val></score>
   <rank><val>level 2</val></rank>
   <node name="riboflavin to FAD">
    <magnitude><val>7</val><val>34</val></magnitude>
    <score><val>-32.4286</val><val>-59.4044</val></score>
    <rank><val>level 3</val></rank>
    <node name="Riboflavin synthase archaeal (EC 2.5.1.9)">
     <magnitude><val>3</val><val>12</val></magnitude>
     <score><val>-42.3333</val><val>-56.875</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="CTP-dependent archaeal riboflavin kinase">
     <score><val>-30</val><val>-70.5</val></score>
     <magnitude><val>2</val><val>17</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="3,4-dihydroxy-2-butanone 4-phosphate synthase">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>5</val></magnitude>
     <score><val>-20</val><val>-27.75</val></score>
    </node>
   </node>
   <node name="Riboflavin, FMN and FAD metabolism">
    <rank><val>level 3</val></rank>
    <score><val>-27.1799</val><val>-49.0408</val></score>
    <magnitude><val>27</val><val>156</val></magnitude>
    <node name="GTP cyclohydrolase II (EC 3.5.4.25)">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>19</val></magnitude>
     <score><val>-17.3333</val><val>-29.4286</val></score>
    </node>
    <node name="DNA-binding HTH domain in riboflavin kinase">
     <magnitude><val>2</val><val>17</val></magnitude>
     <score><val>-30</val><val>-70.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="5-amino-6-(5-phosphoribosylamino)uracil reductase (EC 1.1.1.193)">
     <score><val>0</val><val>-27.7143</val></score>
     <magnitude><val></val><val>9</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Pyrimidine deaminase archaeal predicted (EC 3.5.4.26)">
     <rank><val>function</val></rank>
     <score><val>-27.8571</val><val>-53.2571</val></score>
     <magnitude><val>8</val><val>52</val></magnitude>
    </node>
    <node name="6,7-dimethyl-8-ribityllumazine synthase (EC 2.5.1.9)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-34.1667</val></score>
     <magnitude><val></val><val>11</val></magnitude>
    </node>
    <node name="CTP-dependent archaeal riboflavin kinase">
     <rank><val>function</val></rank>
     <score><val>-30</val><val>-70.5</val></score>
     <magnitude><val>2</val><val>17</val></magnitude>
    </node>
    <node name="3,4-dihydroxy-2-butanone 4-phosphate synthase">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>5</val></magnitude>
     <score><val>-20</val><val>-27.75</val></score>
    </node>
    <node name="FMN adenylyltransferase, type 3 archaeal (EC 2.7.7.2)">
     <magnitude><val>4</val><val>14</val></magnitude>
     <score><val>-30</val><val>-34.1667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Riboflavin synthase archaeal (EC 2.5.1.9)">
     <magnitude><val>3</val><val>12</val></magnitude>
     <score><val>-42.3333</val><val>-56.875</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Lipoic acid">
   <magnitude><val>13</val><val>88</val></magnitude>
   <score><val>-24.2115</val><val>-57.2292</val></score>
   <rank><val>level 2</val></rank>
   <node name="Lipoic acid metabolism">
    <rank><val>level 3</val></rank>
    <magnitude><val>13</val><val>88</val></magnitude>
    <score><val>-24.2115</val><val>-57.2292</val></score>
    <node name="Lipoate synthase">
     <magnitude><val>8</val><val>39</val></magnitude>
     <score><val>-32.625</val><val>-57.3077</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Octanoate-[acyl-carrier-protein]-protein-N-octanoyltransferase">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>49</val></magnitude>
     <score><val>-10.75</val><val>-57.1667</val></score>
    </node>
   </node>
  </node>
 </node>
 <node name="Potassium metabolism">
  <rank><val>level 1</val></rank>
  <score><val>-33.6364</val><val>-88.1945</val></score>
  <magnitude><val>22</val><val>24</val></magnitude>
  <node name="Glutathione-regulated potassium-efflux system and associated functions">
   <rank><val>level 2</val></rank>
   <score><val>-4</val></score>
   <magnitude><val>2</val></magnitude>
   <node name="Potassium efflux system KefA protein / Small-conductance mechanosensitive channel">
    <rank><val>function</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-4</val></score>
   </node>
  </node>
  <node name="Potassium homeostasis">
   <score><val>-36.6</val><val>-88.1945</val></score>
   <magnitude><val>20</val><val>24</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Potassium-transporting ATPase C chain (EC 3.6.3.12) (TC 3.A.3.7.1)">
    <rank><val>function</val></rank>
    <score><val>-11</val><val>-74</val></score>
    <magnitude><val>2</val><val>1</val></magnitude>
   </node>
   <node name="Potassium-transporting ATPase B chain (EC 3.6.3.12) (TC 3.A.3.7.1)">
    <rank><val>function</val></rank>
    <magnitude><val>6</val><val>12</val></magnitude>
    <score><val>-49.8333</val><val>-96.8889</val></score>
   </node>
   <node name="Potassium efflux system KefA">
    <rank><val>function</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-4</val></score>
   </node>
   <node name="Potassium-transporting ATPase A chain (EC 3.6.3.12) (TC 3.A.3.7.1)">
    <rank><val>function</val></rank>
    <magnitude><val>7</val><val>11</val></magnitude>
    <score><val>-56</val><val>-80</val></score>
   </node>
   <node name="Potassium efflux system KefA protein">
    <magnitude><val>2</val></magnitude>
    <score><val>-4</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Potassium channel protein">
    <magnitude><val>1</val></magnitude>
    <score><val>-3</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
 </node>
 <node name="Nitrogen Metabolism">
  <rank><val>level 1</val></rank>
  <magnitude><val>32</val><val>207</val></magnitude>
  <score><val>-48.5156</val><val>-75.9076</val></score>
  <node name="Ammonia assimilation">
   <score><val>-46.0263</val><val>-72.6063</val></score>
   <magnitude><val>19</val><val>164</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Ammonium transporter">
    <magnitude><val>3</val><val>52</val></magnitude>
    <score><val>-29.5</val><val>-76.3793</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="[Protein-PII] uridylyltransferase (EC 2.7.7.59)">
    <rank><val>function</val></rank>
    <score><val>-11</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Glutamine synthetase type I (EC 6.3.1.2)">
    <rank><val>function</val></rank>
    <magnitude><val>9</val><val>37</val></magnitude>
    <score><val>-51.2222</val><val>-79.44</val></score>
   </node>
   <node name="Glutamate synthase [NADPH] large chain (EC 1.4.1.13)">
    <magnitude><val>5</val><val>75</val></magnitude>
    <score><val>-61.4</val><val>-66.619</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Ferredoxin-dependent glutamate synthase (EC 1.4.7.1)">
    <rank><val>function</val></rank>
    <score><val>-7</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
  </node>
  <node name="Nitric oxide synthase">
   <magnitude><val>1</val></magnitude>
   <score><val>-18</val></score>
   <rank><val>level 2</val></rank>
   <node name="putative cytochrome P450 hydroxylase">
    <score><val>-18</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Allantoin Utilization">
   <rank><val>level 2</val></rank>
   <score><val>-25</val></score>
   <magnitude><val>3</val></magnitude>
   <node name="Allantoinase (EC 3.5.2.5)">
    <score><val>-43</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Glyoxylate carboligase (EC 4.1.1.47)">
    <score><val>-25</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Glycerate kinase (EC 2.7.1.31)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-7</val></score>
   </node>
  </node>
  <node name="Nitrogen fixation">
   <score><val>-83.6667</val><val>-94.3114</val></score>
   <magnitude><val>6</val><val>40</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Homocitrate synthase (EC 2.3.3.14)">
    <rank><val>function</val></rank>
    <magnitude><val>4</val><val>38</val></magnitude>
    <score><val>-60.5</val><val>-94.0909</val></score>
   </node>
   <node name="Nitrogenase FeMo-cofactor scaffold and assembly protein NifE">
    <rank><val>function</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-130</val></score>
   </node>
   <node name="Nitrogenase FeMo-cofactor synthesis FeS core scaffold and assembly protein NifB">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>2</val></magnitude>
    <score><val>0</val><val>-98.5</val></score>
   </node>
  </node>
  <node name="Denitrification">
   <magnitude><val>1</val><val>3</val></magnitude>
   <score><val>-11</val><val>-11</val></score>
   <rank><val>level 2</val></rank>
   <node name="Nitrous oxide reductase maturation protein NosF (ATPase)">
    <score><val>-11</val><val>-11</val></score>
    <magnitude><val>1</val><val>3</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Nitrate and nitrite ammonification">
   <magnitude><val>2</val></magnitude>
   <score><val>-36</val></score>
   <rank><val>level 2</val></rank>
   <node name="Respiratory nitrate reductase alpha chain (EC 1.7.99.4)">
    <magnitude><val>1</val></magnitude>
    <score><val>-64</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Ferredoxin--nitrite reductase (EC 1.7.7.1)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-8</val></score>
   </node>
  </node>
 </node>
 <node name="Motility and Chemotaxis">
  <rank><val>level 1</val></rank>
  <magnitude><val>52</val><val>13</val></magnitude>
  <score><val>-31.0481</val><val>-39.6538</val></score>
  <node name="Flagellar motility in Prokaryota">
   <rank><val>level 2</val></rank>
   <score><val>-32.5544</val><val>-33.3333</val></score>
   <magnitude><val>46</val><val>6</val></magnitude>
   <node name="Flagellar motility">
    <score><val>-27.1111</val></score>
    <magnitude><val>9</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Flagellin">
     <rank><val>function</val></rank>
     <score><val>-6</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="RNA polymerase sigma-54 factor RpoN">
     <score><val>-65</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Chemotaxis protein CheV (EC 2.7.3.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Flagellar motor switch protein FliN">
     <rank><val>function</val></rank>
     <score><val>-20</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Flagellar biosynthesis protein FlhB">
     <magnitude><val>2</val></magnitude>
     <score><val>-8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Flagellar biosynthesis protein FlhA">
     <rank><val>function</val></rank>
     <score><val>-62</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
   </node>
   <node name="Flagellum in Campylobacter">
    <rank><val>level 3</val></rank>
    <score><val>-57</val></score>
    <magnitude><val>2</val></magnitude>
    <node name="Cell division protein FtsI [Peptidoglycan synthetase] (EC 2.4.1.129)">
     <magnitude><val>2</val></magnitude>
     <score><val>-57</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Flagellum">
    <rank><val>level 3</val></rank>
    <magnitude><val>14</val><val>1</val></magnitude>
    <score><val>-29.0714</val><val>-6</val></score>
    <node name="RNA polymerase sigma-54 factor RpoN">
     <score><val>-65</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Flagellar biosynthesis protein FlhB">
     <rank><val>function</val></rank>
     <score><val>-8</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Flagellar biosynthesis protein FliP">
     <score><val>-33</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Flagellar biosynthesis protein FlhA">
     <magnitude><val>2</val></magnitude>
     <score><val>-62</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="RNA polymerase sigma factor RpoD">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-41</val></score>
    </node>
    <node name="Flagellar hook-associated protein FlgL">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-9</val></score>
    </node>
    <node name="Flagellar hook protein FlgE">
     <magnitude><val>1</val><val>1</val></magnitude>
     <score><val>-8</val><val>-6</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Flagellar motor switch protein FliN">
     <rank><val>function</val></rank>
     <score><val>-20</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Flagellar hook-length control protein FliK">
     <rank><val>function</val></rank>
     <magnitude><val>3</val></magnitude>
     <score><val>-16.6667</val></score>
    </node>
   </node>
   <node name="Archaeal Flagellum">
    <score><val>-34.881</val><val>-38.8</val></score>
    <magnitude><val>21</val><val>5</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Flagella-related protein FlaJ">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>2</val></magnitude>
     <score><val>-38.5</val><val>-34</val></score>
    </node>
    <node name="Flagellin FlaB">
     <rank><val>function</val></rank>
     <score><val>-6</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Flagella-related protein FlaG">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-25</val></score>
    </node>
    <node name="Flagella-related protein FlaC">
     <rank><val>function</val></rank>
     <score><val>-11.6667</val></score>
     <magnitude><val>3</val></magnitude>
    </node>
    <node name="Flagella-related protein FlaI">
     <magnitude><val>5</val><val>1</val></magnitude>
     <score><val>-65.8</val><val>-26</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Flagellin FlaB2">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-6</val></score>
    </node>
    <node name="Flagella-related protein FlaD">
     <magnitude><val>1</val><val>2</val></magnitude>
     <score><val>-50</val><val>-50</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Bacterial Chemotaxis">
   <rank><val>level 2</val></rank>
   <score><val>-19.5</val><val>-45.0714</val></score>
   <magnitude><val>6</val><val>7</val></magnitude>
   <node name="Flagellar motor switch protein FliN">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-20</val></score>
   </node>
   <node name="Dipeptide-binding ABC transporter, periplasmic substrate-binding component (TC 3.A.1.5.2)">
    <score><val>-38</val><val>-49.75</val></score>
    <magnitude><val>1</val><val>6</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Chemotaxis response regulator protein-glutamate methylesterase CheB (EC 3.1.1.61)">
    <rank><val>function</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-17</val></score>
   </node>
   <node name="Chemotaxis protein CheV (EC 2.7.3.-)">
    <rank><val>function</val></rank>
    <score><val>-7</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Chemotaxis protein methyltransferase CheR (EC 2.1.1.80)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val><val>1</val></magnitude>
    <score><val>-18</val><val>-17</val></score>
   </node>
  </node>
 </node>
 <node name="Clustering-based subsystems">
  <rank><val>level 1</val></rank>
  <score><val>-36.3378</val><val>-65.1919</val></score>
  <magnitude><val>1159</val><val>4324</val></magnitude>
  <node name="CBSS-316057.3.peg.3521">
   <rank><val>level 2</val></rank>
   <magnitude><val>8</val><val>31</val></magnitude>
   <score><val>-32.1875</val><val>-38.8909</val></score>
   <node name="Thiol peroxidase, Bcp-type (EC 1.11.1.15)">
    <rank><val>function</val></rank>
    <magnitude><val>6</val><val>28</val></magnitude>
    <score><val>-33.25</val><val>-40.6471</val></score>
   </node>
   <node name="Large exoproteins involved in heme utilization or adhesion">
    <rank><val>function</val></rank>
    <score><val>-29</val><val>-22.5</val></score>
    <magnitude><val>2</val><val>3</val></magnitude>
   </node>
  </node>
  <node name="Lipopolysaccharide assembly cluster">
   <score><val>-47.5</val></score>
   <magnitude><val>2</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="RNA polymerase sigma-54 factor RpoN">
    <magnitude><val>1</val></magnitude>
    <score><val>-65</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="FIG000506: Predicted P-loop-containing kinase">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-30</val></score>
   </node>
  </node>
  <node name="Ribosomal Protein L28P relates to a set of uncharacterized proteins">
   <score><val>-18.2</val><val>-42</val></score>
   <magnitude><val>5</val><val>31</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="A Gram-positive cluster that relates ribosomal protein L28P to a set of uncharacterized proteins">
    <score><val>-18.2</val><val>-42</val></score>
    <magnitude><val>5</val><val>31</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Ribulose-phosphate 3-epimerase (EC 5.1.3.1)">
     <magnitude><val>3</val><val>31</val></magnitude>
     <score><val>-17.3333</val><val>-42</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydroxyacetone kinase family protein">
     <rank><val>function</val></rank>
     <score><val>-19.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Staphylococcus aureus hypothetical repetitive gene loci">
   <rank><val>level 2</val></rank>
   <magnitude><val>28</val><val>65</val></magnitude>
   <score><val>-16.8129</val><val>-57.4796</val></score>
   <node name="Diphosphomevalonate decarboxylase (EC 4.1.1.33)">
    <rank><val>function</val></rank>
    <score><val>-16.1667</val><val>-46.375</val></score>
    <magnitude><val>8</val><val>22</val></magnitude>
   </node>
   <node name="Alcohol dehydrogenase (EC 1.1.1.1)">
    <score><val>-21.5</val><val>-68.2</val></score>
    <magnitude><val>12</val><val>34</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Deoxyguanosinetriphosphate triphosphohydrolase (EC 3.1.5.1)">
    <rank><val>function</val></rank>
    <magnitude><val>8</val><val>9</val></magnitude>
    <score><val>-10.4286</val><val>-44.125</val></score>
   </node>
  </node>
  <node name="Choline bitartrate degradation, putative">
   <magnitude><val>2</val><val>34</val></magnitude>
   <score><val>-19</val><val>-76.3913</val></score>
   <rank><val>level 2</val></rank>
   <node name="CBSS-344610.3.peg.2335">
    <rank><val>level 3</val></rank>
    <magnitude><val>2</val><val>34</val></magnitude>
    <score><val>-19</val><val>-76.3913</val></score>
    <node name="3-isopropylmalate dehydrogenase (EC 1.1.1.85)">
     <magnitude><val>2</val><val>34</val></magnitude>
     <score><val>-19</val><val>-76.3913</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="CBSS-370552.3.peg.1240">
   <score><val>-10</val></score>
   <magnitude><val>1</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Hypothetical protein DUF194, DegV family">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-10</val></score>
   </node>
  </node>
  <node name="Translation">
   <magnitude><val>14</val><val>28</val></magnitude>
   <score><val>-58.5259</val><val>-61.1245</val></score>
   <rank><val>level 2</val></rank>
   <node name="CBSS-326442.4.peg.1852">
    <rank><val>level 3</val></rank>
    <score><val>-58.5259</val><val>-61.1245</val></score>
    <magnitude><val>14</val><val>28</val></magnitude>
    <node name="Seryl-tRNA synthetase (EC 6.1.1.11)">
     <score><val>-63.3636</val><val>-81.4286</val></score>
     <magnitude><val>12</val><val>17</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU m5C1962 methyltransferase RlmI">
     <rank><val>function</val></rank>
     <score><val>-11</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Cytosine deaminase (EC 3.5.4.1)">
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-34</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA-3-methyladenine glycosylase (EC 3.2.2.20)">
     <rank><val>function</val></rank>
     <score><val>-48</val><val>-28.8</val></score>
     <magnitude><val>1</val><val>9</val></magnitude>
    </node>
   </node>
  </node>
  <node name="LSU ribosomal proteins cluster">
   <rank><val>level 2</val></rank>
   <magnitude><val>2</val></magnitude>
   <score><val>-20.5</val></score>
   <node name="LSU ribosomal protein L7/L12 (P1/P2)">
    <rank><val>function</val></rank>
    <score><val>-25</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Transcription antitermination protein NusG">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-16</val></score>
   </node>
  </node>
  <node name="LMPTP YwlE cluster">
   <rank><val>level 2</val></rank>
   <score><val>-59.6765</val><val>-72.4062</val></score>
   <magnitude><val>17</val><val>96</val></magnitude>
   <node name="Peptide chain release factor 1">
    <score><val>-36</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Serine hydroxymethyltransferase (EC 2.1.2.1)">
    <magnitude><val>11</val><val>67</val></magnitude>
    <score><val>-76</val><val>-79.3077</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="COG0009 Sua5 subfamily, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
    <score><val>-28.5</val><val>-56.4615</val></score>
    <magnitude><val>5</val><val>29</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Putative Isoquinoline 1-oxidoreductase subunit">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val></magnitude>
   <score><val>-13</val></score>
   <node name="CBSS-314267.3.peg.390">
    <score><val>-13</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Isoquinoline 1-oxidoreductase beta subunit (EC 1.3.99.16)">
     <score><val>-13</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Carotenoid biosynthesis">
   <score><val>-27.3333</val></score>
   <magnitude><val>3</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="CBSS-320388.3.peg.3759">
    <rank><val>level 3</val></rank>
    <score><val>-27.3333</val></score>
    <magnitude><val>3</val></magnitude>
    <node name="Squalene--hopene cyclase (EC 5.4.99.17)">
     <magnitude><val>1</val></magnitude>
     <score><val>-34</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phytoene synthase (EC 2.5.1.32)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-24</val></score>
    </node>
   </node>
  </node>
  <node name="Disulphide related cluster">
   <rank><val>level 2</val></rank>
   <magnitude><val></val><val>5</val></magnitude>
   <score><val>0</val><val>-4</val></score>
   <node name="Disulfide bond regulator">
    <score><val>0</val><val>-4</val></score>
    <magnitude><val></val><val>5</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-630.2.peg.3360">
   <score><val>0</val><val>-60.5455</val></score>
   <magnitude><val></val><val>14</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Pyrroline-5-carboxylate reductase (EC 1.5.1.2)">
    <magnitude><val></val><val>14</val></magnitude>
    <score><val>0</val><val>-60.5455</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Fatty acid metabolic cluster">
   <magnitude><val>66</val><val>476</val></magnitude>
   <score><val>-32.1036</val><val>-66.6972</val></score>
   <rank><val>level 2</val></rank>
   <node name="CBSS-246196.1.peg.364">
    <magnitude><val>34</val><val>269</val></magnitude>
    <score><val>-42.2894</val><val>-67.5636</val></score>
    <rank><val>level 3</val></rank>
    <node name="3-ketoacyl-CoA thiolase (EC 2.3.1.16)">
     <score><val>-49.92</val><val>-76.8312</val></score>
     <magnitude><val>27</val><val>112</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="3-oxoacyl-[acyl-carrier protein] reductase (EC 1.1.1.100)">
     <score><val>-12.8571</val><val>-60.9524</val></score>
     <magnitude><val>7</val><val>157</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="COG1399">
    <rank><val>level 3</val></rank>
    <magnitude><val>32</val><val>207</val></magnitude>
    <score><val>-21.2812</val><val>-65.5713</val></score>
    <node name="3-oxoacyl-[acyl-carrier protein] reductase (EC 1.1.1.100)">
     <rank><val>function</val></rank>
     <score><val>-12.8571</val><val>-60.9524</val></score>
     <magnitude><val>7</val><val>157</val></magnitude>
    </node>
    <node name="3-oxoacyl-[acyl-carrier-protein] synthase, KASIII (EC 2.3.1.41)">
     <magnitude><val>1</val></magnitude>
     <score><val>-24</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="3-oxoacyl-[acyl-carrier-protein] synthase, KASII (EC 2.3.1.41)">
     <score><val>-35.5</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP-binding protein Era">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
    </node>
    <node name="Thymidylate kinase (EC 2.7.4.9)">
     <rank><val>function</val></rank>
     <score><val>-25.4286</val><val>-88.0606</val></score>
     <magnitude><val>7</val><val>45</val></magnitude>
    </node>
    <node name="Chromosome partition protein smc">
     <magnitude><val>13</val><val>5</val></magnitude>
     <score><val>-21.6923</val><val>-8.2</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Malonyl CoA-acyl carrier protein transacylase (EC 2.3.1.39)">
     <magnitude><val>1</val></magnitude>
     <score><val>-24</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="CBSS-138119.3.peg.2719">
   <score><val>-31.4325</val><val>-92.871</val></score>
   <magnitude><val>18</val><val>46</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="tRNA pseudouridine synthase B (EC 4.2.1.70)">
    <score><val>-9.66667</val></score>
    <magnitude><val>3</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Translation initiation factor 2">
    <score><val>-35.7857</val><val>-92.871</val></score>
    <magnitude><val>15</val><val>46</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-288681.3.peg.1039">
   <rank><val>level 2</val></rank>
   <score><val>-66.5</val><val>-63.25</val></score>
   <magnitude><val>6</val><val>12</val></magnitude>
   <node name="Superfamily II DNA/RNA helicases, SNF2 family">
    <score><val>-66.5</val><val>-63.25</val></score>
    <magnitude><val>6</val><val>12</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Cell wall related cluster">
   <rank><val>level 2</val></rank>
   <score><val>-32</val></score>
   <magnitude><val>1</val></magnitude>
   <node name="Asparagine synthetase [glutamine-hydrolyzing] (EC 6.3.5.4) AsnH">
    <score><val>-32</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-296591.1.peg.2330">
   <rank><val>level 2</val></rank>
   <score><val>-35.076</val><val>-53.9475</val></score>
   <magnitude><val>24</val><val>66</val></magnitude>
   <node name="NAD-dependent epimerase/dehydratase">
    <score><val>-48.5</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="dTDP-glucose 4,6-dehydratase (EC 4.2.1.46)">
    <magnitude><val>3</val><val>6</val></magnitude>
    <score><val>-35.6667</val><val>-53.1667</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="NAD-dependent epimerase/dehydratase family protein">
    <rank><val>function</val></rank>
    <score><val>-11</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="UDP-glucose 4-epimerase (EC 5.1.3.2)">
    <magnitude><val>18</val><val>60</val></magnitude>
    <score><val>-34.8235</val><val>-54.0256</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="USS-DB-7">
   <rank><val>level 2</val></rank>
   <score><val>-43.5</val></score>
   <magnitude><val>2</val></magnitude>
   <node name="ClpB protein">
    <magnitude><val>2</val></magnitude>
    <score><val>-43.5</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-393121.3.peg.1913">
   <magnitude><val>20</val><val>87</val></magnitude>
   <score><val>-30</val><val>-78.4287</val></score>
   <rank><val>level 2</val></rank>
   <node name="ATP-dependent DNA helicase UvrD/PcrA">
    <rank><val>function</val></rank>
    <magnitude><val>8</val><val>11</val></magnitude>
    <score><val>-23.375</val><val>-108.5</val></score>
   </node>
   <node name="DNA ligase (EC 6.5.1.2)">
    <score><val>-43</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="ATP-dependent DNA helicase pcrA (EC 3.6.1.-)">
    <score><val>-8.33333</val><val>-81.625</val></score>
    <magnitude><val>3</val><val>43</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="(S)-3-O-geranylgeranylglyceryl phosphate synthase">
    <rank><val>function</val></rank>
    <score><val>-43.1429</val><val>-64.24</val></score>
    <magnitude><val>7</val><val>33</val></magnitude>
   </node>
  </node>
  <node name="CBSS-312309.3.peg.1965">
   <score><val>-53.2769</val><val>-63.5333</val></score>
   <magnitude><val>13</val><val>31</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Methionine aminopeptidase (EC 3.4.11.18)">
    <score><val>-56.8</val><val>-63.5333</val></score>
    <magnitude><val>12</val><val>31</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="[Protein-PII] uridylyltransferase (EC 2.7.7.59)">
    <magnitude><val>1</val></magnitude>
    <score><val>-11</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="NusA-TFII Cluster">
   <magnitude><val>17</val><val>47</val></magnitude>
   <score><val>-39.3403</val><val>-94.9589</val></score>
   <rank><val>level 2</val></rank>
   <node name="Translation initiation factor 2">
    <magnitude><val>15</val><val>46</val></magnitude>
    <score><val>-35.7857</val><val>-92.871</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Transcription termination protein NusA">
    <rank><val>function</val></rank>
    <score><val>-66</val><val>-191</val></score>
    <magnitude><val>2</val><val>1</val></magnitude>
   </node>
  </node>
  <node name="Chromosome Replication">
   <rank><val>level 2</val></rank>
   <magnitude><val>3</val></magnitude>
   <score><val>-2</val></score>
   <node name="SeqA and Co-occurring Genes">
    <rank><val>level 3</val></rank>
    <magnitude><val>3</val></magnitude>
    <score><val>-2</val></score>
    <node name="Phosphoglucomutase (EC 5.4.2.2)">
     <rank><val>function</val></rank>
     <score><val>-2</val></score>
     <magnitude><val>3</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Pyruvate kinase associated cluster">
   <magnitude><val>2</val><val>39</val></magnitude>
   <score><val>-11.5</val><val>-80.92</val></score>
   <rank><val>level 2</val></rank>
   <node name="CBSS-288000.5.peg.1793">
    <rank><val>level 3</val></rank>
    <score><val>-11.5</val><val>-80.92</val></score>
    <magnitude><val>2</val><val>39</val></magnitude>
    <node name="Pyruvate kinase (EC 2.7.1.40)">
     <score><val>-11.5</val><val>-80.92</val></score>
     <magnitude><val>2</val><val>39</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="ClpAS cluster">
   <rank><val>level 2</val></rank>
   <score><val>-83.5</val></score>
   <magnitude><val>2</val></magnitude>
   <node name="ATP-dependent Clp protease adaptor protein ClpS">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-4</val></score>
   </node>
   <node name="ATP-dependent Clp protease ATP-binding subunit ClpA">
    <score><val>-163</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-228410.1.peg.134">
   <rank><val>level 2</val></rank>
   <score><val>-27.5</val></score>
   <magnitude><val>2</val></magnitude>
   <node name="Hydroxyacylglutathione hydrolase (EC 3.1.2.6)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-53</val></score>
   </node>
   <node name="DNA polymerase III epsilon subunit (EC 2.7.7.7)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-2</val></score>
   </node>
  </node>
  <node name="Bacterial Cell Division">
   <score><val>-52.0574</val><val>-67.1071</val></score>
   <magnitude><val>45</val><val>116</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Cell division protein FtsI [Peptidoglycan synthetase] (EC 2.4.1.129)">
    <score><val>-57</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Cell division trigger factor (EC 5.2.1.8)">
    <rank><val>function</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-3</val></score>
   </node>
   <node name="Cell division protein FtsJ">
    <magnitude><val>4</val><val>3</val></magnitude>
    <score><val>-13.3333</val><val>-20</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Septum site-determining protein MinD">
    <magnitude><val></val><val>29</val></magnitude>
    <score><val>0</val><val>-68.2</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="GTP-binding protein Era">
    <rank><val>function</val></rank>
    <score><val>-12</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Cell division protein FtsX">
    <magnitude><val>1</val></magnitude>
    <score><val>-21</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Cell division protein FtsH (EC 3.4.24.-)">
    <rank><val>function</val></rank>
    <magnitude><val>10</val><val>36</val></magnitude>
    <score><val>-112.6</val><val>-97.9167</val></score>
   </node>
   <node name="Rod shape-determining protein MreB">
    <rank><val>function</val></rank>
    <score><val>-44</val></score>
    <magnitude><val>3</val></magnitude>
   </node>
   <node name="Septum site-determining protein MinC">
    <rank><val>function</val></rank>
    <score><val>-32</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Cytoplasmic axial filament protein CafA and Ribonuclease G (EC 3.1.4.-)">
    <score><val>-27</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Signal recognition particle receptor protein FtsY (=alpha subunit) (TC 3.A.5.1.1)">
    <rank><val>function</val></rank>
    <score><val>-51.25</val><val>-62.16</val></score>
    <magnitude><val>13</val><val>31</val></magnitude>
   </node>
   <node name="rRNA small subunit methyltransferase H">
    <score><val>-35</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Rod shape-determining protein MreC">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-6</val></score>
   </node>
   <node name="Cell division protein FtsZ (EC 3.4.24.-)">
    <rank><val>function</val></rank>
    <magnitude><val>3</val><val>17</val></magnitude>
    <score><val>-12</val><val>-17.3333</val></score>
   </node>
   <node name="Cell division protein MraZ">
    <rank><val>function</val></rank>
    <score><val>-41</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
  </node>
  <node name="Ribonucleotide reductase cluster">
   <magnitude><val>4</val></magnitude>
   <score><val>-70.5</val></score>
   <rank><val>level 2</val></rank>
   <node name="Ribonucleotide reductase of class Ia (aerobic), beta subunit (EC 1.17.4.1)">
    <magnitude><val>1</val></magnitude>
    <score><val>-32</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Ribonucleotide reductase of class Ia (aerobic), alpha subunit (EC 1.17.4.1)">
    <rank><val>function</val></rank>
    <score><val>-83.3333</val></score>
    <magnitude><val>3</val></magnitude>
   </node>
  </node>
  <node name="EC49-61">
   <score><val>-20.75</val><val>-64.853</val></score>
   <magnitude><val>4</val><val>34</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="4-hydroxythreonine-4-phosphate dehydrogenase (EC 1.1.1.262)">
    <score><val>-27</val><val>-67.6667</val></score>
    <magnitude><val>3</val><val>27</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Dimethyladenosine transferase (EC 2.1.1.-)">
    <rank><val>function</val></rank>
    <score><val>-2</val><val>-54</val></score>
    <magnitude><val>1</val><val>7</val></magnitude>
   </node>
  </node>
  <node name="ClpS  and metallo-protease cluster in Cyanos ">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val></magnitude>
   <score><val>-4</val></score>
   <node name="ATP-dependent Clp protease adaptor protein ClpS">
    <rank><val>function</val></rank>
    <score><val>-4</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
  </node>
  <node name="CBSS-320372.3.peg.6046">
   <rank><val>level 2</val></rank>
   <score><val>-42.1</val><val>-75.0133</val></score>
   <magnitude><val>10</val><val>75</val></magnitude>
   <node name="Putative Holliday junction resolvase (EC 3.1.-.-)">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>1</val></magnitude>
    <score><val>0</val><val>-20</val></score>
   </node>
   <node name="Ornithine carbamoyltransferase (EC 2.1.3.3)">
    <magnitude><val></val><val>50</val></magnitude>
    <score><val>0</val><val>-87</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Aspartate carbamoyltransferase (EC 2.1.3.2)">
    <rank><val>function</val></rank>
    <score><val>-42.1</val><val>-52.3333</val></score>
    <magnitude><val>10</val><val>24</val></magnitude>
   </node>
  </node>
  <node name="CBSS-193567.1.peg.90">
   <rank><val>level 2</val></rank>
   <score><val>0</val><val>-60.5455</val></score>
   <magnitude><val></val><val>14</val></magnitude>
   <node name="Pyrroline-5-carboxylate reductase (EC 1.5.1.2)">
    <score><val>0</val><val>-60.5455</val></score>
    <magnitude><val></val><val>14</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Hypothetical lipase related to Phosphatidate metabolism">
   <magnitude><val>8</val><val>22</val></magnitude>
   <score><val>-18.4286</val><val>-39.1007</val></score>
   <rank><val>level 2</val></rank>
   <node name="CBSS-316407.3.peg.1371">
    <score><val>-18.4286</val><val>-39.1007</val></score>
    <magnitude><val>8</val><val>22</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Phosphatidate cytidylyltransferase (EC 2.7.7.41)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>3</val></magnitude>
     <score><val>0</val><val>-61</val></score>
    </node>
    <node name="CDP-diacylglycerol--glycerol-3-phosphate 3-phosphatidyltransferase (EC 2.7.8.5)">
     <rank><val>function</val></rank>
     <score><val>-18.4286</val><val>-35.6429</val></score>
     <magnitude><val>8</val><val>19</val></magnitude>
    </node>
   </node>
  </node>
  <node name="tRNA sulfuration">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val></magnitude>
   <score><val>-7</val></score>
   <node name="CBSS-89187.3.peg.2957">
    <magnitude><val>1</val></magnitude>
    <score><val>-7</val></score>
    <rank><val>level 3</val></rank>
    <node name="Methionyl-tRNA formyltransferase (EC 2.1.2.9)">
     <score><val>-7</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Conjugative transfer related cluster">
   <rank><val>level 2</val></rank>
   <score><val>-9</val></score>
   <magnitude><val>4</val></magnitude>
   <node name="DNA topoisomerase III (EC 5.99.1.2)">
    <magnitude><val>1</val></magnitude>
    <score><val>-11</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="IncW plasmid conjugative relaxase protein TrwC (TraI homolog)">
    <score><val>-3</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="IncF plasmid conjugative transfer protein TraD">
    <rank><val>function</val></rank>
    <score><val>-11</val></score>
    <magnitude><val>2</val></magnitude>
   </node>
  </node>
  <node name="Hypothetical protein possible functionally linked with Alanyl-tRNA synthetase">
   <rank><val>level 2</val></rank>
   <magnitude><val>12</val><val>61</val></magnitude>
   <score><val>-48.9091</val><val>-87.7275</val></score>
   <node name="CBSS-257314.1.peg.488">
    <score><val>-48.9091</val><val>-87.7275</val></score>
    <magnitude><val>12</val><val>61</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Putative Holliday junction resolvase">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-20</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Alanyl-tRNA synthetase (EC 6.1.1.7)">
     <magnitude><val>12</val><val>59</val></magnitude>
     <score><val>-48.9091</val><val>-90.0233</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Putative Holliday junction resolvase (EC 3.1.-.-)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-20</val></score>
    </node>
   </node>
  </node>
  <node name="Ycd cluster ( putative 2-hydroxyacid dehydrogenase )">
   <rank><val>level 2</val></rank>
   <magnitude><val></val><val>1</val></magnitude>
   <score><val>0</val><val>-7</val></score>
   <node name="Putative hydrolase YcdX (EC 3.1.-.-)">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>1</val></magnitude>
    <score><val>0</val><val>-7</val></score>
   </node>
  </node>
  <node name="Cell Division">
   <rank><val>level 2</val></rank>
   <magnitude><val>33</val><val>198</val></magnitude>
   <score><val>-33.8763</val><val>-64.8573</val></score>
   <node name="Cell Division Cluster">
    <magnitude><val>11</val><val>111</val></magnitude>
    <score><val>-44.5455</val><val>-69.4786</val></score>
    <rank><val>level 3</val></rank>
    <node name="Cell division protein FtsZ (EC 3.4.24.-)">
     <magnitude><val>3</val><val>17</val></magnitude>
     <score><val>-12</val><val>-17.3333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Isoleucyl-tRNA synthetase (EC 6.1.1.5)">
     <magnitude><val>8</val><val>94</val></magnitude>
     <score><val>-56.75</val><val>-78.9091</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="CBSS-393130.3.peg.794">
    <rank><val>level 3</val></rank>
    <magnitude><val>22</val><val>87</val></magnitude>
    <score><val>-28.5417</val><val>-58.9612</val></score>
    <node name="Cysteine desulfurase (EC 2.8.1.7)">
     <rank><val>function</val></rank>
     <score><val>-29.4167</val><val>-52.625</val></score>
     <magnitude><val>15</val><val>38</val></magnitude>
    </node>
    <node name="Thiamine biosynthesis protein ThiI">
     <rank><val>function</val></rank>
     <score><val>-26.6667</val><val>-63.875</val></score>
     <magnitude><val>7</val><val>49</val></magnitude>
    </node>
   </node>
  </node>
  <node name="RNA modification and chromosome partitioning cluster">
   <magnitude><val>35</val><val>43</val></magnitude>
   <score><val>-50.4226</val><val>-39.069</val></score>
   <rank><val>level 2</val></rank>
   <node name="Chromosome (plasmid) partitioning protein ParB">
    <magnitude><val>7</val></magnitude>
    <score><val>-46.4286</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Chromosome (plasmid) partitioning protein ParA">
    <magnitude><val>28</val><val>43</val></magnitude>
    <score><val>-51.4211</val><val>-39.069</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Nucleotidyl-phosphate metabolic cluster">
   <rank><val>level 2</val></rank>
   <magnitude><val>10</val><val>11</val></magnitude>
   <score><val>-33.85</val><val>-40.2727</val></score>
   <node name="CBSS-222523.1.peg.1311">
    <rank><val>level 3</val></rank>
    <magnitude><val>10</val><val>11</val></magnitude>
    <score><val>-33.85</val><val>-40.2727</val></score>
    <node name="NAD kinase (EC 2.7.1.23)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>11</val></magnitude>
     <score><val>-34.5</val><val>-40.2727</val></score>
    </node>
    <node name="Adenylate cyclase">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-28</val></score>
    </node>
   </node>
  </node>
  <node name="CBSS-393124.3.peg.2657">
   <score><val>-39</val><val>-41.4</val></score>
   <magnitude><val>1</val><val>17</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Methylated-DNA--protein-cysteine methyltransferase (EC 2.1.1.63)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val><val>17</val></magnitude>
    <score><val>-39</val><val>-41.4</val></score>
   </node>
  </node>
  <node name="Lysine, threonine, methionine, and cysteine">
   <score><val>-26.1324</val><val>-51.4551</val></score>
   <magnitude><val>17</val><val>39</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="CBSS-84588.1.peg.1247">
    <rank><val>level 3</val></rank>
    <magnitude><val>17</val><val>39</val></magnitude>
    <score><val>-26.1324</val><val>-51.4551</val></score>
    <node name="D-alanyl-D-alanine carboxypeptidase (EC 3.4.16.4)">
     <rank><val>function</val></rank>
     <score><val>-1.5</val><val>-7</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7)">
     <rank><val>function</val></rank>
     <magnitude><val>15</val><val>38</val></magnitude>
     <score><val>-29.4167</val><val>-52.625</val></score>
    </node>
   </node>
  </node>
  <node name="CBSS-176279.3.peg.868">
   <rank><val>level 2</val></rank>
   <magnitude><val>2</val></magnitude>
   <score><val>-15.5</val></score>
   <node name="COG0536: GTP-binding protein Obg">
    <magnitude><val>2</val></magnitude>
    <score><val>-15.5</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Oxidative stress">
   <rank><val>level 2</val></rank>
   <score><val>-30.1667</val><val>-9.69606</val></score>
   <magnitude><val>11</val><val>17</val></magnitude>
   <node name="CoA-disulfide reductase (EC 1.8.1.14) containing cluster">
    <magnitude><val>11</val><val>17</val></magnitude>
    <score><val>-30.1667</val><val>-9.69606</val></score>
    <rank><val>level 3</val></rank>
    <node name="Disulfide bond regulator">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-4</val></score>
     <magnitude><val></val><val>5</val></magnitude>
    </node>
    <node name="CoA-disulfide reductase (EC 1.8.1.14)">
     <score><val>0</val><val>-4</val></score>
     <magnitude><val></val><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="PaaD-like protein (DUF59) involved in Fe-S cluster assembly">
     <rank><val>function</val></rank>
     <score><val>-30.1667</val><val>-17.8333</val></score>
     <magnitude><val>11</val><val>7</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Lysine Biosynthesis">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val></magnitude>
   <score><val>-42</val></score>
   <node name="A Glutathione-dependent Thiol Reductase Associated with a Step in Lysine Biosynthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-42</val></score>
    <node name="N-succinyl-L,L-diaminopimelate desuccinylase (EC 3.5.1.18)">
     <magnitude><val>1</val></magnitude>
     <score><val>-42</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="CBSS-316273.3.peg.2709">
   <score><val>-66.5</val><val>-63.25</val></score>
   <magnitude><val>6</val><val>12</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Superfamily II DNA/RNA helicases, SNF2 family">
    <magnitude><val>6</val><val>12</val></magnitude>
    <score><val>-66.5</val><val>-63.25</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-314269.3.peg.1840">
   <rank><val>level 2</val></rank>
   <magnitude><val>13</val><val>32</val></magnitude>
   <score><val>-35.6154</val><val>-57.4643</val></score>
   <node name="Carbon monoxide dehydrogenase medium chain (EC 1.2.99.2)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val><val>24</val></magnitude>
    <score><val>-23</val><val>-67.1429</val></score>
   </node>
   <node name="Carbon monoxide dehydrogenase small chain (EC 1.2.99.2)">
    <rank><val>function</val></rank>
    <score><val>-38</val><val>-28.4286</val></score>
    <magnitude><val>4</val><val>8</val></magnitude>
   </node>
   <node name="Carbon monoxide dehydrogenase large chain (EC 1.2.99.2)">
    <rank><val>function</val></rank>
    <magnitude><val>3</val></magnitude>
    <score><val>-52.6667</val></score>
   </node>
   <node name="carbon monoxide dehydrogenase E protein">
    <rank><val>function</val></rank>
    <score><val>-42.5</val></score>
    <magnitude><val>2</val></magnitude>
   </node>
   <node name="Carbon monoxide dehydrogenase large chain (EC 1.2.99.2) parolog without usual motifs">
    <magnitude><val>2</val></magnitude>
    <score><val>-8.5</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="carbon monoxide dehydrogenase D protein">
    <rank><val>function</val></rank>
    <score><val>-28</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
  </node>
  <node name="Conserved gene cluster associated with Met-tRNA formyltransferase">
   <score><val>-33.2179</val><val>-56.8117</val></score>
   <magnitude><val>39</val><val>151</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Rossmann fold nucleotide-binding protein Smf possibly involved in DNA uptake">
    <rank><val>function</val></rank>
    <magnitude><val>3</val><val>3</val></magnitude>
    <score><val>-30.5</val><val>-23</val></score>
   </node>
   <node name="Ribosomal-protein-S18p-alanine acetyltransferase (EC 2.3.1.-)">
    <score><val>0</val><val>-52.375</val></score>
    <magnitude><val></val><val>9</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Ribonuclease HII (EC 3.1.26.4)">
    <rank><val>function</val></rank>
    <magnitude><val>4</val><val>26</val></magnitude>
    <score><val>-11</val><val>-35.0714</val></score>
   </node>
   <node name="YgjD/Kae1/Qri7 family, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
    <score><val>-43.4</val><val>-85.1667</val></score>
    <magnitude><val>10</val><val>45</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Methionyl-tRNA formyltransferase (EC 2.1.2.9)">
    <rank><val>function</val></rank>
    <score><val>-7</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="50S ribosomal subunit maturation GTPase RbgA (B. subtilis YlqF)">
    <score><val>-11</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Zn-dependent hydrolase, RNA-metabolising">
    <rank><val>function</val></rank>
    <magnitude><val>3</val></magnitude>
    <score><val>-49.3333</val></score>
   </node>
   <node name="Ribulose-phosphate 3-epimerase (EC 5.1.3.1)">
    <magnitude><val>3</val><val>31</val></magnitude>
    <score><val>-17.3333</val><val>-42</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Cytochrome d ubiquinol oxidase subunit II (EC 1.10.3.-)">
    <magnitude><val>1</val></magnitude>
    <score><val>-53</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="DNA topoisomerase I (EC 5.99.1.2)">
    <magnitude><val>13</val><val>37</val></magnitude>
    <score><val>-35</val><val>-53.8333</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-176280.1.peg.1561">
   <magnitude><val>2</val><val>50</val></magnitude>
   <score><val>-10.5</val><val>-45.453</val></score>
   <rank><val>level 2</val></rank>
   <node name="Acyl-CoA hydrolase (EC 3.1.2.20)">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>20</val></magnitude>
    <score><val>0</val><val>-56.0909</val></score>
   </node>
   <node name="ThiJ/PfpI family protein">
    <score><val>-10.5</val><val>-39.1667</val></score>
    <magnitude><val>2</val><val>29</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="A/G-specific adenine glycosylase (EC 3.2.2.-)">
    <score><val>0</val><val>-15</val></score>
    <magnitude><val></val><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-176299.4.peg.1996B">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val></magnitude>
   <score><val>-163</val></score>
   <node name="ATP-dependent Clp protease ATP-binding subunit ClpA">
    <score><val>-163</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Protein export?">
   <rank><val>level 2</val></rank>
   <magnitude><val>7</val><val>1</val></magnitude>
   <score><val>-53.8572</val><val>-5</val></score>
   <node name="CBSS-393121.3.peg.2760">
    <rank><val>level 3</val></rank>
    <magnitude><val>7</val><val>1</val></magnitude>
    <score><val>-53.8572</val><val>-5</val></score>
    <node name="Protein export cytoplasm protein SecA ATPase RNA helicase (TC 3.A.5.1.1)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val></magnitude>
     <score><val>-94.6667</val></score>
    </node>
    <node name="Peptide chain release factor 2">
     <rank><val>function</val></rank>
     <score><val>-11</val><val>-5</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
    <node name="Cell division protein FtsX">
     <magnitude><val>1</val></magnitude>
     <score><val>-21</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="hypothetical protein BH3604">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-50</val></score>
    </node>
   </node>
  </node>
  <node name="CTP synthase (EC 6.3.4.2) cluster">
   <rank><val>level 2</val></rank>
   <score><val>-39.9375</val><val>-65.4151</val></score>
   <magnitude><val>16</val><val>79</val></magnitude>
   <node name="Inner membrane protein translocase component YidC, long form">
    <magnitude><val>1</val></magnitude>
    <score><val>-9</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="CTP synthase (EC 6.3.4.2)">
    <score><val>-42</val><val>-65.4151</val></score>
    <magnitude><val>15</val><val>79</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Biosynthesis of galactoglycans and related lipopolysacharides">
   <rank><val>level 2</val></rank>
   <magnitude><val>37</val><val>140</val></magnitude>
   <score><val>-15.5135</val><val>-50.1894</val></score>
   <node name="CBSS-258594.1.peg.3339">
    <magnitude><val>26</val><val>106</val></magnitude>
    <score><val>-15.8846</val><val>-46.3896</val></score>
    <rank><val>level 3</val></rank>
    <node name="Glycosyltransferase">
     <rank><val>function</val></rank>
     <score><val>-15.8846</val><val>-46.3896</val></score>
     <magnitude><val>26</val><val>106</val></magnitude>
    </node>
   </node>
   <node name="CBSS-376686.6.peg.291">
    <rank><val>level 3</val></rank>
    <magnitude><val>11</val><val>34</val></magnitude>
    <score><val>-14.6364</val><val>-62.0357</val></score>
    <node name="Glycosyltransferase (EC 2.4.1.-)">
     <rank><val>function</val></rank>
     <score><val>-14.6364</val><val>-62.0357</val></score>
     <magnitude><val>11</val><val>34</val></magnitude>
    </node>
   </node>
  </node>
  <node name="CBSS-251221.1.peg.1863">
   <score><val>-20</val><val>-64.3333</val></score>
   <magnitude><val>4</val><val>30</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Phosphoribosylaminoimidazole-succinocarboxamide synthase (EC 6.3.2.6)">
    <rank><val>function</val></rank>
    <score><val>-20</val><val>-64.3333</val></score>
    <magnitude><val>4</val><val>30</val></magnitude>
   </node>
  </node>
  <node name="CBSS-196620.1.peg.2477">
   <magnitude><val>32</val><val>14</val></magnitude>
   <score><val>-34.1416</val><val>-40.5</val></score>
   <rank><val>level 2</val></rank>
   <node name="Maltose O-acetyltransferase (EC 2.3.1.79)">
    <rank><val>function</val></rank>
    <score><val>-46</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Ferrous iron transport protein B">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-19</val></score>
   </node>
   <node name="Copper-translocating P-type ATPase (EC 3.6.3.4)">
    <magnitude><val>9</val></magnitude>
    <score><val>-32.7778</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Delta-1-pyrroline-5-carboxylate dehydrogenase (EC 1.5.1.12)">
    <score><val>-34.8824</val><val>-40.5</val></score>
    <magnitude><val>21</val><val>14</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Cell division-ribosomal stress proteins cluster">
   <magnitude><val>19</val><val>52</val></magnitude>
   <score><val>-71.7368</val><val>-78.3894</val></score>
   <rank><val>level 2</val></rank>
   <node name="Transcription-repair coupling factor">
    <rank><val>function</val></rank>
    <magnitude><val>2</val><val>2</val></magnitude>
    <score><val>-34</val><val>-64</val></score>
   </node>
   <node name="Cell division protein FtsH (EC 3.4.24.-)">
    <rank><val>function</val></rank>
    <magnitude><val>10</val><val>36</val></magnitude>
    <score><val>-112.6</val><val>-97.9167</val></score>
   </node>
   <node name="SSU ribosomal protein S1p">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>1</val></magnitude>
    <score><val>0</val><val>-69</val></score>
   </node>
   <node name="Peptidyl-tRNA hydrolase, archaeal type (EC 3.1.1.29)">
    <rank><val>function</val></rank>
    <score><val>-16</val><val>-27.25</val></score>
    <magnitude><val>4</val><val>13</val></magnitude>
   </node>
   <node name="Hypoxanthine-guanine phosphoribosyltransferase (EC 2.4.2.8)">
    <rank><val>function</val></rank>
    <magnitude><val>3</val></magnitude>
    <score><val>-35</val></score>
   </node>
  </node>
  <node name="Butyrate metabolism cluster">
   <rank><val>level 2</val></rank>
   <score><val>-43.3636</val><val>-77.6361</val></score>
   <magnitude><val>45</val><val>228</val></magnitude>
   <node name="Acetyl-CoA acetyltransferase (EC 2.3.1.9)">
    <rank><val>function</val></rank>
    <magnitude><val>28</val><val>112</val></magnitude>
    <score><val>-49.3462</val><val>-76.8312</val></score>
   </node>
   <node name="3-hydroxyacyl-CoA dehydrogenase (EC 1.1.1.35)">
    <score><val>-36.6667</val><val>-79.2676</val></score>
    <magnitude><val>13</val><val>110</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Enoyl-CoA hydratase (EC 4.2.1.17)">
    <score><val>-23.25</val><val>-62.75</val></score>
    <magnitude><val>4</val><val>6</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-160492.1.peg.550">
   <magnitude><val>6</val><val>19</val></magnitude>
   <score><val>-36.5</val><val>-41.8571</val></score>
   <rank><val>level 2</val></rank>
   <node name="rRNA small subunit methyltransferase I">
    <score><val>-19</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Phosphoheptose isomerase (EC 5.3.1.-)">
    <score><val>-40</val><val>-41.8571</val></score>
    <magnitude><val>5</val><val>19</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="DNA polymerase III epsilon cluster">
   <score><val>-27.5</val></score>
   <magnitude><val>2</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="CBSS-342610.3.peg.1536">
    <score><val>-27.5</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Hydroxyacylglutathione hydrolase (EC 3.1.2.6)">
     <rank><val>function</val></rank>
     <score><val>-53</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="DNA polymerase III epsilon subunit (EC 2.7.7.7)">
     <rank><val>function</val></rank>
     <score><val>-2</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Putrescine/GABA utilization cluster-temporal,to add to SSs">
   <score><val>0</val><val>-5</val></score>
   <magnitude><val></val><val>1</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="GABA and putrescine metabolism from cluters">
    <score><val>0</val><val>-5</val></score>
    <magnitude><val></val><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Putrescine transport ATP-binding protein PotG (TC 3.A.1.11.2)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Shikimate kinase containing cluster">
   <rank><val>level 2</val></rank>
   <magnitude><val>9</val><val>32</val></magnitude>
   <score><val>-22.8889</val><val>-50.1667</val></score>
   <node name="3-dehydroquinate synthase (EC 4.2.3.4)">
    <rank><val>function</val></rank>
    <magnitude><val>8</val><val>32</val></magnitude>
    <score><val>-24.625</val><val>-50.1667</val></score>
   </node>
   <node name="Shikimate kinase I (EC 2.7.1.71)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-9</val></score>
   </node>
  </node>
  <node name="CBSS-393133.3.peg.2787">
   <rank><val>level 2</val></rank>
   <score><val>-24.25</val><val>-88.0606</val></score>
   <magnitude><val>8</val><val>45</val></magnitude>
   <node name="Thymidylate kinase (EC 2.7.4.9)">
    <magnitude><val>7</val><val>45</val></magnitude>
    <score><val>-25.4286</val><val>-88.0606</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="protein from nitrogen regulatory protein P-II (GLNB) family, ortholog YAAQ B. subtilis">
    <score><val>-16</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Probably organic hydroperoxide resistance related hypothetical protein">
   <rank><val>level 2</val></rank>
   <magnitude><val>4</val><val>36</val></magnitude>
   <score><val>-9.75</val><val>-65.125</val></score>
   <node name="CBSS-269482.1.peg.1294">
    <magnitude><val>4</val><val>36</val></magnitude>
    <score><val>-9.75</val><val>-65.125</val></score>
    <rank><val>level 3</val></rank>
    <node name="Homoserine kinase (EC 2.7.1.39)">
     <rank><val>function</val></rank>
     <score><val>-9.75</val><val>-65.125</val></score>
     <magnitude><val>4</val><val>36</val></magnitude>
    </node>
   </node>
  </node>
  <node name="RNA modification cluster">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val></magnitude>
   <score><val>-9</val></score>
   <node name="Inner membrane protein translocase component YidC, long form">
    <score><val>-9</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Hypothetical in Lysine biosynthetic cluster">
   <rank><val>level 2</val></rank>
   <score><val>-27.5</val><val>-51.2049</val></score>
   <magnitude><val>2</val><val>32</val></magnitude>
   <node name="CBSS-323850.3.peg.3269">
    <score><val>-27.5</val><val>-51.2049</val></score>
    <magnitude><val>2</val><val>32</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Diaminopimelate decarboxylase (EC 4.1.1.20)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>29</val></magnitude>
     <score><val>-54</val><val>-55.7778</val></score>
    </node>
    <node name="Tyrosine recombinase XerC">
     <rank><val>function</val></rank>
     <score><val>-1</val><val>-7</val></score>
     <magnitude><val>1</val><val>3</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Hypothetical Related to Dihydroorate Dehydrogenase">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val><val>7</val></magnitude>
   <score><val>-44</val><val>-51.5714</val></score>
   <node name="Hypothetical Related to Dihydroorotate dehydrogenase">
    <score><val>-44</val><val>-51.5714</val></score>
    <magnitude><val>1</val><val>7</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Dihydroorotate dehydrogenase (EC 1.3.3.1)">
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-44</val><val>-51.5714</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Cytochrome biogenesis">
   <rank><val>level 2</val></rank>
   <magnitude><val>64</val><val>240</val></magnitude>
   <score><val>-45.2027</val><val>-70.7248</val></score>
   <node name="CBSS-196164.1.peg.1690">
    <score><val>-44.9021</val><val>-66.3612</val></score>
    <magnitude><val>59</val><val>198</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Iron-sulfur cluster assembly protein SufD">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>33</val></magnitude>
     <score><val>-36</val><val>-59.35</val></score>
    </node>
    <node name="Heme O synthase, protoheme IX farnesyltransferase (EC 2.5.1.-) COX10-CtaB">
     <rank><val>function</val></rank>
     <score><val>-21</val><val>-45.4783</val></score>
     <magnitude><val>7</val><val>39</val></magnitude>
    </node>
    <node name="Iron-sulfur cluster assembly ATPase protein SufC">
     <magnitude><val>9</val><val>23</val></magnitude>
     <score><val>-53.1111</val><val>-58.2941</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="ABC-type multidrug transport system, permease component">
     <rank><val>function</val></rank>
     <score><val>-44.6429</val><val>-44.8182</val></score>
     <magnitude><val>14</val><val>13</val></magnitude>
    </node>
    <node name="ABC-type multidrug transport system, ATPase component">
     <score><val>-27.4286</val><val>-29.5</val></score>
     <magnitude><val>8</val><val>12</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Iron-sulfur cluster assembly protein SufB">
     <rank><val>function</val></rank>
     <magnitude><val>12</val><val>60</val></magnitude>
     <score><val>-79.5455</val><val>-103.939</val></score>
    </node>
    <node name="Iron-sulfur cluster regulator SufR">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>18</val></magnitude>
     <score><val>-16.25</val><val>-49.6429</val></score>
    </node>
   </node>
   <node name="CBSS-196164.1.peg.461">
    <rank><val>level 3</val></rank>
    <score><val>-48.75</val><val>-91.2963</val></score>
    <magnitude><val>5</val><val>42</val></magnitude>
    <node name="Glutamate-1-semialdehyde aminotransferase (EC 5.4.3.8)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>42</val></magnitude>
     <score><val>-48.75</val><val>-91.2963</val></score>
    </node>
   </node>
  </node>
  <node name="Glycogen metabolism cluster">
   <magnitude><val>11</val><val>16</val></magnitude>
   <score><val>-57.1818</val><val>-43.1875</val></score>
   <rank><val>level 2</val></rank>
   <node name="Glycogen debranching enzyme (EC 3.2.1.-)">
    <score><val>-80.75</val><val>-37</val></score>
    <magnitude><val>4</val><val>5</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Glycogen synthase, ADP-glucose transglucosylase (EC 2.4.1.21)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-11</val></score>
   </node>
   <node name="1,4-alpha-glucan (glycogen) branching enzyme, GH-13-type (EC 2.4.1.18)">
    <rank><val>function</val></rank>
    <score><val>-55</val><val>-49</val></score>
    <magnitude><val>5</val><val>10</val></magnitude>
   </node>
   <node name="Glycogen phosphorylase (EC 2.4.1.1)">
    <magnitude><val>1</val><val>1</val></magnitude>
    <score><val>-20</val><val>-16</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Nucleoside Catabolism cluster">
   <magnitude><val>1</val><val>19</val></magnitude>
   <score><val>-22</val><val>-69.3077</val></score>
   <rank><val>level 2</val></rank>
   <node name="Purine nucleoside phosphorylase (EC 2.4.2.1)">
    <score><val>0</val><val>-69.3077</val></score>
    <magnitude><val></val><val>19</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="UDP-glucose dehydrogenase (EC 1.1.1.22)">
    <magnitude><val>1</val></magnitude>
    <score><val>-22</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="USS-DB-4">
   <magnitude><val>1</val></magnitude>
   <score><val>-117</val></score>
   <rank><val>level 2</val></rank>
   <node name="Transcriptional regulator, TetR (AcrR) family [USSDB4A]">
    <magnitude><val>1</val></magnitude>
    <score><val>-117</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="proteosome related">
   <rank><val>level 2</val></rank>
   <score><val>-41.2222</val><val>-38.1135</val></score>
   <magnitude><val>9</val><val>31</val></magnitude>
   <node name="Cluster-based Subsystem Grouping Hypotheticals - perhaps Proteosome Related">
    <rank><val>level 3</val></rank>
    <magnitude><val>9</val><val>31</val></magnitude>
    <score><val>-41.2222</val><val>-38.1135</val></score>
    <node name="Bacterial proteasome-activating AAA-ATPase (PAN)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-103</val></score>
    </node>
    <node name="Twin-arginine translocation protein TatC">
     <rank><val>function</val></rank>
     <score><val>-47</val><val>-44.1538</val></score>
     <magnitude><val>5</val><val>18</val></magnitude>
    </node>
    <node name="FIG019733: possible DNA-binding protein">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-10</val></score>
    </node>
    <node name="Twin-arginine translocation protein TatA">
     <magnitude><val>1</val><val>13</val></magnitude>
     <score><val>-13</val><val>-29.75</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="CRISPRs and associated hypotheticals">
   <magnitude><val>10</val><val>13</val></magnitude>
   <score><val>-23.9</val><val>-35.4188</val></score>
   <rank><val>level 2</val></rank>
   <node name="CBSS-216592.1.peg.3534">
    <magnitude><val>10</val><val>13</val></magnitude>
    <score><val>-23.9</val><val>-35.4188</val></score>
    <rank><val>level 3</val></rank>
    <node name="CRISPR-associated helicase Cas3, protein">
     <rank><val>function</val></rank>
     <score><val>-45</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="CRISPR-associated protein Cas1">
     <rank><val>function</val></rank>
     <score><val>-19.2</val><val>-38.2222</val></score>
     <magnitude><val>5</val><val>11</val></magnitude>
    </node>
    <node name="putative CRISPR associated hypothetical protein">
     <magnitude><val>1</val><val>1</val></magnitude>
     <score><val>-3</val><val>-3</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="CRISPR-associated protein, CT1975 family">
     <rank><val>function</val></rank>
     <score><val>-31.6667</val><val>-37</val></score>
     <magnitude><val>3</val><val>1</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Methylamine utilization">
   <score><val>-50.1765</val><val>-61.814</val></score>
   <magnitude><val>17</val><val>139</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="CBSS-265072.7.peg.546">
    <rank><val>level 3</val></rank>
    <magnitude><val>3</val><val>27</val></magnitude>
    <score><val>-28.3333</val><val>-24.3125</val></score>
    <node name="PlcB, ORFX, ORFP, ORFB, ORFA, ldh gene">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>27</val></magnitude>
     <score><val>-28.3333</val><val>-24.3125</val></score>
    </node>
   </node>
   <node name="Glutamate-mediated methylamine utilization pathway">
    <score><val>-54.8571</val><val>-70.8545</val></score>
    <magnitude><val>14</val><val>112</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Glutamate synthase [NADPH] large chain (EC 1.4.1.13)">
     <magnitude><val>5</val><val>75</val></magnitude>
     <score><val>-61.4</val><val>-66.619</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glutamine synthetase type I (EC 6.3.1.2)">
     <magnitude><val>9</val><val>37</val></magnitude>
     <score><val>-51.2222</val><val>-79.44</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="CBSS-176299.4.peg.1996A">
   <magnitude><val>1</val><val>29</val></magnitude>
   <score><val>-9</val><val>-47.0588</val></score>
   <rank><val>level 2</val></rank>
   <node name="Glycerophosphoryl diester phosphodiesterase (EC 3.1.4.46)">
    <rank><val>function</val></rank>
    <score><val>-9</val><val>-47.0588</val></score>
    <magnitude><val>1</val><val>29</val></magnitude>
   </node>
  </node>
  <node name="Probably GTP or GMP signaling related">
   <rank><val>level 2</val></rank>
   <magnitude><val>8</val><val>8</val></magnitude>
   <score><val>-10.5</val><val>-95.6667</val></score>
   <node name="CBSS-176299.4.peg.1292">
    <rank><val>level 3</val></rank>
    <magnitude><val>8</val><val>8</val></magnitude>
    <score><val>-10.5</val><val>-95.6667</val></score>
    <node name="GTP-binding protein Era">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
    </node>
    <node name="Holo-[acyl-carrier protein] synthase (EC 2.7.8.7)">
     <rank><val>function</val></rank>
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Signal peptidase I (EC 3.4.21.89)">
     <magnitude><val>6</val><val>8</val></magnitude>
     <score><val>-11.5</val><val>-95.6667</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="CBSS-291331.3.peg.3674">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val></magnitude>
   <score><val>-24</val></score>
   <node name="S-adenosyl-L-methionine dependent methyltransferase, similar to cyclopropane-fatty-acyl-phospholipid synthase">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-24</val></score>
   </node>
  </node>
  <node name="Cluster containing CofD-like protein and co-occuring with DNA repair">
   <magnitude><val>1</val></magnitude>
   <score><val>-30</val></score>
   <rank><val>level 2</val></rank>
   <node name="FIG000506: Predicted P-loop-containing kinase">
    <magnitude><val>1</val></magnitude>
    <score><val>-30</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-83333.1.peg.946">
   <rank><val>level 2</val></rank>
   <magnitude><val>6</val><val>22</val></magnitude>
   <score><val>-15.0833</val><val>-22.0909</val></score>
   <node name="Inner membrane protein YccF">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-18</val></score>
   </node>
   <node name="Cell division inhibitor">
    <magnitude><val>1</val></magnitude>
    <score><val>-27</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Succinyl-CoA synthetase, alpha subunit-related enzymes">
    <score><val>-11.5</val><val>-22.0909</val></score>
    <magnitude><val>3</val><val>22</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="LSU m5C1962 methyltransferase RlmI">
    <magnitude><val>1</val></magnitude>
    <score><val>-11</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-235.1.peg.567">
   <score><val>-21.4367</val><val>-73.3837</val></score>
   <magnitude><val>35</val><val>106</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Dihydroneopterin aldolase (EC 4.1.2.25)">
    <score><val>-13</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="GTP-binding protein">
    <magnitude><val>26</val><val>68</val></magnitude>
    <score><val>-22.8571</val><val>-70.561</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Dihydropteroate synthase (EC 2.5.1.15)">
    <magnitude><val>7</val><val>38</val></magnitude>
    <score><val>-20</val><val>-78.4348</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="2-amino-4-hydroxy-6-hydroxymethyldihydropteridine pyrophosphokinase (EC 2.7.6.3)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-3</val></score>
   </node>
  </node>
  <node name="EC699-706">
   <magnitude><val>3</val><val>3</val></magnitude>
   <score><val>-41.5</val><val>-14.3333</val></score>
   <rank><val>level 2</val></rank>
   <node name="Deoxyribodipyrimidine photolyase (EC 4.1.99.3)">
    <magnitude><val>3</val><val>3</val></magnitude>
    <score><val>-41.5</val><val>-14.3333</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-350688.3.peg.1509">
   <rank><val>level 2</val></rank>
   <magnitude><val>23</val><val>49</val></magnitude>
   <score><val>-34.8168</val><val>-92.3891</val></score>
   <node name="tRNA pseudouridine synthase B (EC 4.2.1.70)">
    <rank><val>function</val></rank>
    <score><val>-9.66667</val></score>
    <magnitude><val>3</val></magnitude>
   </node>
   <node name="DNA polymerase III alpha subunit (EC 2.7.7.7)">
    <rank><val>function</val></rank>
    <magnitude><val>3</val><val>2</val></magnitude>
    <score><val>-34.3333</val><val>-32</val></score>
   </node>
   <node name="Translation initiation factor 2">
    <rank><val>function</val></rank>
    <magnitude><val>15</val><val>46</val></magnitude>
    <score><val>-35.7857</val><val>-92.871</val></score>
   </node>
   <node name="Transcription termination protein NusA">
    <rank><val>function</val></rank>
    <score><val>-66</val><val>-191</val></score>
    <magnitude><val>2</val><val>1</val></magnitude>
   </node>
  </node>
  <node name="CBSS-83332.1.peg.3803">
   <magnitude><val>2</val></magnitude>
   <score><val>-46.5</val></score>
   <rank><val>level 2</val></rank>
   <node name="Probable acyl-CoA dehydrogenase fadE35 (EC 1.3.99.-)">
    <magnitude><val>2</val></magnitude>
    <score><val>-46.5</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-342610.3.peg.1794">
   <rank><val>level 2</val></rank>
   <score><val>-17</val><val>-42.6667</val></score>
   <magnitude><val>1</val><val>15</val></magnitude>
   <node name="ADP-ribose pyrophosphatase (EC 3.6.1.13)">
    <score><val>-17</val><val>-42.6667</val></score>
    <magnitude><val>1</val><val>15</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Spore Coat">
   <magnitude><val>1</val></magnitude>
   <score><val>-87</val></score>
   <rank><val>level 2</val></rank>
   <node name="Stage IV sporulation protein A">
    <score><val>-87</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Clustering-based subsystems">
   <rank><val>level 2</val></rank>
   <magnitude><val>27</val><val>66</val></magnitude>
   <score><val>-28.2492</val><val>-56.7912</val></score>
   <node name="CBSS-292415.3.peg.2341">
    <score><val>-53</val><val>-14</val></score>
    <magnitude><val>1</val><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Hydroxyacylglutathione hydrolase (EC 3.1.2.6)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-53</val></score>
    </node>
    <node name="Major facilitator superfamily (MFS) transporter">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-14</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="CBSS-280355.3.peg.2835">
    <rank><val>level 3</val></rank>
    <magnitude><val>7</val><val>14</val></magnitude>
    <score><val>-20.5714</val><val>-24.9857</val></score>
    <node name="DinG family ATP-dependent helicase YoaA">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-29</val></score>
    </node>
    <node name="tRNA nucleotidyltransferase (EC 2.7.7.21) (EC 2.7.7.25)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>2</val></magnitude>
     <score><val>-5</val><val>-16.5</val></score>
    </node>
    <node name="3-methyl-2-oxobutanoate hydroxymethyltransferase (EC 2.1.2.11)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-15</val></score>
    </node>
    <node name="hypothetical protein BH3604">
     <magnitude><val>1</val></magnitude>
     <score><val>-50</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrodipicolinate reductase (EC 1.3.1.26)">
     <rank><val>function</val></rank>
     <score><val>-15</val><val>-26.4</val></score>
     <magnitude><val>3</val><val>12</val></magnitude>
    </node>
   </node>
   <node name="CBSS-262719.3.peg.410">
    <magnitude><val>12</val><val>51</val></magnitude>
    <score><val>-34.7273</val><val>-66.3611</val></score>
    <rank><val>level 3</val></rank>
    <node name="Adenylosuccinate synthetase (EC 6.3.4.4)">
     <score><val>-34.7273</val><val>-66.3611</val></score>
     <magnitude><val>12</val><val>51</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Putative diaminopropionate ammonia-lyase cluster">
    <score><val>-22.25</val></score>
    <magnitude><val>4</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Xanthine dehydrogenase iron-sulfur subunit (EC 1.17.1.4)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-9.5</val></score>
    </node>
    <node name="Xanthine dehydrogenase, FAD binding subunit (EC 1.17.1.4)">
     <rank><val>function</val></rank>
     <score><val>-5</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Xanthine dehydrogenase, molybdenum binding subunit (EC 1.17.1.4)">
     <score><val>-65</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Sporulation-related Hypotheticals">
    <rank><val>level 3</val></rank>
    <magnitude><val>3</val></magnitude>
    <score><val>-20</val></score>
    <node name="Stage III sporulation protein D">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-25</val></score>
    </node>
    <node name="UDP-N-acetylglucosamine 1-carboxyvinyltransferase (EC 2.5.1.7)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-17.5</val></score>
    </node>
   </node>
  </node>
  <node name="TldD cluster">
   <rank><val>level 2</val></rank>
   <magnitude><val>7</val><val>5</val></magnitude>
   <score><val>-37.7143</val><val>-75.2</val></score>
   <node name="CBSS-354.1.peg.2917">
    <rank><val>level 3</val></rank>
    <magnitude><val>7</val><val>5</val></magnitude>
    <score><val>-37.7143</val><val>-75.2</val></score>
    <node name="TldD protein, part of proposed TldE/TldD proteolytic complex (PMID 12029038)">
     <rank><val>function</val></rank>
     <score><val>-46.2</val><val>-75.2</val></score>
     <magnitude><val>5</val><val>5</val></magnitude>
    </node>
    <node name="Rod shape-determining protein MreC">
     <rank><val>function</val></rank>
     <score><val>-6</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Cytoplasmic axial filament protein CafA and Ribonuclease G (EC 3.1.4.-)">
     <rank><val>function</val></rank>
     <score><val>-27</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
  </node>
  <node name="CBSS-281090.3.peg.464">
   <score><val>-45.4546</val><val>-85.9105</val></score>
   <magnitude><val>13</val><val>74</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Shikimate 5-dehydrogenase I alpha (EC 1.1.1.25)">
    <score><val>-4</val><val>-73.2857</val></score>
    <magnitude><val>1</val><val>14</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Putative Holliday junction resolvase (EC 3.1.-.-)">
    <score><val>0</val><val>-20</val></score>
    <magnitude><val></val><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Alanyl-tRNA synthetase (EC 6.1.1.7)">
    <score><val>-48.9091</val><val>-90.0233</val></score>
    <magnitude><val>12</val><val>59</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-316057.3.peg.563">
   <magnitude><val>14</val><val>84</val></magnitude>
   <score><val>-30.7143</val><val>-76.213</val></score>
   <rank><val>level 2</val></rank>
   <node name="Cytochrome c oxidase polypeptide II (EC 1.9.3.1)">
    <magnitude><val>1</val><val>11</val></magnitude>
    <score><val>-16</val><val>-81.6667</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Heme O synthase, protoheme IX farnesyltransferase (EC 2.5.1.-) COX10-CtaB">
    <rank><val>function</val></rank>
    <score><val>-21</val><val>-45.4783</val></score>
    <magnitude><val>7</val><val>39</val></magnitude>
   </node>
   <node name="Cytochrome c oxidase polypeptide III (EC 1.9.3.1)">
    <magnitude><val>1</val><val>29</val></magnitude>
    <score><val>-36</val><val>-115.652</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="TldD protein, part of proposed TldE/TldD proteolytic complex (PMID 12029038)">
    <rank><val>function</val></rank>
    <score><val>-46.2</val><val>-75.2</val></score>
    <magnitude><val>5</val><val>5</val></magnitude>
   </node>
  </node>
  <node name="CBSS-316273.3.peg.448">
   <magnitude><val>1</val></magnitude>
   <score><val>-38</val></score>
   <rank><val>level 2</val></rank>
   <node name="TldE/PmbA family protein, Actinobacterial subgroup">
    <rank><val>function</val></rank>
    <score><val>-38</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
  </node>
  <node name="Probably Pyrimidine biosynthesis-related">
   <magnitude><val>6</val><val>2</val></magnitude>
   <score><val>-23</val><val>-17.5</val></score>
   <rank><val>level 2</val></rank>
   <node name="CBSS-306254.1.peg.1508">
    <magnitude><val>6</val><val>2</val></magnitude>
    <score><val>-23</val><val>-17.5</val></score>
    <rank><val>level 3</val></rank>
    <node name="Carbamoyl-phosphate synthase small chain (EC 6.3.5.5)">
     <rank><val>function</val></rank>
     <score><val>-23</val><val>-17.5</val></score>
     <magnitude><val>6</val><val>2</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Bacterial RNA-metabolizing Zn-dependent hydrolases">
   <score><val>-46.0588</val><val>-79.7014</val></score>
   <magnitude><val>17</val><val>54</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="YgjD/Kae1/Qri7 family, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
    <score><val>-43.4</val><val>-85.1667</val></score>
    <magnitude><val>10</val><val>45</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Cytochrome d ubiquinol oxidase subunit II (EC 1.10.3.-)">
    <magnitude><val>1</val></magnitude>
    <score><val>-53</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Ribosomal-protein-S18p-alanine acetyltransferase (EC 2.3.1.-)">
    <magnitude><val></val><val>9</val></magnitude>
    <score><val>0</val><val>-52.375</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Zn-dependent hydrolase, RNA-metabolising, CPSF 100 kDa analog">
    <rank><val>function</val></rank>
    <score><val>-49.3333</val></score>
    <magnitude><val>3</val></magnitude>
   </node>
   <node name="Zn-dependent hydrolase, RNA-metabolising">
    <rank><val>function</val></rank>
    <magnitude><val>3</val></magnitude>
    <score><val>-49.3333</val></score>
   </node>
  </node>
  <node name="DNA gyrase subunits">
   <score><val>-44.2077</val><val>-87.928</val></score>
   <magnitude><val>26</val><val>107</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="DNA gyrase subunit A (EC 5.99.1.3)">
    <rank><val>function</val></rank>
    <score><val>-24.2667</val><val>-83</val></score>
    <magnitude><val>15</val><val>53</val></magnitude>
   </node>
   <node name="DNA gyrase subunit B (EC 5.99.1.3)">
    <rank><val>function</val></rank>
    <score><val>-71.4</val><val>-92.7647</val></score>
    <magnitude><val>11</val><val>54</val></magnitude>
   </node>
  </node>
  <node name="Isoprenoid/cell wall biosynthesis: PREDICTED UNDECAPRENYL DIPHOSPHATE PHOSPHATASE">
   <rank><val>level 2</val></rank>
   <score><val>-33.6191</val><val>-58.65</val></score>
   <magnitude><val>12</val><val>28</val></magnitude>
   <node name="CBSS-83331.1.peg.3039">
    <rank><val>level 3</val></rank>
    <score><val>-33.6191</val><val>-58.65</val></score>
    <magnitude><val>12</val><val>28</val></magnitude>
    <node name="Cell division protein FtsI [Peptidoglycan synthetase] (EC 2.4.1.129)">
     <score><val>-57</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="1-deoxy-D-xylulose 5-phosphate reductoisomerase (EC 1.1.1.267)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-22</val></score>
    </node>
    <node name="Undecaprenyl pyrophosphate synthetase (EC 2.5.1.31)">
     <score><val>-29.7143</val><val>-58.65</val></score>
     <magnitude><val>9</val><val>28</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="CBSS-56780.10.peg.1536">
   <score><val>-21</val></score>
   <magnitude><val>1</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Metal-dependent hydrolase YbeY, involved in rRNA and/or ribosome maturation and assembly">
    <rank><val>function</val></rank>
    <score><val>-21</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
  </node>
  <node name="Ribosome-related cluster">
   <rank><val>level 2</val></rank>
   <score><val>-14.963</val><val>-53.6127</val></score>
   <magnitude><val>9</val><val>65</val></magnitude>
   <node name="A Gammaproteobacteria Cluster Relating to Translation">
    <score><val>-14.963</val><val>-53.6127</val></score>
    <magnitude><val>9</val><val>65</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Glutamyl-tRNA reductase (EC 1.2.1.70)">
     <score><val>-10</val><val>-49.5769</val></score>
     <magnitude><val>1</val><val>32</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribose-phosphate pyrophosphokinase (EC 2.7.6.1)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>33</val></magnitude>
     <score><val>-12.6667</val><val>-57.5263</val></score>
    </node>
    <node name="Peptide chain release factor 1">
     <rank><val>function</val></rank>
     <score><val>-36</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
  </node>
  <node name="DNA replication cluster 1">
   <rank><val>level 2</val></rank>
   <magnitude><val>11</val><val>54</val></magnitude>
   <score><val>-71.4</val><val>-92.7647</val></score>
   <node name="DNA gyrase subunit B (EC 5.99.1.3)">
    <score><val>-71.4</val><val>-92.7647</val></score>
    <magnitude><val>11</val><val>54</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Proteasome related clusters">
   <score><val>-41.7917</val><val>-46.6016</val></score>
   <magnitude><val>24</val><val>111</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Proteasome subunit alpha archaeal cluster">
    <magnitude><val>24</val><val>111</val></magnitude>
    <score><val>-41.7917</val><val>-46.6016</val></score>
    <rank><val>level 3</val></rank>
    <node name="LSU ribosomal protein L15e">
     <rank><val>function</val></rank>
     <score><val>-16</val><val>-26.0769</val></score>
     <magnitude><val>5</val><val>18</val></magnitude>
    </node>
    <node name="Shwachman-Bodian-Diamond syndrome protein">
     <magnitude><val>7</val><val>33</val></magnitude>
     <score><val>-62</val><val>-41</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Proteasome subunit alpha (EC 3.4.25.1), archaeal">
     <magnitude><val>2</val><val>24</val></magnitude>
     <score><val>-8</val><val>-57.0833</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Probable exosome complex RNA-binding protein 1">
     <rank><val>function</val></rank>
     <score><val>-53.6667</val><val>-66.6667</val></score>
     <magnitude><val>3</val><val>13</val></magnitude>
    </node>
    <node name="Exosome complex exonuclease 1">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>3</val></magnitude>
     <score><val>-50.5</val><val>-63.3333</val></score>
    </node>
    <node name="LSU ribosomal protein L37Ae">
     <score><val>0</val><val>-4.75</val></score>
     <magnitude><val></val><val>4</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Probable exosome complex exonuclease 2 (EC 3.1.13.-)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>16</val></magnitude>
     <score><val>-42.2</val><val>-56.5455</val></score>
    </node>
   </node>
  </node>
  <node name="Shiga toxin cluster">
   <rank><val>level 2</val></rank>
   <magnitude><val></val><val>4</val></magnitude>
   <score><val>0</val><val>-16.25</val></score>
   <node name="CBSS-194948.1.peg.143">
    <rank><val>level 3</val></rank>
    <score><val>0</val><val>-16.25</val></score>
    <magnitude><val></val><val>4</val></magnitude>
    <node name="P protein">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>4</val></magnitude>
     <score><val>0</val><val>-16.25</val></score>
    </node>
   </node>
  </node>
  <node name="recX and regulatory cluster">
   <rank><val>level 2</val></rank>
   <score><val>-27</val><val>-15</val></score>
   <magnitude><val>1</val><val>1</val></magnitude>
   <node name="CBSS-261594.1.peg.788">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val><val>1</val></magnitude>
    <score><val>-27</val><val>-15</val></score>
    <node name="A/G-specific adenine glycosylase (EC 3.2.2.-)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-15</val></score>
    </node>
    <node name="Cell division inhibitor">
     <score><val>-27</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="CBSS-257314.1.peg.752">
   <magnitude><val>7</val><val>16</val></magnitude>
   <score><val>-49.5714</val><val>-74.1333</val></score>
   <rank><val>level 2</val></rank>
   <node name="Adenine-specific methyltransferase (EC 2.1.1.72)">
    <rank><val>function</val></rank>
    <magnitude><val>6</val><val>16</val></magnitude>
    <score><val>-55.8333</val><val>-74.1333</val></score>
   </node>
   <node name="Thiol peroxidase, Tpx-type (EC 1.11.1.15)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-12</val></score>
   </node>
  </node>
  <node name="CBSS-211586.1.peg.3133">
   <rank><val>level 2</val></rank>
   <magnitude><val>4</val><val>47</val></magnitude>
   <score><val>-13.25</val><val>-68.4412</val></score>
   <node name="tRNA pseudouridine 13 synthase (EC 4.2.1.-)">
    <rank><val>function</val></rank>
    <score><val>-13.25</val><val>-68.4412</val></score>
    <magnitude><val>4</val><val>47</val></magnitude>
   </node>
  </node>
  <node name="Three hypotheticals linked to lipoprotein biosynthesis">
   <score><val>-10.5</val></score>
   <magnitude><val>2</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="CBSS-188.1.peg.9880">
    <score><val>-10.5</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="COG1565: Uncharacterized conserved protein">
     <score><val>-9</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Prolipoprotein diacylglyceryl transferase (EC 2.4.99.-)">
     <rank><val>function</val></rank>
     <score><val>-12</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Sulfatases and sulfatase modifying factor 1 (and a hypothetical)">
   <score><val>-14.5</val></score>
   <magnitude><val>4</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Sulfatases and sulfatase modifying factor 1">
    <rank><val>level 3</val></rank>
    <magnitude><val>4</val></magnitude>
    <score><val>-14.5</val></score>
    <node name="Choline-sulfatase (EC 3.1.6.6)">
     <score><val>-14.5</val></score>
     <magnitude><val>4</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Carbohydrates">
   <score><val>-21.5</val><val>-55.0721</val></score>
   <magnitude><val>30</val><val>68</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Cluster Ytf and putative sugar transporter">
    <rank><val>level 3</val></rank>
    <magnitude><val>11</val><val>33</val></magnitude>
    <score><val>-19.1364</val><val>-56.15</val></score>
    <node name="Fructose-1,6-bisphosphatase, type I (EC 3.1.3.11)">
     <score><val>-8</val></score>
     <magnitude><val>4</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Peptide methionine sulfoxide reductase MsrA (EC 1.8.4.11)">
     <magnitude><val>7</val><val>33</val></magnitude>
     <score><val>-25.5</val><val>-56.15</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Putative sugar ABC transporter (ytf cluster)">
    <rank><val>level 3</val></rank>
    <score><val>-19.1364</val><val>-56.15</val></score>
    <magnitude><val>11</val><val>33</val></magnitude>
    <node name="Fructose-1,6-bisphosphatase, type I (EC 3.1.3.11)">
     <magnitude><val>4</val></magnitude>
     <score><val>-8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Peptide methionine sulfoxide reductase MsrA (EC 1.8.4.11)">
     <magnitude><val>7</val><val>33</val></magnitude>
     <score><val>-25.5</val><val>-56.15</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Beta-lactamase cluster in Streptococcus">
    <rank><val>level 3</val></rank>
    <score><val>-28</val><val>-19.5</val></score>
    <magnitude><val>8</val><val>2</val></magnitude>
    <node name="RecD-like DNA helicase YrrC">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>2</val></magnitude>
     <score><val>-28</val><val>-19.5</val></score>
    </node>
   </node>
  </node>
  <node name="Two related proteases">
   <score><val>-18.2699</val><val>-47.2054</val></score>
   <magnitude><val>9</val><val>40</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="CBSS-257314.1.peg.676">
    <rank><val>level 3</val></rank>
    <magnitude><val>9</val><val>40</val></magnitude>
    <score><val>-18.2699</val><val>-47.2054</val></score>
    <node name="Competence/damage-inducible protein CinA">
     <score><val>-17</val><val>-57.6667</val></score>
     <magnitude><val>1</val><val>21</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="CDP-diacylglycerol--glycerol-3-phosphate 3-phosphatidyltransferase (EC 2.7.8.5)">
     <score><val>-18.4286</val><val>-35.6429</val></score>
     <magnitude><val>8</val><val>19</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="tRNA-methylthiotransferase containing cluster">
   <magnitude><val>10</val><val>32</val></magnitude>
   <score><val>-22.6714</val><val>-40.2083</val></score>
   <rank><val>level 2</val></rank>
   <node name="tRNA-i(6)A37 methylthiotransferase">
    <rank><val>function</val></rank>
    <score><val>-22.8571</val><val>-40.2083</val></score>
    <magnitude><val>9</val><val>32</val></magnitude>
   </node>
   <node name="Metal-dependent hydrolase YbeY, involved in rRNA and/or ribosome maturation and assembly">
    <magnitude><val>1</val></magnitude>
    <score><val>-21</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Yfa cluster">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val><val>21</val></magnitude>
   <score><val>-17</val><val>-57.6667</val></score>
   <node name="Competence/damage-inducible protein CinA">
    <score><val>-17</val><val>-57.6667</val></score>
    <magnitude><val>1</val><val>21</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="CBSS-87626.3.peg.3639">
   <score><val>-20.6591</val><val>-58.8571</val></score>
   <magnitude><val>11</val><val>38</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Xaa-Pro aminopeptidase (EC 3.4.11.9)">
    <rank><val>function</val></rank>
    <score><val>-16</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Aminomethyltransferase (glycine cleavage system T protein) (EC 2.1.2.10)">
    <magnitude><val>10</val><val>38</val></magnitude>
    <score><val>-21.125</val><val>-58.8571</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Putative asociate of RNA polymerase sigma-54 factor rpoN">
   <magnitude><val>4</val></magnitude>
   <score><val>-23.25</val></score>
   <rank><val>level 2</val></rank>
   <node name="CBSS-316057.3.peg.1308">
    <magnitude><val>4</val></magnitude>
    <score><val>-23.25</val></score>
    <rank><val>level 3</val></rank>
    <node name="RNA polymerase sigma-70 factor, ECF subfamily">
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="putative RNA polymerase sigma factor">
     <magnitude><val>1</val></magnitude>
     <score><val>-4</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="RNA polymerase sigma-54 factor RpoN">
     <magnitude><val>1</val></magnitude>
     <score><val>-65</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="RNA polymerase sigma-70 factor">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
    </node>
   </node>
  </node>
  <node name="Conserved cluster around acetyltransferase YpeA in Enterobacteria">
   <magnitude><val>1</val></magnitude>
   <score><val>-51</val></score>
   <rank><val>level 2</val></rank>
   <node name="Predicted iron-dependent peroxidase, Dyp-type family">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-51</val></score>
   </node>
  </node>
 </node>
 <node name="Amino Acids and Derivatives">
  <magnitude><val>1411</val><val>7111</val></magnitude>
  <score><val>-40.7849</val><val>-75.5169</val></score>
  <rank><val>level 1</val></rank>
  <node name="Histidine Metabolism">
   <rank><val>level 2</val></rank>
   <magnitude><val>37</val><val>218</val></magnitude>
   <score><val>-47.5946</val><val>-71.6537</val></score>
   <node name="Histidine Degradation">
    <rank><val>level 3</val></rank>
    <score><val>-55.4286</val><val>-77.4678</val></score>
    <magnitude><val>28</val><val>69</val></magnitude>
    <node name="Glutamate formiminotransferase (EC 2.1.2.5)">
     <score><val>-46</val><val>-37</val></score>
     <magnitude><val>3</val><val>7</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Imidazolonepropionase (EC 3.5.2.7)">
     <rank><val>function</val></rank>
     <score><val>-22</val><val>-20.3333</val></score>
     <magnitude><val>2</val><val>3</val></magnitude>
    </node>
    <node name="Histidine ammonia-lyase (EC 4.3.1.3)">
     <magnitude><val>9</val><val>41</val></magnitude>
     <score><val>-43.5556</val><val>-89.6071</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Urocanate hydratase (EC 4.2.1.49)">
     <rank><val>function</val></rank>
     <magnitude><val>14</val><val>18</val></magnitude>
     <score><val>-69.8571</val><val>-75.0769</val></score>
    </node>
   </node>
   <node name="Histidine Biosynthesis">
    <score><val>-23.2222</val><val>-68.9612</val></score>
    <magnitude><val>9</val><val>149</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Imidazole glycerol phosphate synthase cyclase subunit (EC 4.1.3.-)">
     <magnitude><val>1</val><val>12</val></magnitude>
     <score><val>-38</val><val>-86.875</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosyl-AMP cyclohydrolase (EC 3.5.4.19)">
     <magnitude><val>1</val><val>11</val></magnitude>
     <score><val>-34</val><val>-57.1111</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Imidazole glycerol phosphate synthase amidotransferase subunit (EC 2.4.2.-)">
     <score><val>0</val><val>-61.9</val></score>
     <magnitude><val></val><val>11</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="ATP phosphoribosyltransferase catalytic subunit (EC 2.4.2.17)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>10</val></magnitude>
     <score><val>-43</val><val>-48.7143</val></score>
    </node>
    <node name="Imidazoleglycerol-phosphate dehydratase (EC 4.2.1.19)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-60</val></score>
     <magnitude><val></val><val>8</val></magnitude>
    </node>
    <node name="Phosphoribosyl-ATP pyrophosphatase (EC 3.6.1.31)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>11</val></magnitude>
     <score><val>-34</val><val>-57.1111</val></score>
    </node>
    <node name="Phosphoribosylformimino-5-aminoimidazole carboxamide ribotide isomerase (EC 5.3.1.16)">
     <score><val>-15</val><val>-81</val></score>
     <magnitude><val>2</val><val>15</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Histidinol-phosphate aminotransferase (EC 2.6.1.9)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-59.5</val></score>
     <magnitude><val></val><val>21</val></magnitude>
    </node>
    <node name="Histidinol dehydrogenase (EC 1.1.1.23)">
     <magnitude><val></val><val>25</val></magnitude>
     <score><val>0</val><val>-91.8571</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="ATP phosphoribosyltransferase regulatory subunit (EC 2.4.2.17)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>25</val></magnitude>
     <score><val>-10</val><val>-62.6923</val></score>
    </node>
   </node>
  </node>
  <node name="Lysine, threonine, methionine, and cysteine">
   <magnitude><val>264</val><val>1252</val></magnitude>
   <score><val>-38.3733</val><val>-70.8158</val></score>
   <rank><val>level 2</val></rank>
   <node name="Methionine Degradation">
    <score><val>-60.0528</val><val>-62.3889</val></score>
    <magnitude><val>15</val><val>12</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Methionine gamma-lyase (EC 4.4.1.11)">
     <magnitude><val>1</val></magnitude>
     <score><val>-22</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="S-adenosylmethionine synthetase (EC 2.5.1.6)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>7</val></magnitude>
     <score><val>-49.6667</val><val>-73.1429</val></score>
    </node>
    <node name="Adenosylhomocysteinase (EC 3.3.1.1)">
     <magnitude><val>9</val><val>4</val></magnitude>
     <score><val>-66.125</val><val>-53.6667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Pyruvate-flavodoxin oxidoreductase (EC 1.2.7.-)">
     <score><val>-85</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cystathionine beta-synthase (EC 4.2.1.22)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-22</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Lysine Biosynthesis DAP Pathway">
    <score><val>-23.0471</val><val>-52.8971</val></score>
    <magnitude><val>17</val><val>75</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Dihydrodipicolinate synthase (EC 4.2.1.52)">
     <magnitude><val>6</val><val>6</val></magnitude>
     <score><val>-18.8</val><val>-52</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Aspartate-semialdehyde dehydrogenase (EC 1.2.1.11)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>14</val></magnitude>
     <score><val>-23.3333</val><val>-64.8462</val></score>
    </node>
    <node name="N-succinyl-L,L-diaminopimelate desuccinylase (EC 3.5.1.18)">
     <score><val>-42</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LL-diaminopimelate aminotransferase, predicted alternative">
     <score><val>-22</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrodipicolinate reductase (EC 1.3.1.26)">
     <rank><val>function</val></rank>
     <score><val>-15</val><val>-26.4</val></score>
     <magnitude><val>3</val><val>12</val></magnitude>
    </node>
    <node name="Aspartokinase (EC 2.7.2.4)">
     <magnitude><val>2</val><val>14</val></magnitude>
     <score><val>-23</val><val>-58.0769</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Diaminopimelate decarboxylase (EC 4.1.1.20)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>29</val></magnitude>
     <score><val>-54</val><val>-55.7778</val></score>
    </node>
   </node>
   <node name="Threonine and Homoserine Biosynthesis">
    <score><val>-32.0529</val><val>-66.2012</val></score>
    <magnitude><val>53</val><val>196</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Aspartokinase (EC 2.7.2.4)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>14</val></magnitude>
     <score><val>-23</val><val>-58.0769</val></score>
    </node>
    <node name="Homoserine kinase (EC 2.7.1.39)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>36</val></magnitude>
     <score><val>-9.75</val><val>-65.125</val></score>
    </node>
    <node name="Aspartate aminotransferase (EC 2.6.1.1)">
     <score><val>-48.4737</val><val>-49.2692</val></score>
     <magnitude><val>21</val><val>33</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Threonine synthase (EC 4.2.3.1)">
     <magnitude><val>15</val><val>76</val></magnitude>
     <score><val>-27.1333</val><val>-76.8478</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Aspartate-semialdehyde dehydrogenase (EC 1.2.1.11)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>14</val></magnitude>
     <score><val>-23.3333</val><val>-64.8462</val></score>
    </node>
    <node name="Homoserine dehydrogenase (EC 1.1.1.3)">
     <magnitude><val>8</val><val>23</val></magnitude>
     <score><val>-14.8571</val><val>-62.7692</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Lysine fermentation">
    <rank><val>level 3</val></rank>
    <score><val>-47.5236</val><val>-77.4631</val></score>
    <magnitude><val>62</val><val>258</val></magnitude>
    <node name="Electron transfer flavoprotein, alpha subunit">
     <rank><val>function</val></rank>
     <magnitude><val>13</val><val>85</val></magnitude>
     <score><val>-32.1667</val><val>-83</val></score>
    </node>
    <node name="Acetyl-CoA acetyltransferase (EC 2.3.1.9)">
     <score><val>-49.3462</val><val>-76.8312</val></score>
     <magnitude><val>28</val><val>112</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Butyryl-CoA dehydrogenase (EC 1.3.99.2)">
     <rank><val>function</val></rank>
     <score><val>-54.6</val><val>-56.4444</val></score>
     <magnitude><val>21</val><val>25</val></magnitude>
    </node>
    <node name="Electron transfer flavoprotein, beta subunit">
     <magnitude><val></val><val>36</val></magnitude>
     <score><val>0</val><val>-80.9524</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Methionine Salvage">
    <magnitude><val>22</val><val>99</val></magnitude>
    <score><val>-52.1273</val><val>-60.65</val></score>
    <rank><val>level 3</val></rank>
    <node name="Methylthioribose-1-phosphate isomerase (EC 5.3.1.23)">
     <score><val>-71.3333</val><val>-57.7941</val></score>
     <magnitude><val>12</val><val>59</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="2,3-diketo-5-methylthiopentyl-1-phosphate enolase-phosphatase (EC 3.1.3.77)">
     <score><val>-17.5</val><val>-28.5</val></score>
     <magnitude><val>2</val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="2,3-diketo-5-methylthiopentyl-1-phosphate enolase">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>2</val></magnitude>
     <score><val>-17.5</val><val>-28.5</val></score>
    </node>
    <node name="1,2-dihydroxy-3-keto-5-methylthiopentene dioxygenase (EC 1.13.11.54)">
     <magnitude><val></val><val>4</val></magnitude>
     <score><val>0</val><val>-18.75</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Methylthioribulose-1-phosphate dehydratase (EC 4.2.1.109)">
     <score><val>0</val><val>-3</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="5&#39;-methylthioadenosine phosphorylase (EC 2.4.2.28)">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>31</val></magnitude>
     <score><val>-36.8</val><val>-77.5</val></score>
    </node>
   </node>
   <node name="Lysine biosynthesis AAA pathway 2">
    <score><val>-35.2179</val><val>-82.7661</val></score>
    <magnitude><val>24</val><val>240</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Homoisocitrate dehydrogenase (EC 1.1.1.87)">
     <score><val>0</val><val>-78.9091</val></score>
     <magnitude><val></val><val>33</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Lysine biosynthesis protein LysX">
     <magnitude><val>2</val><val>16</val></magnitude>
     <score><val>-31</val><val>-64.6667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="N-acetyl-gamma-aminoadipyl-phosphate reductase (EC 1.2.1.-)">
     <score><val>-35.7143</val><val>-72.8421</val></score>
     <magnitude><val>9</val><val>31</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="N-acetyl-lysine aminotransferase (EC 2.6.1.-)">
     <score><val>-31.8</val><val>-77.4545</val></score>
     <magnitude><val>6</val><val>33</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Homocitrate synthase (EC 2.3.3.14)">
     <magnitude><val>4</val><val>38</val></magnitude>
     <score><val>-60.5</val><val>-94.0909</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Homoaconitase small subunit (EC 4.2.1.36)">
     <score><val>0</val><val>-54.6</val></score>
     <magnitude><val></val><val>13</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="N-acetyl-lysine deacetylase (EC 3.5.1.-)">
     <magnitude><val>2</val><val>28</val></magnitude>
     <score><val>-8</val><val>-84.3333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetylaminoadipate kinase (EC 2.7.2.-)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-70.3333</val></score>
     <magnitude><val></val><val>22</val></magnitude>
    </node>
    <node name="Homoaconitase large subunit (EC 4.2.1.36)">
     <rank><val>function</val></rank>
     <score><val>-13</val><val>-123.737</val></score>
     <magnitude><val>1</val><val>26</val></magnitude>
    </node>
   </node>
   <node name="Methionine Biosynthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>55</val><val>228</val></magnitude>
    <score><val>-30.5578</val><val>-66.6396</val></score>
    <node name="5,10-methylenetetrahydrofolate reductase (EC 1.5.1.20)">
     <magnitude><val>3</val><val>6</val></magnitude>
     <score><val>-9</val><val>-7.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Homoserine kinase (EC 2.7.1.39)">
     <rank><val>function</val></rank>
     <score><val>-9.75</val><val>-65.125</val></score>
     <magnitude><val>4</val><val>36</val></magnitude>
    </node>
    <node name="Cysteine synthase (EC 2.5.1.47)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-22.5</val></score>
     <magnitude><val></val><val>5</val></magnitude>
    </node>
    <node name="S-adenosylmethionine synthetase (EC 2.5.1.6)">
     <magnitude><val>4</val><val>7</val></magnitude>
     <score><val>-49.6667</val><val>-73.1429</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cystathionine gamma-synthase (EC 2.5.1.48)">
     <rank><val>function</val></rank>
     <score><val>-17.1818</val><val>-62.12</val></score>
     <magnitude><val>13</val><val>42</val></magnitude>
    </node>
    <node name="O-acetylhomoserine sulfhydrylase (EC 2.5.1.49)">
     <score><val>0</val><val>-81.6111</val></score>
     <magnitude><val></val><val>33</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cystathionine beta-synthase (EC 4.2.1.22)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-22</val></score>
    </node>
    <node name="O-succinylhomoserine sulfhydrylase (EC 2.5.1.48)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>33</val></magnitude>
     <score><val>0</val><val>-81.6111</val></score>
    </node>
    <node name="Homoserine O-acetyltransferase (EC 2.3.1.31)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-10</val></score>
    </node>
    <node name="Homoserine dehydrogenase (EC 1.1.1.3)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>23</val></magnitude>
     <score><val>-14.8571</val><val>-62.7692</val></score>
    </node>
    <node name="Archaeal S-adenosylmethionine synthetase (EC 2.5.1.6)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>7</val></magnitude>
     <score><val>-49.6667</val><val>-73.1429</val></score>
    </node>
    <node name="Adenosylhomocysteinase (EC 3.3.1.1)">
     <magnitude><val>9</val><val>4</val></magnitude>
     <score><val>-66.125</val><val>-53.6667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="5-methyltetrahydropteroyltriglutamate--homocysteine methyltransferase (EC 2.1.1.14)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>31</val></magnitude>
     <score><val>-30</val><val>-64.2609</val></score>
    </node>
   </node>
   <node name="Cysteine Biosynthesis">
    <score><val>-34</val><val>-66.2601</val></score>
    <magnitude><val>1</val><val>49</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Cysteine synthase A (EC 2.5.1.47)">
     <score><val>0</val><val>-57.9333</val></score>
     <magnitude><val></val><val>18</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Adenylylsulfate kinase (EC 2.7.1.25)">
     <rank><val>function</val></rank>
     <score><val>-34</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Phosphoadenylyl-sulfate reductase [thioredoxin] (EC 1.8.4.8)">
     <score><val>0</val><val>-82.7778</val></score>
     <magnitude><val></val><val>25</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cystathionine beta-synthase (EC 4.2.1.22)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-22</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cysteine synthase (EC 2.5.1.47)">
     <score><val>0</val><val>-22.5</val></score>
     <magnitude><val></val><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Lysine degradation">
    <magnitude><val>1</val></magnitude>
    <score><val>-9</val></score>
    <rank><val>level 3</val></rank>
    <node name="Lysine decarboxylase (EC 4.1.1.18)">
     <rank><val>function</val></rank>
     <score><val>-9</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Threonine degradation">
    <rank><val>level 3</val></rank>
    <score><val>-33.125</val><val>-70.0285</val></score>
    <magnitude><val>8</val><val>48</val></magnitude>
    <node name="L-threonine 3-dehydrogenase (EC 1.1.1.103)">
     <magnitude><val>2</val></magnitude>
     <score><val>-26.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Threonine dehydratase, catabolic (EC 4.3.1.19)">
     <score><val>-35.3333</val><val>-70.5185</val></score>
     <magnitude><val>6</val><val>47</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Monoamine oxidase (1.4.3.4)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-47</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Threonine anaerobic catabolism gene cluster">
    <magnitude><val>6</val><val>47</val></magnitude>
    <score><val>-35.3333</val><val>-70.5185</val></score>
    <rank><val>level 3</val></rank>
    <node name="Threonine dehydratase, catabolic (EC 4.3.1.19)">
     <rank><val>function</val></rank>
     <score><val>-35.3333</val><val>-70.5185</val></score>
     <magnitude><val>6</val><val>47</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Branched-chain amino acids">
   <score><val>-45.0843</val><val>-84.9705</val></score>
   <magnitude><val>419</val><val>2291</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Branched chain amino acid degradation regulons">
    <rank><val>level 3</val></rank>
    <magnitude><val>63</val><val>358</val></magnitude>
    <score><val>-41.8492</val><val>-90.6738</val></score>
    <node name="Branched-chain alpha-keto acid dehydrogenase, E1 component, alpha subunit (EC 1.2.4.4)">
     <score><val>-41.6667</val><val>-106.417</val></score>
     <magnitude><val>14</val><val>89</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Isovaleryl-CoA dehydrogenase (EC 1.3.99.10)">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>43</val></magnitude>
     <score><val>-40.7</val><val>-68.303</val></score>
    </node>
    <node name="Branched-chain amino acid aminotransferase (EC 2.6.1.42)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-39</val></score>
    </node>
    <node name="Dihydrolipoamide acyltransferase component of branched-chain alpha-keto acid dehydrogenase complex (EC 2.3.1.168)">
     <rank><val>function</val></rank>
     <score><val>-28</val><val>-21.4</val></score>
     <magnitude><val>1</val><val>7</val></magnitude>
    </node>
    <node name="Methylcrotonyl-CoA carboxylase biotin-containing subunit (EC 6.4.1.4)">
     <score><val>-43</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Electron transfer flavoprotein, alpha subunit">
     <rank><val>function</val></rank>
     <score><val>-32.1667</val><val>-83</val></score>
     <magnitude><val>13</val><val>85</val></magnitude>
    </node>
    <node name="Acetoacetyl-CoA synthetase (EC 6.2.1.16)">
     <score><val>-37</val></score>
     <magnitude><val>4</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Succinyl-CoA:3-ketoacid-coenzyme A transferase subunit B (EC 2.8.3.5)">
     <rank><val>function</val></rank>
     <score><val>-79</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Transcriptional regulator BkdR of isoleucine and valine catabolism operon">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-41</val></score>
    </node>
    <node name="Electron transfer flavoprotein, beta subunit">
     <magnitude><val></val><val>36</val></magnitude>
     <score><val>0</val><val>-80.9524</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Branched-chain alpha-keto acid dehydrogenase, E1 component, beta subunit (EC 1.2.4.4)">
     <score><val>-58.3077</val><val>-102.927</val></score>
     <magnitude><val>13</val><val>96</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Hydroxymethylglutaryl-CoA lyase (EC 4.1.3.4)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-47</val></score>
    </node>
    <node name="3-hydroxyisobutyrate dehydrogenase (EC 1.1.1.31)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-12</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
   </node>
   <node name="HMG CoA Synthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>17</val><val>43</val></magnitude>
    <score><val>-40.7059</val><val>-68.303</val></score>
    <node name="Hydroxymethylglutaryl-CoA lyase (EC 4.1.3.4)">
     <score><val>-47</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Isovaleryl-CoA dehydrogenase (EC 1.3.99.10)">
     <magnitude><val>10</val><val>43</val></magnitude>
     <score><val>-40.7</val><val>-68.303</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetoacetyl-CoA synthetase (EC 6.2.1.16)">
     <magnitude><val>4</val></magnitude>
     <score><val>-37</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Methylcrotonyl-CoA carboxylase biotin-containing subunit (EC 6.4.1.4)">
     <rank><val>function</val></rank>
     <score><val>-43</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Leucine Degradation and HMG-CoA Metabolism">
    <score><val>-48.3806</val><val>-88.261</val></score>
    <magnitude><val>67</val><val>309</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Hydroxymethylglutaryl-CoA lyase (EC 4.1.3.4)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-47</val></score>
    </node>
    <node name="Branched-chain alpha-keto acid dehydrogenase, E1 component, beta subunit (EC 1.2.4.4)">
     <magnitude><val>13</val><val>96</val></magnitude>
     <score><val>-58.3077</val><val>-102.927</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Succinyl-CoA:3-ketoacid-coenzyme A transferase subunit B (EC 2.8.3.5)">
     <magnitude><val>1</val></magnitude>
     <score><val>-79</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Hydroxymethylglutaryl-CoA reductase (EC 1.1.1.34)">
     <magnitude><val>5</val><val>17</val></magnitude>
     <score><val>-20.3333</val><val>-47.6154</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrolipoamide dehydrogenase of branched-chain alpha-keto acid dehydrogenase (EC 1.8.1.4)">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>16</val></magnitude>
     <score><val>-45.75</val><val>-46.8571</val></score>
    </node>
    <node name="Methylcrotonyl-CoA carboxylase biotin-containing subunit (EC 6.4.1.4)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-43</val></score>
    </node>
    <node name="Acetoacetyl-CoA synthetase (EC 6.2.1.16)">
     <magnitude><val>4</val></magnitude>
     <score><val>-37</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Branched-chain amino acid aminotransferase (EC 2.6.1.42)">
     <score><val>-39</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Hydroxymethylglutaryl-CoA synthase (EC 2.3.3.10)">
     <magnitude><val>4</val><val>31</val></magnitude>
     <score><val>-116</val><val>-91.2105</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Isovaleryl-CoA dehydrogenase (EC 1.3.99.10)">
     <score><val>-40.7</val><val>-68.303</val></score>
     <magnitude><val>10</val><val>43</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrolipoamide acyltransferase component of branched-chain alpha-keto acid dehydrogenase complex (EC 2.3.1.168)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-28</val><val>-21.4</val></score>
    </node>
    <node name="Dihydrolipoamide dehydrogenase (EC 1.8.1.4)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>10</val></magnitude>
     <score><val>0</val><val>-44.7</val></score>
    </node>
    <node name="Branched-chain alpha-keto acid dehydrogenase, E1 component, alpha subunit (EC 1.2.4.4)">
     <rank><val>function</val></rank>
     <magnitude><val>14</val><val>89</val></magnitude>
     <score><val>-41.6667</val><val>-106.417</val></score>
    </node>
   </node>
   <node name="Isoleucine degradation">
    <rank><val>level 3</val></rank>
    <magnitude><val>110</val><val>494</val></magnitude>
    <score><val>-46.5267</val><val>-84.2364</val></score>
    <node name="Branched-chain alpha-keto acid dehydrogenase, E1 component, beta subunit (EC 1.2.4.4)">
     <rank><val>function</val></rank>
     <magnitude><val>13</val><val>96</val></magnitude>
     <score><val>-58.3077</val><val>-102.927</val></score>
    </node>
    <node name="Transcriptional regulator BkdR of isoleucine and valine catabolism operon">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-41</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="3-hydroxyacyl-CoA dehydrogenase (EC 1.1.1.35)">
     <rank><val>function</val></rank>
     <magnitude><val>13</val><val>110</val></magnitude>
     <score><val>-36.6667</val><val>-79.2676</val></score>
    </node>
    <node name="3-ketoacyl-CoA thiolase (EC 2.3.1.16)">
     <score><val>-49.92</val><val>-76.8312</val></score>
     <magnitude><val>27</val><val>112</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrolipoamide dehydrogenase of branched-chain alpha-keto acid dehydrogenase (EC 1.8.1.4)">
     <rank><val>function</val></rank>
     <score><val>-45.75</val><val>-46.8571</val></score>
     <magnitude><val>10</val><val>16</val></magnitude>
    </node>
    <node name="Acyl-CoA dehydrogenase, short-chain specific (EC 1.3.99.2)">
     <score><val>-29.8</val><val>-69</val></score>
     <magnitude><val>5</val><val>32</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Butyryl-CoA dehydrogenase (EC 1.3.99.2)">
     <rank><val>function</val></rank>
     <score><val>-54.6</val><val>-56.4444</val></score>
     <magnitude><val>21</val><val>25</val></magnitude>
    </node>
    <node name="Dihydrolipoamide acyltransferase component of branched-chain alpha-keto acid dehydrogenase complex (EC 2.3.1.168)">
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-28</val><val>-21.4</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Branched-chain amino acid aminotransferase (EC 2.6.1.42)">
     <magnitude><val>2</val></magnitude>
     <score><val>-39</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Branched-chain alpha-keto acid dehydrogenase, E1 component, alpha subunit (EC 1.2.4.4)">
     <rank><val>function</val></rank>
     <magnitude><val>14</val><val>89</val></magnitude>
     <score><val>-41.6667</val><val>-106.417</val></score>
    </node>
    <node name="Enoyl-CoA hydratase (EC 4.2.1.17)">
     <rank><val>function</val></rank>
     <score><val>-23.25</val><val>-62.75</val></score>
     <magnitude><val>4</val><val>6</val></magnitude>
    </node>
   </node>
   <node name="Leucine Biosynthesis">
    <score><val>-40.7206</val><val>-81.7204</val></score>
    <magnitude><val>21</val><val>245</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="2-isopropylmalate synthase (EC 2.3.3.13)">
     <score><val>-46.4</val><val>-81.1194</val></score>
     <magnitude><val>7</val><val>109</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="3-isopropylmalate dehydratase small subunit (EC 4.2.1.33)">
     <rank><val>function</val></rank>
     <score><val>-24.3333</val><val>-51.0526</val></score>
     <magnitude><val>4</val><val>29</val></magnitude>
    </node>
    <node name="Branched-chain amino acid aminotransferase (EC 2.6.1.42)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-39</val></score>
    </node>
    <node name="3-isopropylmalate dehydrogenase (EC 1.1.1.85)">
     <rank><val>function</val></rank>
     <score><val>-19</val><val>-76.3913</val></score>
     <magnitude><val>2</val><val>34</val></magnitude>
    </node>
    <node name="3-isopropylmalate dehydratase large subunit (EC 4.2.1.33)">
     <magnitude><val>6</val><val>73</val></magnitude>
     <score><val>-52.8333</val><val>-97.283</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Branched-Chain Amino Acid Biosynthesis">
    <rank><val>level 3</val></rank>
    <score><val>-41.0949</val><val>-83.2538</val></score>
    <magnitude><val>33</val><val>379</val></magnitude>
    <node name="(R)-citramalate synthase (EC 2.3.1.182)">
     <rank><val>function</val></rank>
     <score><val>-38</val></score>
     <magnitude><val>4</val></magnitude>
    </node>
    <node name="3-isopropylmalate dehydratase small subunit (EC 4.2.1.33)">
     <rank><val>function</val></rank>
     <score><val>-24.3333</val><val>-51.0526</val></score>
     <magnitude><val>4</val><val>29</val></magnitude>
    </node>
    <node name="Ketol-acid reductoisomerase (EC 1.1.1.86)">
     <magnitude><val>1</val><val>28</val></magnitude>
     <score><val>-97</val><val>-80.2353</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2-isopropylmalate synthase (EC 2.3.3.13)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>109</val></magnitude>
     <score><val>-46.4</val><val>-81.1194</val></score>
    </node>
    <node name="Acetolactate synthase small subunit (EC 2.2.1.6)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-43.5</val></score>
    </node>
    <node name="3-isopropylmalate dehydrogenase (EC 1.1.1.85)">
     <magnitude><val>2</val><val>34</val></magnitude>
     <score><val>-19</val><val>-76.3913</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetolactate synthase small subunit (EC 2.2.1.6), predicted">
     <score><val>0</val><val>-43.5</val></score>
     <magnitude><val></val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydroxy-acid dehydratase (EC 4.2.1.9)">
     <magnitude><val></val><val>53</val></magnitude>
     <score><val>0</val><val>-107.731</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetolactate synthase large subunit (EC 2.2.1.6)">
     <score><val>-40</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Branched-chain amino acid aminotransferase (EC 2.6.1.42)">
     <rank><val>function</val></rank>
     <score><val>-39</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Acetolactate synthase small subunit (EC 2.2.1.6), predicted, Archaeal type">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-43.5</val></score>
    </node>
    <node name="Threonine dehydratase, catabolic (EC 4.3.1.19)">
     <rank><val>function</val></rank>
     <score><val>-35.3333</val><val>-70.5185</val></score>
     <magnitude><val>6</val><val>47</val></magnitude>
    </node>
    <node name="3-isopropylmalate dehydratase large subunit (EC 4.2.1.33)">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>73</val></magnitude>
     <score><val>-52.8333</val><val>-97.283</val></score>
    </node>
   </node>
   <node name="Ketoisovalerate oxidoreductase">
    <score><val>-48.9286</val><val>-71.4049</val></score>
    <magnitude><val>28</val><val>112</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Ketoisovalerate oxidoreductase subunit VorA (EC 1.2.7.7)">
     <rank><val>function</val></rank>
     <score><val>-81.5</val><val>-75</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
    <node name="conserved protein associated with acetyl-CoA C-acyltransferase">
     <score><val>-13</val><val>-26.1667</val></score>
     <magnitude><val>1</val><val>7</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Hydroxymethylglutaryl-CoA synthase (EC 2.3.3.10)">
     <score><val>-116</val><val>-91.2105</val></score>
     <magnitude><val>4</val><val>31</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Ketoisovalerate oxidoreductase subunit VorB (EC 1.2.7.7)">
     <rank><val>function</val></rank>
     <score><val>-30</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Acetyl-coenzyme A synthetase (EC 6.2.1.1)">
     <score><val>-37.5</val><val>-67.283</val></score>
     <magnitude><val>16</val><val>73</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="nonspecific lipid-transfer protein (acethyl CoA synthetase)">
     <magnitude><val>4</val></magnitude>
     <score><val>-25</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Valine degradation">
    <magnitude><val>80</val><val>351</val></magnitude>
    <score><val>-45.2638</val><val>-87.7825</val></score>
    <rank><val>level 3</val></rank>
    <node name="Branched-chain amino acid aminotransferase (EC 2.6.1.42)">
     <score><val>-39</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Butyryl-CoA dehydrogenase (EC 1.3.99.2)">
     <rank><val>function</val></rank>
     <magnitude><val>21</val><val>25</val></magnitude>
     <score><val>-54.6</val><val>-56.4444</val></score>
    </node>
    <node name="Dihydrolipoamide acyltransferase component of branched-chain alpha-keto acid dehydrogenase complex (EC 2.3.1.168)">
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-28</val><val>-21.4</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Enoyl-CoA hydratase (EC 4.2.1.17)">
     <rank><val>function</val></rank>
     <score><val>-23.25</val><val>-62.75</val></score>
     <magnitude><val>4</val><val>6</val></magnitude>
    </node>
    <node name="Branched-chain alpha-keto acid dehydrogenase, E1 component, alpha subunit (EC 1.2.4.4)">
     <magnitude><val>14</val><val>89</val></magnitude>
     <score><val>-41.6667</val><val>-106.417</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Branched-chain alpha-keto acid dehydrogenase, E1 component, beta subunit (EC 1.2.4.4)">
     <score><val>-58.3077</val><val>-102.927</val></score>
     <magnitude><val>13</val><val>96</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Transcriptional regulator BkdR of isoleucine and valine catabolism operon">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-41</val></score>
    </node>
    <node name="3-hydroxyisobutyrate dehydrogenase (EC 1.1.1.31)">
     <magnitude><val>2</val><val>1</val></magnitude>
     <score><val>0</val><val>-12</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrolipoamide dehydrogenase of branched-chain alpha-keto acid dehydrogenase (EC 1.8.1.4)">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>16</val></magnitude>
     <score><val>-45.75</val><val>-46.8571</val></score>
    </node>
    <node name="3-hydroxyacyl-CoA dehydrogenase (EC 1.1.1.35)">
     <score><val>-36.6667</val><val>-79.2676</val></score>
     <magnitude><val>13</val><val>110</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Creatine and Creatinine Degradation">
   <score><val>-36.1429</val><val>-78.8723</val></score>
   <magnitude><val>7</val><val>119</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Creatinine amidohydrolase (EC 3.5.2.10)">
    <score><val>-6</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="N-methylhydantoinase A (EC 3.5.2.14)">
    <score><val>-46</val><val>-87</val></score>
    <magnitude><val>1</val><val>61</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="N-carbamoylsarcosine amidase (EC 3.5.1.59)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-33</val></score>
   </node>
   <node name="Sarcosine dehydrogenase (EC 1.5.99.1)">
    <score><val>-62</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="N-methylhydantoinase B (EC 3.5.2.14)">
    <magnitude><val>2</val><val>56</val></magnitude>
    <score><val>-22</val><val>-71.6216</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Cytosine deaminase (EC 3.5.4.1)">
    <magnitude><val></val><val>2</val></magnitude>
    <score><val>0</val><val>-34</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Alanine, serine, and glycine">
   <rank><val>level 2</val></rank>
   <magnitude><val>192</val><val>719</val></magnitude>
   <score><val>-44.8118</val><val>-66.9762</val></score>
   <node name="Glycine cleavage system">
    <magnitude><val>31</val><val>100</val></magnitude>
    <score><val>-36.9234</val><val>-58.6555</val></score>
    <rank><val>level 3</val></rank>
    <node name="Glycine dehydrogenase [decarboxylating] (glycine cleavage system P1 protein) (EC 1.4.4.2)">
     <magnitude><val>9</val><val>39</val></magnitude>
     <score><val>-40.375</val><val>-71.381</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrolipoamide dehydrogenase (EC 1.8.1.4)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-44.7</val></score>
     <magnitude><val></val><val>10</val></magnitude>
    </node>
    <node name="Glycine dehydrogenase [decarboxylating] (glycine cleavage system P2 protein) (EC 1.4.4.2)">
     <score><val>-57.5714</val><val>-32</val></score>
     <magnitude><val>7</val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycine cleavage system H protein">
     <score><val>-33.4</val><val>-30.375</val></score>
     <magnitude><val>5</val><val>11</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Aminomethyltransferase (glycine cleavage system T protein) (EC 2.1.2.10)">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>38</val></magnitude>
     <score><val>-21.125</val><val>-58.8571</val></score>
    </node>
   </node>
   <node name="Glycine and Serine Utilization">
    <rank><val>level 3</val></rank>
    <magnitude><val>85</val><val>318</val></magnitude>
    <score><val>-46.8822</val><val>-67.9601</val></score>
    <node name="Glycine dehydrogenase [decarboxylating] (glycine cleavage system P1 protein) (EC 1.4.4.2)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>39</val></magnitude>
     <score><val>-40.375</val><val>-71.381</val></score>
    </node>
    <node name="L-serine dehydratase (EC 4.3.1.17)">
     <rank><val>function</val></rank>
     <score><val>-107</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Serine hydroxymethyltransferase (EC 2.1.2.1)">
     <rank><val>function</val></rank>
     <score><val>-76</val><val>-79.3077</val></score>
     <magnitude><val>11</val><val>67</val></magnitude>
    </node>
    <node name="Glycine cleavage system H protein">
     <score><val>-33.4</val><val>-30.375</val></score>
     <magnitude><val>5</val><val>11</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="2-amino-3-ketobutyrate coenzyme A ligase (EC 2.3.1.29)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>7</val></magnitude>
     <score><val>-64.4286</val><val>-48.4</val></score>
    </node>
    <node name="Aminomethyltransferase (glycine cleavage system T protein) (EC 2.1.2.10)">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>38</val></magnitude>
     <score><val>-21.125</val><val>-58.8571</val></score>
    </node>
    <node name="Seryl-tRNA synthetase (EC 6.1.1.11)">
     <rank><val>function</val></rank>
     <magnitude><val>12</val><val>17</val></magnitude>
     <score><val>-63.3636</val><val>-81.4286</val></score>
    </node>
    <node name="Threonine dehydratase, catabolic (EC 4.3.1.19)">
     <rank><val>function</val></rank>
     <score><val>-35.3333</val><val>-70.5185</val></score>
     <magnitude><val>6</val><val>47</val></magnitude>
    </node>
    <node name="Glycine dehydrogenase [decarboxylating] (glycine cleavage system P2 protein) (EC 1.4.4.2)">
     <rank><val>function</val></rank>
     <score><val>-57.5714</val><val>-32</val></score>
     <magnitude><val>7</val><val>2</val></magnitude>
    </node>
    <node name="Cystathionine beta-synthase (EC 4.2.1.22)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-22</val></score>
     <magnitude><val></val><val>1</val></magnitude>
    </node>
    <node name="Monoamine oxidase (1.4.3.4)">
     <score><val>0</val><val>-47</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoserine phosphatase (EC 3.1.3.3)">
     <score><val>-31</val><val>-60.9706</val></score>
     <magnitude><val>11</val><val>53</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycerate kinase (EC 2.7.1.31)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-7</val></score>
    </node>
    <node name="D-3-phosphoglycerate dehydrogenase (EC 1.1.1.95)">
     <magnitude><val>5</val><val>35</val></magnitude>
     <score><val>-25.2</val><val>-72.6071</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Serine Biosynthesis">
    <magnitude><val>28</val><val>171</val></magnitude>
    <score><val>-46.7143</val><val>-72.2155</val></score>
    <rank><val>level 3</val></rank>
    <node name="Serine hydroxymethyltransferase (EC 2.1.2.1)">
     <rank><val>function</val></rank>
     <score><val>-76</val><val>-79.3077</val></score>
     <magnitude><val>11</val><val>67</val></magnitude>
    </node>
    <node name="Phosphoserine aminotransferase, putative (EC 2.6.1.52)">
     <score><val>-5</val><val>-78.9091</val></score>
     <magnitude><val>1</val><val>16</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoserine phosphatase (EC 3.1.3.3)">
     <rank><val>function</val></rank>
     <magnitude><val>11</val><val>53</val></magnitude>
     <score><val>-31</val><val>-60.9706</val></score>
    </node>
    <node name="D-3-phosphoglycerate dehydrogenase (EC 1.1.1.95)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>35</val></magnitude>
     <score><val>-25.2</val><val>-72.6071</val></score>
    </node>
   </node>
   <node name="Glycine Biosynthesis">
    <score><val>-67</val><val>-76.384</val></score>
    <magnitude><val>20</val><val>74</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="L-threonine 3-dehydrogenase (EC 1.1.1.103)">
     <rank><val>function</val></rank>
     <score><val>-26.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Serine hydroxymethyltransferase (EC 2.1.2.1)">
     <magnitude><val>11</val><val>67</val></magnitude>
     <score><val>-76</val><val>-79.3077</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2-amino-3-ketobutyrate coenzyme A ligase (EC 2.3.1.29)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>7</val></magnitude>
     <score><val>-64.4286</val><val>-48.4</val></score>
    </node>
   </node>
   <node name="Alanine biosynthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>28</val><val>56</val></magnitude>
    <score><val>-29.5089</val><val>-47.817</val></score>
    <node name="Branched-chain amino acid aminotransferase (EC 2.6.1.42)">
     <score><val>-39</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Ferredoxin, 2Fe-2S">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7), SufS subfamily">
     <rank><val>function</val></rank>
     <score><val>-31</val><val>-37.6667</val></score>
     <magnitude><val>9</val><val>18</val></magnitude>
    </node>
    <node name="Alanine racemase (EC 5.1.1.1)">
     <score><val>-16</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7)">
     <rank><val>function</val></rank>
     <score><val>-29.4167</val><val>-52.625</val></score>
     <magnitude><val>15</val><val>38</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Arginine; urea cycle, polyamines">
   <rank><val>level 2</val></rank>
   <magnitude><val>151</val><val>816</val></magnitude>
   <score><val>-30.6397</val><val>-76.4654</val></score>
   <node name="Arginine Biosynthesis extended">
    <rank><val>level 3</val></rank>
    <score><val>-33.8016</val><val>-77.6304</val></score>
    <magnitude><val>32</val><val>262</val></magnitude>
    <node name="N-acetyl-gamma-glutamyl-phosphate reductase (EC 1.2.1.38)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>31</val></magnitude>
     <score><val>-35.7143</val><val>-72.8421</val></score>
    </node>
    <node name="Acetylglutamate kinase (EC 2.7.2.8)">
     <magnitude><val></val><val>22</val></magnitude>
     <score><val>0</val><val>-70.3333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Argininosuccinate lyase (EC 4.3.2.1)">
     <score><val>-30.6667</val><val>-72.125</val></score>
     <magnitude><val>3</val><val>27</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetylornithine aminotransferase (EC 2.6.1.11)">
     <rank><val>function</val></rank>
     <score><val>-29.8333</val><val>-75</val></score>
     <magnitude><val>8</val><val>34</val></magnitude>
    </node>
    <node name="N-succinyl-L,L-diaminopimelate desuccinylase (EC 3.5.1.18)">
     <magnitude><val>1</val></magnitude>
     <score><val>-42</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ornithine carbamoyltransferase (EC 2.1.3.3)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>50</val></magnitude>
     <score><val>0</val><val>-87</val></score>
    </node>
    <node name="Argininosuccinate synthase (EC 6.3.4.5)">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>49</val></magnitude>
     <score><val>-36.5556</val><val>-87.9</val></score>
    </node>
    <node name="Acetylornithine deacetylase (EC 3.5.1.16)">
     <magnitude><val>1</val><val>49</val></magnitude>
     <score><val>-22</val><val>-68.9643</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Urease subunits">
    <rank><val>level 3</val></rank>
    <score><val>-9</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="Urease accessory protein UreG">
     <rank><val>function</val></rank>
     <score><val>-9</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Arginine and Ornithine Degradation">
    <rank><val>level 3</val></rank>
    <score><val>-30.4033</val><val>-74.9644</val></score>
    <magnitude><val>62</val><val>294</val></magnitude>
    <node name="Ornithine carbamoyltransferase (EC 2.1.3.3)">
     <score><val>0</val><val>-87</val></score>
     <magnitude><val></val><val>50</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Carbamate kinase (EC 2.7.2.2)">
     <magnitude><val>11</val><val>26</val></magnitude>
     <score><val>-26.6</val><val>-70.0526</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Delta-1-pyrroline-5-carboxylate dehydrogenase (EC 1.5.1.12)">
     <magnitude><val>21</val><val>14</val></magnitude>
     <score><val>-34.8824</val><val>-40.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="NADP-specific glutamate dehydrogenase (EC 1.4.1.4)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>47</val></magnitude>
     <score><val>-62</val><val>-91.3438</val></score>
    </node>
    <node name="Agmatinase (EC 3.5.3.11)">
     <magnitude><val>5</val><val>39</val></magnitude>
     <score><val>-22.8</val><val>-71.9545</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ornithine cyclodeaminase (EC 4.3.1.12)">
     <score><val>-8.85714</val><val>-64.6905</val></score>
     <magnitude><val>7</val><val>60</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="N-carbamoylputrescine amidase (3.5.1.53)">
     <score><val>-54</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Arginine deiminase (EC 3.5.3.6)">
     <score><val>-14.875</val><val>-61.4359</val></score>
     <magnitude><val>9</val><val>48</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Arginine/ornithine antiporter ArcD">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-137.143</val></score>
     <magnitude><val></val><val>10</val></magnitude>
    </node>
   </node>
   <node name="Arginine Deiminase Pathway">
    <score><val>-21.3238</val><val>-78.2964</val></score>
    <magnitude><val>20</val><val>134</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Carbamate kinase (EC 2.7.2.2)">
     <score><val>-26.6</val><val>-70.0526</val></score>
     <magnitude><val>11</val><val>26</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Arginine deiminase (EC 3.5.3.6)">
     <magnitude><val>9</val><val>48</val></magnitude>
     <score><val>-14.875</val><val>-61.4359</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ornithine carbamoyltransferase (EC 2.1.3.3)">
     <score><val>0</val><val>-87</val></score>
     <magnitude><val></val><val>50</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Arginine/ornithine antiporter ArcD">
     <score><val>0</val><val>-137.143</val></score>
     <magnitude><val></val><val>10</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Urea decomposition">
    <score><val>-9.66667</val><val>-6</val></score>
    <magnitude><val>3</val><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Urease accessory protein UreG">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-9</val></score>
    </node>
    <node name="Eukaryotic-type low-affinity urea transporter">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-15</val></score>
    </node>
    <node name="Urea ABC transporter, ATPase protein UrtE">
     <score><val>-5</val><val>-6</val></score>
     <magnitude><val>1</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Polyamine Metabolism">
    <rank><val>level 3</val></rank>
    <magnitude><val>32</val><val>124</val></magnitude>
    <score><val>-36.3583</val><val>-76.6155</val></score>
    <node name="Putrescine transport ATP-binding protein PotG (TC 3.A.1.11.2)">
     <score><val>0</val><val>-5</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Agmatinase (EC 3.5.3.11)">
     <magnitude><val>5</val><val>39</val></magnitude>
     <score><val>-22.8</val><val>-71.9545</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Putrescine/proton symporter, putrescine/ornithine antiporter PotE">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-90</val></score>
    </node>
    <node name="Carbamate kinase (EC 2.7.2.2)">
     <rank><val>function</val></rank>
     <score><val>-26.6</val><val>-70.0526</val></score>
     <magnitude><val>11</val><val>26</val></magnitude>
    </node>
    <node name="Arginine/ornithine antiporter">
     <score><val>0</val><val>-137.143</val></score>
     <magnitude><val></val><val>10</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Arginine/ornithine antiporter ArcD">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-137.143</val></score>
     <magnitude><val></val><val>10</val></magnitude>
    </node>
    <node name="4-aminobutyraldehyde dehydrogenase (EC 1.2.1.19)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-19</val></score>
    </node>
    <node name="Spermidine synthase (EC 2.5.1.16)">
     <magnitude><val>7</val><val>22</val></magnitude>
     <score><val>-55.6667</val><val>-63.6</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="S-adenosylmethionine decarboxylase proenzyme (EC 4.1.1.50), prokaryotic class 1B">
     <magnitude><val>6</val><val>15</val></magnitude>
     <score><val>-22.2</val><val>-47.1111</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="N-carbamoylputrescine amidase (3.5.1.53)">
     <score><val>-54</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Putrescine utilization pathways">
    <rank><val>level 3</val></rank>
    <score><val>0</val><val>-19</val></score>
    <magnitude><val></val><val>1</val></magnitude>
    <node name="4-aminobutyraldehyde dehydrogenase (EC 1.2.1.19)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-19</val></score>
    </node>
   </node>
   <node name="Cyanophycin Metabolism">
    <magnitude><val>1</val></magnitude>
    <score><val>-32</val></score>
    <rank><val>level 3</val></rank>
    <node name="Asparagine synthetase [glutamine-hydrolyzing] (EC 6.3.5.4)">
     <magnitude><val>1</val></magnitude>
     <score><val>-32</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Glutamine, glutamate, aspartate, asparagine; ammonia assimilation">
   <rank><val>level 2</val></rank>
   <score><val>-51.499</val><val>-76.1002</val></score>
   <magnitude><val>110</val><val>527</val></magnitude>
   <node name="Glutamate dehydrogenases">
    <rank><val>level 3</val></rank>
    <magnitude><val>16</val><val>94</val></magnitude>
    <score><val>-62</val><val>-91.3438</val></score>
    <node name="NAD-specific glutamate dehydrogenase (EC 1.4.1.2)">
     <score><val>-62</val><val>-91.3438</val></score>
     <magnitude><val>8</val><val>47</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="NADP-specific glutamate dehydrogenase (EC 1.4.1.4)">
     <rank><val>function</val></rank>
     <score><val>-62</val><val>-91.3438</val></score>
     <magnitude><val>8</val><val>47</val></magnitude>
    </node>
   </node>
   <node name="Glutamine synthetases">
    <rank><val>level 3</val></rank>
    <magnitude><val>9</val><val>37</val></magnitude>
    <score><val>-51.2222</val><val>-79.44</val></score>
    <node name="Glutamine synthetase type I (EC 6.3.1.2)">
     <rank><val>function</val></rank>
     <score><val>-51.2222</val><val>-79.44</val></score>
     <magnitude><val>9</val><val>37</val></magnitude>
    </node>
   </node>
   <node name="Aspartate aminotransferase">
    <rank><val>level 3</val></rank>
    <magnitude><val>21</val><val>33</val></magnitude>
    <score><val>-48.4737</val><val>-49.2692</val></score>
    <node name="Aspartate aminotransferase (EC 2.6.1.1)">
     <rank><val>function</val></rank>
     <magnitude><val>21</val><val>33</val></magnitude>
     <score><val>-48.4737</val><val>-49.2692</val></score>
    </node>
   </node>
   <node name="Glutamate and Aspartate uptake in Bacteria">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-32</val></score>
    <node name="Asparagine synthetase [glutamine-hydrolyzing] (EC 6.3.5.4)">
     <magnitude><val>1</val></magnitude>
     <score><val>-32</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Glutamine, Glutamate, Aspartate and Asparagine Biosynthesis">
    <score><val>-51.9289</val><val>-71.1828</val></score>
    <magnitude><val>57</val><val>300</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Glutamate synthase [NADPH] large chain (EC 1.4.1.13)">
     <rank><val>function</val></rank>
     <score><val>-61.4</val><val>-66.619</val></score>
     <magnitude><val>5</val><val>75</val></magnitude>
    </node>
    <node name="NADP-specific glutamate dehydrogenase (EC 1.4.1.4)">
     <magnitude><val>8</val><val>47</val></magnitude>
     <score><val>-62</val><val>-91.3438</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Aspartate ammonia-lyase (EC 4.3.1.1)">
     <magnitude><val>1</val><val>8</val></magnitude>
     <score><val>-51</val><val>-49.6667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="NAD-specific glutamate dehydrogenase (EC 1.4.1.2)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>47</val></magnitude>
     <score><val>-62</val><val>-91.3438</val></score>
    </node>
    <node name="Aspartate aminotransferase (EC 2.6.1.1)">
     <score><val>-48.4737</val><val>-49.2692</val></score>
     <magnitude><val>21</val><val>33</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="L-asparaginase (EC 3.5.1.1)">
     <rank><val>function</val></rank>
     <score><val>-30</val><val>-54.2</val></score>
     <magnitude><val>2</val><val>38</val></magnitude>
    </node>
    <node name="Asparagine synthetase [glutamine-hydrolyzing] (EC 6.3.5.4)">
     <magnitude><val>1</val></magnitude>
     <score><val>-32</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ferredoxin-dependent glutamate synthase (EC 1.4.7.1)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-7</val></score>
    </node>
    <node name="Aspartate racemase (EC 5.1.1.13)">
     <score><val>0</val><val>-50</val></score>
     <magnitude><val></val><val>15</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Asparagine synthetase [glutamine-hydrolyzing] (EC 6.3.5.4) AsnH">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-32</val></score>
    </node>
    <node name="Glutamine synthetase type I (EC 6.3.1.2)">
     <rank><val>function</val></rank>
     <score><val>-51.2222</val><val>-79.44</val></score>
     <magnitude><val>9</val><val>37</val></magnitude>
    </node>
   </node>
   <node name="Poly-gamma-glutamate biosynthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>6</val><val>63</val></magnitude>
    <score><val>-33.6667</val><val>-88.8649</val></score>
    <node name="Gamma-glutamyltranspeptidase (EC 2.3.2.2)">
     <rank><val>function</val></rank>
     <score><val>-33.6667</val><val>-88.8649</val></score>
     <magnitude><val>6</val><val>63</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Aromatic amino acids and derivatives">
   <rank><val>level 2</val></rank>
   <score><val>-31.7888</val><val>-66.2374</val></score>
   <magnitude><val>200</val><val>1078</val></magnitude>
   <node name="Aromatic amino acid interconversions with aryl acids">
    <rank><val>level 3</val></rank>
    <score><val>-45.2111</val><val>-70.4934</val></score>
    <magnitude><val>18</val><val>95</val></magnitude>
    <node name="Indolepyruvate ferredoxin oxidoreductase, alpha and beta subunits">
     <magnitude><val>1</val></magnitude>
     <score><val>-25</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Indolepyruvate oxidoreductase subunit IorA (EC 1.2.7.8)">
     <rank><val>function</val></rank>
     <magnitude><val>14</val><val>63</val></magnitude>
     <score><val>-45.2</val><val>-77.8298</val></score>
    </node>
    <node name="Phenylacetate-coenzyme A ligase (EC 6.2.1.30)">
     <score><val>-68</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Indolepyruvate oxidoreductase subunit IorB (EC 1.2.7.8)">
     <score><val>-20</val><val>-56.05</val></score>
     <magnitude><val>1</val><val>32</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Indole-pyruvate oxidoreductase complex">
    <score><val>-43.52</val><val>-70.4934</val></score>
    <magnitude><val>15</val><val>95</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Indolepyruvate oxidoreductase subunit IorA (EC 1.2.7.8)">
     <rank><val>function</val></rank>
     <magnitude><val>14</val><val>63</val></magnitude>
     <score><val>-45.2</val><val>-77.8298</val></score>
    </node>
    <node name="Indolepyruvate oxidoreductase subunit IorB (EC 1.2.7.8)">
     <rank><val>function</val></rank>
     <score><val>-20</val><val>-56.05</val></score>
     <magnitude><val>1</val><val>32</val></magnitude>
    </node>
   </node>
   <node name="Phenylalanine and Tyrosine Branches from Chorismate">
    <magnitude><val>11</val><val>63</val></magnitude>
    <score><val>-40.3455</val><val>-61.7209</val></score>
    <rank><val>level 3</val></rank>
    <node name="Biosynthetic Aromatic amino acid aminotransferase alpha (EC 2.6.1.57)">
     <magnitude><val>6</val><val>28</val></magnitude>
     <score><val>-61.8</val><val>-54.5455</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Chorismate mutase I (EC 5.4.99.5)">
     <rank><val>function</val></rank>
     <score><val>-31</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Prephenate and/or arogenate dehydrogenase (unknown specificity) (EC 1.3.1.12)(EC 1.3.1.43)">
     <score><val>0</val><val>-67.5</val></score>
     <magnitude><val></val><val>16</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Prephenate dehydratase (EC 4.2.1.51)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>19</val></magnitude>
     <score><val>-10.5</val><val>-67.4286</val></score>
    </node>
   </node>
   <node name="Aromatic amino acid degradation">
    <rank><val>level 3</val></rank>
    <score><val>-26.6667</val><val>-75.359</val></score>
    <magnitude><val>9</val><val>24</val></magnitude>
    <node name="5-carboxymethyl-2-oxo-hex-3- ene-1,7-dioate decarboxylase (EC 4.1.1.68)">
     <magnitude><val>2</val><val>2</val></magnitude>
     <score><val>-8.5</val><val>-26</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2-hydroxyhepta-2,4-diene-1,7-dioate isomerase (EC 5.3.3.-)">
     <rank><val>function</val></rank>
     <score><val>-8.5</val><val>-26</val></score>
     <magnitude><val>2</val><val>2</val></magnitude>
    </node>
    <node name="4-hydroxyphenylpyruvate dioxygenase (EC 1.13.11.27)">
     <score><val>-74</val><val>-85.2308</val></score>
     <magnitude><val>1</val><val>20</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="5-carboxymethyl-2-hydroxymuconate semialdehyde dehydrogenase (EC 1.2.1.60)">
     <rank><val>function</val></rank>
     <score><val>-37</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Tryptophanase (EC 4.1.99.1)">
     <score><val>-29</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Chorismate Synthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>34</val><val>213</val></magnitude>
    <score><val>-23.6397</val><val>-59.6903</val></score>
    <node name="Chorismate mutase I (EC 5.4.99.5)">
     <rank><val>function</val></rank>
     <score><val>-31</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="3-dehydroquinate dehydratase I (EC 4.2.1.10)">
     <score><val>-6</val><val>-40.5714</val></score>
     <magnitude><val>1</val><val>11</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Shikimate 5-dehydrogenase I alpha (EC 1.1.1.25)">
     <magnitude><val>1</val><val>14</val></magnitude>
     <score><val>-4</val><val>-73.2857</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2-keto-3-deoxy-D-arabino-heptulosonate-7-phosphate synthase II (EC 2.5.1.54)">
     <magnitude><val>2</val></magnitude>
     <score><val>-80</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2-amino-3,7-dideoxy-D-threo-hept-6-ulosonate synthase (EC 2.5.1.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-15</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Prephenate dehydratase (EC 4.2.1.51)">
     <magnitude><val>4</val><val>19</val></magnitude>
     <score><val>-10.5</val><val>-67.4286</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="5-Enolpyruvylshikimate-3-phosphate synthase (EC 2.5.1.19)">
     <magnitude><val>5</val><val>31</val></magnitude>
     <score><val>-10.8</val><val>-65.75</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Shikimate kinase I (EC 2.7.1.71)">
     <score><val>-9</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Prephenate and/or arogenate dehydrogenase (unknown specificity) (EC 1.3.1.12)(EC 1.3.1.43)">
     <magnitude><val></val><val>16</val></magnitude>
     <score><val>0</val><val>-67.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Shikimate kinase II (EC 2.7.1.71)">
     <magnitude><val>1</val><val>12</val></magnitude>
     <score><val>-1</val><val>-34.375</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2-keto-3-deoxy-D-arabino-heptulosonate-7-phosphate synthase I beta (EC 2.5.1.54)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>36</val></magnitude>
     <score><val>-35.25</val><val>-57.4167</val></score>
    </node>
    <node name="3-dehydroquinate synthase (EC 4.2.3.4)">
     <magnitude><val>8</val><val>32</val></magnitude>
     <score><val>-24.625</val><val>-50.1667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Chorismate synthase (EC 4.2.3.5)">
     <magnitude><val>5</val><val>42</val></magnitude>
     <score><val>-28.75</val><val>-65.6552</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Common Pathway For Synthesis of Aromatic Compounds (DAHP synthase to chorismate)">
    <rank><val>level 3</val></rank>
    <score><val>-25.1983</val><val>-58.1623</val></score>
    <magnitude><val>29</val><val>178</val></magnitude>
    <node name="Shikimate kinase I (EC 2.7.1.71)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-9</val></score>
    </node>
    <node name="5-Enolpyruvylshikimate-3-phosphate synthase (EC 2.5.1.19)">
     <magnitude><val>5</val><val>31</val></magnitude>
     <score><val>-10.8</val><val>-65.75</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Shikimate kinase II (EC 2.7.1.71)">
     <magnitude><val>1</val><val>12</val></magnitude>
     <score><val>-1</val><val>-34.375</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Chorismate synthase (EC 4.2.3.5)">
     <rank><val>function</val></rank>
     <score><val>-28.75</val><val>-65.6552</val></score>
     <magnitude><val>5</val><val>42</val></magnitude>
    </node>
    <node name="2-keto-3-deoxy-D-arabino-heptulosonate-7-phosphate synthase I beta (EC 2.5.1.54)">
     <score><val>-35.25</val><val>-57.4167</val></score>
     <magnitude><val>4</val><val>36</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="3-dehydroquinate synthase (EC 4.2.3.4)">
     <score><val>-24.625</val><val>-50.1667</val></score>
     <magnitude><val>8</val><val>32</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="3-dehydroquinate dehydratase I (EC 4.2.1.10)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>11</val></magnitude>
     <score><val>-6</val><val>-40.5714</val></score>
    </node>
    <node name="2-amino-3,7-dideoxy-D-threo-hept-6-ulosonate synthase (EC 2.5.1.-)">
     <score><val>-15</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Shikimate 5-dehydrogenase I alpha (EC 1.1.1.25)">
     <rank><val>function</val></rank>
     <score><val>-4</val><val>-73.2857</val></score>
     <magnitude><val>1</val><val>14</val></magnitude>
    </node>
    <node name="2-keto-3-deoxy-D-arabino-heptulosonate-7-phosphate synthase II (EC 2.5.1.54)">
     <rank><val>function</val></rank>
     <score><val>-80</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
   </node>
   <node name="Chorismate: Intermediate for synthesis of Tryptophan, PAPA antibiotics, PABA, 3-hydroxyanthranilate and more.">
    <rank><val>level 3</val></rank>
    <score><val>-30.5528</val><val>-70.6709</val></score>
    <magnitude><val>46</val><val>229</val></magnitude>
    <node name="Phosphoribosylformimino-5-aminoimidazole carboxamide ribotide isomerase (EC 5.3.1.16)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>15</val></magnitude>
     <score><val>-15</val><val>-81</val></score>
    </node>
    <node name="Tryptophan synthase beta chain (EC 4.2.1.20)">
     <rank><val>function</val></rank>
     <magnitude><val>16</val><val>57</val></magnitude>
     <score><val>-52.7143</val><val>-75.5455</val></score>
    </node>
    <node name="Isochorismatase (EC 3.3.2.1)">
     <magnitude><val>3</val><val>30</val></magnitude>
     <score><val>-12.3333</val><val>-57.3333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Anthranilate synthase, amidotransferase component (EC 4.1.3.27)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>9</val></magnitude>
     <score><val>-23.75</val><val>-47.2857</val></score>
    </node>
    <node name="Isochorismate synthase (EC 5.4.4.2) of siderophore biosynthesis">
     <score><val>-6</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Isochorismate synthase (EC 5.4.4.2)">
     <rank><val>function</val></rank>
     <score><val>-6</val><val>-65.5</val></score>
     <magnitude><val>1</val><val>2</val></magnitude>
    </node>
    <node name="Tryptophan synthase beta chain like (EC 4.2.1.20)">
     <score><val>-59</val><val>-55</val></score>
     <magnitude><val>1</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Indole-3-glycerol phosphate synthase (EC 4.1.1.48)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>22</val></magnitude>
     <score><val>-16</val><val>-79.5882</val></score>
    </node>
    <node name="Anthranilate synthase, aminase component (EC 4.1.3.27)">
     <magnitude><val>7</val><val>26</val></magnitude>
     <score><val>-18.5714</val><val>-65.2857</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Anthranilate phosphoribosyltransferase (EC 2.4.2.18)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>26</val></magnitude>
     <score><val>-21</val><val>-94.2857</val></score>
    </node>
    <node name="Tryptophan synthase alpha chain (EC 4.2.1.20)">
     <score><val>-18</val><val>-63.1667</val></score>
     <magnitude><val>2</val><val>25</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylanthranilate isomerase (EC 5.3.1.24)">
     <rank><val>function</val></rank>
     <score><val>-10</val><val>-53.25</val></score>
     <magnitude><val>1</val><val>16</val></magnitude>
    </node>
   </node>
   <node name="Tryptophan synthesis">
    <score><val>-33.3534</val><val>-72.1692</val></score>
    <magnitude><val>38</val><val>181</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Tryptophan synthase beta chain (EC 4.2.1.20)">
     <rank><val>function</val></rank>
     <score><val>-52.7143</val><val>-75.5455</val></score>
     <magnitude><val>16</val><val>57</val></magnitude>
    </node>
    <node name="Anthranilate synthase, aminase component (EC 4.1.3.27)">
     <magnitude><val>7</val><val>26</val></magnitude>
     <score><val>-18.5714</val><val>-65.2857</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Indole-3-glycerol phosphate synthase (EC 4.1.1.48)">
     <score><val>-16</val><val>-79.5882</val></score>
     <magnitude><val>3</val><val>22</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Anthranilate synthase, amidotransferase component (EC 4.1.3.27)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>9</val></magnitude>
     <score><val>-23.75</val><val>-47.2857</val></score>
    </node>
    <node name="Tryptophan synthase alpha chain (EC 4.2.1.20)">
     <score><val>-18</val><val>-63.1667</val></score>
     <magnitude><val>2</val><val>25</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylanthranilate isomerase (EC 5.3.1.24)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>16</val></magnitude>
     <score><val>-10</val><val>-53.25</val></score>
    </node>
    <node name="Anthranilate phosphoribosyltransferase (EC 2.4.2.18)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>26</val></magnitude>
     <score><val>-21</val><val>-94.2857</val></score>
    </node>
   </node>
  </node>
  <node name="Proline and 4-hydroxyproline">
   <rank><val>level 2</val></rank>
   <magnitude><val>31</val><val>91</val></magnitude>
   <score><val>-40.63</val><val>-72.5872</val></score>
   <node name="Proline, 4-hydroxyproline uptake and utilization">
    <rank><val>level 3</val></rank>
    <magnitude><val>23</val><val>16</val></magnitude>
    <score><val>-33.197</val><val>-38.5625</val></score>
    <node name="Delta-1-pyrroline-5-carboxylate dehydrogenase (EC 1.5.1.12)">
     <rank><val>function</val></rank>
     <magnitude><val>21</val><val>14</val></magnitude>
     <score><val>-34.8824</val><val>-40.5</val></score>
    </node>
    <node name="Proline dehydrogenase (EC 1.5.99.8) (Proline oxidase)">
     <rank><val>function</val></rank>
     <score><val>-15.5</val><val>-25</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
    <node name="Transcriptional repressor of PutA and PutP">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-25</val></score>
    </node>
   </node>
   <node name="Proline Synthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>8</val><val>61</val></magnitude>
    <score><val>-62</val><val>-84.2753</val></score>
    <node name="NADP-specific glutamate dehydrogenase (EC 1.4.1.4)">
     <magnitude><val>8</val><val>47</val></magnitude>
     <score><val>-62</val><val>-91.3438</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Pyrroline-5-carboxylate reductase (EC 1.5.1.2)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>14</val></magnitude>
     <score><val>0</val><val>-60.5455</val></score>
    </node>
   </node>
   <node name="A Hypothetical Protein Related to Proline Metabolism">
    <magnitude><val></val><val>14</val></magnitude>
    <score><val>0</val><val>-60.5455</val></score>
    <rank><val>level 3</val></rank>
    <node name="Pyrroline-5-carboxylate reductase (EC 1.5.1.2)">
     <score><val>0</val><val>-60.5455</val></score>
     <magnitude><val></val><val>14</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
 </node>
 <node name="Sulfur Metabolism">
  <rank><val>level 1</val></rank>
  <score><val>-29.8552</val><val>-63.7527</val></score>
  <magnitude><val>70</val><val>496</val></magnitude>
  <node name="Galactosylceramide and Sulfatide metabolism">
   <magnitude><val>8</val><val>56</val></magnitude>
   <score><val>-32.75</val><val>-36.0043</val></score>
   <rank><val>level 2</val></rank>
   <node name="Beta-galactosidase (EC 3.2.1.23)">
    <rank><val>function</val></rank>
    <magnitude><val>8</val><val>55</val></magnitude>
    <score><val>-32.75</val><val>-36.5135</val></score>
   </node>
   <node name="Alpha-galactosidase (EC 3.2.1.22)">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>1</val></magnitude>
    <score><val>0</val><val>-8</val></score>
   </node>
  </node>
  <node name="Sulfur oxidation">
   <score><val>0</val><val>-15</val></score>
   <magnitude><val></val><val>2</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Sulfide dehydrogenase [flavocytochrome C] flavoprotein chain precursor (EC 1.8.2.-)">
    <rank><val>function</val></rank>
    <score><val>0</val><val>-6</val></score>
    <magnitude><val></val><val>1</val></magnitude>
   </node>
   <node name="Sulfite oxidase">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>1</val></magnitude>
    <score><val>0</val><val>-24</val></score>
   </node>
  </node>
  <node name="Thioredoxin-disulfide reductase">
   <magnitude><val>28</val><val>177</val></magnitude>
   <score><val>-38.4107</val><val>-58.9195</val></score>
   <rank><val>level 2</val></rank>
   <node name="Alkyl hydroperoxide reductase subunit C-like protein">
    <score><val>-47.1111</val><val>-53.725</val></score>
    <magnitude><val>9</val><val>58</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Thiol peroxidase, Bcp-type (EC 1.11.1.15)">
    <magnitude><val>6</val><val>28</val></magnitude>
    <score><val>-33.25</val><val>-40.6471</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Thiol peroxidase, Tpx-type (EC 1.11.1.15)">
    <score><val>-12</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Hydrogen peroxide-inducible genes activator">
    <rank><val>function</val></rank>
    <score><val>-7</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Thioredoxin reductase (EC 1.8.1.9)">
    <score><val>-39.3636</val><val>-67.8525</val></score>
    <magnitude><val>11</val><val>91</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Organic sulfur assimilation">
   <magnitude><val>23</val><val>122</val></magnitude>
   <score><val>-24.1028</val><val>-72.56</val></score>
   <rank><val>level 2</val></rank>
   <node name="Alkanesulfonate assimilation">
    <score><val>-32.25</val><val>-37.9286</val></score>
    <magnitude><val>4</val><val>17</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Putative arylsulfatase regulatory protein">
     <score><val>-16</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="ABC-type nitrate/sulfonate/bicarbonate transport system, ATPase component">
     <score><val>-33</val><val>-37.9286</val></score>
     <magnitude><val>2</val><val>17</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="sulfonate monooxygenase">
     <rank><val>function</val></rank>
     <score><val>-47</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Utilization of glutathione as a sulphur source">
    <rank><val>level 3</val></rank>
    <score><val>-33.6667</val><val>-88.8649</val></score>
    <magnitude><val>6</val><val>63</val></magnitude>
    <node name="Gamma-glutamyltranspeptidase (EC 2.3.2.2)">
     <rank><val>function</val></rank>
     <score><val>-33.6667</val><val>-88.8649</val></score>
     <magnitude><val>6</val><val>63</val></magnitude>
    </node>
   </node>
   <node name="L-Cystine Uptake and Metabolism">
    <magnitude><val>13</val><val>42</val></magnitude>
    <score><val>-17.1818</val><val>-62.12</val></score>
    <rank><val>level 3</val></rank>
    <node name="Cystathionine gamma-synthase (EC 2.5.1.48)">
     <magnitude><val>13</val><val>42</val></magnitude>
     <score><val>-17.1818</val><val>-62.12</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Inorganic sulfur assimilation">
   <magnitude><val>11</val><val>139</val></magnitude>
   <score><val>-18</val><val>-74.0576</val></score>
   <rank><val>level 2</val></rank>
   <node name="Inorganic Sulfur Assimilation">
    <rank><val>level 3</val></rank>
    <magnitude><val>11</val><val>139</val></magnitude>
    <score><val>-18</val><val>-74.0576</val></score>
    <node name="Adenylylsulfate kinase (EC 2.7.1.25)">
     <score><val>-34</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoadenylyl-sulfate reductase [thioredoxin] (EC 1.8.4.8)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-82.7778</val></score>
     <magnitude><val></val><val>25</val></magnitude>
    </node>
    <node name="Ferredoxin">
     <magnitude><val>9</val><val>61</val></magnitude>
     <score><val>-16.7778</val><val>-61.9091</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Adenylyl-sulfate reductase [thioredoxin] (EC 1.8.4.10)">
     <magnitude><val></val><val>25</val></magnitude>
     <score><val>0</val><val>-82.7778</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Sulfate adenylyltransferase, dissimilatory-type (EC 2.7.7.4)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>28</val></magnitude>
     <score><val>-13</val><val>-84.9524</val></score>
    </node>
   </node>
  </node>
 </node>
 <node name="Cell Division and Cell Cycle">
  <rank><val>level 1</val></rank>
  <score><val>-42.7561</val><val>-65.6256</val></score>
  <magnitude><val>184</val><val>466</val></magnitude>
  <node name="Cyanobacterial Circadian Clock">
   <score><val>-10</val><val>-61.7143</val></score>
   <magnitude><val>1</val><val>22</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Circadian clock protein KaiC">
    <rank><val>function</val></rank>
    <magnitude><val>1</val><val>22</val></magnitude>
    <score><val>-10</val><val>-61.7143</val></score>
   </node>
  </node>
  <node name="Macromolecular synthesis operon">
   <rank><val>level 2</val></rank>
   <magnitude><val>35</val><val>105</val></magnitude>
   <score><val>-31.8911</val><val>-81.6918</val></score>
   <node name="YgjD/Kae1/Qri7 family, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
    <magnitude><val>10</val><val>45</val></magnitude>
    <score><val>-43.4</val><val>-85.1667</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Carbamoyl-phosphate synthase small chain (EC 6.3.5.5)">
    <rank><val>function</val></rank>
    <score><val>-23</val><val>-17.5</val></score>
    <magnitude><val>6</val><val>2</val></magnitude>
   </node>
   <node name="RNA polymerase sigma factor RpoD">
    <rank><val>function</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-41</val></score>
   </node>
   <node name="Carbamoyl-phosphate synthase large chain (EC 6.3.5.5)">
    <magnitude><val>17</val><val>58</val></magnitude>
    <score><val>-27.1875</val><val>-81.2093</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Bacterial Cytoskeleton">
   <rank><val>level 2</val></rank>
   <score><val>-44.6332</val><val>-44.4094</val></score>
   <magnitude><val>51</val><val>89</val></magnitude>
   <node name="Chromosome (plasmid) partitioning protein ParB">
    <rank><val>function</val></rank>
    <score><val>-46.4286</val></score>
    <magnitude><val>7</val></magnitude>
   </node>
   <node name="Cell division protein FtsI [Peptidoglycan synthetase] (EC 2.4.1.129)">
    <rank><val>function</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-57</val></score>
   </node>
   <node name="Rod shape-determining protein MreB">
    <rank><val>function</val></rank>
    <magnitude><val>3</val></magnitude>
    <score><val>-44</val></score>
   </node>
   <node name="Septum site-determining protein MinD">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>29</val></magnitude>
    <score><val>0</val><val>-68.2</val></score>
   </node>
   <node name="Rod shape-determining protein RodA">
    <score><val>-24</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Rod shape-determining protein MreC">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-6</val></score>
   </node>
   <node name="Septum site-determining protein MinC">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-32</val></score>
   </node>
   <node name="Stage 0 sporulation protein J">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-53</val></score>
   </node>
   <node name="Cell division protein MraZ">
    <rank><val>function</val></rank>
    <score><val>-41</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Sporulation initiation inhibitor protein Soj">
    <magnitude><val>3</val></magnitude>
    <score><val>-24.5</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Chromosome (plasmid) partitioning protein ParA">
    <magnitude><val>28</val><val>43</val></magnitude>
    <score><val>-51.4211</val><val>-39.069</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Cell division protein FtsZ (EC 3.4.24.-)">
    <magnitude><val>3</val><val>17</val></magnitude>
    <score><val>-12</val><val>-17.3333</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Two cell division clusters relating to chromosome partitioning">
   <rank><val>level 2</val></rank>
   <score><val>-48.0554</val><val>-59.9842</val></score>
   <magnitude><val>73</val><val>127</val></magnitude>
   <node name="Chromosome (plasmid) partitioning protein ParB">
    <rank><val>function</val></rank>
    <magnitude><val>7</val></magnitude>
    <score><val>-46.4286</val></score>
   </node>
   <node name="Signal recognition particle receptor protein FtsY (=alpha subunit) (TC 3.A.5.1.1)">
    <rank><val>function</val></rank>
    <magnitude><val>13</val><val>31</val></magnitude>
    <score><val>-51.25</val><val>-62.16</val></score>
   </node>
   <node name="Signal recognition particle, subunit Ffh SRP54 (TC 3.A.5.1.1)">
    <rank><val>function</val></rank>
    <score><val>-66.25</val><val>-82.7097</val></score>
    <magnitude><val>12</val><val>48</val></magnitude>
   </node>
   <node name="Chromosome (plasmid) partitioning protein ParA">
    <rank><val>function</val></rank>
    <score><val>-51.4211</val><val>-39.069</val></score>
    <magnitude><val>28</val><val>43</val></magnitude>
   </node>
   <node name="Chromosome partition protein smc">
    <score><val>-21.6923</val><val>-8.2</val></score>
    <magnitude><val>13</val><val>5</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="YgjD and YeaZ">
   <magnitude><val>18</val><val>90</val></magnitude>
   <score><val>-46.6111</val><val>-85.1667</val></score>
   <rank><val>level 2</val></rank>
   <node name="p53-regulating protein kinase (human PRPK/yeast YGR262c)">
    <rank><val>function</val></rank>
    <magnitude><val>8</val><val>45</val></magnitude>
    <score><val>-50.625</val><val>-85.1667</val></score>
   </node>
   <node name="YgjD/Kae1/Qri7 family, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
    <score><val>-43.4</val><val>-85.1667</val></score>
    <magnitude><val>10</val><val>45</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Control of cell elongation - division cycle in Bacilli">
   <rank><val>level 2</val></rank>
   <score><val>-19.6</val><val>-42.75</val></score>
   <magnitude><val>6</val><val>33</val></magnitude>
   <node name="Endonuclease III (EC 4.2.99.18)">
    <score><val>-19.6</val><val>-42.75</val></score>
    <magnitude><val>6</val><val>33</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
 </node>
 <node name="Metabolism of Aromatic Compounds">
  <rank><val>level 1</val></rank>
  <score><val>-38.9441</val><val>-76.9081</val></score>
  <magnitude><val>172</val><val>614</val></magnitude>
  <node name="Aromatic Amin Catabolism">
   <score><val>0</val><val>-47</val></score>
   <magnitude><val></val><val>1</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Monoamine oxidase (1.4.3.4)">
    <rank><val>function</val></rank>
    <score><val>0</val><val>-47</val></score>
    <magnitude><val></val><val>1</val></magnitude>
   </node>
  </node>
  <node name="Anaerobic degradation of aromatic compounds">
   <rank><val>level 2</val></rank>
   <score><val>-46.6304</val><val>-78.4356</val></score>
   <magnitude><val>36</val><val>157</val></magnitude>
   <node name="Anaerobic benzoate metabolism">
    <magnitude><val>32</val><val>157</val></magnitude>
    <score><val>-47.5842</val><val>-78.4356</val></score>
    <rank><val>level 3</val></rank>
    <node name="3-hydroxybutyryl-CoA dehydrogenase (EC 1.1.1.157)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>45</val></magnitude>
     <score><val>-35.25</val><val>-82.4286</val></score>
    </node>
    <node name="Acetyl-CoA acetyltransferase (EC 2.3.1.9)">
     <rank><val>function</val></rank>
     <score><val>-49.3462</val><val>-76.8312</val></score>
     <magnitude><val>28</val><val>112</val></magnitude>
    </node>
   </node>
   <node name="Anaerobic toluene and ethylbenzene degradation">
    <score><val>-39</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Acetophenone carboxylase subunit Apc4">
     <magnitude><val>2</val></magnitude>
     <score><val>-39</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Acetophenone carboxylase 1">
    <score><val>-39</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Acetophenone carboxylase subunit Apc4">
     <rank><val>function</val></rank>
     <score><val>-39</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Metabolism of central aromatic intermediates">
   <magnitude><val>38</val><val>85</val></magnitude>
   <score><val>-34.4386</val><val>-77.5135</val></score>
   <rank><val>level 2</val></rank>
   <node name="Homogentisate pathway of aromatic compound degradation">
    <magnitude><val>9</val><val>52</val></magnitude>
    <score><val>-51.4444</val><val>-86.0803</val></score>
    <rank><val>level 3</val></rank>
    <node name="Fumarylacetoacetase (EC 3.7.1.2)">
     <magnitude><val>2</val><val>7</val></magnitude>
     <score><val>-15</val><val>-104.667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Transcriptional regulator, IclR family">
     <score><val>-135</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="4-hydroxyphenylpyruvate dioxygenase (EC 1.13.11.27)">
     <score><val>-74</val><val>-85.2308</val></score>
     <magnitude><val>1</val><val>20</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Maleylacetoacetate isomerase (EC 5.2.1.2)">
     <magnitude><val>1</val></magnitude>
     <score><val>-27</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Homogentisate 1,2-dioxygenase (EC 1.13.11.5)">
     <rank><val>function</val></rank>
     <score><val>-49.25</val><val>-81.5556</val></score>
     <magnitude><val>4</val><val>25</val></magnitude>
    </node>
   </node>
   <node name="Protocatechuate branch of beta-ketoadipate pathway">
    <rank><val>level 3</val></rank>
    <magnitude><val>5</val></magnitude>
    <score><val>-39</val></score>
    <node name="Pca regulon regulatory protein PcaR">
     <score><val>-18.5</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Succinyl-CoA:3-ketoacid-coenzyme A transferase subunit B (EC 2.8.3.5)">
     <rank><val>function</val></rank>
     <score><val>-79</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="3-oxoadipate CoA-transferase subunit B (EC 2.8.3.6)">
     <magnitude><val>1</val></magnitude>
     <score><val>-59</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Beta-ketoadipate enol-lactone hydrolase (EC 3.1.1.24)">
     <rank><val>function</val></rank>
     <score><val>-20</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="N-heterocyclic aromatic compound degradation">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-13</val></score>
    <node name="Isoquinoline 1-oxidoreductase beta subunit (EC 1.3.99.16)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-13</val></score>
    </node>
   </node>
   <node name="Salicylate and gentisate catabolism">
    <rank><val>level 3</val></rank>
    <score><val>-15.8519</val><val>-70.6596</val></score>
    <magnitude><val>9</val><val>28</val></magnitude>
    <node name="Maleylacetoacetate isomerase (EC 5.2.1.2)">
     <rank><val>function</val></rank>
     <score><val>-27</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="4-hydroxybenzoate transporter">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>2</val></magnitude>
     <score><val>-11</val><val>-10.5</val></score>
    </node>
    <node name="Fumarylacetoacetate hydrolase family protein">
     <magnitude><val>4</val><val>18</val></magnitude>
     <score><val>-11.6667</val><val>-67.1</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Gentisate 1,2-dioxygenase (EC 1.13.11.4)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-28</val></score>
    </node>
    <node name="Fumarylacetoacetase (EC 3.7.1.2)">
     <rank><val>function</val></rank>
     <score><val>-15</val><val>-104.667</val></score>
     <magnitude><val>2</val><val>7</val></magnitude>
    </node>
    <node name="salicylate esterase">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-17</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="4-Hydroxyphenylacetic acid catabolic pathway">
    <rank><val>level 3</val></rank>
    <score><val>-18</val><val>-26</val></score>
    <magnitude><val>6</val><val>4</val></magnitude>
    <node name="5-carboxymethyl-2-oxo-hex-3- ene-1,7-dioate decarboxylase (EC 4.1.1.68)">
     <rank><val>function</val></rank>
     <score><val>-8.5</val><val>-26</val></score>
     <magnitude><val>2</val><val>2</val></magnitude>
    </node>
    <node name="5-carboxymethyl-2-hydroxymuconate semialdehyde dehydrogenase (EC 1.2.1.60)">
     <rank><val>function</val></rank>
     <score><val>-37</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="2-hydroxyhepta-2,4-diene-1,7-dioate isomerase (EC 5.3.3.-)">
     <rank><val>function</val></rank>
     <score><val>-8.5</val><val>-26</val></score>
     <magnitude><val>2</val><val>2</val></magnitude>
    </node>
   </node>
   <node name="Catechol branch of beta-ketoadipate pathway">
    <magnitude><val>5</val><val>1</val></magnitude>
    <score><val>-48.2</val><val>-30</val></score>
    <rank><val>level 3</val></rank>
    <node name="Succinyl-CoA:3-ketoacid-coenzyme A transferase subunit B (EC 2.8.3.5)">
     <magnitude><val>1</val></magnitude>
     <score><val>-79</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="3-oxoadipate CoA-transferase subunit B (EC 2.8.3.6)">
     <magnitude><val>1</val></magnitude>
     <score><val>-59</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="mandelate racemase/muconate lactonizing enzyme family protein">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>1</val></magnitude>
     <score><val>-27</val><val>-30</val></score>
    </node>
    <node name="Beta-ketoadipate enol-lactone hydrolase (EC 3.1.1.24)">
     <score><val>-20</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Muconate cycloisomerase (EC 5.5.1.1)">
     <score><val>-56</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Central meta-cleavage pathway of aromatic compound degradation">
    <magnitude><val>3</val></magnitude>
    <score><val>-48.6667</val></score>
    <rank><val>level 3</val></rank>
    <node name="5-carboxymethyl-2-hydroxymuconate semialdehyde dehydrogenase (EC 1.2.1.60)">
     <magnitude><val>2</val></magnitude>
     <score><val>-37</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Protocatechuate 4,5-dioxygenase beta chain (EC 1.13.11.8)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-72</val></score>
    </node>
   </node>
  </node>
  <node name="Peripheral pathways for catabolism of aromatic compounds">
   <rank><val>level 2</val></rank>
   <magnitude><val>84</val><val>339</val></magnitude>
   <score><val>-40.9686</val><val>-78.5553</val></score>
   <node name="Salicylate ester degradation">
    <magnitude><val></val><val>1</val></magnitude>
    <score><val>0</val><val>-17</val></score>
    <rank><val>level 3</val></rank>
    <node name="salicylate esterase">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-17</val></score>
     <magnitude><val></val><val>1</val></magnitude>
    </node>
   </node>
   <node name="Quinate degradation">
    <magnitude><val>1</val><val>11</val></magnitude>
    <score><val>-6</val><val>-40.5714</val></score>
    <rank><val>level 3</val></rank>
    <node name="3-dehydroquinate dehydratase I (EC 4.2.1.10)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>11</val></magnitude>
     <score><val>-6</val><val>-40.5714</val></score>
    </node>
   </node>
   <node name="p-Hydroxybenzoate degradation">
    <rank><val>level 3</val></rank>
    <magnitude><val>2</val><val>2</val></magnitude>
    <score><val>-13</val><val>-10.5</val></score>
    <node name="4-hydroxybenzoate transporter">
     <score><val>-11</val><val>-10.5</val></score>
     <magnitude><val>1</val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="P-hydroxybenzoate hydroxylase (EC 1.14.13.2)">
     <magnitude><val>1</val></magnitude>
     <score><val>-15</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Benzoate degradation">
    <rank><val>level 3</val></rank>
    <magnitude><val>3</val><val>41</val></magnitude>
    <score><val>-23</val><val>-103.259</val></score>
    <node name="benzoate MFS transporter BenK">
     <score><val>-7</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Benzoylformate decarboxylase (EC 4.1.1.7)">
     <magnitude><val>1</val><val>41</val></magnitude>
     <score><val>-55</val><val>-103.259</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Toluene degradation">
    <rank><val>level 3</val></rank>
    <score><val>-4</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="toluenesulfonate zinc-independent alcohol dehydrogenase">
     <magnitude><val>1</val></magnitude>
     <score><val>-4</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Chloroaromatic degradation pathway">
    <rank><val>level 3</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-39.5</val></score>
    <node name="3-oxoadipate CoA-transferase subunit B (EC 2.8.3.6)">
     <magnitude><val>1</val></magnitude>
     <score><val>-59</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Beta-ketoadipate enol-lactone hydrolase (EC 3.1.1.24)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-20</val></score>
    </node>
   </node>
   <node name="Chlorobenzoate degradation">
    <score><val>-56</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Muconate cycloisomerase (EC 5.5.1.1)">
     <rank><val>function</val></rank>
     <score><val>-56</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Phenylpropanoid compound degradation">
    <magnitude><val>1</val><val>5</val></magnitude>
    <score><val>-11</val><val>-23.2</val></score>
    <rank><val>level 3</val></rank>
    <node name="Chlorogenate esterase">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phenylpropionate dioxygenase and related ring-hydroxylating dioxygenases, large terminal subunit">
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-44</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="4-hydroxybenzoate transporter">
     <rank><val>function</val></rank>
     <score><val>-11</val><val>-10.5</val></score>
     <magnitude><val>1</val><val>2</val></magnitude>
    </node>
   </node>
   <node name="Phenol hydroxylase">
    <rank><val>level 3</val></rank>
    <magnitude><val>4</val></magnitude>
    <score><val>-58.5</val></score>
    <node name="Phenol hydroxylase, FAD- and [2Fe-2S]-containing reductase component DmpP">
     <rank><val>function</val></rank>
     <score><val>-6</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Phenol hydroxylase, P1 oxygenase component DmpL (EC 1.14.13.7)">
     <rank><val>function</val></rank>
     <score><val>-18</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Phenol hydroxylase, P3 oxygenase component DmpN (EC 1.14.13.7)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-105</val></score>
    </node>
   </node>
   <node name="Benzoate catabolism">
    <rank><val>level 3</val></rank>
    <score><val>-56</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="Muconate cycloisomerase (EC 5.5.1.1)">
     <score><val>-56</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="n-Phenylalkanoic acid degradation">
    <rank><val>level 3</val></rank>
    <score><val>-42.6524</val><val>-78.123</val></score>
    <magnitude><val>68</val><val>279</val></magnitude>
    <node name="Enoyl-CoA hydratase (EC 4.2.1.17)">
     <score><val>-23.25</val><val>-62.75</val></score>
     <magnitude><val>4</val><val>6</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="3-hydroxyacyl-CoA dehydrogenase (EC 1.1.1.35)">
     <score><val>-36.6667</val><val>-79.2676</val></score>
     <magnitude><val>13</val><val>110</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Long-chain-fatty-acid--CoA ligase (EC 6.2.1.3)">
     <score><val>-40.9524</val><val>-80.3</val></score>
     <magnitude><val>24</val><val>51</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="3-ketoacyl-CoA thiolase (EC 2.3.1.16)">
     <score><val>-49.92</val><val>-76.8312</val></score>
     <magnitude><val>27</val><val>112</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Benzoate transport and degradation cluster">
   <rank><val>level 2</val></rank>
   <magnitude><val>5</val><val>12</val></magnitude>
   <score><val>-21.8</val><val>-34.375</val></score>
   <node name="benzoyl-CoA-dihydrodiol lyase (BoxC)">
    <magnitude><val>1</val></magnitude>
    <score><val>-87</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Shikimate kinase I (EC 2.7.1.71)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-9</val></score>
   </node>
   <node name="benzoate degradation ring-cleavage hydrolase">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-2</val></score>
   </node>
   <node name="2-keto-4-pentenoate hydratase/2-oxohepta-3-ene-1,7-dioic acid hydratase (catechol pathway)">
    <score><val>-10</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Shikimate kinase II (EC 2.7.1.71)">
    <score><val>-1</val><val>-34.375</val></score>
    <magnitude><val>1</val><val>12</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Toluene 4-monooxygenase (T4MO)">
   <score><val>-24</val></score>
   <magnitude><val>2</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Toluene-4-monooxygenase, subunit TmoE">
    <magnitude><val>2</val></magnitude>
    <score><val>-24</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Gentisare degradation">
   <score><val>-16.0953</val><val>-61.44</val></score>
   <magnitude><val>7</val><val>20</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Gentisate 1,2-dioxygenase (EC 1.13.11.4)">
    <score><val>-28</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Fumarylacetoacetate hydrolase family protein">
    <rank><val>function</val></rank>
    <score><val>-11.6667</val><val>-67.1</val></score>
    <magnitude><val>4</val><val>18</val></magnitude>
   </node>
   <node name="4-hydroxybenzoate transporter">
    <score><val>-11</val><val>-10.5</val></score>
    <magnitude><val>1</val><val>2</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Maleylacetoacetate isomerase (EC 5.2.1.2)">
    <score><val>-27</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
 </node>
 <node name="Virulence, Disease and Defense">
  <rank><val>level 1</val></rank>
  <score><val>-31.7383</val><val>-69.8135</val></score>
  <magnitude><val>192</val><val>552</val></magnitude>
  <node name="Resistance to antibiotics and toxic compounds">
   <rank><val>level 2</val></rank>
   <magnitude><val>126</val><val>314</val></magnitude>
   <score><val>-31.3635</val><val>-66.4319</val></score>
   <node name="Resistance to fluoroquinolones">
    <rank><val>level 3</val></rank>
    <score><val>-44.2077</val><val>-87.928</val></score>
    <magnitude><val>26</val><val>107</val></magnitude>
    <node name="DNA gyrase subunit A (EC 5.99.1.3)">
     <rank><val>function</val></rank>
     <score><val>-24.2667</val><val>-83</val></score>
     <magnitude><val>15</val><val>53</val></magnitude>
    </node>
    <node name="DNA gyrase subunit B (EC 5.99.1.3)">
     <magnitude><val>11</val><val>54</val></magnitude>
     <score><val>-71.4</val><val>-92.7647</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Cobalt-zinc-cadmium resistance">
    <rank><val>level 3</val></rank>
    <magnitude><val>15</val><val>4</val></magnitude>
    <score><val>-46.2722</val><val>-87.25</val></score>
    <node name="Cobalt-zinc-cadmium resistance protein CzcA">
     <rank><val>function</val></rank>
     <magnitude><val>4</val></magnitude>
     <score><val>-55.6667</val></score>
    </node>
    <node name="Probable Co/Zn/Cd efflux system membrane fusion protein">
     <rank><val>function</val></rank>
     <score><val>-13</val><val>-46</val></score>
     <magnitude><val>1</val><val>1</val></magnitude>
    </node>
    <node name="Cation efflux system protein CusA">
     <rank><val>function</val></rank>
     <magnitude><val>4</val></magnitude>
     <score><val>-55.6667</val></score>
    </node>
    <node name="Heavy metal RND efflux outer membrane protein, CzcC family">
     <score><val>0</val><val>-138</val></score>
     <magnitude><val></val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cobalt-zinc-cadmium resistance protein CzcD">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
    </node>
    <node name="Cobalt/zinc/cadmium efflux RND transporter, membrane fusion protein, CzcB family">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-27</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cobalt-zinc-cadmium resistance protein">
     <rank><val>function</val></rank>
     <magnitude><val>5</val></magnitude>
     <score><val>-44.75</val></score>
    </node>
   </node>
   <node name="Multidrug Resistance Operon mdtRP of Bacillus">
    <rank><val>level 3</val></rank>
    <score><val>0</val><val>-68</val></score>
    <magnitude><val></val><val>1</val></magnitude>
    <node name="Multidrug efflux transporter MdtP">
     <score><val>0</val><val>-68</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="BlaR1 Family Regulatory Sensor-transducer Disambiguation">
    <rank><val>level 3</val></rank>
    <score><val>-27.8264</val><val>-44.6471</val></score>
    <magnitude><val>18</val><val>34</val></magnitude>
    <node name="Beta-lactamase (EC 3.5.2.6)">
     <score><val>-22.875</val><val>-44.6471</val></score>
     <magnitude><val>9</val><val>34</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Copper-translocating P-type ATPase (EC 3.6.3.4)">
     <rank><val>function</val></rank>
     <score><val>-32.7778</val></score>
     <magnitude><val>9</val></magnitude>
    </node>
   </node>
   <node name="Mercury resistance operon">
    <magnitude><val>12</val><val>31</val></magnitude>
    <score><val>-20.2727</val><val>-70.9167</val></score>
    <rank><val>level 3</val></rank>
    <node name="Mercuric ion reductase (EC 1.16.1.1)">
     <magnitude><val>12</val><val>31</val></magnitude>
     <score><val>-20.2727</val><val>-70.9167</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Erythromycin resistance">
    <rank><val>level 3</val></rank>
    <score><val>-2</val><val>-54</val></score>
    <magnitude><val>1</val><val>7</val></magnitude>
    <node name="Dimethyladenosine transferase (EC 2.1.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-2</val><val>-54</val></score>
    </node>
   </node>
   <node name="Copper homeostasis">
    <rank><val>level 3</val></rank>
    <score><val>-33.2</val></score>
    <magnitude><val>10</val></magnitude>
    <node name="Cytochrome c heme lyase subunit CcmF">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-37</val></score>
    </node>
    <node name="Copper-translocating P-type ATPase (EC 3.6.3.4)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val></magnitude>
     <score><val>-32.7778</val></score>
    </node>
   </node>
   <node name="Multidrug Resistance Efflux Pumps">
    <magnitude><val>1</val><val>1</val></magnitude>
    <score><val>-120</val><val>-20</val></score>
    <rank><val>level 3</val></rank>
    <node name="Macrolide export ATP-binding/permease protein MacB (EC 3.6.3.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-120</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acriflavin resistance protein">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-20</val></score>
    </node>
   </node>
   <node name="Beta-lactamase">
    <score><val>-22.875</val><val>-44.6471</val></score>
    <magnitude><val>18</val><val>68</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Beta-lactamase (EC 3.5.2.6)">
     <magnitude><val>9</val><val>34</val></magnitude>
     <score><val>-22.875</val><val>-44.6471</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Beta-lactamase">
     <score><val>-22.875</val><val>-44.6471</val></score>
     <magnitude><val>9</val><val>34</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Methicillin resistance in Staphylococci">
    <magnitude><val>3</val></magnitude>
    <score><val>-9.33333</val></score>
    <rank><val>level 3</val></rank>
    <node name="Undecaprenyl-phosphate N-acetylglucosaminyl 1-phosphate transferase (EC 2.7.8.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-10</val></score>
    </node>
    <node name="RNA polymerase sigma factor SigB">
     <score><val>-8</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="UDP-N-acetylmuramoylalanyl-D-glutamate--2,6-diaminopimelate ligase (EC 6.3.2.13)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-10</val></score>
    </node>
   </node>
   <node name="Mercuric reductase">
    <magnitude><val>12</val><val>31</val></magnitude>
    <score><val>-20.2727</val><val>-70.9167</val></score>
    <rank><val>level 3</val></rank>
    <node name="Mercuric ion reductase (EC 1.16.1.1)">
     <rank><val>function</val></rank>
     <score><val>-20.2727</val><val>-70.9167</val></score>
     <magnitude><val>12</val><val>31</val></magnitude>
    </node>
   </node>
   <node name="Arsenic resistance">
    <magnitude><val>10</val><val>30</val></magnitude>
    <score><val>-22.7143</val><val>-56.1824</val></score>
    <rank><val>level 3</val></rank>
    <node name="Arsenate reductase (EC 1.20.4.1)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-3</val></score>
    </node>
    <node name="Arsenic efflux pump protein">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-19</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Arsenical resistance operon repressor">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>1</val></magnitude>
     <score><val>-7</val><val>-13</val></score>
    </node>
    <node name="Arsenical pump-driving ATPase (EC 3.6.3.16)">
     <magnitude><val>8</val><val>28</val></magnitude>
     <score><val>-27.1429</val><val>-59.0526</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Streptococcus agalactiae virulome">
   <magnitude><val>1</val></magnitude>
   <score><val>-12</val></score>
   <rank><val>level 2</val></rank>
   <node name="Tyrosine-protein kinase EpsD (EC 2.7.10.2)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-12</val></score>
   </node>
  </node>
  <node name="Streptococcus pyogenes Virulome">
   <score><val>-23.5</val></score>
   <magnitude><val>2</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="N protein">
    <magnitude><val>1</val></magnitude>
    <score><val>-27</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Protein F">
    <score><val>-20</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Bacterial cyanide production and tolerance mechanisms">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val></magnitude>
   <score><val>-10</val></score>
   <node name="Formate dehydrogenase O alpha subunit (EC 1.2.1.2)">
    <score><val>-10</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Toxins and superantigens">
   <score><val>-18</val><val>-74.55</val></score>
   <magnitude><val>10</val><val>28</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Diphtheria toxin">
    <rank><val>level 3</val></rank>
    <score><val>-18</val><val>-74.55</val></score>
    <magnitude><val>10</val><val>28</val></magnitude>
    <node name="diptheria toxin resistance protein required for diphthamide biosynthesis (Saccharomyces)-like 1">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>28</val></magnitude>
     <score><val>-18</val><val>-74.55</val></score>
    </node>
   </node>
  </node>
  <node name="C jejuni colonization of chick caeca">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val></magnitude>
   <score><val>-65</val></score>
   <node name="RNA polymerase sigma-54 factor RpoN">
    <score><val>-65</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Detection">
   <rank><val>level 2</val></rank>
   <score><val>-40.4875</val><val>-73.3896</val></score>
   <magnitude><val>43</val><val>202</val></magnitude>
   <node name="MLST">
    <magnitude><val>43</val><val>202</val></magnitude>
    <score><val>-40.4875</val><val>-73.3896</val></score>
    <rank><val>level 3</val></rank>
    <node name="Glycerol kinase (EC 2.7.1.30)">
     <score><val>0</val><val>-76.1</val></score>
     <magnitude><val></val><val>41</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetyl-CoA acetyltransferase (EC 2.3.1.9)">
     <score><val>-49.3462</val><val>-76.8312</val></score>
     <magnitude><val>28</val><val>112</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Carbamate kinase (EC 2.7.2.2)">
     <magnitude><val>11</val><val>26</val></magnitude>
     <score><val>-26.6</val><val>-70.0526</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Triosephosphate isomerase (EC 5.3.1.1)">
     <magnitude><val>4</val><val>23</val></magnitude>
     <score><val>-16.6667</val><val>-55.5714</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Adhesion">
   <rank><val>level 2</val></rank>
   <score><val>-10.875</val><val>-95.6667</val></score>
   <magnitude><val>8</val><val>8</val></magnitude>
   <node name="Adhesion of Campylobacter">
    <score><val>-9</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Fibronectin/fibrinogen-binding protein">
     <score><val>-9</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Streptococcus pyogenes recombinatorial zone">
    <rank><val>level 3</val></rank>
    <magnitude><val>6</val><val>8</val></magnitude>
    <score><val>-11.5</val><val>-95.6667</val></score>
    <node name="Signal peptidase I">
     <magnitude><val>6</val><val>8</val></magnitude>
     <score><val>-11.5</val><val>-95.6667</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
 </node>
 <node name="Dormancy and Sporulation">
  <score><val>-25.6667</val><val>-28.6667</val></score>
  <magnitude><val>15</val><val>6</val></magnitude>
  <rank><val>level 1</val></rank>
  <node name="Spore DNA protection">
   <rank><val>level 2</val></rank>
   <score><val>-25</val></score>
   <magnitude><val>1</val></magnitude>
   <node name="Dipicolinate Synthesis">
    <rank><val>level 3</val></rank>
    <score><val>-25</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="Dipicolinate synthase subunit B">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-25</val></score>
    </node>
   </node>
  </node>
  <node name="Sporulation Cluster">
   <rank><val>level 2</val></rank>
   <magnitude><val>4</val><val>3</val></magnitude>
   <score><val>-23</val><val>-46.6667</val></score>
   <node name="FIG006789: Stage V sporulation protein">
    <rank><val>function</val></rank>
    <score><val>0</val><val>-12</val></score>
    <magnitude><val></val><val>1</val></magnitude>
   </node>
   <node name="Nucleoside triphosphate pyrophosphohydrolase MazG">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-14</val></score>
   </node>
   <node name="Transcription-repair coupling factor">
    <score><val>-34</val><val>-64</val></score>
    <magnitude><val>2</val><val>2</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Stage V sporulation protein T, AbrB family transcriptional regulator (SpoVT)">
    <rank><val>function</val></rank>
    <score><val>-10</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
  </node>
  <node name="Sporulation Cluster III A">
   <magnitude><val></val><val>1</val></magnitude>
   <score><val>0</val><val>-10</val></score>
   <rank><val>level 2</val></rank>
   <node name="Stage III sporulation protein AA">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>1</val></magnitude>
    <score><val>0</val><val>-10</val></score>
   </node>
  </node>
  <node name="Sporulation draft">
   <score><val>0</val><val>-12</val></score>
   <magnitude><val></val><val>1</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="FIG006789: Stage V sporulation protein">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>1</val></magnitude>
    <score><val>0</val><val>-12</val></score>
   </node>
  </node>
  <node name="Persister Cells">
   <magnitude><val>2</val></magnitude>
   <score><val>-15</val></score>
   <rank><val>level 2</val></rank>
   <node name="HipA protein">
    <rank><val>function</val></rank>
    <score><val>-3</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Cell division inhibitor">
    <score><val>-27</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Spore germination">
   <rank><val>level 2</val></rank>
   <score><val>-6</val></score>
   <magnitude><val>1</val></magnitude>
   <node name="Spore germination protein GerKA">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-6</val></score>
   </node>
  </node>
  <node name="Sporulation gene orphans">
   <rank><val>level 2</val></rank>
   <score><val>-33.1429</val><val>-10</val></score>
   <magnitude><val>7</val><val>1</val></magnitude>
   <node name="Stage III sporulation protein AA">
    <score><val>0</val><val>-10</val></score>
    <magnitude><val></val><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Stage V sporulation protein AF (SpoVAF)">
    <score><val>-52</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Stage V sporulation protein T, AbrB family transcriptional regulator (SpoVT)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-10</val></score>
   </node>
   <node name="Stage IV sporulation pro-sigma-K processing enzyme (SpoIVFB)">
    <score><val>-3</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Stage III sporulation protein D">
    <rank><val>function</val></rank>
    <score><val>-25</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="RNA polymerase sporulation specific sigma factor SigF">
    <rank><val>function</val></rank>
    <score><val>-36</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Stage IV sporulation protein B">
    <magnitude><val>1</val></magnitude>
    <score><val>-19</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Stage IV sporulation protein A">
    <score><val>-87</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
 </node>
 <node name="Membrane Transport">
  <rank><val>level 1</val></rank>
  <score><val>-41.3699</val><val>-69.0873</val></score>
  <magnitude><val>107</val><val>380</val></magnitude>
  <node name="Choline Transport">
   <magnitude><val>4</val></magnitude>
   <score><val>-14.5</val></score>
   <rank><val>level 2</val></rank>
   <node name="Choline-sulfatase (EC 3.1.6.6)">
    <rank><val>function</val></rank>
    <magnitude><val>4</val></magnitude>
    <score><val>-14.5</val></score>
   </node>
  </node>
  <node name="ABC transporters">
   <rank><val>level 2</val></rank>
   <magnitude><val>19</val><val>76</val></magnitude>
   <score><val>-35.5439</val><val>-92.2575</val></score>
   <node name="ABC transporter oligopeptide (TC 3.A.1.5.1)">
    <rank><val>level 3</val></rank>
    <score><val>-22.8</val></score>
    <magnitude><val>5</val></magnitude>
    <node name="Oligopeptide transport system permease protein OppC (TC 3.A.1.5.1)">
     <rank><val>function</val></rank>
     <score><val>-23</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Oligopeptide transport ATP-binding protein OppF (TC 3.A.1.5.1)">
     <rank><val>function</val></rank>
     <score><val>-32</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Oligopeptide transport system permease protein OppB (TC 3.A.1.5.1)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-16</val></score>
    </node>
    <node name="Oligopeptide transport ATP-binding protein OppD (TC 3.A.1.5.1)">
     <rank><val>function</val></rank>
     <score><val>-20</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="ABC transporter dipeptide (TC 3.A.1.5.2)">
    <magnitude><val>8</val><val>24</val></magnitude>
    <score><val>-50.5416</val><val>-38.9306</val></score>
    <rank><val>level 3</val></rank>
    <node name="Dipeptide-binding ABC transporter, periplasmic substrate-binding component (TC 3.A.1.5.2)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>6</val></magnitude>
     <score><val>-38</val><val>-49.75</val></score>
    </node>
    <node name="Dipeptide transport system permease protein DppB (TC 3.A.1.5.2)">
     <magnitude><val>4</val><val>8</val></magnitude>
     <score><val>-79.3333</val><val>-60</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dipeptide transport ATP-binding protein DppF (TC 3.A.1.5.2)">
     <magnitude><val>1</val><val>5</val></magnitude>
     <score><val>-16</val><val>-27.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dipeptide transport system permease protein DppC (TC 3.A.1.5.2)">
     <rank><val>function</val></rank>
     <score><val>-16.5</val><val>-3.66667</val></score>
     <magnitude><val>2</val><val>5</val></magnitude>
    </node>
   </node>
   <node name="ABC transporter branched-chain amino acid (TC 3.A.1.4.1)">
    <score><val>-26.1667</val><val>-40</val></score>
    <magnitude><val>6</val><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Branched-chain amino acid transport system permease protein LivM (TC 3.A.1.4.1)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-40</val></score>
    </node>
    <node name="Branched-chain amino acid transport ATP-binding protein LivG (TC 3.A.1.4.1)">
     <score><val>-24</val></score>
     <magnitude><val>4</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Branched-chain amino acid transport ATP-binding protein LivF (TC 3.A.1.4.1)">
     <score><val>-30.5</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="ATP-dependent efflux pump transporter Ybh">
    <rank><val>level 3</val></rank>
    <magnitude><val></val><val>51</val></magnitude>
    <score><val>0</val><val>-118.377</val></score>
    <node name="ABC transporter multidrug efflux pump, fused ATP-binding domains">
     <magnitude><val></val><val>8</val></magnitude>
     <score><val>0</val><val>-111.4</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Predicted membrane fusion protein (MFP) component of efflux pump, membrane anchor protein YbhG">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-100.333</val></score>
     <magnitude><val></val><val>17</val></magnitude>
    </node>
    <node name="ABC transport system, permease component YbhS">
     <score><val>0</val><val>-145.625</val></score>
     <magnitude><val></val><val>23</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Transcriptional regulator YbiH, TetR family">
     <magnitude><val></val><val>3</val></magnitude>
     <score><val>0</val><val>-30.3333</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Protein secretion system, Type I">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val></magnitude>
   <score><val>-9</val></score>
   <node name="Type I secretion system for aggregation">
    <rank><val>level 3</val></rank>
    <score><val>-9</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="type I secretion system ATPase, LssB family (LapB)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-9</val></score>
    </node>
   </node>
  </node>
  <node name="Transport of Molybdenum">
   <rank><val>level 2</val></rank>
   <magnitude><val></val><val>2</val></magnitude>
   <score><val>0</val><val>-20</val></score>
   <node name="Molybdenum transport system permease protein ModB (TC 3.A.1.8.1)">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>2</val></magnitude>
    <score><val>0</val><val>-20</val></score>
   </node>
  </node>
  <node name="Transport of Manganese">
   <rank><val>level 2</val></rank>
   <score><val>-64</val></score>
   <magnitude><val>1</val></magnitude>
   <node name="Manganese transport protein MntH">
    <rank><val>function</val></rank>
    <score><val>-64</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
  </node>
  <node name="Transport of Zinc">
   <score><val>-6</val></score>
   <magnitude><val>1</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Zinc ABC transporter, ATP-binding protein ZnuC">
    <rank><val>function</val></rank>
    <score><val>-6</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
  </node>
  <node name="Protein and nucleoprotein secretion system, Type IV">
   <rank><val>level 2</val></rank>
   <score><val>-24.6364</val><val>-48.1518</val></score>
   <magnitude><val>22</val><val>45</val></magnitude>
   <node name="pVir Plasmid of Campylobacter">
    <rank><val>level 3</val></rank>
    <magnitude><val>15</val><val>37</val></magnitude>
    <score><val>-30.4667</val><val>-53.8333</val></score>
    <node name="DNA topoisomerase I (EC 5.99.1.2)">
     <rank><val>function</val></rank>
     <score><val>-35</val><val>-53.8333</val></score>
     <magnitude><val>13</val><val>37</val></magnitude>
    </node>
    <node name="Single-stranded DNA-binding protein">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-1</val></score>
    </node>
   </node>
   <node name="Conjugative transfer">
    <rank><val>level 3</val></rank>
    <score><val>-14.4</val><val>-26.4</val></score>
    <magnitude><val>5</val><val>5</val></magnitude>
    <node name="Conjugative transfer protein TrbE">
     <score><val>-23.5</val><val>-9</val></score>
     <magnitude><val>2</val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="IncF plasmid conjugative transfer protein TraD">
     <score><val>-11</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="IncQ plasmid conjugative transfer protein TraB">
     <magnitude><val></val><val>3</val></magnitude>
     <score><val>0</val><val>-38</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="IncW plasmid conjugative relaxase protein TrwC (TraI homolog)">
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Type 4 conjugative transfer system, IncI1 type">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-7</val></score>
    <node name="Shufflon-specific DNA recombinase">
     <rank><val>function</val></rank>
     <score><val>-7</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Dot-Icm type IV secretion system">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val><val>3</val></magnitude>
    <score><val>-6</val><val>-14.3333</val></score>
    <node name="IcmB (DotO) protein">
     <magnitude><val>1</val><val>3</val></magnitude>
     <score><val>-6</val><val>-14.3333</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Protein translocation across cytoplasmic membrane">
   <rank><val>level 2</val></rank>
   <magnitude><val>41</val><val>185</val></magnitude>
   <score><val>-62.0793</val><val>-75.8079</val></score>
   <node name="Bacterial signal recognition particle (SRP)">
    <score><val>-58.45</val><val>-74.6459</val></score>
    <magnitude><val>25</val><val>79</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Signal recognition particle receptor protein FtsY (=alpha subunit) (TC 3.A.5.1.1)">
     <rank><val>function</val></rank>
     <magnitude><val>13</val><val>31</val></magnitude>
     <score><val>-51.25</val><val>-62.16</val></score>
    </node>
    <node name="Signal recognition particle, subunit Ffh SRP54 (TC 3.A.5.1.1)">
     <rank><val>function</val></rank>
     <score><val>-66.25</val><val>-82.7097</val></score>
     <magnitude><val>12</val><val>48</val></magnitude>
    </node>
   </node>
   <node name="Twin-arginine translocation system">
    <rank><val>level 3</val></rank>
    <magnitude><val>6</val><val>31</val></magnitude>
    <score><val>-41.3333</val><val>-38.1135</val></score>
    <node name="Twin-arginine translocation protein TatC">
     <magnitude><val>5</val><val>18</val></magnitude>
     <score><val>-47</val><val>-44.1538</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Twin-arginine translocation protein TatA">
     <rank><val>function</val></rank>
     <score><val>-13</val><val>-29.75</val></score>
     <magnitude><val>1</val><val>13</val></magnitude>
    </node>
   </node>
   <node name="HtrA and Sec secretion">
    <magnitude><val>10</val><val>75</val></magnitude>
    <score><val>-83.6</val><val>-92.6124</val></score>
    <rank><val>level 3</val></rank>
    <node name="Protein-export membrane protein SecF (TC 3.A.5.1.1)">
     <magnitude><val></val><val>32</val></magnitude>
     <score><val>0</val><val>-85.5263</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Preprotein translocase secY subunit (TC 3.A.5.1.1)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>43</val></magnitude>
     <score><val>-78.8571</val><val>-97.8857</val></score>
    </node>
    <node name="Protein export cytoplasm protein SecA ATPase RNA helicase (TC 3.A.5.1.1)">
     <magnitude><val>3</val></magnitude>
     <score><val>-94.6667</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Protein secretion system, Type VI">
   <score><val>-43.5</val></score>
   <magnitude><val>2</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Type VI secretion systems">
    <magnitude><val>2</val></magnitude>
    <score><val>-43.5</val></score>
    <rank><val>level 3</val></rank>
    <node name="ClpB protein">
     <rank><val>function</val></rank>
     <score><val>-43.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Protein secretion system, Type II">
   <score><val>-25.5714</val><val>-46.4381</val></score>
   <magnitude><val>14</val><val>35</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Predicted secretion system W clustering with cell division proteins">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-21</val></score>
    <node name="Cell division protein FtsX">
     <magnitude><val>1</val></magnitude>
     <score><val>-21</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="General Secretion Pathway">
    <rank><val>level 3</val></rank>
    <score><val>-18</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="General secretion pathway protein D">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-18</val></score>
    </node>
   </node>
   <node name="Widespread colonization island">
    <rank><val>level 3</val></rank>
    <score><val>-39.5</val></score>
    <magnitude><val>2</val></magnitude>
    <node name="Type II/IV secretion system ATP hydrolase TadA/VirB11/CpaF, TadA subfamily">
     <rank><val>function</val></rank>
     <score><val>-39.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
   </node>
   <node name="Type IV pilus">
    <rank><val>level 3</val></rank>
    <magnitude><val>10</val><val>35</val></magnitude>
    <score><val>-24</val><val>-46.4381</val></score>
    <node name="3-dehydroquinate synthase (EC 4.2.3.4)">
     <score><val>-24.625</val><val>-50.1667</val></score>
     <magnitude><val>8</val><val>32</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Multimodular transpeptidase-transglycosylase (EC 2.4.1.129) (EC 3.4.-.-)">
     <score><val>-4</val><val>-6</val></score>
     <magnitude><val>1</val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Twitching motility protein PilT">
     <score><val>-39</val><val>-8</val></score>
     <magnitude><val>1</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Transport of Nickel and Cobalt">
   <magnitude><val>2</val></magnitude>
   <score><val>-41</val></score>
   <rank><val>level 2</val></rank>
   <node name="HoxN/HupN/NixA family nickel/cobalt transporter">
    <magnitude><val>2</val></magnitude>
    <score><val>-41</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="ECF class transporters">
   <rank><val>level 2</val></rank>
   <score><val>0</val><val>-37.4319</val></score>
   <magnitude><val></val><val>37</val></magnitude>
   <node name="Transmembrane component SS1135 of energizing module of predicted ECF transporter">
    <magnitude><val></val><val>7</val></magnitude>
    <score><val>0</val><val>-29</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Substrate-specific component ST1137 of predicted ECF transporter">
    <magnitude><val></val><val>7</val></magnitude>
    <score><val>0</val><val>-34.3333</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Duplicated ATPase component SS1136 of energizing module of predicted ECF transporter">
    <rank><val>function</val></rank>
    <score><val>0</val><val>-40.9412</val></score>
    <magnitude><val></val><val>23</val></magnitude>
   </node>
  </node>
 </node>
 <node name="Secondary Metabolism">
  <rank><val>level 1</val></rank>
  <magnitude><val>35</val><val>181</val></magnitude>
  <score><val>-46.1723</val><val>-72.5933</val></score>
  <node name="Plant Alkaloids">
   <rank><val>level 2</val></rank>
   <magnitude><val>3</val><val>41</val></magnitude>
   <score><val>-25.6667</val><val>-65.7407</val></score>
   <node name="Alkaloid biosynthesis from L-lysine">
    <magnitude><val>3</val><val>41</val></magnitude>
    <score><val>-25.6667</val><val>-65.7407</val></score>
    <rank><val>level 3</val></rank>
    <node name="Deoxyhypusine synthase">
     <rank><val>function</val></rank>
     <score><val>-25.6667</val><val>-65.7407</val></score>
     <magnitude><val>3</val><val>41</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Plant Hormones">
   <magnitude><val>31</val><val>138</val></magnitude>
   <score><val>-49.2913</val><val>-75.5291</val></score>
   <rank><val>level 2</val></rank>
   <node name="Auxin biosynthesis">
    <magnitude><val>25</val><val>125</val></magnitude>
    <score><val>-40.0972</val><val>-73.8855</val></score>
    <rank><val>level 3</val></rank>
    <node name="Monoamine oxidase (1.4.3.4)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-47</val></score>
     <magnitude><val></val><val>1</val></magnitude>
    </node>
    <node name="Tryptophan synthase beta chain (EC 4.2.1.20)">
     <rank><val>function</val></rank>
     <score><val>-52.7143</val><val>-75.5455</val></score>
     <magnitude><val>16</val><val>57</val></magnitude>
    </node>
    <node name="Anthranilate phosphoribosyltransferase (EC 2.4.2.18)">
     <magnitude><val>5</val><val>26</val></magnitude>
     <score><val>-21</val><val>-94.2857</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Tryptophan synthase alpha chain (EC 4.2.1.20)">
     <magnitude><val>2</val><val>25</val></magnitude>
     <score><val>-18</val><val>-63.1667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylanthranilate isomerase (EC 5.3.1.24)">
     <score><val>-10</val><val>-53.25</val></score>
     <magnitude><val>1</val><val>16</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Indole-acetamide hydrolase">
     <score><val>-8</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Auxin degradation">
    <rank><val>level 3</val></rank>
    <score><val>-87.6</val><val>-91.3333</val></score>
    <magnitude><val>6</val><val>13</val></magnitude>
    <node name="Peroxidase (EC 1.11.1.7)">
     <magnitude><val>6</val><val>13</val></magnitude>
     <score><val>-87.6</val><val>-91.3333</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Aromatic amino acids and derivatives">
   <magnitude><val>1</val><val>2</val></magnitude>
   <score><val>-11</val><val>-10.5</val></score>
   <rank><val>level 2</val></rank>
   <node name="Cinnamic Acid Degradation">
    <rank><val>level 3</val></rank>
    <score><val>-11</val><val>-10.5</val></score>
    <magnitude><val>1</val><val>2</val></magnitude>
    <node name="4-hydroxybenzoate transporter">
     <magnitude><val>1</val><val>2</val></magnitude>
     <score><val>-11</val><val>-10.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
 </node>
 <node name="RNA Metabolism">
  <rank><val>level 1</val></rank>
  <magnitude><val>852</val><val>3887</val></magnitude>
  <score><val>-32.5188</val><val>-60.2184</val></score>
  <node name="Group II intron-associated genes">
   <magnitude><val>10</val><val>4</val></magnitude>
   <score><val>-43.6667</val><val>-12.5</val></score>
   <rank><val>level 2</val></rank>
   <node name="Retron-type reverse transcriptase">
    <score><val>-43.6667</val><val>-12.5</val></score>
    <magnitude><val>10</val><val>4</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Transcription">
   <rank><val>level 2</val></rank>
   <score><val>-51.03</val><val>-64.119</val></score>
   <magnitude><val>131</val><val>662</val></magnitude>
   <node name="Transcription elongation factors, archaeal">
    <magnitude><val>7</val><val>42</val></magnitude>
    <score><val>-32.2857</val><val>-58.8755</val></score>
    <rank><val>level 3</val></rank>
    <node name="Putative transcription antitermination protein NusG">
     <magnitude><val>5</val><val>23</val></magnitude>
     <score><val>-32</val><val>-86.0625</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="NusA protein homolog, archaeal">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>16</val></magnitude>
     <score><val>-33</val><val>-27.8333</val></score>
    </node>
    <node name="Transcription factor S">
     <score><val>0</val><val>-16</val></score>
     <magnitude><val></val><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="RNA polymerase II initiation factors">
    <rank><val>level 3</val></rank>
    <score><val>-40.8</val><val>-38.5789</val></score>
    <magnitude><val>9</val><val>36</val></magnitude>
    <node name="TATA-box binding protein">
     <score><val>-40.8</val><val>-38.5789</val></score>
     <magnitude><val>9</val><val>36</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Transcription initiation, bacterial sigma factors">
    <rank><val>level 3</val></rank>
    <magnitude><val>6</val></magnitude>
    <score><val>-33.8333</val></score>
    <node name="RNA polymerase sigma factor RpoD">
     <score><val>-41</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="RNA polymerase sigma-54 factor RpoN">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-65</val></score>
    </node>
    <node name="RNA polymerase sigma-70 factor">
     <score><val>-12</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="RNA polymerase sigma factor SigB">
     <magnitude><val>1</val></magnitude>
     <score><val>-8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="RNA polymerase sporulation specific sigma factor SigF">
     <magnitude><val>1</val></magnitude>
     <score><val>-36</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Rrf2 family transcriptional regulators">
    <magnitude><val>1</val></magnitude>
    <score><val>-10</val></score>
    <rank><val>level 3</val></rank>
    <node name="Predicted transcriptional regulator of sulfate adenylyltransferase, Rrf2 family">
     <score><val>-10</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="RNA polymerase III initiation factors">
    <magnitude><val>9</val><val>36</val></magnitude>
    <score><val>-40.8</val><val>-38.5789</val></score>
    <rank><val>level 3</val></rank>
    <node name="TATA-box binding protein">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>36</val></magnitude>
     <score><val>-40.8</val><val>-38.5789</val></score>
    </node>
   </node>
   <node name="RNA polymerase archaeal">
    <rank><val>level 3</val></rank>
    <magnitude><val>59</val><val>413</val></magnitude>
    <score><val>-57.339</val><val>-74.7368</val></score>
    <node name="DNA-directed RNA polymerase subunit A&#39; (EC 2.7.7.6)">
     <rank><val>function</val></rank>
     <score><val>-69.9412</val><val>-83.3607</val></score>
     <magnitude><val>17</val><val>90</val></magnitude>
    </node>
    <node name="DNA-directed RNA polymerase subunit H (EC 2.7.7.6)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>17</val></magnitude>
     <score><val>-22.5</val><val>-29.4545</val></score>
    </node>
    <node name="DNA-directed RNA polymerase subunit N (EC 2.7.7.6)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-16.8</val></score>
     <magnitude><val></val><val>5</val></magnitude>
    </node>
    <node name="DNA-directed RNA polymerase subunit E&#39; (EC 2.7.7.6)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>29</val></magnitude>
     <score><val>-33</val><val>-45.8095</val></score>
    </node>
    <node name="DNA-directed RNA polymerase subunit K (EC 2.7.7.6)">
     <score><val>-13.3333</val><val>-15.625</val></score>
     <magnitude><val>6</val><val>9</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA-directed RNA polymerase subunit B (EC 2.7.7.6)">
     <magnitude><val>15</val><val>156</val></magnitude>
     <score><val>-91.4</val><val>-97.9118</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA-directed RNA polymerase subunit F (EC 2.7.7.6)">
     <score><val>0</val><val>-42.6</val></score>
     <magnitude><val></val><val>15</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA-directed RNA polymerase subunit D (EC 2.7.7.6)">
     <score><val>-57.5</val><val>-55.25</val></score>
     <magnitude><val>2</val><val>19</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA-directed RNA polymerase subunit L (EC 2.7.7.6)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>18</val></magnitude>
     <score><val>-22</val><val>-26.8889</val></score>
    </node>
    <node name="DNA-directed RNA polymerase subunit A&#39;&#39; (EC 2.7.7.6)">
     <magnitude><val>8</val><val>49</val></magnitude>
     <score><val>-41.25</val><val>-77.6944</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA-directed RNA polymerase subunit E&#39;&#39;">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>4</val></magnitude>
     <score><val>0</val><val>-11</val></score>
    </node>
    <node name="DNA-directed RNA polymerase subunit P (EC 2.7.7.6)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-6</val></score>
    </node>
   </node>
   <node name="RNA polymerase bacterial">
    <rank><val>level 3</val></rank>
    <magnitude><val>4</val></magnitude>
    <score><val>-110</val></score>
    <node name="DNA-directed RNA polymerase beta subunit (EC 2.7.7.6)">
     <score><val>-96</val></score>
     <magnitude><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA-directed RNA polymerase beta&#39; subunit (EC 2.7.7.6)">
     <rank><val>function</val></rank>
     <score><val>-152</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="RNA polymerase archaeal initiation factors">
    <magnitude><val>29</val><val>132</val></magnitude>
    <score><val>-47.9494</val><val>-45.5379</val></score>
    <rank><val>level 3</val></rank>
    <node name="Transcription initiation factor B">
     <rank><val>function</val></rank>
     <magnitude><val>20</val><val>96</val></magnitude>
     <score><val>-51.1667</val><val>-48.1475</val></score>
    </node>
    <node name="TATA-box binding protein">
     <magnitude><val>9</val><val>36</val></magnitude>
     <score><val>-40.8</val><val>-38.5789</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Transcription factors cyanobacterial RpoD-like sigma factors">
    <rank><val>level 3</val></rank>
    <score><val>-41</val></score>
    <magnitude><val>2</val></magnitude>
    <node name="RNA polymerase sigma factor RpoD">
     <score><val>-41</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Transcription factors bacterial">
    <magnitude><val>5</val><val>3</val></magnitude>
    <score><val>-43.2</val><val>-106.333</val></score>
    <rank><val>level 3</val></rank>
    <node name="Transcription termination protein NusA">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>1</val></magnitude>
     <score><val>-66</val><val>-191</val></score>
    </node>
    <node name="Transcription-repair coupling factor">
     <rank><val>function</val></rank>
     <score><val>-34</val><val>-64</val></score>
     <magnitude><val>2</val><val>2</val></magnitude>
    </node>
    <node name="Transcription antitermination protein NusG">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-16</val></score>
    </node>
   </node>
  </node>
  <node name="RNA processing and modification">
   <score><val>-28.9514</val><val>-59.476</val></score>
   <magnitude><val>711</val><val>3221</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="mnm5U34 biosynthesis bacteria">
    <magnitude><val>25</val><val>77</val></magnitude>
    <score><val>-29.37</val><val>-38.382</val></score>
    <rank><val>level 3</val></rank>
    <node name="tRNA 5-methylaminomethyl-2-thiouridine synthase TusA">
     <rank><val>function</val></rank>
     <score><val>-14</val><val>-13.2222</val></score>
     <magnitude><val>1</val><val>21</val></magnitude>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7)">
     <score><val>-29.4167</val><val>-52.625</val></score>
     <magnitude><val>15</val><val>38</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7), SufS subfamily">
     <magnitude><val>9</val><val>18</val></magnitude>
     <score><val>-31</val><val>-37.6667</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Polyadenylation bacterial">
    <rank><val>level 3</val></rank>
    <score><val>-5</val><val>-16.5</val></score>
    <magnitude><val>1</val><val>2</val></magnitude>
    <node name="tRNA nucleotidyltransferase (EC 2.7.7.21) (EC 2.7.7.25)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>2</val></magnitude>
     <score><val>-5</val><val>-16.5</val></score>
    </node>
   </node>
   <node name="eukaryotic rRNA modification and related functions">
    <rank><val>level 3</val></rank>
    <score><val>-21.5714</val><val>-42.9091</val></score>
    <magnitude><val>7</val><val>15</val></magnitude>
    <node name="RNA helicase, putative">
     <magnitude><val>2</val></magnitude>
     <score><val>-9</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="NADH dehydrogenase subunit 5">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-2</val></score>
    </node>
    <node name="SSU ribosomal protein S14e (S11p)">
     <rank><val>function</val></rank>
     <score><val>-32.75</val><val>-42.9091</val></score>
     <magnitude><val>4</val><val>15</val></magnitude>
    </node>
   </node>
   <node name="Ribonuclease H">
    <magnitude><val>4</val><val>26</val></magnitude>
    <score><val>-11</val><val>-35.0714</val></score>
    <rank><val>level 3</val></rank>
    <node name="Ribonuclease HII (EC 3.1.26.4)">
     <magnitude><val>4</val><val>26</val></magnitude>
     <score><val>-11</val><val>-35.0714</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA nucleotidyltransferase">
    <rank><val>level 3</val></rank>
    <magnitude><val>4</val><val>32</val></magnitude>
    <score><val>-10.75</val><val>-62.8672</val></score>
    <node name="tRNA nucleotidyltransferase, archaeal type (EC 2.7.7.21) (EC 2.7.7.25)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>30</val></magnitude>
     <score><val>-12.6667</val><val>-65.9583</val></score>
    </node>
    <node name="tRNA nucleotidyltransferase (EC 2.7.7.21) (EC 2.7.7.25)">
     <magnitude><val>1</val><val>2</val></magnitude>
     <score><val>-5</val><val>-16.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNAmodification position 34">
    <score><val>-17</val><val>-56.9583</val></score>
    <magnitude><val>7</val><val>36</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="n-type ATP pyrophosphatase superfamily / TilS and TtcA-like">
     <rank><val>function</val></rank>
     <score><val>-27</val><val>-56.9583</val></score>
     <magnitude><val>2</val><val>36</val></magnitude>
    </node>
    <node name="tRNA-guanine transglycosylase (EC 2.4.2.29)">
     <score><val>-26</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="S-adenosylmethionine:tRNA ribosyltransferase-isomerase (EC 5.-.-.-)">
     <score><val>-4.33333</val></score>
     <magnitude><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="rRNA modification Bacteria">
    <magnitude><val>5</val><val>7</val></magnitude>
    <score><val>-12.2</val><val>-54</val></score>
    <rank><val>level 3</val></rank>
    <node name="Ribosomal large subunit pseudouridine synthase D (EC 4.2.1.70)">
     <score><val>-5</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="23S rRNA (guanosine-2&#39;-O-) -methyltransferase rlmB (EC 2.1.1.-)">
     <score><val>-13</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Metal-dependent hydrolase YbeY, involved in rRNA and/or ribosome maturation and assembly">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-21</val></score>
    </node>
    <node name="Ribosomal RNA small subunit methyltransferase C (EC 2.1.1.52)">
     <magnitude><val>1</val></magnitude>
     <score><val>-20</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dimethyladenosine transferase (EC 2.1.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-2</val><val>-54</val></score>
    </node>
   </node>
   <node name="mcm5s2U biosynthesis in tRNA">
    <magnitude><val>24</val><val>70</val></magnitude>
    <score><val>-34.8854</val><val>-50.9107</val></score>
    <rank><val>level 3</val></rank>
    <node name="histone acetyltransferase, ELP3 family">
     <rank><val>function</val></rank>
     <score><val>-44</val><val>-48.875</val></score>
     <magnitude><val>9</val><val>32</val></magnitude>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7)">
     <rank><val>function</val></rank>
     <score><val>-29.4167</val><val>-52.625</val></score>
     <magnitude><val>15</val><val>38</val></magnitude>
    </node>
   </node>
   <node name="tRNA processing">
    <rank><val>level 3</val></rank>
    <magnitude><val>33</val><val>159</val></magnitude>
    <score><val>-25.2186</val><val>-60.0339</val></score>
    <node name="tRNA pseudouridine synthase B (EC 4.2.1.70)">
     <score><val>-9.66667</val></score>
     <magnitude><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribonuclease PH (EC 2.7.7.56)">
     <rank><val>function</val></rank>
     <score><val>-42</val><val>-63</val></score>
     <magnitude><val>1</val><val>13</val></magnitude>
    </node>
    <node name="tRNA pseudouridine 55 synthase (EC 4.2.1.70)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>31</val></magnitude>
     <score><val>-28.5</val><val>-60.875</val></score>
    </node>
    <node name="tRNA pseudouridine 13 synthase (EC 4.2.1.-)">
     <magnitude><val>4</val><val>47</val></magnitude>
     <score><val>-13.25</val><val>-68.4412</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA delta(2)-isopentenylpyrophosphate transferase (EC 2.5.1.8)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-20</val></score>
    </node>
    <node name="tRNA pseudouridine synthase A (EC 4.2.1.70)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>14</val></magnitude>
     <score><val>0</val><val>-56.4545</val></score>
    </node>
    <node name="Ribonuclease Z (EC 3.1.26.11)">
     <rank><val>function</val></rank>
     <score><val>-35.375</val><val>-70.25</val></score>
     <magnitude><val>8</val><val>22</val></magnitude>
    </node>
    <node name="tRNA-i(6)A37 methylthiotransferase">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>32</val></magnitude>
     <score><val>-22.8571</val><val>-40.2083</val></score>
    </node>
   </node>
   <node name="16S rRNA modification within P site of ribosome">
    <rank><val>level 3</val></rank>
    <score><val>-40.6667</val></score>
    <magnitude><val>6</val></magnitude>
    <node name="Cell division protein FtsI [Peptidoglycan synthetase] (EC 2.4.1.129)">
     <rank><val>function</val></rank>
     <score><val>-57</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="rRNA small subunit methyltransferase H">
     <rank><val>function</val></rank>
     <score><val>-35</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Cell division protein MraZ">
     <rank><val>function</val></rank>
     <score><val>-41</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="rRNA small subunit methyltransferase I">
     <score><val>-19</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA modification Archaea">
    <magnitude><val>216</val><val>1009</val></magnitude>
    <score><val>-28.0647</val><val>-57.9957</val></score>
    <rank><val>level 3</val></rank>
    <node name="tRNA pseudouridine 55 synthase Archaea">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>17</val></magnitude>
     <score><val>-15.3333</val><val>-61.7333</val></score>
    </node>
    <node name="Archease">
     <magnitude><val></val><val>10</val></magnitude>
     <score><val>0</val><val>-10.1429</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="COG2047: Uncharacterized protein (ATP-grasp superfamily)">
     <magnitude><val>6</val><val>13</val></magnitude>
     <score><val>-25.6</val><val>-34.625</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="OB-fold nucleic acid binding domain protein COG1571">
     <magnitude><val>3</val><val>17</val></magnitude>
     <score><val>-9.33333</val><val>-89.2667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Putative RNA-binding protein COG1094">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>26</val></magnitude>
     <score><val>-21</val><val>-47.0526</val></score>
    </node>
    <node name="histone acetyltransferase, ELP3 family">
     <magnitude><val>9</val><val>32</val></magnitude>
     <score><val>-44</val><val>-48.875</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Predicted RNA methylase COG2263">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>2</val></magnitude>
     <score><val>-11.5</val><val>-24</val></score>
    </node>
    <node name="Thiamine biosynthesis protein ThiI">
     <magnitude><val>7</val><val>49</val></magnitude>
     <score><val>-26.6667</val><val>-63.875</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7), SufS subfamily">
     <score><val>-31</val><val>-37.6667</val></score>
     <magnitude><val>9</val><val>18</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Predicted RNA-binding protein COG1491">
     <magnitude><val>6</val><val>20</val></magnitude>
     <score><val>-43.8333</val><val>-47.1538</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Rhodanese-like domain protein">
     <score><val>-7</val><val>-21</val></score>
     <magnitude><val>1</val><val>4</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA pseudouridine synthase A (EC 4.2.1.70)">
     <score><val>0</val><val>-56.4545</val></score>
     <magnitude><val></val><val>14</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Uncharacterized protein MJ0319">
     <magnitude><val>3</val><val>7</val></magnitude>
     <score><val>-21.6667</val><val>-27</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="conserved hypothetical protein [Pyrococcus horikoshii]; COG2102: Predicted ATPases of PP-loop superfamily; IPR002761: Domain of unknown function DUF71">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>25</val></magnitude>
     <score><val>-12.5714</val><val>-61.0588</val></score>
    </node>
    <node name="COG0009 Sua5 subfamily, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
     <rank><val>function</val></rank>
     <score><val>-28.5</val><val>-56.4615</val></score>
     <magnitude><val>5</val><val>29</val></magnitude>
    </node>
    <node name="archaeosine tRNA-ribosyltransferase (EC 2.4.2.-) type 1">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>5</val></magnitude>
     <score><val>-25.5714</val><val>-46.6</val></score>
    </node>
    <node name="Asparagine--tRNA ligase related protein">
     <rank><val>function</val></rank>
     <score><val>-33.6667</val><val>-73.75</val></score>
     <magnitude><val>6</val><val>39</val></magnitude>
    </node>
    <node name="tRNA pseudouridine 55 synthase (EC 4.2.1.70)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>31</val></magnitude>
     <score><val>-28.5</val><val>-60.875</val></score>
    </node>
    <node name="Alanyl-tRNA synthetase domain protein">
     <rank><val>function</val></rank>
     <score><val>-23.6667</val><val>-46.6857</val></score>
     <magnitude><val>10</val><val>50</val></magnitude>
    </node>
    <node name="Hypothetical ATP-binding protein, containing DUF265 domain">
     <rank><val>function</val></rank>
     <score><val>-38</val><val>-79</val></score>
     <magnitude><val>2</val><val>17</val></magnitude>
    </node>
    <node name="Methylthioadenosine deaminase">
     <rank><val>function</val></rank>
     <score><val>-15.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="tRNA:Cm32/Um32 methyltransferase">
     <magnitude><val>1</val><val>19</val></magnitude>
     <score><val>-20</val><val>-77.6154</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA (Guanine37-N1) -methyltransferase (EC 2.1.1.31)">
     <score><val>-19.25</val><val>-79.9412</val></score>
     <magnitude><val>5</val><val>26</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="conserved protein with predicted RNA binding PUA domain">
     <magnitude><val>2</val><val>14</val></magnitude>
     <score><val>-6</val><val>-25.1818</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="COG1537: Predicted RNA-binding proteins">
     <rank><val>function</val></rank>
     <score><val>-5</val><val>-48.9048</val></score>
     <magnitude><val>1</val><val>26</val></magnitude>
    </node>
    <node name="tRNA N2,N2-dimethyl(Guanine26-N2)-methyltransferase (EC 2.1.1.32)">
     <rank><val>function</val></rank>
     <score><val>-23.8</val><val>-71.2273</val></score>
     <magnitude><val>5</val><val>30</val></magnitude>
    </node>
    <node name="GTP-binding protein Era">
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA-binding protein MTH1615">
     <magnitude><val>4</val><val>29</val></magnitude>
     <score><val>-16.25</val><val>-29.4667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Fe-S OXIDOREDUCTASE (1.8.-.-) Wyeosine biosynthesis">
     <rank><val>function</val></rank>
     <score><val>-86.3333</val><val>-81.4444</val></score>
     <magnitude><val>9</val><val>33</val></magnitude>
    </node>
    <node name="tRNA-(G10-N2) methyltransferase; tRNA-(G10-N2) dimethyltransferase">
     <score><val>-12</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="archaeosine tRNA-ribosyltransferase (EC 2.4.2.-) PUA domain">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>5</val></magnitude>
     <score><val>-25.5714</val><val>-46.6</val></score>
    </node>
    <node name="tRNA pseudouridine 13 synthase (EC 4.2.1.-)">
     <score><val>-13.25</val><val>-68.4412</val></score>
     <magnitude><val>4</val><val>47</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="TilS and TtcA-like">
     <score><val>-27</val><val>-56.9583</val></score>
     <magnitude><val>2</val><val>36</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="COG1180: Radical SAM, Pyruvate-formate lyase-activating enzyme like">
     <score><val>-28.5</val><val>-37.4</val></score>
     <magnitude><val>2</val><val>8</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="COG1355, Predicted dioxygenase">
     <score><val>-11.6</val><val>-96.7143</val></score>
     <magnitude><val>6</val><val>13</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Asparaginyl-tRNA synthetase (EC 6.1.1.22)">
     <magnitude><val>13</val><val>20</val></magnitude>
     <score><val>-37.6667</val><val>-43.2</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="COG2078: Uncharacterized ACR">
     <magnitude><val>3</val><val>5</val></magnitude>
     <score><val>-13.3333</val><val>-27</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Protein RtcB">
     <magnitude><val>3</val><val>40</val></magnitude>
     <score><val>-22</val><val>-86.9</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7)">
     <score><val>-29.4167</val><val>-52.625</val></score>
     <magnitude><val>15</val><val>38</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA-i(6)A37 methylthiotransferase">
     <score><val>-22.8571</val><val>-40.2083</val></score>
     <magnitude><val>9</val><val>32</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA (adenine57/58-N1)-methyltransferase (EC 2.1.1.36)">
     <magnitude><val></val><val>34</val></magnitude>
     <score><val>0</val><val>-44</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA cytosine methylase Pos56">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>39</val></magnitude>
     <score><val>-35.5</val><val>-62.9286</val></score>
    </node>
    <node name="archaeal ATPase, fused to C-terminal DUF234 domain">
     <magnitude><val></val><val>20</val></magnitude>
     <score><val>0</val><val>-35.5294</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="archaeosine tRNA-ribosyltransferase (EC 2.4.2.-) type 2">
     <score><val>-26.7143</val><val>-57.087</val></score>
     <magnitude><val>8</val><val>39</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="COG1293, Predicted RNA-binding protein homologous to eukaryotic snRNP">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>31</val></magnitude>
     <score><val>-23.1667</val><val>-75.8182</val></score>
    </node>
   </node>
   <node name="Ribonuclease P archaeal and eukaryal">
    <rank><val>level 3</val></rank>
    <score><val>0</val><val>-20.3333</val></score>
    <magnitude><val></val><val>3</val></magnitude>
    <node name="Ribonuclease P protein component 4 (EC 3.1.26.5)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-6</val></score>
    </node>
    <node name="Ribonuclease P protein component 1 (EC 3.1.26.5)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-49</val></score>
     <magnitude><val></val><val>1</val></magnitude>
    </node>
   </node>
   <node name="tRNA modification Bacteria">
    <magnitude><val>96</val><val>447</val></magnitude>
    <score><val>-34.1399</val><val>-60.0047</val></score>
    <rank><val>level 3</val></rank>
    <node name="tRNA pseudouridine 13 synthase (EC 4.2.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>47</val></magnitude>
     <score><val>-13.25</val><val>-68.4412</val></score>
    </node>
    <node name="Queuosine biosynthesis QueD, PTPS-I">
     <rank><val>function</val></rank>
     <score><val>-9</val><val>-34.9</val></score>
     <magnitude><val>2</val><val>16</val></magnitude>
    </node>
    <node name="tRNA 5-methylaminomethyl-2-thiouridine synthase TusA">
     <rank><val>function</val></rank>
     <score><val>-14</val><val>-13.2222</val></score>
     <magnitude><val>1</val><val>21</val></magnitude>
    </node>
    <node name="Iron-sulfur cluster assembly ATPase protein SufC">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>23</val></magnitude>
     <score><val>-53.1111</val><val>-58.2941</val></score>
    </node>
    <node name="Thiamine biosynthesis protein ThiI">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>49</val></magnitude>
     <score><val>-26.6667</val><val>-63.875</val></score>
    </node>
    <node name="Queuosine Biosynthesis QueC ATPase">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>34</val></magnitude>
     <score><val>-29.8</val><val>-30.1</val></score>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7)">
     <score><val>-29.4167</val><val>-52.625</val></score>
     <magnitude><val>15</val><val>38</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Ferredoxin, 2Fe-2S">
     <score><val>-12</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cysteine desulfurase (EC 2.8.1.7), SufS subfamily">
     <magnitude><val>9</val><val>18</val></magnitude>
     <score><val>-31</val><val>-37.6667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>16</val></magnitude>
     <score><val>-64</val><val>-47.7778</val></score>
    </node>
    <node name="tRNA-i(6)A37 methylthiotransferase">
     <score><val>-22.8571</val><val>-40.2083</val></score>
     <magnitude><val>9</val><val>32</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA pseudouridine synthase A (EC 4.2.1.70)">
     <score><val>0</val><val>-56.4545</val></score>
     <magnitude><val></val><val>14</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Queuosine Biosynthesis QueE Radical SAM">
     <magnitude><val>1</val><val>1</val></magnitude>
     <score><val>-14</val><val>-8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA delta(2)-isopentenylpyrophosphate transferase (EC 2.5.1.8)">
     <score><val>-20</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA-guanine transglycosylase (EC 2.4.2.29)">
     <magnitude><val>2</val></magnitude>
     <score><val>-26</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA (Guanine37-N1) -methyltransferase (EC 2.1.1.31)">
     <magnitude><val>5</val><val>26</val></magnitude>
     <score><val>-19.25</val><val>-79.9412</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA:Cm32/Um32 methyltransferase">
     <score><val>-20</val><val>-77.6154</val></score>
     <magnitude><val>1</val><val>19</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA pseudouridine synthase B (EC 4.2.1.70)">
     <magnitude><val>3</val></magnitude>
     <score><val>-9.66667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Iron-sulfur cluster assembly protein SufD">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>33</val></magnitude>
     <score><val>-36</val><val>-59.35</val></score>
    </node>
    <node name="Iron-sulfur cluster assembly protein SufB">
     <rank><val>function</val></rank>
     <magnitude><val>12</val><val>60</val></magnitude>
     <score><val>-79.5455</val><val>-103.939</val></score>
    </node>
    <node name="S-adenosylmethionine:tRNA ribosyltransferase-isomerase (EC 5.-.-.-)">
     <rank><val>function</val></rank>
     <score><val>-4.33333</val></score>
     <magnitude><val>3</val></magnitude>
    </node>
    <node name="Selenide,water dikinase (EC 2.7.9.3)">
     <magnitude><val>1</val></magnitude>
     <score><val>-34</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Wyeosine-MimG Biosynthesis">
    <rank><val>level 3</val></rank>
    <score><val>-47.675</val><val>-69.0711</val></score>
    <magnitude><val>30</val><val>192</val></magnitude>
    <node name="Fe-S OXIDOREDUCTASE (1.8.-.-) Wyeosine biosynthesis">
     <magnitude><val>9</val><val>33</val></magnitude>
     <score><val>-86.3333</val><val>-81.4444</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Hypothetical ATP-binding protein, containing DUF265 domain">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>17</val></magnitude>
     <score><val>-38</val><val>-79</val></score>
    </node>
    <node name="Thioredoxin reductase (EC 1.8.1.9)">
     <score><val>-39.3636</val><val>-67.8525</val></score>
     <magnitude><val>11</val><val>91</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA (Guanine37-N1) -methyltransferase (EC 2.1.1.31)">
     <score><val>-19.25</val><val>-79.9412</val></score>
     <magnitude><val>5</val><val>26</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA methylase Trm12p Wyeosine biosynthesis">
     <score><val>-16</val><val>-39.1176</val></score>
     <magnitude><val>3</val><val>25</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Methylthiotransferases">
    <rank><val>level 3</val></rank>
    <magnitude><val>20</val><val>64</val></magnitude>
    <score><val>-22.5714</val><val>-40.2083</val></score>
    <node name="tRNA N6-threonylcarbamoyladenosine 2-methylthiotransferase (putative)">
     <rank><val>function</val></rank>
     <score><val>-22.8571</val><val>-40.2083</val></score>
     <magnitude><val>9</val><val>32</val></magnitude>
    </node>
    <node name="Ribosomal protein S12p Asp88 (E. coli) methylthiotransferase">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-20</val></score>
    </node>
    <node name="tRNA-i(6)A37 methylthiotransferase">
     <magnitude><val>9</val><val>32</val></magnitude>
     <score><val>-22.8571</val><val>-40.2083</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Queuosine-Archaeosine Biosynthesis">
    <score><val>-22.5918</val><val>-43.8117</val></score>
    <magnitude><val>53</val><val>179</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="COG2047: Uncharacterized protein (ATP-grasp superfamily)">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>13</val></magnitude>
     <score><val>-25.6</val><val>-34.625</val></score>
    </node>
    <node name="Queuosine Biosynthesis QueC ATPase">
     <magnitude><val>5</val><val>34</val></magnitude>
     <score><val>-29.8</val><val>-30.1</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA-guanine transglycosylase (EC 2.4.2.29)">
     <rank><val>function</val></rank>
     <score><val>-26</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Queuosine biosynthesis QueD, PTPS-I">
     <score><val>-9</val><val>-34.9</val></score>
     <magnitude><val>2</val><val>16</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="archaeosine tRNA-ribosyltransferase (EC 2.4.2.-) type 1">
     <score><val>-25.5714</val><val>-46.6</val></score>
     <magnitude><val>9</val><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
     <rank><val>function</val></rank>
     <score><val>-64</val><val>-47.7778</val></score>
     <magnitude><val>1</val><val>16</val></magnitude>
    </node>
    <node name="Queuosine Biosynthesis QueE Radical SAM">
     <magnitude><val>1</val><val>1</val></magnitude>
     <score><val>-14</val><val>-8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 2">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>5</val></magnitude>
     <score><val>-8</val><val>-7</val></score>
    </node>
    <node name="Permease of the drug/metabolite transporter (DMT) superfamily">
     <rank><val>function</val></rank>
     <score><val>-21.9091</val><val>-50.1053</val></score>
     <magnitude><val>12</val><val>27</val></magnitude>
    </node>
    <node name="Peptidyl-prolyl cis-trans isomerase (EC 5.2.1.8)">
     <score><val>-5</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Inosine-uridine preferring nucleoside hydrolase (EC 3.2.2.1)">
     <score><val>-6</val><val>-51.7692</val></score>
     <magnitude><val>1</val><val>23</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="archaeosine tRNA-ribosyltransferase (EC 2.4.2.-) type 2">
     <score><val>-26.7143</val><val>-57.087</val></score>
     <magnitude><val>8</val><val>39</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="S-adenosylmethionine:tRNA ribosyltransferase-isomerase (EC 5.-.-.-)">
     <rank><val>function</val></rank>
     <score><val>-4.33333</val></score>
     <magnitude><val>3</val></magnitude>
    </node>
   </node>
   <node name="RNA 3&#39;-terminal phosphate cyclase">
    <rank><val>level 3</val></rank>
    <score><val>-22</val><val>-86.9</val></score>
    <magnitude><val>3</val><val>40</val></magnitude>
    <node name="Protein RtcB">
     <magnitude><val>3</val><val>40</val></magnitude>
     <score><val>-22</val><val>-86.9</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="ATP-dependent RNA helicases, bacterial">
    <rank><val>level 3</val></rank>
    <score><val>-41.5</val><val>-8</val></score>
    <magnitude><val>2</val><val>2</val></magnitude>
    <node name="ATP-dependent RNA helicase VCA0061">
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="ATP-dependent RNA helicase RhlE">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-15</val></score>
    </node>
    <node name="Cold-shock DEAD-box protein A">
     <rank><val>function</val></rank>
     <score><val>-68</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Polyadenylation specificity factor">
    <rank><val>level 3</val></rank>
    <magnitude><val>17</val><val>106</val></magnitude>
    <score><val>-49.5882</val><val>-83.4614</val></score>
    <node name="Protein similar to polyadenylation specificity factor, MJ0047 type">
     <magnitude><val>8</val><val>33</val></magnitude>
     <score><val>-38</val><val>-67.4583</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Protein similar to polyadenylation specificity factor, MJ1236 type">
     <rank><val>function</val></rank>
     <score><val>-59.8889</val><val>-90.6957</val></score>
     <magnitude><val>9</val><val>73</val></magnitude>
    </node>
   </node>
   <node name="RNA methylation">
    <rank><val>level 3</val></rank>
    <score><val>-16.225</val><val>-75.5993</val></score>
    <magnitude><val>10</val><val>52</val></magnitude>
    <node name="Dimethyladenosine transferase (EC 2.1.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-2</val><val>-54</val></score>
    </node>
    <node name="Ribosomal RNA small subunit methyltransferase C (EC 2.1.1.52)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-20</val></score>
    </node>
    <node name="tRNA (Guanine37-N1) -methyltransferase (EC 2.1.1.31)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>26</val></magnitude>
     <score><val>-19.25</val><val>-79.9412</val></score>
    </node>
    <node name="tRNA:Cm32/Um32 methyltransferase">
     <magnitude><val>1</val><val>19</val></magnitude>
     <score><val>-20</val><val>-77.6154</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU m5C1962 methyltransferase RlmI">
     <score><val>-11</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="23S rRNA (guanosine-2&#39;-O-) -methyltransferase rlmB (EC 2.1.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-13</val></score>
    </node>
   </node>
   <node name="Ribosome biogenesis archaeal">
    <score><val>-29</val><val>-58.1429</val></score>
    <magnitude><val>3</val><val>21</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Fibrillarin">
     <score><val>-29</val><val>-58.1429</val></score>
     <magnitude><val>3</val><val>21</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA modification yeast mitochondrial">
    <score><val>-30.6439</val><val>-70.6438</val></score>
    <magnitude><val>33</val><val>175</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="tRNA pseudouridine synthase A (EC 4.2.1.70)">
     <magnitude><val></val><val>14</val></magnitude>
     <score><val>0</val><val>-56.4545</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA (Guanine37-N1) -methyltransferase (EC 2.1.1.31)">
     <magnitude><val>5</val><val>26</val></magnitude>
     <score><val>-19.25</val><val>-79.9412</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA delta(2)-isopentenylpyrophosphate transferase (EC 2.5.1.8)">
     <magnitude><val>1</val></magnitude>
     <score><val>-20</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA N2,N2-dimethyl(Guanine26-N2)-methyltransferase (EC 2.1.1.32)">
     <rank><val>function</val></rank>
     <score><val>-23.8</val><val>-71.2273</val></score>
     <magnitude><val>5</val><val>30</val></magnitude>
    </node>
    <node name="tRNA pseudouridine 55 synthase (EC 4.2.1.70)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>31</val></magnitude>
     <score><val>-28.5</val><val>-60.875</val></score>
    </node>
    <node name="COG0009 Sua5 subfamily, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
     <magnitude><val>5</val><val>29</val></magnitude>
     <score><val>-28.5</val><val>-56.4615</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="YgjD/Kae1/Qri7 family, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>45</val></magnitude>
     <score><val>-43.4</val><val>-85.1667</val></score>
    </node>
   </node>
   <node name="tRNA modification yeast cytoplasmic">
    <rank><val>level 3</val></rank>
    <score><val>-31.0397</val><val>-64.6148</val></score>
    <magnitude><val>63</val><val>328</val></magnitude>
    <node name="Cysteine desulfurase (EC 2.8.1.7)">
     <score><val>-29.4167</val><val>-52.625</val></score>
     <magnitude><val>15</val><val>38</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA N2,N2-dimethyl(Guanine26-N2)-methyltransferase (EC 2.1.1.32)">
     <score><val>-23.8</val><val>-71.2273</val></score>
     <magnitude><val>5</val><val>30</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA pseudouridine synthase A (EC 4.2.1.70)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-56.4545</val></score>
     <magnitude><val></val><val>14</val></magnitude>
    </node>
    <node name="tRNA delta(2)-isopentenylpyrophosphate transferase (EC 2.5.1.8)">
     <score><val>-20</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA pseudouridine 13 synthase (EC 4.2.1.-)">
     <magnitude><val>4</val><val>47</val></magnitude>
     <score><val>-13.25</val><val>-68.4412</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="histone acetyltransferase, ELP3 family">
     <rank><val>function</val></rank>
     <score><val>-44</val><val>-48.875</val></score>
     <magnitude><val>9</val><val>32</val></magnitude>
    </node>
    <node name="COG0009 Sua5 subfamily, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>29</val></magnitude>
     <score><val>-28.5</val><val>-56.4615</val></score>
    </node>
    <node name="YgjD/Kae1/Qri7 family, required for N6-threonylcarbamoyl adenosine t(6)A37 modification in tRNA">
     <rank><val>function</val></rank>
     <score><val>-43.4</val><val>-85.1667</val></score>
     <magnitude><val>10</val><val>45</val></magnitude>
    </node>
    <node name="tRNA pseudouridine 55 synthase (EC 4.2.1.70)">
     <score><val>-28.5</val><val>-60.875</val></score>
     <magnitude><val>7</val><val>31</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="n-type ATP pyrophosphatase superfamily / TilS and TtcA-like">
     <magnitude><val>2</val><val>36</val></magnitude>
     <score><val>-27</val><val>-56.9583</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA (Guanine37-N1) -methyltransferase (EC 2.1.1.31)">
     <score><val>-19.25</val><val>-79.9412</val></score>
     <magnitude><val>5</val><val>26</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="tRNA splicing">
    <score><val>-13.1071</val><val>-53.8929</val></score>
    <magnitude><val>7</val><val>14</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="ADP-ribose 1"-phosphate phophatase">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-6</val></score>
    </node>
    <node name="RNA:NAD 2&#39;-phosphotransferase">
     <rank><val>function</val></rank>
     <score><val>-24</val><val>-9</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
    <node name="tRNA-intron endonuclease (EC 3.1.27.9)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>12</val></magnitude>
     <score><val>-8.75</val><val>-61.625</val></score>
    </node>
   </node>
   <node name="RNA processing and degradation, bacterial">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-27</val></score>
    <node name="Cytoplasmic axial filament protein CafA and Ribonuclease G (EC 3.1.4.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-27</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="rRNA modification Archaea">
    <rank><val>level 3</val></rank>
    <score><val>-19.8513</val><val>-51.7123</val></score>
    <magnitude><val>37</val><val>151</val></magnitude>
    <node name="Putative RNA-binding protein COG1094">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>26</val></magnitude>
     <score><val>-21</val><val>-47.0526</val></score>
    </node>
    <node name="Predicted RNA methylase COG2263">
     <rank><val>function</val></rank>
     <score><val>-11.5</val><val>-24</val></score>
     <magnitude><val>4</val><val>2</val></magnitude>
    </node>
    <node name="tRNA pseudouridine 55 synthase (EC 4.2.1.70)">
     <rank><val>function</val></rank>
     <score><val>-28.5</val><val>-60.875</val></score>
     <magnitude><val>7</val><val>31</val></magnitude>
    </node>
    <node name="Ribosomal RNA large subunit methyltransferase J (EC 2.1.1.-)">
     <score><val>-13.3333</val><val>-20</val></score>
     <magnitude><val>4</val><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dimethyladenosine transferase (EC 2.1.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-2</val><val>-54</val></score>
    </node>
    <node name="tRNA:Cm32/Um32 methyltransferase">
     <rank><val>function</val></rank>
     <score><val>-20</val><val>-77.6154</val></score>
     <magnitude><val>1</val><val>19</val></magnitude>
    </node>
    <node name="Predicted Zn-ribbon RNA-binding protein with a function in translation">
     <rank><val>function</val></rank>
     <score><val>-9</val><val>-9.75</val></score>
     <magnitude><val>1</val><val>5</val></magnitude>
    </node>
    <node name="Fibrillarin">
     <rank><val>function</val></rank>
     <score><val>-29</val><val>-58.1429</val></score>
     <magnitude><val>3</val><val>21</val></magnitude>
    </node>
    <node name="N1-specific pseudouridine methyltransferase NEP1, biosynthesis of m1acp3-psi in 18S/16S rRNA in Eukaryotes and Archaea">
     <score><val>-8.33333</val><val>-41.1538</val></score>
     <magnitude><val>8</val><val>17</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="LSU ribosomal protein L7Ae">
     <rank><val>function</val></rank>
     <score><val>-34.8333</val><val>-38.4</val></score>
     <magnitude><val>6</val><val>20</val></magnitude>
    </node>
   </node>
   <node name="RNA pseudouridine syntheses">
    <magnitude><val>4</val><val>14</val></magnitude>
    <score><val>-8.5</val><val>-56.4545</val></score>
    <rank><val>level 3</val></rank>
    <node name="Ribosomal large subunit pseudouridine synthase D (EC 4.2.1.70)">
     <rank><val>function</val></rank>
     <score><val>-5</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="tRNA pseudouridine synthase A (EC 4.2.1.70)">
     <magnitude><val></val><val>14</val></magnitude>
     <score><val>0</val><val>-56.4545</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="tRNA pseudouridine synthase B (EC 4.2.1.70)">
     <magnitude><val>3</val></magnitude>
     <score><val>-9.66667</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
 </node>
 <node name="Phages, Prophages, Transposable elements, Plasmids">
  <rank><val>level 1</val></rank>
  <magnitude><val>100</val><val>441</val></magnitude>
  <score><val>-47.0155</val><val>-70.4107</val></score>
  <node name="Pathogenicity islands">
   <magnitude><val>12</val><val>118</val></magnitude>
   <score><val>-70.8333</val><val>-87.9258</val></score>
   <rank><val>level 2</val></rank>
   <node name="Staphylococcal pathogenicity islands SaPI">
    <score><val>-70.8333</val><val>-87.9258</val></score>
    <magnitude><val>12</val><val>118</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="GMP synthase [glutamine-hydrolyzing] (EC 6.3.5.2)">
     <score><val>-3</val><val>-78.5238</val></score>
     <magnitude><val>1</val><val>31</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Heat shock protein 60 family chaperone GroEL">
     <score><val>-77</val><val>-91.2759</val></score>
     <magnitude><val>11</val><val>87</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Transposable elements">
   <rank><val>level 2</val></rank>
   <score><val>-23.0795</val><val>-44.6471</val></score>
   <magnitude><val>11</val><val>34</val></magnitude>
   <node name="CBSS-203122.12.peg.188">
    <rank><val>level 3</val></rank>
    <score><val>-24</val></score>
    <magnitude><val>2</val></magnitude>
    <node name="MG(2+) CHELATASE FAMILY PROTEIN">
     <magnitude><val>1</val></magnitude>
     <score><val>-24</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="ComM-related protein">
     <score><val>-24</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Tn552">
    <score><val>-22.875</val><val>-44.6471</val></score>
    <magnitude><val>9</val><val>34</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Beta-lactamase (EC 3.5.2.6)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>34</val></magnitude>
     <score><val>-22.875</val><val>-44.6471</val></score>
    </node>
   </node>
  </node>
  <node name="Plasmid related functions">
   <score><val>-82.5556</val></score>
   <magnitude><val>9</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="F plasmid replication">
    <magnitude><val>9</val></magnitude>
    <score><val>-82.5556</val></score>
    <rank><val>level 3</val></rank>
    <node name="Plasmid partition protein B">
     <score><val>-53</val></score>
     <magnitude><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Replication initiation protein RepE">
     <magnitude><val>4</val></magnitude>
     <score><val>-119.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Phages, Prophages">
   <rank><val>level 2</val></rank>
   <score><val>-41.9806</val><val>-66.2902</val></score>
   <magnitude><val>68</val><val>289</val></magnitude>
   <node name="Phage tail proteins 2">
    <score><val>0</val><val>-102</val></score>
    <magnitude><val></val><val>3</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Phage tail assembly protein">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-102</val></score>
    </node>
    <node name="Phage tail assembly protein I">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-102</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phage tail assembly">
     <score><val>0</val><val>-102</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Phage integration and excision">
    <rank><val>level 3</val></rank>
    <magnitude><val>24</val><val>142</val></magnitude>
    <score><val>-37.6597</val><val>-52.3883</val></score>
    <node name="Phage integrase">
     <score><val>-15.5</val><val>-53.5</val></score>
     <magnitude><val>2</val><val>27</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Tyrosine recombinase XerC">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>3</val></magnitude>
     <score><val>-1</val><val>-7</val></score>
    </node>
    <node name="Site-specific recombinase">
     <magnitude><val>13</val><val>42</val></magnitude>
     <score><val>-59.1667</val><val>-48.6923</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Integrase">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>70</val></magnitude>
     <score><val>-12.8333</val><val>-56.1224</val></score>
    </node>
   </node>
   <node name="Phage replication">
    <rank><val>level 3</val></rank>
    <magnitude><val>19</val><val>58</val></magnitude>
    <score><val>-35.5</val><val>-68.8342</val></score>
    <node name="DNA polymerase I (EC 2.7.7.7)">
     <rank><val>function</val></rank>
     <score><val>-35.5</val><val>-69.7436</val></score>
     <magnitude><val>19</val><val>57</val></magnitude>
    </node>
    <node name="DNA helicase (EC 3.6.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-17</val></score>
    </node>
   </node>
   <node name="Phage packaging machinery">
    <score><val>0</val><val>-35</val></score>
    <magnitude><val></val><val>2</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Phage terminase">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-35</val></score>
     <magnitude><val></val><val>1</val></magnitude>
    </node>
    <node name="Phage terminase large subunit">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-35</val></score>
    </node>
   </node>
   <node name="Phage DNA synthesis">
    <magnitude><val>6</val><val>22</val></magnitude>
    <score><val>-32.8</val><val>-48.3636</val></score>
    <rank><val>level 3</val></rank>
    <node name="Thymidine kinase">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-53</val></score>
     <magnitude><val></val><val>19</val></magnitude>
    </node>
    <node name="Glutaredoxin">
     <score><val>-32.8</val><val>-19</val></score>
     <magnitude><val>6</val><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Phage regulation of gene expression">
    <rank><val>level 3</val></rank>
    <score><val>-23.2</val></score>
    <magnitude><val>5</val></magnitude>
    <node name="RNA polymerase sigma factor">
     <score><val>-23.2</val></score>
     <magnitude><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="r1t-like streptococcal phages">
    <rank><val>level 3</val></rank>
    <magnitude><val>2</val><val>2</val></magnitude>
    <score><val>-4.5</val><val>-25</val></score>
    <node name="Phage protein">
     <rank><val>function</val></rank>
     <score><val>-4.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Phage terminase">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-35</val></score>
     <magnitude><val></val><val>1</val></magnitude>
    </node>
    <node name="Deoxyuridine 5&#39;-triphosphate nucleotidohydrolase (EC 3.6.1.23)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-15</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Staphylococcal phi-Mu50B-like prophages">
    <magnitude><val>12</val><val>60</val></magnitude>
    <score><val>-79.5455</val><val>-103.939</val></score>
    <rank><val>level 3</val></rank>
    <node name="Iron-sulfur cluster assembly protein SufB">
     <rank><val>function</val></rank>
     <magnitude><val>12</val><val>60</val></magnitude>
     <score><val>-79.5455</val><val>-103.939</val></score>
    </node>
   </node>
  </node>
 </node>
 <node name="Cell Wall and Capsule">
  <rank><val>level 1</val></rank>
  <score><val>-28.3506</val><val>-60.7837</val></score>
  <magnitude><val>251</val><val>730</val></magnitude>
  <node name="Recycling of Peptidoglycan Amino Sugars">
   <magnitude><val>2</val></magnitude>
   <score><val>-11</val></score>
   <rank><val>level 2</val></rank>
   <node name="Anhydro-N-acetylmuramic acid kinase (EC 2.7.1.-)">
    <score><val>-11</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="tRNA-dependent amino acid transfers">
   <magnitude><val>1</val></magnitude>
   <score><val>-5</val></score>
   <rank><val>level 2</val></rank>
   <node name="tRNA-dependent lipid II--amino acid ligase">
    <rank><val>function</val></rank>
    <score><val>-5</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
  </node>
  <node name="Gram-Positive cell wall components">
   <rank><val>level 2</val></rank>
   <score><val>-12</val><val>-11</val></score>
   <magnitude><val>3</val><val>1</val></magnitude>
   <node name="Teichoic and lipoteichoic acids biosynthesis">
    <rank><val>level 3</val></rank>
    <score><val>-7</val><val>-11</val></score>
    <magnitude><val>2</val><val>1</val></magnitude>
    <node name="Undecaprenyl-phosphate N-acetylglucosaminyl 1-phosphate transferase (EC 2.7.8.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-10</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="diglucosyldiacylglycerol synthase (LTA membrane anchor synthesis)">
     <magnitude><val>1</val></magnitude>
     <score><val>-4</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Poly(glycerol-phosphate) alpha-glucosyltransferase (EC 2.4.1.52)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-11</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Teichuronic acid biosynthesis">
    <magnitude><val>1</val></magnitude>
    <score><val>-22</val></score>
    <rank><val>level 3</val></rank>
    <node name="UDP-glucose dehydrogenase (EC 1.1.1.22)">
     <rank><val>function</val></rank>
     <score><val>-22</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Murein Hydrolases">
   <rank><val>level 2</val></rank>
   <score><val>-21.6667</val><val>-84.2258</val></score>
   <magnitude><val>9</val><val>31</val></magnitude>
   <node name="N-acetylmuramoyl-L-alanine amidase (EC 3.5.1.28)">
    <rank><val>function</val></rank>
    <score><val>-5</val></score>
    <magnitude><val>2</val></magnitude>
   </node>
   <node name="D-alanyl-D-alanine carboxypeptidase (EC 3.4.16.4)">
    <rank><val>function</val></rank>
    <score><val>-1.5</val><val>-7</val></score>
    <magnitude><val>2</val><val>1</val></magnitude>
   </node>
   <node name="Muramoyltetrapeptide carboxypeptidase (EC 3.4.17.13)">
    <rank><val>function</val></rank>
    <score><val>-36.4</val><val>-86.8</val></score>
    <magnitude><val>5</val><val>30</val></magnitude>
   </node>
  </node>
  <node name="Capsular and extracellular polysacchrides">
   <rank><val>level 2</val></rank>
   <score><val>-30.2737</val><val>-55.9142</val></score>
   <magnitude><val>126</val><val>319</val></magnitude>
   <node name="Extracellular Polysaccharide Biosynthesis of Streptococci">
    <rank><val>level 3</val></rank>
    <score><val>-12</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="Tyrosine-protein kinase EpsD (EC 2.7.10.2)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
    </node>
   </node>
   <node name="Exopolysaccharide Biosynthesis">
    <rank><val>level 3</val></rank>
    <score><val>-12</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="Tyrosine-protein kinase EpsD (EC 2.7.10.2)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
    </node>
   </node>
   <node name="Streptococcal Hyaluronic Acid Capsule">
    <rank><val>level 3</val></rank>
    <score><val>-22</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="UDP-glucose dehydrogenase (EC 1.1.1.22)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-22</val></score>
    </node>
   </node>
   <node name="Alginate metabolism">
    <magnitude><val>14</val><val>35</val></magnitude>
    <score><val>-41.7214</val><val>-78.92</val></score>
    <rank><val>level 3</val></rank>
    <node name="Phosphomannomutase (EC 5.4.2.8)">
     <rank><val>function</val></rank>
     <score><val>-44.7</val><val>-78.92</val></score>
     <magnitude><val>13</val><val>35</val></magnitude>
    </node>
    <node name="Mannose-6-phosphate isomerase (EC 5.3.1.8)">
     <rank><val>function</val></rank>
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Sialic Acid Metabolism">
    <rank><val>level 3</val></rank>
    <score><val>-34.9077</val><val>-74.9655</val></score>
    <magnitude><val>13</val><val>29</val></magnitude>
    <node name="Phosphoglucosamine mutase (EC 5.4.2.10)">
     <magnitude><val>4</val><val>1</val></magnitude>
     <score><val>-54</val><val>-11</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="N-acetylglucosamine-1-phosphate uridyltransferase (EC 2.7.7.23)">
     <score><val>-8</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glucosamine--fructose-6-phosphate aminotransferase [isomerizing] (EC 2.6.1.16)">
     <magnitude><val>6</val><val>28</val></magnitude>
     <score><val>-35.8</val><val>-77.25</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="N-acetylglucosamine-6-phosphate deacetylase (EC 3.5.1.25)">
     <rank><val>function</val></rank>
     <score><val>-7</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Glucosamine-1-phosphate N-acetyltransferase (EC 2.3.1.157)">
     <score><val>-8</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Polysaccharide deacetylases">
    <score><val>-39</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Polysaccharide deacetylase">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-39</val></score>
    </node>
   </node>
   <node name="Rhamnose containing glycans">
    <rank><val>level 3</val></rank>
    <magnitude><val>44</val><val>118</val></magnitude>
    <score><val>-29.5047</val><val>-51.3511</val></score>
    <node name="dTDP-4-dehydrorhamnose reductase (EC 1.1.1.133)">
     <magnitude><val>4</val><val>4</val></magnitude>
     <score><val>-17.5</val><val>-36</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glucose-1-phosphate thymidylyltransferase (EC 2.7.7.24)">
     <magnitude><val>14</val><val>37</val></magnitude>
     <score><val>-21.3846</val><val>-48.3846</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="UDP-glucose 4-epimerase (EC 5.1.3.2)">
     <rank><val>function</val></rank>
     <score><val>-34.8235</val><val>-54.0256</val></score>
     <magnitude><val>18</val><val>60</val></magnitude>
    </node>
    <node name="dTDP-4-dehydrorhamnose 3,5-epimerase (EC 5.1.3.13)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>11</val></magnitude>
     <score><val>-39</val><val>-51.3333</val></score>
    </node>
    <node name="dTDP-glucose 4,6-dehydratase (EC 4.2.1.46)">
     <magnitude><val>3</val><val>6</val></magnitude>
     <score><val>-35.6667</val><val>-53.1667</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="O-Methyl Phosphoramidate Capsule Modification in Campylobacter">
    <rank><val>level 3</val></rank>
    <score><val>-34</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="Adenylylsulfate kinase (EC 2.7.1.25)">
     <score><val>-34</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="dTDP-rhamnose synthesis">
    <rank><val>level 3</val></rank>
    <score><val>-29.7719</val><val>-52.1292</val></score>
    <magnitude><val>29</val><val>65</val></magnitude>
    <node name="dTDP-4-dehydrorhamnose 3,5-epimerase (EC 5.1.3.13)">
     <magnitude><val>5</val><val>11</val></magnitude>
     <score><val>-39</val><val>-51.3333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glucose-1-phosphate thymidylyltransferase (EC 2.7.7.24)">
     <rank><val>function</val></rank>
     <magnitude><val>14</val><val>37</val></magnitude>
     <score><val>-21.3846</val><val>-48.3846</val></score>
    </node>
    <node name="dTDP-4-dehydrorhamnose reductase (EC 1.1.1.133)">
     <rank><val>function</val></rank>
     <score><val>-17.5</val><val>-36</val></score>
     <magnitude><val>4</val><val>4</val></magnitude>
    </node>
    <node name="Glucose-1-phosphate cytidylyltransferase (EC 2.7.7.33)">
     <rank><val>function</val></rank>
     <score><val>-64</val><val>-81.5</val></score>
     <magnitude><val>3</val><val>7</val></magnitude>
    </node>
    <node name="dTDP-glucose 4,6-dehydratase (EC 4.2.1.46)">
     <magnitude><val>3</val><val>6</val></magnitude>
     <score><val>-35.6667</val><val>-53.1667</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Capsular heptose biosynthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>21</val><val>72</val></magnitude>
    <score><val>-23.619</val><val>-47.9527</val></score>
    <node name="Phosphoheptose isomerase 1 (EC 5.3.1.-)">
     <rank><val>function</val></rank>
     <score><val>-41</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Phosphoheptose isomerase (EC 5.3.1.-)">
     <rank><val>function</val></rank>
     <score><val>-40</val><val>-41.8571</val></score>
     <magnitude><val>5</val><val>19</val></magnitude>
    </node>
    <node name="dTDP-4-dehydrorhamnose 3,5-epimerase (EC 5.1.3.13)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>11</val></magnitude>
     <score><val>-39</val><val>-51.3333</val></score>
    </node>
    <node name="D-glycero-D-manno-heptose 1,7-bisphosphate phosphatase (EC 3.1.1.-)">
     <magnitude><val>6</val><val>19</val></magnitude>
     <score><val>-2.33333</val><val>-27.8235</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="D,D-heptose 7-phosphate kinase">
     <magnitude><val>4</val><val>23</val></magnitude>
     <score><val>-11.5</val><val>-68</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Gram-Negative cell wall components">
   <score><val>-29.66</val><val>-53.8831</val></score>
   <magnitude><val>35</val><val>81</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="LOS core oligosaccharide biosynthesis">
    <rank><val>level 3</val></rank>
    <score><val>-20</val><val>-34.8403</val></score>
    <magnitude><val>13</val><val>38</val></magnitude>
    <node name="Phosphoheptose isomerase 1 (EC 5.3.1.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-41</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Beta-1,3-glucosyltransferase">
     <magnitude><val>1</val></magnitude>
     <score><val>-5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="D-glycero-D-manno-heptose 1,7-bisphosphate phosphatase (EC 3.1.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>19</val></magnitude>
     <score><val>-2.33333</val><val>-27.8235</val></score>
    </node>
    <node name="Phosphoheptose isomerase (EC 5.3.1.-)">
     <rank><val>function</val></rank>
     <score><val>-40</val><val>-41.8571</val></score>
     <magnitude><val>5</val><val>19</val></magnitude>
    </node>
   </node>
   <node name="Lipid A-Ara4N pathway ( Polymyxin resistance )">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-22</val></score>
    <node name="UDP-glucose dehydrogenase (EC 1.1.1.22)">
     <rank><val>function</val></rank>
     <score><val>-22</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Lipid A modifications">
    <score><val>-1</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Sensor protein PhoQ (EC 2.7.13.3)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-1</val></score>
    </node>
   </node>
   <node name="Perosamine Synthesis Vibrio">
    <magnitude><val>14</val><val>35</val></magnitude>
    <score><val>-44.5071</val><val>-78.92</val></score>
    <rank><val>level 3</val></rank>
    <node name="O-antigen export system, ATP-binding protein">
     <score><val>-42</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphomannomutase (EC 5.4.2.8)">
     <rank><val>function</val></rank>
     <score><val>-44.7</val><val>-78.92</val></score>
     <magnitude><val>13</val><val>35</val></magnitude>
    </node>
   </node>
   <node name="Lipopolysaccharide-related cluster in Alphaproteobacteria">
    <magnitude><val>4</val><val>8</val></magnitude>
    <score><val>-14</val><val>-34.8</val></score>
    <rank><val>level 3</val></rank>
    <node name="TldE/PmbA protein, part of proposed TldE/TldD proteolytic complex (PMID 12029038)">
     <magnitude><val>4</val><val>8</val></magnitude>
     <score><val>-14</val><val>-34.8</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Vibrio Core Oligosaccharide Biosynthesis">
    <rank><val>level 3</val></rank>
    <score><val>-39</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="Polysaccharide deacetylase">
     <score><val>-39</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="KDO2-Lipid A biosynthesis">
    <rank><val>level 3</val></rank>
    <score><val>-37</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="Acyl-[acyl-carrier-protein]--UDP-N-acetylglucosamine O-acyltransferase (EC 2.3.1.129)">
     <score><val>-37</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Cell wall of Mycobacteria">
   <magnitude><val>14</val><val>157</val></magnitude>
   <score><val>-21.7143</val><val>-60.9524</val></score>
   <rank><val>level 2</val></rank>
   <node name="mycolic acid synthesis">
    <magnitude><val>14</val><val>157</val></magnitude>
    <score><val>-21.7143</val><val>-60.9524</val></score>
    <rank><val>level 3</val></rank>
    <node name="3-oxoacyl-[acyl-carrier protein] reductase (EC 1.1.1.100)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>157</val></magnitude>
     <score><val>-12.8571</val><val>-60.9524</val></score>
    </node>
    <node name="3-oxoacyl-[acyl-carrier-protein] synthase, KASII (EC 2.3.1.41)">
     <magnitude><val>2</val></magnitude>
     <score><val>-35.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="acyl-CoA synthetase">
     <score><val>-30</val></score>
     <magnitude><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Enoyl-[acyl-carrier-protein] reductase [NADH] (EC 1.3.1.9)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-29</val></score>
    </node>
    <node name="Malonyl CoA-acyl carrier protein transacylase (EC 2.3.1.39)">
     <score><val>-24</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="UDP-N-acetylmuramate from Fructose-6-phosphate Biosynthesis">
   <magnitude><val>14</val><val>29</val></magnitude>
   <score><val>-34.4143</val><val>-74.9655</val></score>
   <rank><val>level 2</val></rank>
   <node name="Glucosamine-1-phosphate N-acetyltransferase (EC 2.3.1.157)">
    <score><val>-8</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Glucosamine--fructose-6-phosphate aminotransferase [isomerizing] (EC 2.6.1.16)">
    <rank><val>function</val></rank>
    <magnitude><val>6</val><val>28</val></magnitude>
    <score><val>-35.8</val><val>-77.25</val></score>
   </node>
   <node name="UDP-N-acetylglucosamine 1-carboxyvinyltransferase (EC 2.5.1.7)">
    <rank><val>function</val></rank>
    <score><val>-17.5</val></score>
    <magnitude><val>2</val></magnitude>
   </node>
   <node name="N-acetylglucosamine-1-phosphate uridyltransferase (EC 2.7.7.23)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-8</val></score>
   </node>
   <node name="Phosphoglucosamine mutase (EC 5.4.2.10)">
    <rank><val>function</val></rank>
    <score><val>-54</val><val>-11</val></score>
    <magnitude><val>4</val><val>1</val></magnitude>
   </node>
  </node>
  <node name="Peptidoglycan Biosynthesis">
   <magnitude><val>22</val><val>40</val></magnitude>
   <score><val>-33.4091</val><val>-73.957</val></score>
   <rank><val>level 2</val></rank>
   <node name="N-acetylglucosamine-1-phosphate uridyltransferase (EC 2.7.7.23)">
    <score><val>-8</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Multimodular transpeptidase-transglycosylase (EC 2.4.1.129) (EC 3.4.-.-)">
    <rank><val>function</val></rank>
    <score><val>-4</val><val>-6</val></score>
    <magnitude><val>1</val><val>2</val></magnitude>
   </node>
   <node name="Glucosamine-1-phosphate N-acetyltransferase (EC 2.3.1.157)">
    <magnitude><val>1</val></magnitude>
    <score><val>-8</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="UDP-N-acetylmuramoylalanyl-D-glutamate--2,6-diaminopimelate ligase (EC 6.3.2.13)">
    <score><val>-10</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="UDP-N-acetylglucosamine 1-carboxyvinyltransferase (EC 2.5.1.7)">
    <rank><val>function</val></rank>
    <score><val>-17.5</val></score>
    <magnitude><val>2</val></magnitude>
   </node>
   <node name="Rod shape-determining protein RodA">
    <rank><val>function</val></rank>
    <score><val>-24</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="UDP-N-acetylmuramate--alanine ligase (EC 6.3.2.8)">
    <magnitude><val>2</val></magnitude>
    <score><val>-34</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="D-alanyl-D-alanine carboxypeptidase (EC 3.4.16.4)">
    <score><val>-1.5</val><val>-7</val></score>
    <magnitude><val>2</val><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Glutamine synthetase type I (EC 6.3.1.2)">
    <rank><val>function</val></rank>
    <score><val>-51.2222</val><val>-79.44</val></score>
    <magnitude><val>9</val><val>37</val></magnitude>
   </node>
   <node name="Cell division protein FtsI [Peptidoglycan synthetase] (EC 2.4.1.129)">
    <rank><val>function</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-57</val></score>
   </node>
  </node>
  <node name="YjeE">
   <magnitude><val>15</val><val>42</val></magnitude>
   <score><val>-14.3077</val><val>-53.4074</val></score>
   <rank><val>level 2</val></rank>
   <node name="YjeF protein, function unknown">
    <score><val>-14.3077</val><val>-53.4074</val></score>
    <magnitude><val>15</val><val>42</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Recycling of Peptidoglycan Amino Acids">
   <rank><val>level 2</val></rank>
   <magnitude><val>7</val><val>30</val></magnitude>
   <score><val>-27.4286</val><val>-86.8</val></score>
   <node name="Muramoyltetrapeptide carboxypeptidase (EC 3.4.17.13)">
    <score><val>-36.4</val><val>-86.8</val></score>
    <magnitude><val>5</val><val>30</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="N-acetylmuramoyl-L-alanine amidase (EC 3.5.1.28)">
    <rank><val>function</val></rank>
    <score><val>-5</val></score>
    <magnitude><val>2</val></magnitude>
   </node>
  </node>
  <node name="Peptidoglycan biosynthesis--gjo">
   <rank><val>level 2</val></rank>
   <score><val>-26</val></score>
   <magnitude><val>3</val></magnitude>
   <node name="UDP-N-acetylmuramoylalanyl-D-glutamate--2,6-diaminopimelate ligase (EC 6.3.2.13)">
    <magnitude><val>1</val></magnitude>
    <score><val>-10</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="UDP-N-acetylmuramate--alanine ligase (EC 6.3.2.8)">
    <score><val>-34</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
 </node>
 <node name="Phosphorus Metabolism">
  <magnitude><val>25</val><val>18</val></magnitude>
  <score><val>-38.5867</val><val>-33.8291</val></score>
  <rank><val>level 1</val></rank>
  <node name="High affinity phosphate transporter and control of PHO regulon">
   <magnitude><val>2</val></magnitude>
   <score><val>-49</val></score>
   <rank><val>level 2</val></rank>
   <node name="Phosphate transport system permease protein PstA (TC 3.A.1.7.1)">
    <rank><val>function</val></rank>
    <score><val>-8</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Phosphate transport ATP-binding protein PstB (TC 3.A.1.7.1)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-90</val></score>
   </node>
  </node>
  <node name="Phosphate metabolism">
   <magnitude><val>21</val><val>18</val></magnitude>
   <score><val>-36.6032</val><val>-33.8291</val></score>
   <rank><val>level 2</val></rank>
   <node name="response regulator in two-component regulatory system with PhoQ">
    <score><val>-23</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="NAD(P) transhydrogenase alpha subunit (EC 1.6.1.2)">
    <rank><val>function</val></rank>
    <magnitude><val>7</val><val>1</val></magnitude>
    <score><val>-35.1429</val><val>-10</val></score>
   </node>
   <node name="Phosphate transport system permease protein PstA (TC 3.A.1.7.1)">
    <score><val>-8</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="NAD(P) transhydrogenase subunit beta (EC 1.6.1.2)">
    <rank><val>function</val></rank>
    <score><val>-36</val></score>
    <magnitude><val>2</val></magnitude>
   </node>
   <node name="PhoQ">
    <score><val>-12</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Phosphate transport ATP-binding protein PstB (TC 3.A.1.7.1)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-90</val></score>
   </node>
   <node name="Inorganic pyrophosphatase (EC 3.6.1.1)">
    <magnitude><val>7</val><val>17</val></magnitude>
    <score><val>-43.6667</val><val>-35.2308</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="P uptake (cyanobacteria)">
   <rank><val>level 2</val></rank>
   <magnitude><val>2</val></magnitude>
   <score><val>-49</val></score>
   <node name="Phosphate transport system permease protein PstA (TC 3.A.1.7.1)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-8</val></score>
   </node>
   <node name="Phosphate transport ATP-binding protein PstB (TC 3.A.1.7.1)">
    <magnitude><val>1</val></magnitude>
    <score><val>-90</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
 </node>
 <node name="DNA Metabolism">
  <magnitude><val>467</val><val>1917</val></magnitude>
  <score><val>-37.4174</val><val>-73.1692</val></score>
  <rank><val>level 1</val></rank>
  <node name="DNA recombination">
   <rank><val>level 2</val></rank>
   <score><val>-69.3254</val><val>-60.4211</val></score>
   <magnitude><val>14</val><val>38</val></magnitude>
   <node name="RuvABC plus a hypothetical">
    <rank><val>level 3</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-71.5</val></score>
    <node name="Holliday junction DNA helicase RuvB">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-71.5</val></score>
    </node>
   </node>
   <node name="DNA recombination, archaeal">
    <magnitude><val>12</val><val>38</val></magnitude>
    <score><val>-68.963</val><val>-60.4211</val></score>
    <rank><val>level 3</val></rank>
    <node name="DNA repair and recombination protein RadB">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>8</val></magnitude>
     <score><val>-21</val><val>-14.5</val></score>
    </node>
    <node name="DNA repair and recombination protein RadA">
     <rank><val>function</val></rank>
     <score><val>-78.5556</val><val>-72.6667</val></score>
     <magnitude><val>10</val><val>30</val></magnitude>
    </node>
   </node>
  </node>
  <node name="DNA phosphorothioation">
   <score><val>-49</val></score>
   <magnitude><val>2</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="DNA sulfur modification protein DndB">
    <score><val>-29</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="3&#39;-phosphoadenosine 5&#39;-phosphosulfate sulfurtransferase DndC">
    <magnitude><val>1</val></magnitude>
    <score><val>-69</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Type I Restriction-Modification">
   <score><val>-47.5963</val><val>-89.8921</val></score>
   <magnitude><val>23</val><val>116</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Type I restriction-modification system, DNA-methyltransferase subunit M (EC 2.1.1.72)">
    <score><val>-80.7143</val><val>-90.5676</val></score>
    <magnitude><val>8</val><val>48</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Type I restriction-modification system, specificity subunit S (EC 3.1.21.3)">
    <rank><val>function</val></rank>
    <magnitude><val>8</val><val>15</val></magnitude>
    <score><val>-22.75</val><val>-59.4667</val></score>
   </node>
   <node name="Type I restriction-modification system, restriction subunit R (EC 3.1.21.3)">
    <rank><val>function</val></rank>
    <score><val>-38.1429</val><val>-97.8913</val></score>
    <magnitude><val>7</val><val>53</val></magnitude>
   </node>
  </node>
  <node name="DNA replication">
   <rank><val>level 2</val></rank>
   <magnitude><val>256</val><val>1056</val></magnitude>
   <score><val>-37.5932</val><val>-72.2605</val></score>
   <node name="DNA-replication">
    <rank><val>level 3</val></rank>
    <score><val>-41.5167</val><val>-82.6936</val></score>
    <magnitude><val>30</val><val>102</val></magnitude>
    <node name="Transcription-repair coupling factor">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>2</val></magnitude>
     <score><val>-34</val><val>-64</val></score>
    </node>
    <node name="DNA polymerase III epsilon subunit (EC 2.7.7.7)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-2</val></score>
    </node>
    <node name="Holliday junction DNA helicase RuvB">
     <rank><val>function</val></rank>
     <score><val>-71.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="DNA polymerase III alpha subunit (EC 2.7.7.7)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>2</val></magnitude>
     <score><val>-34.3333</val><val>-32</val></score>
    </node>
    <node name="Replicative DNA helicase (EC 3.6.1.-)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-17</val></score>
     <magnitude><val></val><val>1</val></magnitude>
    </node>
    <node name="DNA polymerase I (EC 2.7.7.7)">
     <score><val>-35.5</val><val>-69.7436</val></score>
     <magnitude><val>19</val><val>57</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA polymerase III subunits gamma and tau (EC 2.7.7.7)">
     <score><val>-44</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA repair protein RecN">
     <magnitude><val>1</val></magnitude>
     <score><val>-174</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Single-stranded-DNA-specific exonuclease RecJ (EC 3.1.-.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>40</val></magnitude>
     <score><val>-37</val><val>-106.259</val></score>
    </node>
   </node>
   <node name="Plasmid replication">
    <score><val>-50.4386</val><val>-39.069</val></score>
    <magnitude><val>36</val><val>43</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Chromosome (plasmid) partitioning protein ParA">
     <rank><val>function</val></rank>
     <score><val>-51.4211</val><val>-39.069</val></score>
     <magnitude><val>28</val><val>43</val></magnitude>
    </node>
    <node name="Chromosome (plasmid) partitioning protein ParB">
     <score><val>-46.4286</val></score>
     <magnitude><val>7</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Plasmid replication protein RepC">
     <rank><val>function</val></rank>
     <score><val>-51</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="DNA topoisomerases, Type I, ATP-independent">
    <rank><val>level 3</val></rank>
    <score><val>-31.75</val><val>-52.6535</val></score>
    <magnitude><val>16</val><val>38</val></magnitude>
    <node name="DNA topoisomerase I (EC 5.99.1.2)">
     <rank><val>function</val></rank>
     <score><val>-35</val><val>-53.8333</val></score>
     <magnitude><val>13</val><val>37</val></magnitude>
    </node>
    <node name="DNA topoisomerase III (EC 5.99.1.2)">
     <score><val>-11</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA topoisomerase V">
     <rank><val>function</val></rank>
     <score><val>-21</val><val>-9</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
   </node>
   <node name="DNA replication, archaeal">
    <magnitude><val>137</val><val>715</val></magnitude>
    <score><val>-32.2869</val><val>-71.4726</val></score>
    <rank><val>level 3</val></rank>
    <node name="DNA replication helicase protein MCM">
     <rank><val>function</val></rank>
     <magnitude><val>18</val><val>112</val></magnitude>
     <score><val>-41.3125</val><val>-74.5909</val></score>
    </node>
    <node name="Archaeal DNA polymerase II small subunit (EC 2.7.7.7)">
     <magnitude><val>13</val><val>41</val></magnitude>
     <score><val>-39.9091</val><val>-114.381</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Archaeal DNA polymerase II large subunit (EC 2.7.7.7)">
     <rank><val>function</val></rank>
     <magnitude><val>14</val><val>116</val></magnitude>
     <score><val>-55.2308</val><val>-71.7821</val></score>
    </node>
    <node name="Cell division control protein 6">
     <rank><val>function</val></rank>
     <score><val>-24.4</val><val>-55.4667</val></score>
     <magnitude><val>12</val><val>64</val></magnitude>
    </node>
    <node name="Origin of replication recognition protein">
     <magnitude><val>12</val><val>64</val></magnitude>
     <score><val>-24.4</val><val>-55.4667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Flap structure-specific endonuclease (EC 3.-.-.-)">
     <magnitude><val>6</val><val>20</val></magnitude>
     <score><val>-24.1667</val><val>-77.3889</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Archaeal DNA polymerase I (EC 2.7.7.7)">
     <rank><val>function</val></rank>
     <magnitude><val>17</val><val>57</val></magnitude>
     <score><val>-37.5714</val><val>-69.7436</val></score>
    </node>
    <node name="Replication factor A (ssDNA-binding protein)">
     <score><val>-18</val><val>-57</val></score>
     <magnitude><val>5</val><val>14</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA primase small subunit (EC 2.7.7.-)">
     <magnitude><val>13</val><val>30</val></magnitude>
     <score><val>-13.7778</val><val>-50.1176</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Replication factor C large subunit">
     <rank><val>function</val></rank>
     <score><val>-14</val><val>-97.4074</val></score>
     <magnitude><val>1</val><val>39</val></magnitude>
    </node>
    <node name="ATP-dependent DNA ligase (EC 6.5.1.1)">
     <rank><val>function</val></rank>
     <score><val>-29.2</val><val>-89.3947</val></score>
     <magnitude><val>11</val><val>63</val></magnitude>
    </node>
    <node name="DNA primase large subunit (EC 2.7.7.-)">
     <score><val>0</val><val>-88.5385</val></score>
     <magnitude><val></val><val>15</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Replication factor C small subunit">
     <score><val>-24.8</val><val>-56.6452</val></score>
     <magnitude><val>5</val><val>39</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribonuclease HII (EC 3.1.26.4)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>26</val></magnitude>
     <score><val>-11</val><val>-35.0714</val></score>
    </node>
    <node name="DNA polymerase sliding clamp protein PCNA">
     <score><val>-41</val><val>-61.875</val></score>
     <magnitude><val>6</val><val>15</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="DNA topoisomerases, Type II, ATP-dependent">
    <rank><val>level 3</val></rank>
    <magnitude><val>28</val><val>108</val></magnitude>
    <score><val>-42.55</val><val>-87.1972</val></score>
    <node name="DNA gyrase subunit A (EC 5.99.1.3)">
     <magnitude><val>15</val><val>53</val></magnitude>
     <score><val>-24.2667</val><val>-83</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA gyrase subunit B (EC 5.99.1.3)">
     <rank><val>function</val></rank>
     <magnitude><val>11</val><val>54</val></magnitude>
     <score><val>-71.4</val><val>-92.7647</val></score>
    </node>
    <node name="DNA topoisomerase VI subunit A (EC 5.99.1.3)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-9</val></score>
    </node>
    <node name="DNA topoisomerase VI subunit B (EC 5.99.1.3)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-21</val></score>
    </node>
   </node>
   <node name="DNA Helicase of Unknown Function">
    <magnitude><val>9</val><val>50</val></magnitude>
    <score><val>-48.875</val><val>-73.426</val></score>
    <rank><val>level 3</val></rank>
    <node name="DNA helicase (Rad25 homolog)">
     <magnitude><val>9</val><val>43</val></magnitude>
     <score><val>-48.875</val><val>-73.7667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Hypothetical protein (associated with DNA helicase - Rad25 homolog)">
     <magnitude><val></val><val>7</val></magnitude>
     <score><val>0</val><val>-71.3333</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="CRISPs">
   <rank><val>level 2</val></rank>
   <score><val>-23.5625</val><val>-31.7191</val></score>
   <magnitude><val>16</val><val>18</val></magnitude>
   <node name="CRISPRs">
    <rank><val>level 3</val></rank>
    <magnitude><val>16</val><val>18</val></magnitude>
    <score><val>-23.5625</val><val>-31.7191</val></score>
    <node name="CRISPR-associated protein Cas1">
     <magnitude><val>5</val><val>11</val></magnitude>
     <score><val>-19.2</val><val>-38.2222</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="CRISPR-associated protein, CT1972 family">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-18</val></score>
    </node>
    <node name="CRISPR-associated helicase Cas3, protein">
     <score><val>-45</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="CRISPR-associated protein, TM1802 family">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-37</val></score>
    </node>
    <node name="CRISPR-associated protein, MJ0381 family">
     <rank><val>function</val></rank>
     <score><val>-8</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="CRISPR-associated RecB family exonuclease Cas4b">
     <magnitude><val>2</val><val>3</val></magnitude>
     <score><val>-28</val><val>-17.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="CRISPR-associated protein, CT1975 family">
     <rank><val>function</val></rank>
     <score><val>-31.6667</val><val>-37</val></score>
     <magnitude><val>3</val><val>1</val></magnitude>
    </node>
    <node name="CRISPR-associated protein Cas2">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-12</val></score>
    </node>
    <node name="CRISPR-associated helicase Cas3">
     <rank><val>function</val></rank>
     <score><val>-19.6667</val></score>
     <magnitude><val>3</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Restriction-Modification System">
   <score><val>-44.3086</val><val>-83.8196</val></score>
   <magnitude><val>25</val><val>127</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Putative predicted metal-dependent hydrolase">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>5</val></magnitude>
    <score><val>0</val><val>-20</val></score>
   </node>
   <node name="Type I restriction-modification system, restriction subunit R (EC 3.1.21.3)">
    <rank><val>function</val></rank>
    <score><val>-38.1429</val><val>-97.8913</val></score>
    <magnitude><val>7</val><val>53</val></magnitude>
   </node>
   <node name="Type I restriction-modification system, specificity subunit S (EC 3.1.21.3)">
    <rank><val>function</val></rank>
    <score><val>-22.75</val><val>-59.4667</val></score>
    <magnitude><val>8</val><val>15</val></magnitude>
   </node>
   <node name="Type III restriction-modification system methylation subunit (EC 2.1.1.72)">
    <score><val>-6.5</val><val>-19.6</val></score>
    <magnitude><val>2</val><val>6</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Type I restriction-modification system, DNA-methyltransferase subunit M (EC 2.1.1.72)">
    <rank><val>function</val></rank>
    <score><val>-80.7143</val><val>-90.5676</val></score>
    <magnitude><val>8</val><val>48</val></magnitude>
   </node>
  </node>
  <node name="DNA repair">
   <rank><val>level 2</val></rank>
   <score><val>-32.1479</val><val>-71.2077</val></score>
   <magnitude><val>129</val><val>562</val></magnitude>
   <node name="DNA repair, bacterial MutL-MutS system">
    <rank><val>level 3</val></rank>
    <score><val>-22.1852</val><val>-61.3918</val></score>
    <magnitude><val>9</val><val>38</val></magnitude>
    <node name="MutS domain protein, family 4">
     <score><val>-19</val><val>-56.1053</val></score>
     <magnitude><val>1</val><val>24</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA mismatch repair protein MutL">
     <score><val>-14.6667</val></score>
     <magnitude><val>4</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="MutS domain protein, family 5">
     <magnitude><val>3</val><val>14</val></magnitude>
     <score><val>-31.3333</val><val>-70.4545</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA mismatch repair protein MutS">
     <score><val>-28</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Uracil-DNA glycosylase">
    <rank><val>level 3</val></rank>
    <score><val>-29.8571</val><val>-40.8577</val></score>
    <magnitude><val>7</val><val>23</val></magnitude>
    <node name="Uracil-DNA glycosylase, family 5">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>17</val></magnitude>
     <score><val>-37.6</val><val>-47.4545</val></score>
    </node>
    <node name="Uracil-DNA glycosylase, family 4">
     <rank><val>function</val></rank>
     <score><val>-10.5</val><val>-22.1667</val></score>
     <magnitude><val>2</val><val>6</val></magnitude>
    </node>
   </node>
   <node name="DNA repair, bacterial DinG and relatives">
    <score><val>-29</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="DinG family ATP-dependent helicase YoaA">
     <rank><val>function</val></rank>
     <score><val>-29</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="DNA repair, bacterial RecFOR pathway">
    <magnitude><val>3</val><val>40</val></magnitude>
    <score><val>-13</val><val>-106.259</val></score>
    <rank><val>level 3</val></rank>
    <node name="Single-stranded DNA-binding protein">
     <magnitude><val>2</val></magnitude>
     <score><val>-1</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Single-stranded-DNA-specific exonuclease RecJ (EC 3.1.-.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>40</val></magnitude>
     <score><val>-37</val><val>-106.259</val></score>
    </node>
   </node>
   <node name="DNA repair, bacterial photolyase">
    <rank><val>level 3</val></rank>
    <magnitude><val>3</val><val>3</val></magnitude>
    <score><val>-41.5</val><val>-14.3333</val></score>
    <node name="Deoxyribodipyrimidine photolyase (EC 4.1.99.3)">
     <score><val>-41.5</val><val>-14.3333</val></score>
     <magnitude><val>3</val><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="DNA Repair Base Excision">
    <rank><val>level 3</val></rank>
    <magnitude><val>48</val><val>223</val></magnitude>
    <score><val>-30.1521</val><val>-76.5845</val></score>
    <node name="Single-stranded-DNA-specific exonuclease RecJ (EC 3.1.-.-)">
     <magnitude><val>1</val><val>40</val></magnitude>
     <score><val>-37</val><val>-106.259</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="ATP-dependent DNA ligase (EC 6.5.1.1) clustered with Ku protein, LigD">
     <score><val>-7</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Flap structure-specific endonuclease (EC 3.-.-.-)">
     <score><val>-24.1667</val><val>-77.3889</val></score>
     <magnitude><val>6</val><val>20</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA-3-methyladenine glycosylase (EC 3.2.2.20)">
     <score><val>-48</val><val>-28.8</val></score>
     <magnitude><val>1</val><val>9</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Formamidopyrimidine-DNA glycosylase (EC 3.2.2.23)">
     <score><val>-11</val><val>-3</val></score>
     <magnitude><val>1</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA polymerase I (EC 2.7.7.7)">
     <magnitude><val>19</val><val>57</val></magnitude>
     <score><val>-35.5</val><val>-69.7436</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Endonuclease III (EC 4.2.99.18)">
     <rank><val>function</val></rank>
     <score><val>-19.6</val><val>-42.75</val></score>
     <magnitude><val>6</val><val>33</val></magnitude>
    </node>
    <node name="DNA ligase (EC 6.5.1.2)">
     <score><val>-43</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="ATP-dependent DNA ligase (EC 6.5.1.1)">
     <rank><val>function</val></rank>
     <score><val>-29.2</val><val>-89.3947</val></score>
     <magnitude><val>11</val><val>63</val></magnitude>
    </node>
   </node>
   <node name="DNA repair, bacterial RecBCD pathway">
    <magnitude><val>8</val><val>3</val></magnitude>
    <score><val>-28</val><val>-13.6667</val></score>
    <rank><val>level 3</val></rank>
    <node name="RecD-like DNA helicase YrrC">
     <score><val>-28</val><val>-19.5</val></score>
     <magnitude><val>8</val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="RecD-like DNA helicase Atu2026">
     <score><val>0</val><val>-2</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Nonhomologous End-Joining in Bacteria">
    <score><val>-7</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="ATP-dependent DNA ligase (EC 6.5.1.1) clustered with Ku protein, LigD">
     <score><val>-7</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="DNA repair, bacterial UvrD and related helicases">
    <rank><val>level 3</val></rank>
    <magnitude><val>13</val><val>54</val></magnitude>
    <score><val>-17.3846</val><val>-87.0995</val></score>
    <node name="ATP-dependent DNA helicase pcrA (EC 3.6.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>43</val></magnitude>
     <score><val>-8.33333</val><val>-81.625</val></score>
    </node>
    <node name="ATP-dependent DNA helicase UvrD/PcrA, actinomycete paralog">
     <magnitude><val>1</val></magnitude>
     <score><val>-6</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="ATP-dependent DNA helicase UvrD/PcrA">
     <magnitude><val>8</val><val>11</val></magnitude>
     <score><val>-23.375</val><val>-108.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="putative ATP-dependent DNA helicase YjcD">
     <magnitude><val>1</val></magnitude>
     <score><val>-8</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="2-phosphoglycolate salvage">
    <magnitude><val>2</val><val>17</val></magnitude>
    <score><val>-7</val><val>-38.5455</val></score>
    <rank><val>level 3</val></rank>
    <node name="Phosphoglycolate phosphatase, archaeal type (EC 3.1.3.18)">
     <score><val>-7</val><val>-38.5455</val></score>
     <magnitude><val>2</val><val>17</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="DNA repair, UvrABC system">
    <rank><val>level 3</val></rank>
    <magnitude><val>9</val></magnitude>
    <score><val>-70.1111</val></score>
    <node name="Excinuclease ABC subunit A">
     <rank><val>function</val></rank>
     <magnitude><val>3</val></magnitude>
     <score><val>-62.3333</val></score>
    </node>
    <node name="Excinuclease ABC subunit B">
     <score><val>-74</val></score>
     <magnitude><val>6</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="DNA repair, bacterial">
    <rank><val>level 3</val></rank>
    <score><val>-39.8646</val><val>-61.9551</val></score>
    <magnitude><val>25</val><val>161</val></magnitude>
    <node name="Methylated-DNA--protein-cysteine methyltransferase (EC 2.1.1.63)">
     <magnitude><val>1</val><val>17</val></magnitude>
     <score><val>-39</val><val>-41.4</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Endonuclease IV (EC 3.1.21.2)">
     <score><val>-41.6154</val><val>-66.9048</val></score>
     <magnitude><val>14</val><val>62</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Endonuclease V (EC 3.1.21.7)">
     <rank><val>function</val></rank>
     <score><val>-28</val><val>-18</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
    <node name="DNA repair protein RecN">
     <score><val>-174</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA recombination protein RmuC">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-87.1</val></score>
     <magnitude><val></val><val>19</val></magnitude>
    </node>
    <node name="Exodeoxyribonuclease III (EC 3.1.11.2)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>30</val></magnitude>
     <score><val>-27.3333</val><val>-40.7059</val></score>
    </node>
    <node name="A/G-specific adenine glycosylase (EC 3.2.2.-)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-15</val></score>
     <magnitude><val></val><val>1</val></magnitude>
    </node>
    <node name="DNA-cytosine methyltransferase (EC 2.1.1.37)">
     <score><val>0</val><val>-31.5</val></score>
     <magnitude><val></val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Single-stranded DNA-binding protein">
     <rank><val>function</val></rank>
     <score><val>-1</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Exodeoxyribonuclease VII large subunit (EC 3.1.11.6)">
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Very-short-patch mismatch repair endonuclease (G-T specific)">
     <magnitude><val>1</val><val>2</val></magnitude>
     <score><val>-49</val><val>-44</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA polymerase IV (EC 2.7.7.7)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-76.4</val></score>
     <magnitude><val></val><val>27</val></magnitude>
    </node>
   </node>
  </node>
  <node name="DNA uptake, competence">
   <magnitude><val>2</val></magnitude>
   <score><val>-27.5</val></score>
   <rank><val>level 2</val></rank>
   <node name="DNA processing cluster">
    <rank><val>level 3</val></rank>
    <score><val>-27.5</val></score>
    <magnitude><val>2</val></magnitude>
    <node name="DNA topoisomerase III (EC 5.99.1.2)">
     <magnitude><val>1</val></magnitude>
     <score><val>-11</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="DNA polymerase III subunits gamma and tau (EC 2.7.7.7)">
     <score><val>-44</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
 </node>
 <node name="Carbohydrates">
  <rank><val>level 1</val></rank>
  <magnitude><val>1704</val><val>7739</val></magnitude>
  <score><val>-44.2026</val><val>-71.9241</val></score>
  <node name="CO2 fixation">
   <rank><val>level 2</val></rank>
   <score><val>-44.7523</val><val>-64.7636</val></score>
   <magnitude><val>135</val><val>453</val></magnitude>
   <node name="Carboxysome">
    <rank><val>level 3</val></rank>
    <score><val>-14.4444</val><val>-53.875</val></score>
    <magnitude><val>9</val><val>8</val></magnitude>
    <node name="NADH dehydrogenase (EC 1.6.99.3)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>8</val></magnitude>
     <score><val>-10</val><val>-53.875</val></score>
    </node>
    <node name="Ribulose bisphosphate carboxylase large chain (EC 4.1.1.39)">
     <rank><val>function</val></rank>
     <score><val>-50</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Calvin-Benson cycle">
    <rank><val>level 3</val></rank>
    <score><val>-47.6821</val><val>-64.7667</val></score>
    <magnitude><val>65</val><val>223</val></magnitude>
    <node name="Ribulose-phosphate 3-epimerase (EC 5.1.3.1)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>31</val></magnitude>
     <score><val>-17.3333</val><val>-42</val></score>
    </node>
    <node name="Triosephosphate isomerase (EC 5.3.1.1)">
     <rank><val>function</val></rank>
     <score><val>-16.6667</val><val>-55.5714</val></score>
     <magnitude><val>4</val><val>23</val></magnitude>
    </node>
    <node name="Transketolase (EC 2.2.1.1)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-17</val></score>
    </node>
    <node name="Ribose 5-phosphate isomerase A (EC 5.3.1.6)">
     <magnitude><val>7</val><val>24</val></magnitude>
     <score><val>-16.1667</val><val>-49</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Fructose-bisphosphate aldolase class I (EC 4.1.2.13)">
     <score><val>-8</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="NAD(P)-dependent glyceraldehyde 3-phosphate dehydrogenase archaeal (EC 1.2.1.59)">
     <rank><val>function</val></rank>
     <score><val>-42.5</val><val>-57.9444</val></score>
     <magnitude><val>5</val><val>29</val></magnitude>
    </node>
    <node name="Ribulose bisphosphate carboxylase large chain (EC 4.1.1.39)">
     <magnitude><val>1</val></magnitude>
     <score><val>-50</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Fructose-1,6-bisphosphatase, type I (EC 3.1.3.11)">
     <score><val>-8</val></score>
     <magnitude><val>4</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Transketolase, C-terminal section (EC 2.2.1.1)">
     <score><val>-79.8333</val><val>-86.7273</val></score>
     <magnitude><val>12</val><val>32</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Transketolase, N-terminal section (EC 2.2.1.1)">
     <magnitude><val>7</val><val>33</val></magnitude>
     <score><val>-79.1429</val><val>-76.0455</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Fructose-1,6-bisphosphatase, type V, archaeal (EC 3.1.3.11)">
     <magnitude><val>10</val><val>38</val></magnitude>
     <score><val>-70.1</val><val>-76.1905</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Fructose-1,6-bisphosphatase, GlpX type (EC 3.1.3.11)">
     <score><val>-127</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="NAD-dependent glyceraldehyde-3-phosphate dehydrogenase (EC 1.2.1.12)">
     <score><val>-11</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoglycerate kinase (EC 2.7.2.3)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>13</val></magnitude>
     <score><val>-25.7143</val><val>-63.5714</val></score>
    </node>
   </node>
   <node name="Photorespiration (oxidative C2 cycle)">
    <score><val>-46.0886</val><val>-65.1528</val></score>
    <magnitude><val>59</val><val>222</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Glycine cleavage system H protein">
     <rank><val>function</val></rank>
     <score><val>-33.4</val><val>-30.375</val></score>
     <magnitude><val>5</val><val>11</val></magnitude>
    </node>
    <node name="Glycolate dehydrogenase (EC 1.1.99.14), subunit GlcD">
     <magnitude><val>2</val><val>1</val></magnitude>
     <score><val>-10.5</val><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Aminomethyltransferase (glycine cleavage system T protein) (EC 2.1.2.10)">
     <magnitude><val>10</val><val>38</val></magnitude>
     <score><val>-21.125</val><val>-58.8571</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Serine hydroxymethyltransferase (EC 2.1.2.1)">
     <rank><val>function</val></rank>
     <score><val>-76</val><val>-79.3077</val></score>
     <magnitude><val>11</val><val>67</val></magnitude>
    </node>
    <node name="Glycine dehydrogenase [decarboxylating] (glycine cleavage system P1 protein) (EC 1.4.4.2)">
     <rank><val>function</val></rank>
     <score><val>-40.375</val><val>-71.381</val></score>
     <magnitude><val>9</val><val>39</val></magnitude>
    </node>
    <node name="Ribulose bisphosphate carboxylase large chain (EC 4.1.1.39)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-50</val></score>
    </node>
    <node name="Malate synthase G (EC 2.3.3.9)">
     <score><val>-39</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="L-alanine:glyoxylate aminotransferase (EC 2.6.1.44)">
     <rank><val>function</val></rank>
     <score><val>-33</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Hydroxypyruvate reductase (EC 1.1.1.81)">
     <magnitude><val></val><val>29</val></magnitude>
     <score><val>0</val><val>-60.0833</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Serine--glyoxylate aminotransferase (EC 2.6.1.45)">
     <magnitude><val>4</val><val>12</val></magnitude>
     <score><val>-9.5</val><val>-29</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycerate kinase (EC 2.7.1.31)">
     <magnitude><val>1</val></magnitude>
     <score><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycine dehydrogenase [decarboxylating] (glycine cleavage system P2 protein) (EC 1.4.4.2)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>2</val></magnitude>
     <score><val>-57.5714</val><val>-32</val></score>
    </node>
    <node name="Glyoxylate carboligase (EC 4.1.1.47)">
     <score><val>-25</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrolipoamide dehydrogenase (EC 1.8.1.4)">
     <score><val>0</val><val>-44.7</val></score>
     <magnitude><val></val><val>10</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Catalase (EC 1.11.1.6)">
     <score><val>-87.6</val><val>-91.3333</val></score>
     <magnitude><val>6</val><val>13</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="CO2 uptake, carboxysome">
    <rank><val>level 3</val></rank>
    <score><val>-46.5</val></score>
    <magnitude><val>2</val></magnitude>
    <node name="Ribulose bisphosphate carboxylase large chain (EC 4.1.1.39)">
     <score><val>-50</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="carboxysome shell protein CsoS1">
     <score><val>-43</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Lacto-N-Biose I and Galacto-N-Biose Metabolic Pathway">
   <magnitude><val>18</val><val>60</val></magnitude>
   <score><val>-34.8235</val><val>-54.0256</val></score>
   <rank><val>level 2</val></rank>
   <node name="UDP-glucose 4-epimerase (EC 5.1.3.2)">
    <magnitude><val>18</val><val>60</val></magnitude>
    <score><val>-34.8235</val><val>-54.0256</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Glycoside hydrolases">
   <rank><val>level 2</val></rank>
   <score><val>-39</val></score>
   <magnitude><val>1</val></magnitude>
   <node name="Predicted carbohydrate hydrolases">
    <score><val>-39</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Polysaccharide deacetylase">
     <score><val>-39</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Aminosugars">
   <magnitude><val>14</val><val>6</val></magnitude>
   <score><val>-16.5</val><val>-19.625</val></score>
   <rank><val>level 2</val></rank>
   <node name="N-Acetyl-Galactosamine and Galactosamine Utilization">
    <magnitude><val>1</val></magnitude>
    <score><val>-35</val></score>
    <rank><val>level 3</val></rank>
    <node name="6-phosphofructokinase (EC 2.7.1.11)">
     <magnitude><val>1</val></magnitude>
     <score><val>-35</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Chitin and N-acetylglucosamine utilization">
    <rank><val>level 3</val></rank>
    <magnitude><val>13</val><val>5</val></magnitude>
    <score><val>-15.0769</val><val>-15.75</val></score>
    <node name="N-Acetyl-D-glucosamine ABC transport system, permease protein 1">
     <magnitude><val>2</val></magnitude>
     <score><val>-10.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="N-Acetyl-D-glucosamine ABC transport system, sugar-binding protein">
     <rank><val>function</val></rank>
     <score><val>-9</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="N-Acetyl-D-glucosamine ABC transport system, permease protein 2">
     <magnitude><val>2</val></magnitude>
     <score><val>-28.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Predicted transcriptional regulator of N-Acetylglucosamine utilization, GntR family">
     <score><val>-18</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="N-acetylglucosamine-6-phosphate deacetylase (EC 3.5.1.25)">
     <magnitude><val>1</val></magnitude>
     <score><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="N-acetylglucosamine kinase of eukaryotic type (EC 2.7.1.59)">
     <rank><val>function</val></rank>
     <score><val>-15</val><val>-15.75</val></score>
     <magnitude><val>5</val><val>5</val></magnitude>
    </node>
   </node>
   <node name="(GlcNAc)2 Catabolic Operon">
    <rank><val>level 3</val></rank>
    <score><val>0</val><val>-39</val></score>
    <magnitude><val></val><val>1</val></magnitude>
    <node name="(GlcNAc)2 ABC transporter, ATP-binding component 2">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-39</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="One-carbon Metabolism">
   <rank><val>level 2</val></rank>
   <score><val>-47.2997</val><val>-75.2644</val></score>
   <magnitude><val>270</val><val>1206</val></magnitude>
   <node name="One-carbon metabolism by tetrahydropterines">
    <score><val>-30.0057</val><val>-68.12</val></score>
    <magnitude><val>25</val><val>125</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Methenyltetrahydrofolate cyclohydrolase (EC 3.5.4.9)">
     <rank><val>function</val></rank>
     <score><val>-14.2</val><val>-53.5</val></score>
     <magnitude><val>5</val><val>17</val></magnitude>
    </node>
    <node name="Formate--tetrahydrofolate ligase (EC 6.3.4.3)">
     <score><val>-67.1429</val><val>-86.5909</val></score>
     <magnitude><val>8</val><val>64</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Methylenetetrahydrofolate dehydrogenase (NADP+) (EC 1.5.1.5)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>17</val></magnitude>
     <score><val>-14.2</val><val>-53.5</val></score>
    </node>
    <node name="Formiminotetrahydrofolate cyclodeaminase (EC 4.3.1.4)">
     <score><val>-11</val><val>-52.8182</val></score>
     <magnitude><val>4</val><val>21</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="5,10-methylenetetrahydrofolate reductase (EC 1.5.1.20)">
     <rank><val>function</val></rank>
     <score><val>-9</val><val>-7.5</val></score>
     <magnitude><val>3</val><val>6</val></magnitude>
    </node>
   </node>
   <node name="Serine-glyoxylate cycle">
    <score><val>-49.2285</val><val>-76.0905</val></score>
    <magnitude><val>244</val><val>1081</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Methylmalonyl-CoA mutase (EC 5.4.99.2)">
     <rank><val>function</val></rank>
     <magnitude><val>16</val><val>46</val></magnitude>
     <score><val>-102</val><val>-92.2258</val></score>
    </node>
    <node name="ethylmalonyl-CoA mutase">
     <rank><val>function</val></rank>
     <score><val>-102</val><val>-92.2258</val></score>
     <magnitude><val>16</val><val>46</val></magnitude>
    </node>
    <node name="Succinate dehydrogenase flavoprotein subunit (EC 1.3.99.1)">
     <magnitude><val>13</val><val>53</val></magnitude>
     <score><val>-65.7692</val><val>-125.667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Succinyl-CoA ligase [ADP-forming] beta chain (EC 6.2.1.5)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>27</val></magnitude>
     <score><val>-37</val><val>-80</val></score>
    </node>
    <node name="Serine hydroxymethyltransferase (EC 2.1.2.1)">
     <rank><val>function</val></rank>
     <score><val>-76</val><val>-79.3077</val></score>
     <magnitude><val>11</val><val>67</val></magnitude>
    </node>
    <node name="Acetyl-CoA acetyltransferase (EC 2.3.1.9)">
     <rank><val>function</val></rank>
     <magnitude><val>28</val><val>112</val></magnitude>
     <score><val>-49.3462</val><val>-76.8312</val></score>
    </node>
    <node name="Serine--pyruvate aminotransferase (EC 2.6.1.51)">
     <magnitude><val>1</val></magnitude>
     <score><val>-33</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Methenyltetrahydrofolate cyclohydrolase (EC 3.5.4.9)">
     <magnitude><val>5</val><val>17</val></magnitude>
     <score><val>-14.2</val><val>-53.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycerate kinase (EC 2.7.1.31)">
     <magnitude><val>1</val></magnitude>
     <score><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Citrate synthase (si) (EC 2.3.3.1)">
     <magnitude><val>4</val><val>37</val></magnitude>
     <score><val>-20.75</val><val>-78.8966</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Serine--glyoxylate aminotransferase (EC 2.6.1.45)">
     <score><val>-9.5</val><val>-29</val></score>
     <magnitude><val>4</val><val>12</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Formate--tetrahydrofolate ligase (EC 6.3.4.3)">
     <score><val>-67.1429</val><val>-86.5909</val></score>
     <magnitude><val>8</val><val>64</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Aconitate hydratase (EC 4.2.1.3)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>64</val></magnitude>
     <score><val>-76</val><val>-103.438</val></score>
    </node>
    <node name="Isocitrate lyase (EC 4.1.3.1)">
     <rank><val>function</val></rank>
     <score><val>-41</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Malate dehydrogenase (EC 1.1.1.37)">
     <rank><val>function</val></rank>
     <score><val>-73.2</val><val>-62.6071</val></score>
     <magnitude><val>8</val><val>51</val></magnitude>
    </node>
    <node name="Succinyl-CoA:3-ketoacid-coenzyme A transferase subunit B (EC 2.8.3.5)">
     <score><val>-79</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="5,10-methylenetetrahydrofolate reductase (EC 1.5.1.20)">
     <score><val>-9</val><val>-7.5</val></score>
     <magnitude><val>3</val><val>6</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="3-ketoacyl-CoA thiolase (EC 2.3.1.16)">
     <score><val>-49.92</val><val>-76.8312</val></score>
     <magnitude><val>27</val><val>112</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="decarboxylase">
     <rank><val>function</val></rank>
     <magnitude><val>57</val><val>266</val></magnitude>
     <score><val>-17.0612</val><val>-58.4724</val></score>
    </node>
    <node name="Methylcrotonyl-CoA carboxylase biotin-containing subunit (EC 6.4.1.4)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-43</val></score>
    </node>
    <node name="Succinate dehydrogenase iron-sulfur protein (EC 1.3.99.1)">
     <magnitude><val>3</val><val>18</val></magnitude>
     <score><val>-30.6667</val><val>-94.9231</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Succinyl-CoA ligase [ADP-forming] alpha chain (EC 6.2.1.5)">
     <score><val>-89.5714</val><val>-79.2</val></score>
     <magnitude><val>7</val><val>7</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Formiminotetrahydrofolate cyclodeaminase (EC 4.3.1.4)">
     <score><val>-11</val><val>-52.8182</val></score>
     <magnitude><val>4</val><val>21</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Enolase (EC 4.2.1.11)">
     <magnitude><val>16</val><val>38</val></magnitude>
     <score><val>-46.875</val><val>-79.2917</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Methylenetetrahydrofolate dehydrogenase (NADP+) (EC 1.5.1.5)">
     <magnitude><val>5</val><val>17</val></magnitude>
     <score><val>-14.2</val><val>-53.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Methanogenesis">
    <rank><val>level 3</val></rank>
    <score><val>-9</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="F420-dependent N(5),N(10)-methylenetetrahydromethanopterin reductase (EC 1.5.99.11)">
     <magnitude><val>1</val></magnitude>
     <score><val>-9</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Organic acids">
   <rank><val>level 2</val></rank>
   <score><val>-65.8105</val><val>-85.3188</val></score>
   <magnitude><val>109</val><val>704</val></magnitude>
   <node name="Lactate utilization">
    <score><val>-28</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Predicted L-lactate dehydrogenase, Iron-sulfur cluster-binding subunit YkgF">
     <magnitude><val>1</val></magnitude>
     <score><val>-21</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="L-lactate dehydrogenase (EC 1.1.2.3)">
     <score><val>-35</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Propionyl-CoA to Succinyl-CoA Module">
    <rank><val>level 3</val></rank>
    <score><val>-98</val><val>-85.4003</val></score>
    <magnitude><val>17</val><val>57</val></magnitude>
    <node name="B12 binding domain of Methylmalonyl-CoA mutase (EC 5.4.99.2)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>11</val></magnitude>
     <score><val>-34</val><val>-56.8571</val></score>
    </node>
    <node name="Methylmalonyl-CoA mutase (EC 5.4.99.2)">
     <score><val>-102</val><val>-92.2258</val></score>
     <magnitude><val>16</val><val>46</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Isobutyryl-CoA to Propionyl-CoA Module">
    <score><val>-49.8522</val><val>-54.735</val></score>
    <magnitude><val>23</val><val>26</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="3-hydroxyisobutyrate dehydrogenase (EC 1.1.1.31)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>1</val></magnitude>
     <score><val>0</val><val>-12</val></score>
    </node>
    <node name="Butyryl-CoA dehydrogenase (EC 1.3.99.2)">
     <magnitude><val>21</val><val>25</val></magnitude>
     <score><val>-54.6</val><val>-56.4444</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Methylcitrate cycle">
    <score><val>-72.2991</val><val>-88.283</val></score>
    <magnitude><val>28</val><val>235</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="2-methylcitrate synthase (EC 2.3.3.5)">
     <rank><val>function</val></rank>
     <score><val>-49</val><val>-78.85</val></score>
     <magnitude><val>2</val><val>36</val></magnitude>
    </node>
    <node name="2-methylisocitrate dehydratase (EC 4.2.1.99)">
     <magnitude><val></val><val>58</val></magnitude>
     <score><val>0</val><val>-108.467</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Propionate--CoA ligase (EC 6.2.1.17)">
     <rank><val>function</val></rank>
     <score><val>-93.5714</val><val>-88.7037</val></score>
     <magnitude><val>14</val><val>83</val></magnitude>
    </node>
    <node name="2-methylcitrate dehydratase (EC 4.2.1.79)">
     <magnitude><val>3</val><val>45</val></magnitude>
     <score><val>-33.3333</val><val>-82</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Methylisocitrate lyase (EC 4.1.3.30)">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>13</val></magnitude>
     <score><val>-57.375</val><val>-43.4167</val></score>
    </node>
   </node>
   <node name="Propionate-CoA to Succinate Module">
    <rank><val>level 3</val></rank>
    <magnitude><val>30</val><val>299</val></magnitude>
    <score><val>-72.5458</val><val>-91.5269</val></score>
    <node name="Methylisocitrate lyase (EC 4.1.3.30)">
     <magnitude><val>9</val><val>13</val></magnitude>
     <score><val>-57.375</val><val>-43.4167</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Aconitate hydratase (EC 4.2.1.3)">
     <rank><val>function</val></rank>
     <score><val>-76</val><val>-103.438</val></score>
     <magnitude><val>2</val><val>64</val></magnitude>
    </node>
    <node name="2-methylisocitrate dehydratase (EC 4.2.1.99)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-108.467</val></score>
     <magnitude><val></val><val>58</val></magnitude>
    </node>
    <node name="Propionate--CoA ligase (EC 6.2.1.17)">
     <rank><val>function</val></rank>
     <score><val>-93.5714</val><val>-88.7037</val></score>
     <magnitude><val>14</val><val>83</val></magnitude>
    </node>
    <node name="2-methylcitrate dehydratase (EC 4.2.1.79)">
     <score><val>-33.3333</val><val>-82</val></score>
     <magnitude><val>3</val><val>45</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="2-methylcitrate synthase (EC 2.3.3.5)">
     <score><val>-49</val><val>-78.85</val></score>
     <magnitude><val>2</val><val>36</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Glycerate metabolism">
    <rank><val>level 3</val></rank>
    <score><val>-11.5556</val><val>-65.0628</val></score>
    <magnitude><val>9</val><val>87</val></magnitude>
    <node name="Hydroxypyruvate reductase (EC 1.1.1.81)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>29</val></magnitude>
     <score><val>0</val><val>-60.0833</val></score>
    </node>
    <node name="D-glycerate 2-kinase (EC 2.7.1.-)">
     <rank><val>function</val></rank>
     <score><val>-11</val><val>-59.1667</val></score>
     <magnitude><val>1</val><val>7</val></magnitude>
    </node>
    <node name="Glyoxylate carboligase (EC 4.1.1.47)">
     <score><val>-25</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Pyruvate kinase (EC 2.7.1.40)">
     <score><val>-11.5</val><val>-80.92</val></score>
     <magnitude><val>2</val><val>39</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycerate kinase (EC 2.7.1.31)">
     <rank><val>function</val></rank>
     <score><val>-7</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Serine--glyoxylate aminotransferase (EC 2.6.1.45)">
     <score><val>-9.5</val><val>-29</val></score>
     <magnitude><val>4</val><val>12</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Sugar utilization in Thermotogales">
   <magnitude><val>115</val><val>542</val></magnitude>
   <score><val>-40.7524</val><val>-59.815</val></score>
   <rank><val>level 2</val></rank>
   <node name="Alpha-glucosidase (EC 3.2.1.20)">
    <magnitude><val>11</val><val>14</val></magnitude>
    <score><val>-18.6</val><val>-18.75</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Mannonate dehydratase (EC 4.2.1.8)">
    <magnitude><val></val><val>2</val></magnitude>
    <score><val>0</val><val>-124</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Predicted transcriptional regulator of N-Acetylglucosamine utilization, GntR family">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-18</val></score>
   </node>
   <node name="N-acetylglucosamine-6-phosphate deacetylase (EC 3.5.1.25)">
    <score><val>-7</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Xylulose kinase (EC 2.7.1.17)">
    <score><val>-12</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Ribulose-phosphate 3-epimerase (EC 5.1.3.1)">
    <rank><val>function</val></rank>
    <score><val>-17.3333</val><val>-42</val></score>
    <magnitude><val>3</val><val>31</val></magnitude>
   </node>
   <node name="UDP-glucose 4-epimerase (EC 5.1.3.2)">
    <magnitude><val>18</val><val>60</val></magnitude>
    <score><val>-34.8235</val><val>-54.0256</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Glycerol-3-phosphate ABC transporter, ATP-binding protein UgpC (TC 3.A.1.1.3)">
    <magnitude><val></val><val>1</val></magnitude>
    <score><val>0</val><val>-50</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Phosphomannomutase (EC 5.4.2.8)">
    <magnitude><val>13</val><val>35</val></magnitude>
    <score><val>-44.7</val><val>-78.92</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Transketolase, N-terminal section (EC 2.2.1.1)">
    <magnitude><val>7</val><val>33</val></magnitude>
    <score><val>-79.1429</val><val>-76.0455</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Glucose-6-phosphate 1-dehydrogenase (EC 1.1.1.49)">
    <rank><val>function</val></rank>
    <magnitude><val>5</val></magnitude>
    <score><val>-48</val></score>
   </node>
   <node name="Xylose-responsive transcription regulator, ROK family">
    <magnitude><val>2</val></magnitude>
    <score><val>-6</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Beta-galactosidase (EC 3.2.1.23)">
    <rank><val>function</val></rank>
    <magnitude><val>8</val><val>55</val></magnitude>
    <score><val>-32.75</val><val>-36.5135</val></score>
   </node>
   <node name="Glycerol kinase (EC 2.7.1.30)">
    <magnitude><val></val><val>41</val></magnitude>
    <score><val>0</val><val>-76.1</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="2-dehydro-3-deoxyphosphogluconate aldolase (EC 4.1.2.14)">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>23</val></magnitude>
    <score><val>0</val><val>-68</val></score>
   </node>
   <node name="Galactokinase (EC 2.7.1.6)">
    <rank><val>function</val></rank>
    <score><val>-39.3333</val><val>-64.5556</val></score>
    <magnitude><val>3</val><val>29</val></magnitude>
   </node>
   <node name="Beta-glucosidase (EC 3.2.1.21)">
    <magnitude><val>7</val><val>46</val></magnitude>
    <score><val>-30.5714</val><val>-32.7419</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Mannose-6-phosphate isomerase (EC 5.3.1.8)">
    <rank><val>function</val></rank>
    <score><val>-3</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Transketolase (EC 2.2.1.1)">
    <rank><val>function</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-17</val></score>
   </node>
   <node name="Transaldolase (EC 2.2.1.2)">
    <rank><val>function</val></rank>
    <magnitude><val>6</val><val>43</val></magnitude>
    <score><val>-50.3333</val><val>-54.6333</val></score>
   </node>
   <node name="6-phosphogluconate dehydrogenase, decarboxylating (EC 1.1.1.44)">
    <rank><val>function</val></rank>
    <score><val>-45</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Maltose/maltodextrin ABC transporter, permease protein MalG">
    <rank><val>function</val></rank>
    <score><val>-71.3333</val><val>-36</val></score>
    <magnitude><val>3</val><val>3</val></magnitude>
   </node>
   <node name="Ribose ABC transport system, permease protein RbsC (TC 3.A.1.2.1)">
    <magnitude><val>1</val></magnitude>
    <score><val>-30</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Neopullulanase (EC 3.2.1.135)">
    <magnitude><val></val><val>6</val></magnitude>
    <score><val>0</val><val>-19.6667</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Glycerol-3-phosphate dehydrogenase (EC 1.1.5.3)">
    <rank><val>function</val></rank>
    <magnitude><val>2</val><val>27</val></magnitude>
    <score><val>-31</val><val>-101.833</val></score>
   </node>
   <node name="Transketolase, C-terminal section (EC 2.2.1.1)">
    <magnitude><val>12</val><val>32</val></magnitude>
    <score><val>-79.8333</val><val>-86.7273</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="D-glycerate 2-kinase (EC 2.7.1.-)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val><val>7</val></magnitude>
    <score><val>-11</val><val>-59.1667</val></score>
   </node>
   <node name="Alpha-galactosidase (EC 3.2.1.22)">
    <magnitude><val></val><val>1</val></magnitude>
    <score><val>0</val><val>-8</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Ribokinase (EC 2.7.1.15)">
    <magnitude><val>3</val><val>53</val></magnitude>
    <score><val>-4</val><val>-65.1176</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Fructokinase (EC 2.7.1.4)">
    <score><val>-101</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Glycerol-3-phosphate ABC transporter, periplasmic glycerol-3-phosphate-binding protein (TC 3.A.1.1.3)">
    <score><val>-13</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Galactose-1-phosphate uridylyltransferase (EC 2.7.7.10)">
    <score><val>0</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Monosaccharides">
   <rank><val>level 2</val></rank>
   <magnitude><val>81</val><val>351</val></magnitude>
   <score><val>-28.0356</val><val>-59.0801</val></score>
   <node name="Xylose utilization">
    <score><val>-10.4</val></score>
    <magnitude><val>5</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Putative xylulose kinase (EC 2.7.1.17)">
     <score><val>-14</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Xylose-responsive transcription regulator, ROK family">
     <rank><val>function</val></rank>
     <score><val>-6</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Xylulose kinase (EC 2.7.1.17)">
     <rank><val>function</val></rank>
     <score><val>-12</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Mannose Metabolism">
    <rank><val>level 3</val></rank>
    <score><val>-39.1476</val><val>-64.1238</val></score>
    <magnitude><val>21</val><val>72</val></magnitude>
    <node name="Hexokinase (EC 2.7.1.1)">
     <magnitude><val>5</val><val>36</val></magnitude>
     <score><val>-41.8</val><val>-50.7143</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphomannomutase (EC 5.4.2.8)">
     <score><val>-44.7</val><val>-78.92</val></score>
     <magnitude><val>13</val><val>35</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Mannose-6-phosphate isomerase (EC 5.3.1.8)">
     <rank><val>function</val></rank>
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Mannose-1-phosphate guanylyltransferase (EC 2.7.7.13 )">
     <score><val>-14.5</val><val>-29</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="D-gluconate and ketogluconates metabolism">
    <score><val>-34.8704</val><val>-52.2495</val></score>
    <magnitude><val>23</val><val>77</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Glucose 1-dehydrogenase (EC 1.1.1.47)">
     <rank><val>function</val></rank>
     <magnitude><val>12</val><val>32</val></magnitude>
     <score><val>-30.8182</val><val>-42.1538</val></score>
    </node>
    <node name="5-keto-D-gluconate 5-reductase (EC 1.1.1.69)">
     <magnitude><val>1</val></magnitude>
     <score><val>-28</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Gluconate dehydratase (EC 4.2.1.39)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>45</val></magnitude>
     <score><val>-38.4</val><val>-59.4286</val></score>
    </node>
    <node name="6-phosphogluconate dehydrogenase, decarboxylating (EC 1.1.1.44)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-45</val></score>
    </node>
    <node name="2,5-diketo-D-gluconic acid reductase B (EC 1.1.1.274)">
     <score><val>-52</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="D-galactarate, D-glucarate and D-glycerate catabolism">
    <score><val>-9</val><val>-59.1667</val></score>
    <magnitude><val>2</val><val>7</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="D-glycerate 2-kinase (EC 2.7.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-11</val><val>-59.1667</val></score>
    </node>
    <node name="Glycerate kinase (EC 2.7.1.31)">
     <rank><val>function</val></rank>
     <score><val>-7</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="D-ribose utilization">
    <score><val>-14.1061</val><val>-60.0939</val></score>
    <magnitude><val>11</val><val>77</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Ribokinase (EC 2.7.1.15)">
     <magnitude><val>3</val><val>53</val></magnitude>
     <score><val>-4</val><val>-65.1176</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribose ABC transport system, permease protein RbsC (TC 3.A.1.2.1)">
     <rank><val>function</val></rank>
     <score><val>-30</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Ribose 5-phosphate isomerase A (EC 5.3.1.6)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>24</val></magnitude>
     <score><val>-16.1667</val><val>-49</val></score>
    </node>
   </node>
   <node name="Deoxyribose and Deoxynucleoside Catabolism">
    <rank><val>level 3</val></rank>
    <score><val>-4</val><val>-66.2233</val></score>
    <magnitude><val>3</val><val>72</val></magnitude>
    <node name="Ribokinase (EC 2.7.1.15)">
     <magnitude><val>3</val><val>53</val></magnitude>
     <score><val>-4</val><val>-65.1176</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Purine nucleoside phosphorylase (EC 2.4.2.1)">
     <magnitude><val></val><val>19</val></magnitude>
     <score><val>0</val><val>-69.3077</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="D-Galacturonate and D-Glucuronate Utilization">
    <score><val>-18.6</val><val>-52.3875</val></score>
    <magnitude><val>11</val><val>40</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Uronate isomerase, family BH0493 (EC 5.3.1.12)">
     <score><val>0</val><val>-21</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Mannonate dehydratase (EC 4.2.1.8)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-124</val></score>
     <magnitude><val></val><val>2</val></magnitude>
    </node>
    <node name="Alpha-glucosidase (EC 3.2.1.20)">
     <rank><val>function</val></rank>
     <magnitude><val>11</val><val>14</val></magnitude>
     <score><val>-18.6</val><val>-18.75</val></score>
    </node>
    <node name="2-dehydro-3-deoxyphosphogluconate aldolase (EC 4.1.2.14)">
     <magnitude><val></val><val>23</val></magnitude>
     <score><val>0</val><val>-68</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="L-rhamnose utilization">
    <rank><val>level 3</val></rank>
    <magnitude><val>3</val><val>6</val></magnitude>
    <score><val>-23</val><val>-32</val></score>
    <node name="L-rhamnose mutarotase">
     <score><val>0</val><val>-32</val></score>
     <magnitude><val></val><val>6</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Predicted L-lactate dehydrogenase, Iron-sulfur cluster-binding subunit YkgF">
     <score><val>-21</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Transcriptional regulator of rhamnose utilization, LacI family">
     <score><val>-13</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="L-lactate dehydrogenase (EC 1.1.2.3)">
     <score><val>-35</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="D-Tagatose and Galactitol Utilization">
    <rank><val>level 3</val></rank>
    <score><val>-35</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="6-phosphofructokinase (EC 2.7.1.11)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-35</val></score>
    </node>
   </node>
   <node name="Fructose utilization">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-101</val></score>
    <node name="Fructokinase (EC 2.7.1.4)">
     <rank><val>function</val></rank>
     <score><val>-101</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Polysaccharides">
   <score><val>-46.3571</val><val>-60.7603</val></score>
   <magnitude><val>14</val><val>55</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Glycogen metabolism">
    <rank><val>level 3</val></rank>
    <score><val>-46.3571</val><val>-60.7603</val></score>
    <magnitude><val>14</val><val>55</val></magnitude>
    <node name="Glycogen branching enzyme, GH-57-type, archaeal (EC 2.4.1.18)">
     <score><val>0</val><val>-26</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycogen phosphorylase (EC 2.4.1.1)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>1</val></magnitude>
     <score><val>-20</val><val>-16</val></score>
    </node>
    <node name="1,4-alpha-glucan (glycogen) branching enzyme, GH-13-type (EC 2.4.1.18)">
     <rank><val>function</val></rank>
     <score><val>-55</val><val>-49</val></score>
     <magnitude><val>5</val><val>10</val></magnitude>
    </node>
    <node name="Putative glycogen debranching enzyme, archaeal type, TIGR01561">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>38</val></magnitude>
     <score><val>-6.66667</val><val>-69.0741</val></score>
    </node>
    <node name="Glycogen synthase, ADP-glucose transglucosylase (EC 2.4.1.21)">
     <score><val>-11</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycogen debranching enzyme (EC 3.2.1.-)">
     <score><val>-80.75</val><val>-37</val></score>
     <magnitude><val>4</val><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Unknown carbohydrate utilization ( cluster Yeg )">
   <score><val>-8</val></score>
   <magnitude><val>1</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Fructose-bisphosphate aldolase class I (EC 4.1.2.13)">
    <magnitude><val>1</val></magnitude>
    <score><val>-8</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Central carbohydrate metabolism">
   <score><val>-45.5286</val><val>-76.1769</val></score>
   <magnitude><val>509</val><val>2430</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Pyruvate:ferredoxin oxidoreductase">
    <rank><val>level 3</val></rank>
    <score><val>-85</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="Pyruvate-flavodoxin oxidoreductase (EC 1.2.7.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-85</val></score>
    </node>
   </node>
   <node name="Pentose phosphate pathway">
    <score><val>-53.5139</val><val>-61.2774</val></score>
    <magnitude><val>60</val><val>196</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Ribose 5-phosphate isomerase A (EC 5.3.1.6)">
     <rank><val>function</val></rank>
     <score><val>-16.1667</val><val>-49</val></score>
     <magnitude><val>7</val><val>24</val></magnitude>
    </node>
    <node name="Xylulose-5-phosphate phosphoketolase (EC 4.1.2.9)">
     <score><val>-82.4</val></score>
     <magnitude><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Fructose-6-phosphate phosphoketolase (EC 4.1.2.22)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val></magnitude>
     <score><val>-82.4</val></score>
    </node>
    <node name="Transketolase (EC 2.2.1.1)">
     <score><val>-17</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribulose-phosphate 3-epimerase (EC 5.1.3.1)">
     <magnitude><val>3</val><val>31</val></magnitude>
     <score><val>-17.3333</val><val>-42</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Transaldolase (EC 2.2.1.2)">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>43</val></magnitude>
     <score><val>-50.3333</val><val>-54.6333</val></score>
    </node>
    <node name="6-phosphogluconate dehydrogenase, decarboxylating (EC 1.1.1.44)">
     <rank><val>function</val></rank>
     <score><val>-45</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Transketolase, N-terminal section (EC 2.2.1.1)">
     <magnitude><val>7</val><val>33</val></magnitude>
     <score><val>-79.1429</val><val>-76.0455</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Transketolase, C-terminal section (EC 2.2.1.1)">
     <score><val>-79.8333</val><val>-86.7273</val></score>
     <magnitude><val>12</val><val>32</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glucose-6-phosphate 1-dehydrogenase (EC 1.1.1.49)">
     <magnitude><val>5</val></magnitude>
     <score><val>-48</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribose-phosphate pyrophosphokinase (EC 2.7.6.1)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>33</val></magnitude>
     <score><val>-12.6667</val><val>-57.5263</val></score>
    </node>
   </node>
   <node name="Dehydrogenase complexes">
    <rank><val>level 3</val></rank>
    <score><val>-44.3296</val><val>-97.1192</val></score>
    <magnitude><val>45</val><val>296</val></magnitude>
    <node name="Pyruvate dehydrogenase E1 component beta subunit (EC 1.2.4.1)">
     <score><val>-34</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrolipoamide succinyltransferase component (E2) of 2-oxoglutarate dehydrogenase complex (EC 2.3.1.61)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>1</val></magnitude>
     <score><val>-35</val><val>-20</val></score>
    </node>
    <node name="Cytosol aminopeptidase PepA (EC 3.4.11.1)">
     <rank><val>function</val></rank>
     <score><val>-9.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Branched-chain alpha-keto acid dehydrogenase, E1 component, beta subunit (EC 1.2.4.4)">
     <rank><val>function</val></rank>
     <magnitude><val>13</val><val>96</val></magnitude>
     <score><val>-58.3077</val><val>-102.927</val></score>
    </node>
    <node name="Dihydrolipoamide dehydrogenase of branched-chain alpha-keto acid dehydrogenase (EC 1.8.1.4)">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>16</val></magnitude>
     <score><val>-45.75</val><val>-46.8571</val></score>
    </node>
    <node name="Dihydrolipoamide dehydrogenase of acetoin dehydrogenase (EC 1.8.1.4)">
     <score><val>0</val><val>-116.562</val></score>
     <magnitude><val></val><val>23</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrolipoamide acetyltransferase component (E2) of acetoin dehydrogenase complex (EC 2.3.1.-)">
     <score><val>0</val><val>-72.6</val></score>
     <magnitude><val></val><val>21</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrolipoamide acyltransferase component of branched-chain alpha-keto acid dehydrogenase complex (EC 2.3.1.168)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-28</val><val>-21.4</val></score>
    </node>
    <node name="Branched-chain alpha-keto acid dehydrogenase, E1 component, alpha subunit (EC 1.2.4.4)">
     <magnitude><val>14</val><val>89</val></magnitude>
     <score><val>-41.6667</val><val>-106.417</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2-oxoglutarate dehydrogenase E1 component (EC 1.2.4.2)">
     <score><val>-35</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetoin dehydrogenase E1 component beta-subunit (EC 1.2.4.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>15</val></magnitude>
     <score><val>-11</val><val>-92.875</val></score>
    </node>
    <node name="Acetoin dehydrogenase E1 component alpha-subunit (EC 1.2.4.-)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-102.75</val></score>
     <magnitude><val></val><val>28</val></magnitude>
    </node>
   </node>
   <node name="Pyruvate metabolism I: anaplerotic reactions, PEP">
    <magnitude><val>36</val><val>189</val></magnitude>
    <score><val>-58.5444</val><val>-80.1199</val></score>
    <rank><val>level 3</val></rank>
    <node name="Phosphoenolpyruvate carboxylase, archaeal (EC 4.1.1.31)">
     <magnitude><val></val><val>37</val></magnitude>
     <score><val>0</val><val>-60.12</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoenolpyruvate synthase (EC 2.7.9.2)">
     <score><val>-61</val><val>-103.909</val></score>
     <magnitude><val>3</val><val>44</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Pyruvate kinase (EC 2.7.1.40)">
     <score><val>-11.5</val><val>-80.92</val></score>
     <magnitude><val>2</val><val>39</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="NADP-dependent malic enzyme (EC 1.1.1.40)">
     <magnitude><val>14</val><val>38</val></magnitude>
     <score><val>-65.3</val><val>-91.7308</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoenolpyruvate carboxykinase [GTP] (EC 4.1.1.32)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>15</val></magnitude>
     <score><val>-56.6</val><val>-74.3846</val></score>
    </node>
    <node name="Pyruvate,phosphate dikinase (EC 2.7.9.1)">
     <rank><val>function</val></rank>
     <magnitude><val>12</val><val>16</val></magnitude>
     <score><val>-58.7</val><val>-36.8</val></score>
    </node>
   </node>
   <node name="Pyruvate Alanine Serine Interconversions">
    <magnitude><val>6</val></magnitude>
    <score><val>-55</val></score>
    <rank><val>level 3</val></rank>
    <node name="Branched-chain amino acid aminotransferase (EC 2.6.1.42)">
     <magnitude><val>2</val></magnitude>
     <score><val>-39</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Alanine dehydrogenase (EC 1.4.1.1)">
     <magnitude><val>1</val></magnitude>
     <score><val>-96</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="L-alanine:glyoxylate aminotransferase (EC 2.6.1.44)">
     <magnitude><val>1</val></magnitude>
     <score><val>-33</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Alanine racemase (EC 5.1.1.1)">
     <magnitude><val>1</val></magnitude>
     <score><val>-16</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="L-serine dehydratase (EC 4.3.1.17)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-107</val></score>
    </node>
   </node>
   <node name="Soluble methane monooxygenase (sMMO)">
    <magnitude><val>1</val></magnitude>
    <score><val>-19</val></score>
    <rank><val>level 3</val></rank>
    <node name="Methane monooxygenase component C (EC 1.14.13.25)">
     <score><val>-19</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Glycolysis and Gluconeogenesis">
    <score><val>-39.0167</val><val>-68.9075</val></score>
    <magnitude><val>64</val><val>250</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="2,3-bisphosphoglycerate-independent phosphoglycerate mutase (EC 5.4.2.1)">
     <magnitude><val>1</val></magnitude>
     <score><val>-31</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Fructose-1,6-bisphosphatase, GlpX type (EC 3.1.3.11)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-127</val></score>
    </node>
    <node name="Pyruvate,phosphate dikinase (EC 2.7.9.1)">
     <score><val>-58.7</val><val>-36.8</val></score>
     <magnitude><val>12</val><val>16</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Pyruvate kinase (EC 2.7.1.40)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>39</val></magnitude>
     <score><val>-11.5</val><val>-80.92</val></score>
    </node>
    <node name="Fructose-bisphosphate aldolase class I (EC 4.1.2.13)">
     <rank><val>function</val></rank>
     <score><val>-8</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Enolase (EC 4.2.1.11)">
     <rank><val>function</val></rank>
     <magnitude><val>16</val><val>38</val></magnitude>
     <score><val>-46.875</val><val>-79.2917</val></score>
    </node>
    <node name="Phosphoglycerate mutase (EC 5.4.2.1)">
     <magnitude><val></val><val>24</val></magnitude>
     <score><val>0</val><val>-46.2308</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoenolpyruvate synthase (EC 2.7.9.2)">
     <rank><val>function</val></rank>
     <score><val>-61</val><val>-103.909</val></score>
     <magnitude><val>3</val><val>44</val></magnitude>
    </node>
    <node name="NAD-dependent glyceraldehyde-3-phosphate dehydrogenase (EC 1.2.1.12)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-11</val></score>
    </node>
    <node name="Glucose-6-phosphate isomerase, archaeal II (EC 5.3.1.9)">
     <rank><val>function</val></rank>
     <score><val>-24.5</val><val>-50.4286</val></score>
     <magnitude><val>4</val><val>17</val></magnitude>
    </node>
    <node name="Hexokinase (EC 2.7.1.1)">
     <magnitude><val>5</val><val>36</val></magnitude>
     <score><val>-41.8</val><val>-50.7143</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoglycerate kinase (EC 2.7.2.3)">
     <magnitude><val>7</val><val>13</val></magnitude>
     <score><val>-25.7143</val><val>-63.5714</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Fructose-1,6-bisphosphatase, type I (EC 3.1.3.11)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val></magnitude>
     <score><val>-8</val></score>
    </node>
    <node name="Glucose-6-phosphate isomerase (EC 5.3.1.9)">
     <rank><val>function</val></rank>
     <score><val>-19.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="6-phosphofructokinase (EC 2.7.1.11)">
     <score><val>-35</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Triosephosphate isomerase (EC 5.3.1.1)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>23</val></magnitude>
     <score><val>-16.6667</val><val>-55.5714</val></score>
    </node>
   </node>
   <node name="Glycolate, glyoxylate interconversions">
    <magnitude><val>4</val><val>105</val></magnitude>
    <score><val>-8.75</val><val>-56.0907</val></score>
    <rank><val>level 3</val></rank>
    <node name="Glyoxylate reductase (EC 1.1.1.79)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-60.0833</val></score>
     <magnitude><val></val><val>29</val></magnitude>
    </node>
    <node name="Phosphoglycolate phosphatase, archaeal type (EC 3.1.3.18)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>17</val></magnitude>
     <score><val>-7</val><val>-38.5455</val></score>
    </node>
    <node name="Glyoxylate reductase (EC 1.1.1.26)">
     <magnitude><val></val><val>29</val></magnitude>
     <score><val>0</val><val>-60.0833</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Hydroxypyruvate reductase (EC 1.1.1.81)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>29</val></magnitude>
     <score><val>0</val><val>-60.0833</val></score>
    </node>
    <node name="Glycolate dehydrogenase (EC 1.1.99.14), subunit GlcD">
     <rank><val>function</val></rank>
     <score><val>-10.5</val><val>-7</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
   </node>
   <node name="Glycolysis and Gluconeogenesis, including Archaeal enzymes">
    <rank><val>level 3</val></rank>
    <magnitude><val>87</val><val>348</val></magnitude>
    <score><val>-42.1253</val><val>-68.5413</val></score>
    <node name="Fructose-1,6-bisphosphatase, type I (EC 3.1.3.11)">
     <magnitude><val>4</val></magnitude>
     <score><val>-8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glucose-6-phosphate isomerase (EC 5.3.1.9)">
     <magnitude><val>2</val></magnitude>
     <score><val>-19.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoglycerate kinase (EC 2.7.2.3)">
     <score><val>-25.7143</val><val>-63.5714</val></score>
     <magnitude><val>7</val><val>13</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Hexokinase (EC 2.7.1.1)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>36</val></magnitude>
     <score><val>-41.8</val><val>-50.7143</val></score>
    </node>
    <node name="Glucose-6-phosphate isomerase, archaeal II (EC 5.3.1.9)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>17</val></magnitude>
     <score><val>-24.5</val><val>-50.4286</val></score>
    </node>
    <node name="Triosephosphate isomerase (EC 5.3.1.1)">
     <magnitude><val>4</val><val>23</val></magnitude>
     <score><val>-16.6667</val><val>-55.5714</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="6-phosphofructokinase (EC 2.7.1.11)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-35</val></score>
    </node>
    <node name="Fructose-1,6-bisphosphatase, type V, archaeal (EC 3.1.3.11)">
     <rank><val>function</val></rank>
     <score><val>-70.1</val><val>-76.1905</val></score>
     <magnitude><val>10</val><val>38</val></magnitude>
    </node>
    <node name="2,3-bisphosphoglycerate-independent phosphoglycerate mutase (EC 5.4.2.1)">
     <rank><val>function</val></rank>
     <score><val>-31</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Pyruvate,phosphate dikinase (EC 2.7.9.1)">
     <magnitude><val>12</val><val>16</val></magnitude>
     <score><val>-58.7</val><val>-36.8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Fructose-1,6-bisphosphatase, GlpX type (EC 3.1.3.11)">
     <rank><val>function</val></rank>
     <score><val>-127</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Phosphoenolpyruvate synthase (EC 2.7.9.2)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>44</val></magnitude>
     <score><val>-61</val><val>-103.909</val></score>
    </node>
    <node name="Phosphoglycerate mutase (EC 5.4.2.1)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-46.2308</val></score>
     <magnitude><val></val><val>24</val></magnitude>
    </node>
    <node name="Enolase (EC 4.2.1.11)">
     <rank><val>function</val></rank>
     <score><val>-46.875</val><val>-79.2917</val></score>
     <magnitude><val>16</val><val>38</val></magnitude>
    </node>
    <node name="Pyruvate kinase (EC 2.7.1.40)">
     <score><val>-11.5</val><val>-80.92</val></score>
     <magnitude><val>2</val><val>39</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="NAD(P)-dependent glyceraldehyde 3-phosphate dehydrogenase archaeal (EC 1.2.1.59)">
     <magnitude><val>5</val><val>29</val></magnitude>
     <score><val>-42.5</val><val>-57.9444</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2,3-bisphosphoglycerate-independent phosphoglycerate mutase, archaeal type (EC 5.4.2.1)">
     <rank><val>function</val></rank>
     <magnitude><val>10</val><val>31</val></magnitude>
     <score><val>-27.3333</val><val>-66.125</val></score>
    </node>
   </node>
   <node name="TCA Cycle">
    <rank><val>level 3</val></rank>
    <magnitude><val>62</val><val>320</val></magnitude>
    <score><val>-64.3077</val><val>-88.3113</val></score>
    <node name="Succinate dehydrogenase flavoprotein subunit (EC 1.3.99.1)">
     <score><val>-65.7692</val><val>-125.667</val></score>
     <magnitude><val>13</val><val>53</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Succinyl-CoA ligase [ADP-forming] beta chain (EC 6.2.1.5)">
     <score><val>-37</val><val>-80</val></score>
     <magnitude><val>2</val><val>27</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Succinate dehydrogenase iron-sulfur protein (EC 1.3.99.1)">
     <score><val>-30.6667</val><val>-94.9231</val></score>
     <magnitude><val>3</val><val>18</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Malate dehydrogenase (EC 1.1.1.37)">
     <score><val>-73.2</val><val>-62.6071</val></score>
     <magnitude><val>8</val><val>51</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrolipoamide succinyltransferase component (E2) of 2-oxoglutarate dehydrogenase complex (EC 2.3.1.61)">
     <score><val>-35</val><val>-20</val></score>
     <magnitude><val>1</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Isocitrate dehydrogenase [NADP] (EC 1.1.1.42)">
     <rank><val>function</val></rank>
     <score><val>-79.7778</val><val>-78.8929</val></score>
     <magnitude><val>12</val><val>46</val></magnitude>
    </node>
    <node name="Dihydrolipoamide dehydrogenase (EC 1.8.1.4)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-44.7</val></score>
     <magnitude><val></val><val>10</val></magnitude>
    </node>
    <node name="Aconitate hydratase (EC 4.2.1.3)">
     <rank><val>function</val></rank>
     <score><val>-76</val><val>-103.438</val></score>
     <magnitude><val>2</val><val>64</val></magnitude>
    </node>
    <node name="2-oxoglutarate dehydrogenase E1 component (EC 1.2.4.2)">
     <magnitude><val>1</val></magnitude>
     <score><val>-35</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Fumarate hydratase class II (EC 4.2.1.2)">
     <rank><val>function</val></rank>
     <score><val>-54.5714</val><val>-58</val></score>
     <magnitude><val>9</val><val>6</val></magnitude>
    </node>
    <node name="Citrate synthase (si) (EC 2.3.3.1)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>37</val></magnitude>
     <score><val>-20.75</val><val>-78.8966</val></score>
    </node>
    <node name="Succinyl-CoA ligase [ADP-forming] alpha chain (EC 6.2.1.5)">
     <magnitude><val>7</val><val>7</val></magnitude>
     <score><val>-89.5714</val><val>-79.2</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Entner-Doudoroff Pathway">
    <score><val>-34.7336</val><val>-62.9596</val></score>
    <magnitude><val>70</val><val>330</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="NAD-dependent glyceraldehyde-3-phosphate dehydrogenase (EC 1.2.1.12)">
     <magnitude><val>1</val></magnitude>
     <score><val>-11</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Hexokinase (EC 2.7.1.1)">
     <rank><val>function</val></rank>
     <score><val>-41.8</val><val>-50.7143</val></score>
     <magnitude><val>5</val><val>36</val></magnitude>
    </node>
    <node name="Gluconate dehydratase (EC 4.2.1.39)">
     <score><val>-38.4</val><val>-59.4286</val></score>
     <magnitude><val>8</val><val>45</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoglycerate kinase (EC 2.7.2.3)">
     <score><val>-25.7143</val><val>-63.5714</val></score>
     <magnitude><val>7</val><val>13</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="2-dehydro-3-deoxygluconate aldolase (EC 4.1.2-)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>23</val></magnitude>
     <score><val>0</val><val>-68</val></score>
    </node>
    <node name="Gluconolactonase (EC 3.1.1.17)">
     <score><val>-13</val><val>-64.7778</val></score>
     <magnitude><val>2</val><val>19</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="2,3-bisphosphoglycerate-independent phosphoglycerate mutase (EC 5.4.2.1)">
     <score><val>-31</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glucose-6-phosphate 1-dehydrogenase (EC 1.1.1.49)">
     <score><val>-48</val></score>
     <magnitude><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="D-glycerate 2-kinase (EC 2.7.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-11</val><val>-59.1667</val></score>
    </node>
    <node name="Glucose 1-dehydrogenase (EC 1.1.1.47)">
     <rank><val>function</val></rank>
     <magnitude><val>12</val><val>32</val></magnitude>
     <score><val>-30.8182</val><val>-42.1538</val></score>
    </node>
    <node name="2,3-bisphosphoglycerate-independent phosphoglycerate mutase, archaeal type (EC 5.4.2.1)">
     <score><val>-27.3333</val><val>-66.125</val></score>
     <magnitude><val>10</val><val>31</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Pyruvate kinase (EC 2.7.1.40)">
     <magnitude><val>2</val><val>39</val></magnitude>
     <score><val>-11.5</val><val>-80.92</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Enolase (EC 4.2.1.11)">
     <rank><val>function</val></rank>
     <score><val>-46.875</val><val>-79.2917</val></score>
     <magnitude><val>16</val><val>38</val></magnitude>
    </node>
    <node name="Phosphoglycerate mutase (EC 5.4.2.1)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>24</val></magnitude>
     <score><val>0</val><val>-46.2308</val></score>
    </node>
    <node name="2-dehydro-3-deoxyphosphogluconate aldolase (EC 4.1.2.14)">
     <score><val>0</val><val>-68</val></score>
     <magnitude><val></val><val>23</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Glyoxylate bypass">
    <score><val>-56.2875</val><val>-83.7643</val></score>
    <magnitude><val>16</val><val>152</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Malate synthase G (EC 2.3.3.9)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-39</val></score>
    </node>
    <node name="Aconitate hydratase (EC 4.2.1.3)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>64</val></magnitude>
     <score><val>-76</val><val>-103.438</val></score>
    </node>
    <node name="Isocitrate lyase (EC 4.1.3.1)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-41</val></score>
    </node>
    <node name="Malate dehydrogenase (EC 1.1.1.37)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>51</val></magnitude>
     <score><val>-73.2</val><val>-62.6071</val></score>
    </node>
    <node name="Citrate synthase (si) (EC 2.3.3.1)">
     <score><val>-20.75</val><val>-78.8966</val></score>
     <magnitude><val>4</val><val>37</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="HPr kinase and hprK operon in Gram-positive organisms">
    <rank><val>level 3</val></rank>
    <score><val>-12</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="Prolipoprotein diacylglyceryl transferase (EC 2.4.99.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-12</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Methylglyoxal Metabolism">
    <rank><val>level 3</val></rank>
    <score><val>-42.5</val><val>-100.689</val></score>
    <magnitude><val>10</val><val>43</val></magnitude>
    <node name="Hydroxyacylglutathione hydrolase (EC 3.1.2.6)">
     <magnitude><val>1</val></magnitude>
     <score><val>-53</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Aldehyde dehydrogenase B (EC 1.2.1.22)">
     <magnitude><val>7</val><val>8</val></magnitude>
     <score><val>-44.5714</val><val>-55.2857</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Aldehyde dehydrogenase (EC 1.2.1.3)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>35</val></magnitude>
     <score><val>-30</val><val>-111.067</val></score>
    </node>
   </node>
   <node name="Pyruvate metabolism II: acetyl-CoA, acetogenesis from pyruvate">
    <magnitude><val>44</val><val>201</val></magnitude>
    <score><val>-32.6773</val><val>-80.3117</val></score>
    <rank><val>level 3</val></rank>
    <node name="Aldehyde dehydrogenase (EC 1.2.1.3)">
     <magnitude><val>2</val><val>35</val></magnitude>
     <score><val>-30</val><val>-111.067</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acylphosphate phosphohydrolase (EC 3.6.1.7), putative">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Pyruvate dehydrogenase E1 component beta subunit (EC 1.2.4.1)">
     <magnitude><val>2</val></magnitude>
     <score><val>-34</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetyl-coenzyme A synthetase (EC 6.2.1.1)">
     <magnitude><val>16</val><val>73</val></magnitude>
     <score><val>-37.5</val><val>-67.283</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetyl-CoA synthetase (ADP-forming) alpha and beta chains, putative">
     <magnitude><val>11</val><val>70</val></magnitude>
     <score><val>-25.6</val><val>-85.9792</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Lactate 2-monooxygenase (EC 1.13.12.4)">
     <rank><val>function</val></rank>
     <score><val>-37.2</val><val>-59.1111</val></score>
     <magnitude><val>5</val><val>10</val></magnitude>
    </node>
    <node name="Protein acetyltransferase">
     <score><val>-12</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetate permease ActP (cation/acetate symporter)">
     <magnitude><val>1</val></magnitude>
     <score><val>-25</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Pyruvate oxidase [ubiquinone, cytochrome] (EC 1.2.2.2)">
     <magnitude><val>6</val><val>12</val></magnitude>
     <score><val>-34.2</val><val>-60.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Ethylmalonyl-CoA pathway of C2 assimilation">
    <rank><val>level 3</val></rank>
    <score><val>-18</val></score>
    <magnitude><val>2</val></magnitude>
    <node name="L-malyl-CoA/beta-methylmalyl-CoA lyase (EC 4.1.3.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-18</val></score>
    </node>
    <node name="L-malyl-CoA/beta-methylmalyl-CoA lyase (EC 4.1.3.-), actinobacterial type">
     <magnitude><val>1</val></magnitude>
     <score><val>-18</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Fermentation">
   <score><val>-43.5163</val><val>-76.9166</val></score>
   <magnitude><val>258</val><val>1176</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Acetyl-CoA fermentation to Butyrate">
    <rank><val>level 3</val></rank>
    <score><val>-42.4289</val><val>-77.7386</val></score>
    <magnitude><val>92</val><val>463</val></magnitude>
    <node name="Butyryl-CoA dehydrogenase (EC 1.3.99.2)">
     <magnitude><val>21</val><val>25</val></magnitude>
     <score><val>-54.6</val><val>-56.4444</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetyl-CoA acetyltransferase (EC 2.3.1.9)">
     <rank><val>function</val></rank>
     <magnitude><val>28</val><val>112</val></magnitude>
     <score><val>-49.3462</val><val>-76.8312</val></score>
    </node>
    <node name="3-hydroxybutyryl-CoA dehydratase (EC 4.2.1.55)">
     <magnitude><val>1</val></magnitude>
     <score><val>-17</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetyl-CoA:acetoacetyl-CoA transferase, alpha subunit (EC 2.8.3.8)">
     <score><val>0</val><val>-40</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="3-hydroxybutyryl-CoA dehydrogenase (EC 1.1.1.157)">
     <magnitude><val>4</val><val>45</val></magnitude>
     <score><val>-35.25</val><val>-82.4286</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ferredoxin-like protein">
     <score><val>-38</val><val>-26.8</val></score>
     <magnitude><val>1</val><val>7</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Enoyl-CoA hydratase (EC 4.2.1.17)">
     <magnitude><val>4</val><val>6</val></magnitude>
     <score><val>-23.25</val><val>-62.75</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Probable electron transfer flavoprotein-quinone oxidoreductase FixC (EC 1.5.5.-)">
     <magnitude><val>7</val><val>36</val></magnitude>
     <score><val>-27.3333</val><val>-82.6296</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Electron transfer flavoprotein, beta subunit">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>36</val></magnitude>
     <score><val>0</val><val>-80.9524</val></score>
    </node>
    <node name="3-hydroxyacyl-CoA dehydrogenase (EC 1.1.1.35)">
     <score><val>-36.6667</val><val>-79.2676</val></score>
     <magnitude><val>13</val><val>110</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Electron transfer flavoprotein, alpha subunit">
     <score><val>-32.1667</val><val>-83</val></score>
     <magnitude><val>13</val><val>85</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Fermentations: Lactate">
    <rank><val>level 3</val></rank>
    <magnitude><val>11</val></magnitude>
    <score><val>-75.7273</val></score>
    <node name="Xylulose-5-phosphate phosphoketolase (EC 4.1.2.9)">
     <score><val>-82.4</val></score>
     <magnitude><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Fructose-6-phosphate phosphoketolase (EC 4.1.2.22)">
     <rank><val>function</val></rank>
     <score><val>-82.4</val></score>
     <magnitude><val>5</val></magnitude>
    </node>
    <node name="D-lactate dehydrogenase (EC 1.1.1.28)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-9</val></score>
    </node>
   </node>
   <node name="Fermentations: Mixed acid">
    <score><val>-18.5294</val><val>-59.2422</val></score>
    <magnitude><val>17</val><val>94</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Alcohol dehydrogenase (EC 1.1.1.1)">
     <rank><val>function</val></rank>
     <score><val>-21.5</val><val>-68.2</val></score>
     <magnitude><val>12</val><val>34</val></magnitude>
    </node>
    <node name="Phosphoenolpyruvate carboxylase, archaeal (EC 4.1.1.31)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-60.12</val></score>
     <magnitude><val></val><val>37</val></magnitude>
    </node>
    <node name="D-lactate dehydrogenase (EC 1.1.1.28)">
     <score><val>-9</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Sugar/maltose fermentation stimulation protein homolog">
     <magnitude><val>4</val><val>23</val></magnitude>
     <score><val>-12</val><val>-44.5882</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Butanol Biosynthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>57</val><val>188</val></magnitude>
    <score><val>-48.4613</val><val>-75.0106</val></score>
    <node name="Acetyl-CoA acetyltransferase (EC 2.3.1.9)">
     <rank><val>function</val></rank>
     <magnitude><val>28</val><val>112</val></magnitude>
     <score><val>-49.3462</val><val>-76.8312</val></score>
    </node>
    <node name="Butyryl-CoA dehydrogenase (EC 1.3.99.2)">
     <rank><val>function</val></rank>
     <score><val>-54.6</val><val>-56.4444</val></score>
     <magnitude><val>21</val><val>25</val></magnitude>
    </node>
    <node name="Enoyl-CoA hydratase (EC 4.2.1.17)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>6</val></magnitude>
     <score><val>-23.25</val><val>-62.75</val></score>
    </node>
    <node name="3-hydroxybutyryl-CoA dehydrogenase (EC 1.1.1.157)">
     <score><val>-35.25</val><val>-82.4286</val></score>
     <magnitude><val>4</val><val>45</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Acetone Butanol Ethanol Synthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>79</val><val>338</val></magnitude>
    <score><val>-42.5628</val><val>-77.0816</val></score>
    <node name="3-hydroxybutyryl-CoA dehydrogenase (EC 1.1.1.157)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>45</val></magnitude>
     <score><val>-35.25</val><val>-82.4286</val></score>
    </node>
    <node name="Alcohol dehydrogenase (EC 1.1.1.1)">
     <magnitude><val>12</val><val>34</val></magnitude>
     <score><val>-21.5</val><val>-68.2</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="3-hydroxybutyryl-CoA dehydratase (EC 4.2.1.55)">
     <magnitude><val>1</val></magnitude>
     <score><val>-17</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetyl-CoA:acetoacetyl-CoA transferase, alpha subunit (EC 2.8.3.8)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-40</val></score>
     <magnitude><val></val><val>1</val></magnitude>
    </node>
    <node name="Acetyl-CoA acetyltransferase (EC 2.3.1.9)">
     <rank><val>function</val></rank>
     <score><val>-49.3462</val><val>-76.8312</val></score>
     <magnitude><val>28</val><val>112</val></magnitude>
    </node>
    <node name="Butyryl-CoA dehydrogenase (EC 1.3.99.2)">
     <rank><val>function</val></rank>
     <magnitude><val>21</val><val>25</val></magnitude>
     <score><val>-54.6</val><val>-56.4444</val></score>
    </node>
    <node name="Electron transfer flavoprotein, alpha subunit">
     <magnitude><val>13</val><val>85</val></magnitude>
     <score><val>-32.1667</val><val>-83</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Electron transfer flavoprotein, beta subunit">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-80.9524</val></score>
     <magnitude><val></val><val>36</val></magnitude>
    </node>
   </node>
   <node name="Acetoin, butanediol metabolism">
    <magnitude><val>2</val><val>93</val></magnitude>
    <score><val>-25.5</val><val>-93.9425</val></score>
    <rank><val>level 3</val></rank>
    <node name="Acetolactate synthase small subunit (EC 2.2.1.6)">
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-43.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydrolipoamide dehydrogenase of acetoin dehydrogenase (EC 1.8.1.4)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-116.562</val></score>
     <magnitude><val></val><val>23</val></magnitude>
    </node>
    <node name="Acetolactate synthase small subunit (EC 2.2.1.6), predicted">
     <score><val>0</val><val>-43.5</val></score>
     <magnitude><val></val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetolactate synthase large subunit (EC 2.2.1.6)">
     <rank><val>function</val></rank>
     <score><val>-40</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Acetoin dehydrogenase E1 component beta-subunit (EC 1.2.4.-)">
     <magnitude><val>1</val><val>15</val></magnitude>
     <score><val>-11</val><val>-92.875</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetolactate synthase small subunit (EC 2.2.1.6), predicted, Archaeal type">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-43.5</val></score>
     <magnitude><val></val><val>2</val></magnitude>
    </node>
    <node name="Acetoin dehydrogenase E1 component alpha-subunit (EC 1.2.4.-)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>28</val></magnitude>
     <score><val>0</val><val>-102.75</val></score>
    </node>
    <node name="Dihydrolipoamide acetyltransferase component (E2) of acetoin dehydrogenase complex (EC 2.3.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>21</val></magnitude>
     <score><val>0</val><val>-72.6</val></score>
    </node>
   </node>
  </node>
  <node name="Unknown carbohydrate utilization ( cluster Ydj )">
   <rank><val>level 2</val></rank>
   <magnitude><val>2</val></magnitude>
   <score><val>-26</val></score>
   <node name="Peptide methionine sulfoxide reductase MsrB (EC 1.8.4.12)">
    <score><val>-26</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Sugar alcohols">
   <magnitude><val>18</val><val>142</val></magnitude>
   <score><val>-27.7222</val><val>-64.2063</val></score>
   <rank><val>level 2</val></rank>
   <node name="Glycerol fermenation to 1,3-propanediol">
    <rank><val>level 3</val></rank>
    <magnitude><val>4</val><val>21</val></magnitude>
    <score><val>-25</val><val>-43</val></score>
    <node name="Cob(I)alamin adenosyltransferase PduO (EC 2.5.1.17)">
     <score><val>-25</val><val>-43</val></score>
     <magnitude><val>4</val><val>21</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Mannitol Utilization">
    <magnitude><val>1</val></magnitude>
    <score><val>-101</val></score>
    <rank><val>level 3</val></rank>
    <node name="Fructokinase (EC 2.7.1.4)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-101</val></score>
    </node>
   </node>
   <node name="Erythritol utilization">
    <score><val>-53</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Predicted erythritol ABC transporter 1, ATP-binding component 2">
     <rank><val>function</val></rank>
     <score><val>-53</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Ethanolamine utilization">
    <score><val>-15</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Ethanolamine utilization polyhedral-body-like protein EutN">
     <rank><val>function</val></rank>
     <score><val>-15</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Propanediol utilization">
    <score><val>-25</val><val>-43</val></score>
    <magnitude><val>4</val><val>21</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Cob(I)alamin adenosyltransferase PduO (EC 2.5.1.17)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>21</val></magnitude>
     <score><val>-25</val><val>-43</val></score>
    </node>
   </node>
   <node name="Glycerol and Glycerol-3-phosphate Uptake and Utilization">
    <magnitude><val>6</val><val>99</val></magnitude>
    <score><val>-19.6667</val><val>-73.6495</val></score>
    <rank><val>level 3</val></rank>
    <node name="Glycerol-3-phosphate ABC transporter, periplasmic glycerol-3-phosphate-binding protein (TC 3.A.1.1.3)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-13</val></score>
    </node>
    <node name="Glycerol-3-phosphate ABC transporter, ATP-binding protein UgpC (TC 3.A.1.1.3)">
     <score><val>0</val><val>-50</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Aerobic glycerol-3-phosphate dehydrogenase (EC 1.1.5.3)">
     <rank><val>function</val></rank>
     <score><val>-17</val><val>-7</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
    </node>
    <node name="Glycerol-3-phosphate dehydrogenase (EC 1.1.5.3)">
     <magnitude><val>2</val><val>27</val></magnitude>
     <score><val>-31</val><val>-101.833</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycerol kinase (EC 2.7.1.30)">
     <magnitude><val></val><val>41</val></magnitude>
     <score><val>0</val><val>-76.1</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycerophosphoryl diester phosphodiesterase (EC 3.1.4.46)">
     <score><val>-9</val><val>-47.0588</val></score>
     <magnitude><val>1</val><val>29</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Inositol catabolism">
    <rank><val>level 3</val></rank>
    <score><val>-12</val><val>-20</val></score>
    <magnitude><val>1</val><val>1</val></magnitude>
    <node name="Myo-inositol 2-dehydrogenase (EC 1.1.1.18)">
     <score><val>-12</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Methylmalonate-semialdehyde dehydrogenase [inositol] (EC 1.2.1.27)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-20</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Di- and oligosaccharides">
   <rank><val>level 2</val></rank>
   <score><val>-36.9305</val><val>-51.972</val></score>
   <magnitude><val>159</val><val>614</val></magnitude>
   <node name="Lactose utilization">
    <score><val>-32.75</val><val>-36.5135</val></score>
    <magnitude><val>8</val><val>55</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Beta-galactosidase (EC 3.2.1.23)">
     <score><val>-32.75</val><val>-36.5135</val></score>
     <magnitude><val>8</val><val>55</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Trehalose Biosynthesis">
    <rank><val>level 3</val></rank>
    <score><val>-43.3665</val><val>-61.2349</val></score>
    <magnitude><val>62</val><val>177</val></magnitude>
    <node name="Trehalose-6-phosphate phosphatase (EC 3.1.3.12)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>15</val></magnitude>
     <score><val>0</val><val>-46.4</val></score>
    </node>
    <node name="Malto-oligosyltrehalose synthase (EC 5.4.99.15)">
     <rank><val>function</val></rank>
     <score><val>-11</val><val>-8.5</val></score>
     <magnitude><val>4</val><val>3</val></magnitude>
    </node>
    <node name="Trehalose phosphorylase (EC 2.4.1.64)">
     <magnitude><val>2</val></magnitude>
     <score><val>-33</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Malto-oligosyltrehalose trehalohydrolase (EC 3.2.1.141)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>3</val></magnitude>
     <score><val>-50.5</val><val>-34.6667</val></score>
    </node>
    <node name="Glycogen debranching enzyme (EC 3.2.1.-)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>5</val></magnitude>
     <score><val>-80.75</val><val>-37</val></score>
    </node>
    <node name="Alpha-amylase (EC 3.2.1.1)">
     <rank><val>function</val></rank>
     <score><val>-36</val><val>-63.3846</val></score>
     <magnitude><val>5</val><val>20</val></magnitude>
    </node>
    <node name="1,4-alpha-glucan (glycogen) branching enzyme, GH-13-type (EC 2.4.1.18)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>10</val></magnitude>
     <score><val>-55</val><val>-49</val></score>
    </node>
    <node name="Trehalose synthase (EC 5.4.99.16)">
     <rank><val>function</val></rank>
     <magnitude><val>14</val><val>10</val></magnitude>
     <score><val>-64.4286</val><val>-47</val></score>
    </node>
    <node name="Glucoamylase (EC 3.2.1.3)">
     <score><val>-30.4</val><val>-74.0429</val></score>
     <magnitude><val>12</val><val>96</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Alpha,alpha-trehalose-phosphate synthase [UDP-forming] (EC 2.4.1.15)">
     <rank><val>function</val></rank>
     <magnitude><val>14</val><val>15</val></magnitude>
     <score><val>-30.9231</val><val>-32.8182</val></score>
    </node>
   </node>
   <node name="Trehalose Uptake and Utilization">
    <rank><val>level 3</val></rank>
    <score><val>-33</val></score>
    <magnitude><val>2</val></magnitude>
    <node name="Trehalose phosphorylase (EC 2.4.1.64)">
     <score><val>-33</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Sucrose utilization Shewanella">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-101</val></score>
    <node name="Fructokinase (EC 2.7.1.4)">
     <magnitude><val>1</val></magnitude>
     <score><val>-101</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Beta-Glucoside Metabolism">
    <rank><val>level 3</val></rank>
    <score><val>-30.5714</val><val>-32.7419</val></score>
    <magnitude><val>7</val><val>46</val></magnitude>
    <node name="Beta-glucosidase (EC 3.2.1.21)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>46</val></magnitude>
     <score><val>-30.5714</val><val>-32.7419</val></score>
    </node>
   </node>
   <node name="Lactose and Galactose Uptake and Utilization">
    <rank><val>level 3</val></rank>
    <score><val>-33.5608</val><val>-49.1717</val></score>
    <magnitude><val>30</val><val>145</val></magnitude>
    <node name="UDP-glucose 4-epimerase (EC 5.1.3.2)">
     <score><val>-34.8235</val><val>-54.0256</val></score>
     <magnitude><val>18</val><val>60</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Galactokinase (EC 2.7.1.6)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>29</val></magnitude>
     <score><val>-39.3333</val><val>-64.5556</val></score>
    </node>
    <node name="Galactose-1-phosphate uridylyltransferase (EC 2.7.7.10)">
     <magnitude><val>1</val></magnitude>
     <score><val>0</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Beta-galactosidase (EC 3.2.1.23)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>55</val></magnitude>
     <score><val>-32.75</val><val>-36.5135</val></score>
    </node>
    <node name="Alpha-galactosidase (EC 3.2.1.22)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-8</val></score>
     <magnitude><val></val><val>1</val></magnitude>
    </node>
   </node>
   <node name="Fructooligosaccharides(FOS) and Raffinose Utilization">
    <score><val>-30.5</val><val>-32.2155</val></score>
    <magnitude><val>8</val><val>47</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Alpha-galactosidase (EC 3.2.1.22)">
     <score><val>0</val><val>-8</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Beta-glucosidase (EC 3.2.1.21)">
     <score><val>-30.5714</val><val>-32.7419</val></score>
     <magnitude><val>7</val><val>46</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Multiple sugar ABC transporter, substrate-binding protein">
     <rank><val>function</val></rank>
     <score><val>-30</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Melibiose Utilization">
    <rank><val>level 3</val></rank>
    <magnitude><val></val><val>1</val></magnitude>
    <score><val>0</val><val>-8</val></score>
    <node name="Alpha-galactosidase (EC 3.2.1.22)">
     <score><val>0</val><val>-8</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Sucrose utilization">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-101</val></score>
    <node name="Fructokinase (EC 2.7.1.4)">
     <magnitude><val>1</val></magnitude>
     <score><val>-101</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Maltose and Maltodextrin Utilization">
    <rank><val>level 3</val></rank>
    <magnitude><val>40</val><val>143</val></magnitude>
    <score><val>-29.71</val><val>-62.2784</val></score>
    <node name="Maltose/maltodextrin transport ATP-binding protein MalK (EC 3.6.3.19)">
     <magnitude><val>1</val></magnitude>
     <score><val>-49</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Malto-oligosyltrehalose synthase (EC 5.4.99.15)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>3</val></magnitude>
     <score><val>-11</val><val>-8.5</val></score>
    </node>
    <node name="Maltose/maltodextrin ABC transporter, permease protein MalG">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>3</val></magnitude>
     <score><val>-71.3333</val><val>-36</val></score>
    </node>
    <node name="Maltose O-acetyltransferase (EC 2.3.1.79)">
     <score><val>-46</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Neopullulanase (EC 3.2.1.135)">
     <score><val>0</val><val>-19.6667</val></score>
     <magnitude><val></val><val>6</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Alpha-glucosidase (EC 3.2.1.20)">
     <magnitude><val>11</val><val>14</val></magnitude>
     <score><val>-18.6</val><val>-18.75</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glucoamylase (EC 3.2.1.3)">
     <score><val>-30.4</val><val>-74.0429</val></score>
     <magnitude><val>12</val><val>96</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Maltose phosphorylase (EC 2.4.1.8)">
     <score><val>-33</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycogen phosphorylase (EC 2.4.1.1)">
     <rank><val>function</val></rank>
     <score><val>-20</val><val>-16</val></score>
     <magnitude><val>1</val><val>1</val></magnitude>
    </node>
    <node name="Alpha-amylase (EC 3.2.1.1)">
     <rank><val>function</val></rank>
     <score><val>-36</val><val>-63.3846</val></score>
     <magnitude><val>5</val><val>20</val></magnitude>
    </node>
   </node>
  </node>
 </node>
 <node name="Iron acquisition and metabolism">
  <score><val>-24.1667</val><val>-57.1925</val></score>
  <magnitude><val>12</val><val>127</val></magnitude>
  <rank><val>level 1</val></rank>
  <node name="Campylobacter Iron Metabolism">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val><val>1</val></magnitude>
   <score><val>-19</val><val>-14</val></score>
   <node name="Ferrous iron transport protein B">
    <rank><val>function</val></rank>
    <score><val>-19</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Ferric iron ABC transporter, permease protein">
    <rank><val>function</val></rank>
    <score><val>0</val><val>-14</val></score>
    <magnitude><val></val><val>1</val></magnitude>
   </node>
  </node>
  <node name="Encapsulating protein for DyP-type peroxidase and ferritin-like protein oligomers">
   <magnitude><val>1</val><val>20</val></magnitude>
   <score><val>-51</val><val>-13.35</val></score>
   <rank><val>level 2</val></rank>
   <node name="Predicted iron-dependent peroxidase, Dyp-type family">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-51</val></score>
   </node>
   <node name="Encapsulating protein for a DyP-type peroxidase or ferritin-like protein oligomers">
    <magnitude><val></val><val>8</val></magnitude>
    <score><val>0</val><val>-10.75</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Uncharacterized protein COG3461">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>4</val></magnitude>
    <score><val>0</val><val>-23.75</val></score>
   </node>
   <node name="Ferritin-like di-iron-carboxylate protein">
    <magnitude><val></val><val>8</val></magnitude>
    <score><val>0</val><val>-10.75</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Heme, hemin uptake and utilization systems in GramNegatives">
   <rank><val>level 2</val></rank>
   <magnitude><val>2</val><val>45</val></magnitude>
   <score><val>-33.5</val><val>-78.8619</val></score>
   <node name="Electron transfer flavoprotein, beta subunit">
    <score><val>0</val><val>-80.9524</val></score>
    <magnitude><val></val><val>36</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Ferrichrome transport ATP-binding protein FhuC (TC 3.A.1.14.3)">
    <rank><val>function</val></rank>
    <score><val>-33.5</val><val>-70.5</val></score>
    <magnitude><val>2</val><val>9</val></magnitude>
   </node>
  </node>
  <node name="Iron acquisition in Streptococcus">
   <rank><val>level 2</val></rank>
   <magnitude><val></val><val>1</val></magnitude>
   <score><val>0</val><val>-14</val></score>
   <node name="Ferric iron ABC transporter, permease protein">
    <score><val>0</val><val>-14</val></score>
    <magnitude><val></val><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Heme, hemin uptake and utilization systems in GramPositives">
   <score><val>-5</val><val>-40.2222</val></score>
   <magnitude><val>1</val><val>16</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Zn-dependent hydrolase YycJ/WalJ, required for cell wall metabolism and coordination of cell division with DNA replication">
    <score><val>-5</val><val>-40.2222</val></score>
    <magnitude><val>1</val><val>16</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Siderophores">
   <rank><val>level 2</val></rank>
   <score><val>-6</val><val>-120.367</val></score>
   <magnitude><val>1</val><val>10</val></magnitude>
   <node name="Siderophore Enterobactin">
    <rank><val>level 3</val></rank>
    <score><val>0</val><val>-120.367</val></score>
    <magnitude><val></val><val>10</val></magnitude>
    <node name="Enterobactin exporter EntS">
     <score><val>0</val><val>-156</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Ferric enterobactin-binding periplasmic protein FepB (TC 3.A.1.14.2)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-96</val></score>
    </node>
    <node name="Enterobactin esterase">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-56</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Isochorismate synthase (EC 5.4.4.2) [enterobactin] siderophore">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-65.5</val></score>
     <magnitude><val></val><val>2</val></magnitude>
    </node>
    <node name="Ferric enterobactin transport system permease protein FepG (TC 3.A.1.14.2)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-134</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Enterobactin synthetase component F, serine activating enzyme (EC 2.7.7.-)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-157.667</val></score>
     <magnitude><val></val><val>4</val></magnitude>
    </node>
   </node>
   <node name="Siderophore assembly kit">
    <magnitude><val>1</val></magnitude>
    <score><val>-6</val></score>
    <rank><val>level 3</val></rank>
    <node name="Isochorismate synthase (EC 5.4.4.2) of siderophore biosynthesis">
     <score><val>-6</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Iron acquisition in Vibrio">
   <rank><val>level 2</val></rank>
   <score><val>-24.6</val><val>-47.2254</val></score>
   <magnitude><val>5</val><val>33</val></magnitude>
   <node name="Ferrichrome transport ATP-binding protein FhuC (TC 3.A.1.14.3)">
    <score><val>-33.5</val><val>-70.5</val></score>
    <magnitude><val>2</val><val>9</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Ferric iron ABC transporter, permease protein">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>1</val></magnitude>
    <score><val>0</val><val>-14</val></score>
   </node>
   <node name="Ferrous iron transport protein B">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-19</val></score>
   </node>
   <node name="Isochorismate synthase (EC 5.4.4.2) of siderophore biosynthesis">
    <rank><val>function</val></rank>
    <score><val>-6</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Bacterioferritin">
    <rank><val>function</val></rank>
    <magnitude><val>1</val><val>23</val></magnitude>
    <score><val>-31</val><val>-39.5625</val></score>
   </node>
  </node>
  <node name="Transport of Iron">
   <magnitude><val>1</val><val>1</val></magnitude>
   <score><val>-19</val><val>-14</val></score>
   <rank><val>level 2</val></rank>
   <node name="Ferric iron ABC transporter, permease protein">
    <rank><val>function</val></rank>
    <score><val>0</val><val>-14</val></score>
    <magnitude><val></val><val>1</val></magnitude>
   </node>
   <node name="Ferrous iron transport protein B">
    <score><val>-19</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
 </node>
 <node name="Regulation and Cell signaling">
  <rank><val>level 1</val></rank>
  <magnitude><val>62</val><val>174</val></magnitude>
  <score><val>-27.82</val><val>-56.4102</val></score>
  <node name="Cell envelope-associated LytR-CpsA-Psr transcriptional attenuators">
   <rank><val>level 2</val></rank>
   <score><val>-8</val></score>
   <magnitude><val>1</val></magnitude>
   <node name="Cell envelope-associated transcriptional attenuator LytR-CpsA-Psr, subfamily M (as in PMID19099556)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-8</val></score>
   </node>
  </node>
  <node name="DNA-binding regulatory proteins, strays">
   <rank><val>level 2</val></rank>
   <score><val>-7</val></score>
   <magnitude><val>1</val></magnitude>
   <node name="Hydrogen peroxide-inducible genes activator">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-7</val></score>
   </node>
  </node>
  <node name="Programmed Cell Death and Toxin-antitoxin Systems">
   <rank><val>level 2</val></rank>
   <score><val>-22.8518</val><val>-9.35089</val></score>
   <magnitude><val>9</val><val>19</val></magnitude>
   <node name="Murein hydrolase regulation and cell death">
    <score><val>-30</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="LrgA-associated membrane protein LrgB">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-30</val></score>
    </node>
   </node>
   <node name="Phd-Doc, YdcE-YdcD toxin-antitoxin (programmed cell death) systems">
    <magnitude><val>5</val><val>4</val></magnitude>
    <score><val>-27.3333</val><val>-14</val></score>
    <rank><val>level 3</val></rank>
    <node name="Death on curing protein, Doc toxin">
     <score><val>-27.3333</val><val>-14</val></score>
     <magnitude><val>5</val><val>4</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Toxin-antitoxin systems (other than RelBE and MazEF)">
    <magnitude><val></val><val>5</val></magnitude>
    <score><val>0</val><val>-4</val></score>
    <rank><val>level 3</val></rank>
    <node name="VapB protein (antitoxin to VapC)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>5</val></magnitude>
     <score><val>0</val><val>-4</val></score>
    </node>
   </node>
   <node name="Toxin-antitoxin replicon stabilization systems">
    <score><val>-13</val><val>-10.1667</val></score>
    <magnitude><val>3</val><val>10</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="RelE/StbE replicon stabilization toxin">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>10</val></magnitude>
     <score><val>-13</val><val>-10.1667</val></score>
    </node>
   </node>
  </node>
  <node name="Regulation of virulence">
   <score><val>-18.4286</val><val>-35.6429</val></score>
   <magnitude><val>8</val><val>19</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="BarA-UvrY(SirA) two-component regulatory system">
    <score><val>-18.4286</val><val>-35.6429</val></score>
    <magnitude><val>8</val><val>19</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="CDP-diacylglycerol--glycerol-3-phosphate 3-phosphatidyltransferase (EC 2.7.8.5)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>19</val></magnitude>
     <score><val>-18.4286</val><val>-35.6429</val></score>
    </node>
   </node>
  </node>
  <node name="Zinc regulated enzymes">
   <rank><val>level 2</val></rank>
   <score><val>-31.9435</val><val>-65.4922</val></score>
   <magnitude><val>34</val><val>129</val></magnitude>
   <node name="Dihydroorotase (EC 3.5.2.3)">
    <rank><val>function</val></rank>
    <score><val>-37.5385</val><val>-72.9545</val></score>
    <magnitude><val>15</val><val>31</val></magnitude>
   </node>
   <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 2">
    <magnitude><val>2</val><val>5</val></magnitude>
    <score><val>-8</val><val>-7</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="GTP cyclohydrolase I (EC 3.5.4.16) type 1">
    <score><val>-64</val><val>-47.7778</val></score>
    <magnitude><val>1</val><val>16</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Porphobilinogen synthase (EC 4.2.1.24)">
    <magnitude><val>7</val><val>29</val></magnitude>
    <score><val>-24.2857</val><val>-70.4375</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Phosphoribosyl-AMP cyclohydrolase (EC 3.5.4.19)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val><val>11</val></magnitude>
    <score><val>-34</val><val>-57.1111</val></score>
   </node>
   <node name="Queuosine biosynthesis QueD, PTPS-I">
    <rank><val>function</val></rank>
    <magnitude><val>2</val><val>16</val></magnitude>
    <score><val>-9</val><val>-34.9</val></score>
   </node>
   <node name="Cysteinyl-tRNA synthetase (EC 6.1.1.16)">
    <score><val>-52.75</val><val>-102.769</val></score>
    <magnitude><val>4</val><val>21</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="N-acetylmuramoyl-L-alanine amidase (EC 3.5.1.28)">
    <rank><val>function</val></rank>
    <score><val>-5</val></score>
    <magnitude><val>2</val></magnitude>
   </node>
  </node>
  <node name="WhiB and WhiB-type regulatory proteins ">
   <score><val>-9</val></score>
   <magnitude><val>2</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="WhiB-like transcription regulator">
    <score><val>-8</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Sporulation regulatory protein WhiB">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-10</val></score>
   </node>
  </node>
  <node name="Oxygen and light sensor PpaA-PpsR">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val></magnitude>
   <score><val>-18</val></score>
   <node name="Transcriptional regulator, PpsR">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-18</val></score>
   </node>
  </node>
  <node name="Quorum sensing and biofilm formation">
   <magnitude><val>5</val><val>7</val></magnitude>
   <score><val>-41.3334</val><val>-73.1429</val></score>
   <rank><val>level 2</val></rank>
   <node name="Biofilm formation in Staphylococcus">
    <rank><val>level 3</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-8</val></score>
    <node name="RNA polymerase sigma factor SigB">
     <magnitude><val>1</val></magnitude>
     <score><val>-8</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Quorum Sensing: Autoinducer-2 Synthesis">
    <rank><val>level 3</val></rank>
    <score><val>-49.6667</val><val>-73.1429</val></score>
    <magnitude><val>4</val><val>7</val></magnitude>
    <node name="S-adenosylmethionine synthetase (EC 2.5.1.6)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>7</val></magnitude>
     <score><val>-49.6667</val><val>-73.1429</val></score>
    </node>
   </node>
  </node>
  <node name="cAMP signaling in bacteria">
   <magnitude><val>1</val></magnitude>
   <score><val>-28</val></score>
   <rank><val>level 2</val></rank>
   <node name="Adenylate cyclase (EC 4.6.1.1)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-28</val></score>
   </node>
  </node>
 </node>
 <node name="Stress Response">
  <score><val>-48.174</val><val>-64.9938</val></score>
  <magnitude><val>177</val><val>562</val></magnitude>
  <rank><val>level 1</val></rank>
  <node name="Osmotic stress">
   <rank><val>level 2</val></rank>
   <score><val>-13.6</val><val>-19.5</val></score>
   <magnitude><val>5</val><val>2</val></magnitude>
   <node name="Choline and Betaine Uptake and Betaine Biosynthesis">
    <rank><val>level 3</val></rank>
    <score><val>-14.5</val></score>
    <magnitude><val>4</val></magnitude>
    <node name="Choline-sulfatase (EC 3.1.6.6)">
     <magnitude><val>4</val></magnitude>
     <score><val>-14.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Osmoregulation">
    <score><val>-10</val><val>-19.5</val></score>
    <magnitude><val>1</val><val>2</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Aquaporin Z">
     <score><val>-10</val><val>-19.5</val></score>
     <magnitude><val>1</val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="SigmaB stress responce regulation">
   <score><val>-8</val></score>
   <magnitude><val>1</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="RNA polymerase sigma factor SigB">
    <magnitude><val>1</val></magnitude>
    <score><val>-8</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Cold shock">
   <score><val>-19</val></score>
   <magnitude><val>1</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Cold shock, CspA family of proteins">
    <score><val>-19</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Cold shock protein CspA">
     <score><val>-19</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Bacterial hemoglobins">
   <score><val>-20</val></score>
   <magnitude><val>2</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="diguanylate cyclase/phosphodiesterase (GGDEF & EAL domains) with PAS/PAC sensor(s)">
    <score><val>-20</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Oxidative stress">
   <rank><val>level 2</val></rank>
   <score><val>-46.7944</val><val>-63.7123</val></score>
   <magnitude><val>142</val><val>372</val></magnitude>
   <node name="Glutathione: Biosynthesis and gamma-glutamyl cycle">
    <rank><val>level 3</val></rank>
    <score><val>-33.6667</val><val>-88.8649</val></score>
    <magnitude><val>6</val><val>63</val></magnitude>
    <node name="Gamma-glutamyltranspeptidase (EC 2.3.2.2)">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>63</val></magnitude>
     <score><val>-33.6667</val><val>-88.8649</val></score>
    </node>
   </node>
   <node name="Regulation of Oxidative Stress Response">
    <rank><val>level 3</val></rank>
    <magnitude><val>18</val><val>46</val></magnitude>
    <score><val>-46.0333</val><val>-65.6029</val></score>
    <node name="Superoxide dismutase [Mn] (EC 1.15.1.1)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>27</val></magnitude>
     <score><val>-41</val><val>-61.5333</val></score>
    </node>
    <node name="Superoxide dismutase [Fe] (EC 1.15.1.1)">
     <magnitude><val>4</val><val>6</val></magnitude>
     <score><val>-16</val><val>-28.1667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Peroxidase (EC 1.11.1.7)">
     <score><val>-87.6</val><val>-91.3333</val></score>
     <magnitude><val>6</val><val>13</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="RNA polymerase sigma factor">
     <score><val>-23.2</val></score>
     <magnitude><val>5</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Redox-dependent regulation of nucleus processes">
    <rank><val>level 3</val></rank>
    <magnitude><val>14</val><val>37</val></magnitude>
    <score><val>-25.6571</val><val>-48.8091</val></score>
    <node name="Nicotinamidase (EC 3.5.1.19)">
     <magnitude><val>2</val><val>8</val></magnitude>
     <score><val>-13.5</val><val>-42.2857</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Nicotinate phosphoribosyltransferase (EC 2.4.2.11)">
     <magnitude><val>11</val><val>29</val></magnitude>
     <score><val>-29.2</val><val>-50.6087</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="NAD-dependent glyceraldehyde-3-phosphate dehydrogenase (EC 1.2.1.12)">
     <score><val>-11</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Glutathione: Non-redox reactions">
    <magnitude><val>2</val></magnitude>
    <score><val>-40</val></score>
    <rank><val>level 3</val></rank>
    <node name="Hydroxyacylglutathione hydrolase (EC 3.1.2.6)">
     <rank><val>function</val></rank>
     <score><val>-53</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Glutathione S-transferase, zeta (EC 2.5.1.18)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-27</val></score>
    </node>
   </node>
   <node name="CoA disulfide thiol-disulfide redox system">
    <rank><val>level 3</val></rank>
    <score><val>0</val><val>-4</val></score>
    <magnitude><val></val><val>5</val></magnitude>
    <node name="CoA-disulfide reductase (EC 1.8.1.14)">
     <magnitude><val></val><val>5</val></magnitude>
     <score><val>0</val><val>-4</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Glutathione: Redox cycle">
    <score><val>-29.4</val><val>-16.5</val></score>
    <magnitude><val>7</val><val>4</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Glutathione reductase (EC 1.8.1.7)">
     <score><val>-9</val><val>-9</val></score>
     <magnitude><val>1</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glutaredoxin">
     <rank><val>function</val></rank>
     <score><val>-32.8</val><val>-19</val></score>
     <magnitude><val>6</val><val>3</val></magnitude>
    </node>
   </node>
   <node name="Oxidative stress">
    <score><val>-51.983</val><val>-61.1662</val></score>
    <magnitude><val>47</val><val>121</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Transcriptional regulator, FUR family">
     <magnitude><val>2</val><val>1</val></magnitude>
     <score><val>-5</val><val>-5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Redox-sensitive transcriptional regulator (AT-rich DNA-binding protein)">
     <score><val>-14</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Superoxide dismutase [Mn] (EC 1.15.1.1)">
     <rank><val>function</val></rank>
     <score><val>-41</val><val>-61.5333</val></score>
     <magnitude><val>3</val><val>27</val></magnitude>
    </node>
    <node name="Catalase (EC 1.11.1.6)">
     <rank><val>function</val></rank>
     <score><val>-87.6</val><val>-91.3333</val></score>
     <magnitude><val>6</val><val>13</val></magnitude>
    </node>
    <node name="Alkyl hydroperoxide reductase subunit C-like protein">
     <magnitude><val>9</val><val>58</val></magnitude>
     <score><val>-47.1111</val><val>-53.725</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Rubrerythrin">
     <rank><val>function</val></rank>
     <score><val>-41</val><val>-25</val></score>
     <magnitude><val>12</val><val>3</val></magnitude>
    </node>
    <node name="Hydrogen peroxide-inducible genes activator">
     <score><val>-7</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Probable peroxiredoxin (EC 1.11.1.15)">
     <rank><val>function</val></rank>
     <score><val>-86</val></score>
     <magnitude><val>3</val></magnitude>
    </node>
    <node name="Superoxide dismutase [Fe] (EC 1.15.1.1)">
     <magnitude><val>4</val><val>6</val></magnitude>
     <score><val>-16</val><val>-28.1667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Peroxidase (EC 1.11.1.7)">
     <rank><val>function</val></rank>
     <score><val>-87.6</val><val>-91.3333</val></score>
     <magnitude><val>6</val><val>13</val></magnitude>
    </node>
   </node>
   <node name="Glutathione analogs: mycothiol">
    <score><val>-20</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Maleylpyruvate isomerase, mycothiol-dependent (EC 5.2.1.4)">
     <rank><val>function</val></rank>
     <score><val>-20</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
   </node>
   <node name="Glutaredoxins">
    <score><val>-32.8</val><val>-19</val></score>
    <magnitude><val>6</val><val>3</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Glutaredoxin">
     <rank><val>function</val></rank>
     <score><val>-32.8</val><val>-19</val></score>
     <magnitude><val>6</val><val>3</val></magnitude>
    </node>
   </node>
   <node name="Rubrerythrin">
    <rank><val>level 3</val></rank>
    <score><val>-48.9167</val><val>-52.3123</val></score>
    <magnitude><val>24</val><val>61</val></magnitude>
    <node name="Alkyl hydroperoxide reductase subunit C-like protein">
     <rank><val>function</val></rank>
     <magnitude><val>9</val><val>58</val></magnitude>
     <score><val>-47.1111</val><val>-53.725</val></score>
    </node>
    <node name="Probable peroxiredoxin (EC 1.11.1.15)">
     <magnitude><val>3</val></magnitude>
     <score><val>-86</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Rubrerythrin">
     <score><val>-41</val><val>-25</val></score>
     <magnitude><val>12</val><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Protection from Reactive Oxygen Species">
    <rank><val>level 3</val></rank>
    <score><val>-69.7</val><val>-79.4896</val></score>
    <magnitude><val>16</val><val>32</val></magnitude>
    <node name="Catalase (EC 1.11.1.6)">
     <rank><val>function</val></rank>
     <score><val>-87.6</val><val>-91.3333</val></score>
     <magnitude><val>6</val><val>13</val></magnitude>
    </node>
    <node name="Superoxide dismutase [Fe] (EC 1.15.1.1)">
     <magnitude><val>4</val><val>6</val></magnitude>
     <score><val>-16</val><val>-28.1667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Peroxidase (EC 1.11.1.7)">
     <score><val>-87.6</val><val>-91.3333</val></score>
     <magnitude><val>6</val><val>13</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Heat shock">
   <rank><val>level 2</val></rank>
   <magnitude><val>24</val><val>172</val></magnitude>
   <score><val>-71.1665</val><val>-70.1542</val></score>
   <node name="Heat shock dnaK gene cluster extended">
    <rank><val>level 3</val></rank>
    <magnitude><val>24</val><val>172</val></magnitude>
    <score><val>-71.1665</val><val>-70.1542</val></score>
    <node name="Chaperone protein DnaK">
     <magnitude><val>14</val><val>90</val></magnitude>
     <score><val>-103.214</val><val>-86.6364</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Chaperone protein DnaJ">
     <magnitude><val>5</val><val>46</val></magnitude>
     <score><val>-30.4</val><val>-57.5862</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribonuclease PH (EC 2.7.7.56)">
     <magnitude><val>1</val><val>13</val></magnitude>
     <score><val>-42</val><val>-63</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Xanthosine/inosine triphosphate pyrophosphatase">
     <magnitude><val>2</val><val>21</val></magnitude>
     <score><val>-18</val><val>-37.7273</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Heat shock protein GrpE">
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-4.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ribosomal protein L11 methyltransferase (EC 2.1.1.-)">
     <score><val>-14</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="rRNA small subunit methyltransferase I">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-19</val></score>
    </node>
   </node>
  </node>
  <node name="Universal stress protein family">
   <rank><val>level 2</val></rank>
   <magnitude><val>1</val></magnitude>
   <score><val>-3</val></score>
   <node name="Universal stress protein family">
    <magnitude><val>1</val></magnitude>
    <score><val>-3</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Phage shock protein (psp) operon">
   <rank><val>level 2</val></rank>
   <score><val>-36</val></score>
   <magnitude><val>1</val></magnitude>
   <node name="Phage shock protein A">
    <score><val>-36</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Acid stress">
   <rank><val>level 2</val></rank>
   <score><val>0</val><val>-45</val></score>
   <magnitude><val></val><val>16</val></magnitude>
   <node name="Acid resistance mechanisms">
    <rank><val>level 3</val></rank>
    <magnitude><val></val><val>16</val></magnitude>
    <score><val>0</val><val>-45</val></score>
    <node name="Glutamate decarboxylase (EC 4.1.1.15)">
     <magnitude><val></val><val>16</val></magnitude>
     <score><val>0</val><val>-45</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
 </node>
 <node name="Nucleosides and Nucleotides">
  <rank><val>level 1</val></rank>
  <score><val>-33.7519</val><val>-69.4364</val></score>
  <magnitude><val>317</val><val>1483</val></magnitude>
  <node name="Adenosyl nucleosidases">
   <rank><val>level 2</val></rank>
   <score><val>0</val><val>-69.3077</val></score>
   <magnitude><val></val><val>19</val></magnitude>
   <node name="Purine nucleoside phosphorylase (EC 2.4.2.1)">
    <score><val>0</val><val>-69.3077</val></score>
    <magnitude><val></val><val>19</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Detoxification">
   <score><val>-15.5</val><val>-35.7895</val></score>
   <magnitude><val>2</val><val>19</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="Housecleaning nucleoside triphosphate pyrophosphatases">
    <score><val>0</val><val>-15</val></score>
    <magnitude><val></val><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Deoxyuridine 5&#39;-triphosphate nucleotidohydrolase (EC 3.6.1.23)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-15</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Nudix proteins (nucleoside triphosphate hydrolases)">
    <rank><val>level 3</val></rank>
    <score><val>-17</val><val>-36.9445</val></score>
    <magnitude><val>1</val><val>18</val></magnitude>
    <node name="Mutator mutT protein (7,8-dihydro-8-oxoguanine-triphosphatase) (EC 3.6.1.-)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Thiamin-phosphate pyrophosphorylase-like protein">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-5</val></score>
    </node>
    <node name="ADP-ribose pyrophosphatase (EC 3.6.1.13)">
     <rank><val>function</val></rank>
     <score><val>-17</val><val>-42.6667</val></score>
     <magnitude><val>1</val><val>15</val></magnitude>
    </node>
    <node name="Deoxyuridine 5&#39;-triphosphate nucleotidohydrolase (EC 3.6.1.23)">
     <magnitude><val></val><val>1</val></magnitude>
     <score><val>0</val><val>-15</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Nucleoside triphosphate pyrophosphohydrolase MazG">
    <score><val>-14</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Nucleoside triphosphate pyrophosphohydrolase MazG">
     <score><val>-14</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Hydantoin metabolism">
   <rank><val>level 2</val></rank>
   <score><val>-29.5</val><val>-80.5476</val></score>
   <magnitude><val>4</val><val>160</val></magnitude>
   <node name="D-hydantoinase (EC 3.5.2.2)">
    <magnitude><val></val><val>39</val></magnitude>
    <score><val>0</val><val>-89.5333</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="N-methylhydantoinase A (EC 3.5.2.14)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val><val>61</val></magnitude>
    <score><val>-46</val><val>-87</val></score>
   </node>
   <node name="Dihydropyrimidinase (EC 3.5.2.2)">
    <magnitude><val>1</val><val>4</val></magnitude>
    <score><val>-28</val><val>-19.5</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="N-methylhydantoinase B (EC 3.5.2.14)">
    <magnitude><val>2</val><val>56</val></magnitude>
    <score><val>-22</val><val>-71.6216</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Pyrimidines">
   <rank><val>level 2</val></rank>
   <magnitude><val>123</val><val>502</val></magnitude>
   <score><val>-30.7583</val><val>-62.8013</val></score>
   <node name="Pyrimidine utilization">
    <score><val>-28</val><val>-40.2</val></score>
    <magnitude><val>1</val><val>10</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Dihydropyrimidine dehydrogenase [NADP+] (EC 1.3.1.2)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-54</val></score>
     <magnitude><val></val><val>6</val></magnitude>
    </node>
    <node name="Dihydropyrimidinase (EC 3.5.2.2)">
     <magnitude><val>1</val><val>4</val></magnitude>
     <score><val>-28</val><val>-19.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Novel non-oxidative pathway of Uracil catabolism">
    <rank><val>level 3</val></rank>
    <score><val>-28</val><val>-40.2</val></score>
    <magnitude><val>1</val><val>10</val></magnitude>
    <node name="Dihydropyrimidine dehydrogenase [NADP+] (EC 1.3.1.2)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>6</val></magnitude>
     <score><val>0</val><val>-54</val></score>
    </node>
    <node name="Dihydropyrimidinase (EC 3.5.2.2)">
     <magnitude><val>1</val><val>4</val></magnitude>
     <score><val>-28</val><val>-19.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="pyrimidine conversions">
    <rank><val>level 3</val></rank>
    <magnitude><val>48</val><val>280</val></magnitude>
    <score><val>-31.5937</val><val>-66.8698</val></score>
    <node name="Nucleoside diphosphate kinase (EC 2.7.4.6)">
     <rank><val>function</val></rank>
     <score><val>-33.5</val><val>-28.8</val></score>
     <magnitude><val>5</val><val>7</val></magnitude>
    </node>
    <node name="Thioredoxin reductase (EC 1.8.1.9)">
     <magnitude><val>11</val><val>91</val></magnitude>
     <score><val>-39.3636</val><val>-67.8525</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Purine nucleoside phosphorylase (EC 2.4.2.1)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-69.3077</val></score>
     <magnitude><val></val><val>19</val></magnitude>
    </node>
    <node name="CTP synthase (EC 6.3.4.2)">
     <rank><val>function</val></rank>
     <magnitude><val>15</val><val>79</val></magnitude>
     <score><val>-42</val><val>-65.4151</val></score>
    </node>
    <node name="Thymidylate kinase (EC 2.7.4.9)">
     <score><val>-25.4286</val><val>-88.0606</val></score>
     <magnitude><val>7</val><val>45</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Thymidine kinase (EC 2.7.1.21)">
     <score><val>0</val><val>-53</val></score>
     <magnitude><val></val><val>19</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cytosine deaminase (EC 3.5.4.1)">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>2</val></magnitude>
     <score><val>0</val><val>-34</val></score>
    </node>
    <node name="Uridine phosphorylase (EC 2.4.2.3)">
     <magnitude><val>2</val></magnitude>
     <score><val>-3.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cytidylate kinase (EC 2.7.4.14)">
     <rank><val>function</val></rank>
     <score><val>-8</val><val>-51</val></score>
     <magnitude><val>1</val><val>15</val></magnitude>
    </node>
    <node name="Deoxycytidine triphosphate deaminase (EC 3.5.4.13)">
     <magnitude><val>7</val><val>3</val></magnitude>
     <score><val>-13.2857</val><val>-20</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="De Novo Pyrimidine Synthesis">
    <rank><val>level 3</val></rank>
    <magnitude><val>73</val><val>202</val></magnitude>
    <score><val>-30.2845</val><val>-59.3995</val></score>
    <node name="Aspartate carbamoyltransferase regulatory chain (PyrI)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>2</val></magnitude>
     <score><val>-18.5</val><val>-20.5</val></score>
    </node>
    <node name="Carbamoyl-phosphate synthase large chain (EC 6.3.5.5)">
     <rank><val>function</val></rank>
     <score><val>-27.1875</val><val>-81.2093</val></score>
     <magnitude><val>17</val><val>58</val></magnitude>
    </node>
    <node name="Dihydroorotate dehydrogenase (EC 1.3.3.1)">
     <magnitude><val>1</val><val>7</val></magnitude>
     <score><val>-44</val><val>-51.5714</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Orotidine 5&#39;-phosphate decarboxylase (EC 4.1.1.23)">
     <score><val>-11.6667</val><val>-27</val></score>
     <magnitude><val>3</val><val>14</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Carbamoyl-phosphate synthase small chain (EC 6.3.5.5)">
     <score><val>-23</val><val>-17.5</val></score>
     <magnitude><val>6</val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dihydroorotate dehydrogenase electron transfer subunit (EC 1.3.3.1)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>14</val></magnitude>
     <score><val>-19.5</val><val>-36.4167</val></score>
    </node>
    <node name="Orotate phosphoribosyltransferase (EC 2.4.2.10)">
     <rank><val>function</val></rank>
     <score><val>-20</val><val>-37.75</val></score>
     <magnitude><val>4</val><val>32</val></magnitude>
    </node>
    <node name="Dihydroorotase (EC 3.5.2.3)">
     <rank><val>function</val></rank>
     <score><val>-37.5385</val><val>-72.9545</val></score>
     <magnitude><val>15</val><val>31</val></magnitude>
    </node>
    <node name="Aspartate carbamoyltransferase (EC 2.1.3.2)">
     <rank><val>function</val></rank>
     <score><val>-42.1</val><val>-52.3333</val></score>
     <magnitude><val>10</val><val>24</val></magnitude>
    </node>
    <node name="Dihydroorotate dehydrogenase, catalytic subunit (EC 1.3.3.1)">
     <rank><val>function</val></rank>
     <magnitude><val>5</val><val>18</val></magnitude>
     <score><val>-47.5</val><val>-68.7857</val></score>
    </node>
   </node>
  </node>
  <node name="Ribonucleotide reduction">
   <rank><val>level 2</val></rank>
   <magnitude><val>22</val><val>17</val></magnitude>
   <score><val>-61.8068</val><val>-70.375</val></score>
   <node name="Ribonucleotide reductase of class II (coenzyme B12-dependent) (EC 1.17.4.1)">
    <rank><val>function</val></rank>
    <magnitude><val>18</val><val>17</val></magnitude>
    <score><val>-59.875</val><val>-70.375</val></score>
   </node>
   <node name="Ribonucleotide reductase of class Ia (aerobic), beta subunit (EC 1.17.4.1)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-32</val></score>
   </node>
   <node name="Ribonucleotide reductase of class Ia (aerobic), alpha subunit (EC 1.17.4.1)">
    <magnitude><val>3</val></magnitude>
    <score><val>-83.3333</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Purines">
   <rank><val>level 2</val></rank>
   <magnitude><val>166</val><val>766</val></magnitude>
   <score><val>-32.5744</val><val>-72.2809</val></score>
   <node name="Purine conversions">
    <magnitude><val>59</val><val>267</val></magnitude>
    <score><val>-31.1784</val><val>-76.9642</val></score>
    <rank><val>level 3</val></rank>
    <node name="Nucleoside diphosphate kinase (EC 2.7.4.6)">
     <rank><val>function</val></rank>
     <score><val>-33.5</val><val>-28.8</val></score>
     <magnitude><val>5</val><val>7</val></magnitude>
    </node>
    <node name="Deoxyguanosinetriphosphate triphosphohydrolase (EC 3.1.5.1)">
     <rank><val>function</val></rank>
     <score><val>-10.4286</val><val>-44.125</val></score>
     <magnitude><val>8</val><val>9</val></magnitude>
    </node>
    <node name="Adenylosuccinate lyase (EC 4.3.2.2)">
     <rank><val>function</val></rank>
     <magnitude><val>16</val><val>39</val></magnitude>
     <score><val>-41.8667</val><val>-78.8571</val></score>
    </node>
    <node name="GMP synthase [glutamine-hydrolyzing] (EC 6.3.5.2)">
     <score><val>-3</val><val>-78.5238</val></score>
     <magnitude><val>1</val><val>31</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Hypoxanthine-guanine phosphoribosyltransferase (EC 2.4.2.8)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val></magnitude>
     <score><val>-35</val></score>
    </node>
    <node name="Inosine-5&#39;-monophosphate dehydrogenase (EC 1.1.1.205)">
     <score><val>-33.4</val><val>-106.826</val></score>
     <magnitude><val>10</val><val>43</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Nucleotide pyrophosphatase (EC 3.6.1.9)">
     <score><val>0</val><val>-93.3235</val></score>
     <magnitude><val></val><val>42</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Purine nucleoside phosphorylase (EC 2.4.2.1)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-69.3077</val></score>
     <magnitude><val></val><val>19</val></magnitude>
    </node>
    <node name="Inosine-uridine preferring nucleoside hydrolase (EC 3.2.2.1)">
     <magnitude><val>1</val><val>23</val></magnitude>
     <score><val>-6</val><val>-51.7692</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Xanthine-guanine phosphoribosyltransferase (EC 2.4.2.22)">
     <magnitude><val>3</val><val>3</val></magnitude>
     <score><val>-18</val><val>-12</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Adenylosuccinate synthetase (EC 6.3.4.4)">
     <rank><val>function</val></rank>
     <magnitude><val>12</val><val>51</val></magnitude>
     <score><val>-34.7273</val><val>-66.3611</val></score>
    </node>
   </node>
   <node name="Purine Utilization">
    <magnitude><val>8</val><val>32</val></magnitude>
    <score><val>-17.25</val><val>-90.6316</val></score>
    <rank><val>level 3</val></rank>
    <node name="Xanthine dehydrogenase, molybdenum binding subunit (EC 1.17.1.4)">
     <score><val>-65</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Xanthine dehydrogenase, FAD binding subunit (EC 1.17.1.4)">
     <rank><val>function</val></rank>
     <score><val>-5</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="XdhC protein (assists in molybdopterin insertion into xanthine dehydrogenase)">
     <rank><val>function</val></rank>
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Cytosine/purine/uracil/thiamine/allantoin permease family protein">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>32</val></magnitude>
     <score><val>-34</val><val>-90.6316</val></score>
    </node>
    <node name="Xanthine and CO dehydrogenases maturation factor, XdhC/CoxF family">
     <rank><val>function</val></rank>
     <score><val>-6</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Xanthine dehydrogenase iron-sulfur subunit (EC 1.17.1.4)">
     <score><val>-9.5</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="De Novo Purine Biosynthesis">
    <rank><val>level 3</val></rank>
    <score><val>-34.6446</val><val>-68.3458</val></score>
    <magnitude><val>99</val><val>467</val></magnitude>
    <node name="Phosphoribosylaminoimidazole carboxylase catalytic subunit (EC 4.1.1.21)">
     <score><val>-48.3333</val><val>-47.0588</val></score>
     <magnitude><val>6</val><val>23</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylaminoimidazole-succinocarboxamide synthase (EC 6.3.2.6)">
     <magnitude><val>4</val><val>30</val></magnitude>
     <score><val>-20</val><val>-64.3333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylformylglycinamidine synthase, PurS subunit (EC 6.3.5.3)">
     <magnitude><val></val><val>10</val></magnitude>
     <score><val>0</val><val>-16</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylglycinamide formyltransferase (EC 2.1.2.2)">
     <magnitude><val>2</val><val>21</val></magnitude>
     <score><val>-25</val><val>-38.3571</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylformylglycinamidine cyclo-ligase (EC 6.3.3.1)">
     <magnitude><val>12</val><val>48</val></magnitude>
     <score><val>-67.8571</val><val>-79.9259</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylformylglycinamidine synthase, synthetase subunit (EC 6.3.5.3)">
     <magnitude><val>8</val><val>81</val></magnitude>
     <score><val>-33</val><val>-82.8333</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylaminoimidazolecarboxamide formyltransferase (EC 2.1.2.3)">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>32</val></magnitude>
     <score><val>-17.6</val><val>-65.4545</val></score>
    </node>
    <node name="Phosphoribosylaminoimidazolecarboxamide formyltransferase [alternate form]">
     <score><val>-39</val></score>
     <magnitude><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Possible subunit variant of phosphoribosylaminoimidazolecarboxamide formyltransferase [alternate form]">
     <score><val>-24</val></score>
     <magnitude><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylaminoimidazole carboxylase ATPase subunit (EC 4.1.1.21)">
     <magnitude><val>3</val><val>31</val></magnitude>
     <score><val>-20</val><val>-76.2381</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phosphoribosylformylglycinamidine synthase, glutamine amidotransferase subunit (EC 6.3.5.3)">
     <rank><val>function</val></rank>
     <score><val>-14</val><val>-26.1</val></score>
     <magnitude><val>1</val><val>18</val></magnitude>
    </node>
    <node name="IMP cyclohydrolase (EC 3.5.4.10)">
     <rank><val>function</val></rank>
     <score><val>-17.6</val><val>-65.4545</val></score>
     <magnitude><val>6</val><val>32</val></magnitude>
    </node>
    <node name="Phosphoribosylamine--glycine ligase (EC 6.3.4.13)">
     <rank><val>function</val></rank>
     <score><val>-23.7</val><val>-79.5789</val></score>
     <magnitude><val>14</val><val>27</val></magnitude>
    </node>
    <node name="Adenylosuccinate lyase (EC 4.3.2.2)">
     <magnitude><val>16</val><val>39</val></magnitude>
     <score><val>-41.8667</val><val>-78.8571</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Amidophosphoribosyltransferase (EC 2.4.2.14)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>42</val></magnitude>
     <score><val>-45.875</val><val>-77.3571</val></score>
    </node>
    <node name="Ribose-phosphate pyrophosphokinase (EC 2.7.6.1)">
     <rank><val>function</val></rank>
     <score><val>-12.6667</val><val>-57.5263</val></score>
     <magnitude><val>7</val><val>33</val></magnitude>
    </node>
   </node>
  </node>
 </node>
 <node name="Fatty Acids, Lipids, and Isoprenoids">
  <rank><val>level 1</val></rank>
  <magnitude><val>597</val><val>2335</val></magnitude>
  <score><val>-35.6404</val><val>-70.2403</val></score>
  <node name="Phospholipids">
   <rank><val>level 2</val></rank>
   <magnitude><val>46</val><val>235</val></magnitude>
   <score><val>-30.9538</val><val>-75.7813</val></score>
   <node name="Cardiolipin synthesis">
    <score><val>0</val><val>-49</val></score>
    <magnitude><val></val><val>12</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Cardiolipin synthetase (EC 2.7.8.-)">
     <score><val>0</val><val>-49</val></score>
     <magnitude><val></val><val>12</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Glycerolipid and Glycerophospholipid Metabolism in Bacteria">
    <rank><val>level 3</val></rank>
    <score><val>-30.9538</val><val>-77.2224</val></score>
    <magnitude><val>46</val><val>223</val></magnitude>
    <node name="Cardiolipin synthetase (EC 2.7.8.-)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-49</val></score>
     <magnitude><val></val><val>12</val></magnitude>
    </node>
    <node name="CDP-diacylglycerol--glycerol-3-phosphate 3-phosphatidyltransferase (EC 2.7.8.5)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>19</val></magnitude>
     <score><val>-18.4286</val><val>-35.6429</val></score>
    </node>
    <node name="Phosphatidate cytidylyltransferase (EC 2.7.7.41)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-61</val></score>
     <magnitude><val></val><val>3</val></magnitude>
    </node>
    <node name="Aerobic glycerol-3-phosphate dehydrogenase (EC 1.1.5.3)">
     <score><val>-17</val><val>-7</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycerol-3-phosphate dehydrogenase (EC 1.1.5.3)">
     <rank><val>function</val></rank>
     <score><val>-31</val><val>-101.833</val></score>
     <magnitude><val>2</val><val>27</val></magnitude>
    </node>
    <node name="Dihydroxyacetone kinase family protein">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-19.5</val></score>
    </node>
    <node name="Alcohol dehydrogenase (EC 1.1.1.1)">
     <rank><val>function</val></rank>
     <score><val>-21.5</val><val>-68.2</val></score>
     <magnitude><val>12</val><val>34</val></magnitude>
    </node>
    <node name="Aldehyde dehydrogenase B (EC 1.2.1.22)">
     <score><val>-44.5714</val><val>-55.2857</val></score>
     <magnitude><val>7</val><val>8</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycerol kinase (EC 2.7.1.30)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-76.1</val></score>
     <magnitude><val></val><val>41</val></magnitude>
    </node>
    <node name="Glycerate kinase (EC 2.7.1.31)">
     <magnitude><val>1</val></magnitude>
     <score><val>-7</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Aldehyde dehydrogenase (EC 1.2.1.3)">
     <rank><val>function</val></rank>
     <score><val>-30</val><val>-111.067</val></score>
     <magnitude><val>2</val><val>35</val></magnitude>
    </node>
    <node name="Glycerol-1-phosphate dehydrogenase [NAD(P)] (EC 1.1.1.261)">
     <score><val>-50.4444</val><val>-75.52</val></score>
     <magnitude><val>10</val><val>43</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Polyhydroxybutyrate metabolism">
   <magnitude><val>81</val><val>385</val></magnitude>
   <score><val>-44.5087</val><val>-77.9621</val></score>
   <rank><val>level 2</val></rank>
   <node name="3-hydroxyacyl-CoA dehydrogenase (EC 1.1.1.35)">
    <rank><val>function</val></rank>
    <score><val>-36.6667</val><val>-79.2676</val></score>
    <magnitude><val>13</val><val>110</val></magnitude>
   </node>
   <node name="3-hydroxybutyryl-CoA dehydrogenase (EC 1.1.1.157)">
    <score><val>-35.25</val><val>-82.4286</val></score>
    <magnitude><val>4</val><val>45</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="3-ketoacyl-CoA thiolase (EC 2.3.1.16)">
    <rank><val>function</val></rank>
    <magnitude><val>27</val><val>112</val></magnitude>
    <score><val>-49.92</val><val>-76.8312</val></score>
   </node>
   <node name="Acetoacetyl-CoA synthetase (EC 6.2.1.16)">
    <rank><val>function</val></rank>
    <magnitude><val>4</val></magnitude>
    <score><val>-37</val></score>
   </node>
   <node name="Enoyl-CoA hydratase (EC 4.2.1.17)">
    <magnitude><val>4</val><val>6</val></magnitude>
    <score><val>-23.25</val><val>-62.75</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Acetyl-CoA acetyltransferase (EC 2.3.1.9)">
    <magnitude><val>28</val><val>112</val></magnitude>
    <score><val>-49.3462</val><val>-76.8312</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="3-hydroxybutyryl-CoA dehydratase (EC 4.2.1.55)">
    <rank><val>function</val></rank>
    <score><val>-17</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
  </node>
  <node name="Fatty acids">
   <rank><val>level 2</val></rank>
   <magnitude><val>170</val><val>688</val></magnitude>
   <score><val>-41.4815</val><val>-71.2378</val></score>
   <node name="Acyl-CoA thioesterase II">
    <score><val>-26</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="TesB-like acyl-CoA thioesterase 3">
     <magnitude><val>1</val></magnitude>
     <score><val>-26</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Fatty acid degradation regulons">
    <rank><val>level 3</val></rank>
    <score><val>-44.3505</val><val>-70.519</val></score>
    <magnitude><val>87</val><val>250</val></magnitude>
    <node name="Butyryl-CoA dehydrogenase (EC 1.3.99.2)">
     <score><val>-54.6</val><val>-56.4444</val></score>
     <magnitude><val>21</val><val>25</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Long-chain-fatty-acid--CoA ligase (EC 6.2.1.3)">
     <magnitude><val>24</val><val>51</val></magnitude>
     <score><val>-40.9524</val><val>-80.3</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="3-ketoacyl-CoA thiolase (EC 2.3.1.16)">
     <magnitude><val>27</val><val>112</val></magnitude>
     <score><val>-49.92</val><val>-76.8312</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="2,4-dienoyl-CoA reductase [NADPH] (EC 1.3.1.34)">
     <rank><val>function</val></rank>
     <score><val>-26.2</val><val>-56.1026</val></score>
     <magnitude><val>11</val><val>56</val></magnitude>
    </node>
    <node name="Enoyl-CoA hydratase (EC 4.2.1.17)">
     <magnitude><val>4</val><val>6</val></magnitude>
     <score><val>-23.25</val><val>-62.75</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Fatty Acid Biosynthesis FASII">
    <magnitude><val>14</val><val>159</val></magnitude>
    <score><val>-19.0714</val><val>-60.2863</val></score>
    <rank><val>level 3</val></rank>
    <node name="3-oxoacyl-[acyl-carrier protein] reductase (EC 1.1.1.100)">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>157</val></magnitude>
     <score><val>-12.8571</val><val>-60.9524</val></score>
    </node>
    <node name="Enoyl-[acyl-carrier-protein] reductase [NADH] (EC 1.3.1.9)">
     <score><val>-29</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetyl-coenzyme A carboxyl transferase alpha chain (EC 6.4.1.2)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-8</val></score>
     <magnitude><val></val><val>2</val></magnitude>
    </node>
    <node name="3-oxoacyl-[acyl-carrier-protein] synthase, KASII (EC 2.3.1.41)">
     <rank><val>function</val></rank>
     <score><val>-35.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="3-oxoacyl-[acyl-carrier-protein] synthase, KASIII (EC 2.3.1.41)">
     <score><val>-24</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Enoyl-[acyl-carrier-protein] reductase [FMN] (EC 1.3.1.9)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-26</val></score>
    </node>
    <node name="Holo-[acyl-carrier protein] synthase (EC 2.7.8.7)">
     <score><val>-3</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Malonyl CoA-acyl carrier protein transacylase (EC 2.3.1.39)">
     <score><val>-24</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Fatty acid metabolism cluster">
    <rank><val>level 3</val></rank>
    <score><val>-42.6524</val><val>-78.123</val></score>
    <magnitude><val>68</val><val>279</val></magnitude>
    <node name="Enoyl-CoA hydratase (EC 4.2.1.17)">
     <magnitude><val>4</val><val>6</val></magnitude>
     <score><val>-23.25</val><val>-62.75</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Long-chain-fatty-acid--CoA ligase (EC 6.2.1.3)">
     <score><val>-40.9524</val><val>-80.3</val></score>
     <magnitude><val>24</val><val>51</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="3-ketoacyl-CoA thiolase (EC 2.3.1.16)">
     <rank><val>function</val></rank>
     <magnitude><val>27</val><val>112</val></magnitude>
     <score><val>-49.92</val><val>-76.8312</val></score>
    </node>
    <node name="3-hydroxyacyl-CoA dehydrogenase (EC 1.1.1.35)">
     <score><val>-36.6667</val><val>-79.2676</val></score>
     <magnitude><val>13</val><val>110</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Isoprenoids">
   <rank><val>level 2</val></rank>
   <score><val>-30.6545</val><val>-65.4094</val></score>
   <magnitude><val>300</val><val>1027</val></magnitude>
   <node name="Dolichol metabolism">
    <magnitude><val>3</val><val>15</val></magnitude>
    <score><val>-8.66667</val><val>-68.375</val></score>
    <rank><val>level 3</val></rank>
    <node name="Dolichol-phosphate mannosyltransferase (EC 2.4.1.83)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>15</val></magnitude>
     <score><val>-8.66667</val><val>-68.375</val></score>
    </node>
   </node>
   <node name="Polyprenyl Diphosphate Biosynthesis">
    <score><val>-21.7414</val><val>-60.295</val></score>
    <magnitude><val>50</val><val>152</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Geranylgeranyl pyrophosphate synthetase (EC 2.5.1.29)">
     <magnitude><val>11</val><val>28</val></magnitude>
     <score><val>-19.3333</val><val>-49.85</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Octaprenyl-diphosphate synthase (EC 2.5.1.-)">
     <magnitude><val>3</val><val>6</val></magnitude>
     <score><val>-20.6667</val><val>-24.8</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Geranylgeranyl reductase (EC 1.3.1.83)">
     <score><val>-20.7692</val><val>-73.907</val></score>
     <magnitude><val>16</val><val>62</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Undecaprenyl pyrophosphate synthetase (EC 2.5.1.31)">
     <rank><val>function</val></rank>
     <score><val>-29.7143</val><val>-58.65</val></score>
     <magnitude><val>9</val><val>28</val></magnitude>
    </node>
    <node name="Geranyltranstransferase (farnesyldiphosphate synthase) (EC 2.5.1.10)">
     <rank><val>function</val></rank>
     <magnitude><val>11</val><val>28</val></magnitude>
     <score><val>-19.3333</val><val>-49.85</val></score>
    </node>
   </node>
   <node name="Archaeal lipids">
    <magnitude><val>117</val><val>456</val></magnitude>
    <score><val>-35.0873</val><val>-67.2617</val></score>
    <rank><val>level 3</val></rank>
    <node name="Digeranylgeranylglycerophospholipid reductase">
     <score><val>-32</val><val>-85.6667</val></score>
     <magnitude><val>1</val><val>25</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Diphosphomevalonate decarboxylase (EC 4.1.1.33)">
     <rank><val>function</val></rank>
     <magnitude><val>8</val><val>22</val></magnitude>
     <score><val>-16.1667</val><val>-46.375</val></score>
    </node>
    <node name="Hydroxymethylglutaryl-CoA reductase (EC 1.1.1.34)">
     <rank><val>function</val></rank>
     <score><val>-20.3333</val><val>-47.6154</val></score>
     <magnitude><val>5</val><val>17</val></magnitude>
    </node>
    <node name="Geranylgeranyl pyrophosphate synthetase (EC 2.5.1.29)">
     <magnitude><val>11</val><val>28</val></magnitude>
     <score><val>-19.3333</val><val>-49.85</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Geranyltranstransferase (farnesyldiphosphate synthase) (EC 2.5.1.10)">
     <rank><val>function</val></rank>
     <magnitude><val>11</val><val>28</val></magnitude>
     <score><val>-19.3333</val><val>-49.85</val></score>
    </node>
    <node name="Dimethylallyltransferase (EC 2.5.1.1)">
     <magnitude><val>11</val><val>28</val></magnitude>
     <score><val>-19.3333</val><val>-49.85</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="(S)-3-O-geranylgeranylglyceryl phosphate synthase">
     <rank><val>function</val></rank>
     <magnitude><val>7</val><val>33</val></magnitude>
     <score><val>-43.1429</val><val>-64.24</val></score>
    </node>
    <node name="Hydroxymethylglutaryl-CoA synthase (EC 2.3.3.10)">
     <rank><val>function</val></rank>
     <score><val>-116</val><val>-91.2105</val></score>
     <magnitude><val>4</val><val>31</val></magnitude>
    </node>
    <node name="Glycerol-1-phosphate dehydrogenase [NAD(P)] (EC 1.1.1.261)">
     <score><val>-50.4444</val><val>-75.52</val></score>
     <magnitude><val>10</val><val>43</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Acetyl-CoA acetyltransferase (EC 2.3.1.9)">
     <magnitude><val>28</val><val>112</val></magnitude>
     <score><val>-49.3462</val><val>-76.8312</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="(S)-2,3-di-O-geranylgeranylglyceryl phosphate synthase">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>21</val></magnitude>
     <score><val>-22.4</val><val>-55.5714</val></score>
    </node>
    <node name="Isopentenyl phosphate kinase">
     <score><val>-6.5</val><val>-49.5833</val></score>
     <magnitude><val>3</val><val>24</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Isopentenyl-diphosphate delta-isomerase, FMN-dependent (EC 5.3.3.2)">
     <magnitude><val>12</val><val>44</val></magnitude>
     <score><val>-33.1818</val><val>-76.2647</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="polyprenyl synthesis">
    <rank><val>level 3</val></rank>
    <score><val>-36.5</val></score>
    <magnitude><val>2</val></magnitude>
    <node name="4-hydroxy-3-methylbut-2-enyl diphosphate reductase (EC 1.17.1.2)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-51</val></score>
    </node>
    <node name="1-deoxy-D-xylulose 5-phosphate reductoisomerase (EC 1.1.1.267)">
     <rank><val>function</val></rank>
     <score><val>-22</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
   </node>
   <node name="Carotenoids">
    <magnitude><val>28</val><val>57</val></magnitude>
    <score><val>-19.6547</val><val>-49.3789</val></score>
    <rank><val>level 3</val></rank>
    <node name="Squalene--hopene cyclase (EC 5.4.99.17)">
     <score><val>-34</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Geranyltranstransferase (farnesyldiphosphate synthase) (EC 2.5.1.10)">
     <rank><val>function</val></rank>
     <score><val>-19.3333</val><val>-49.85</val></score>
     <magnitude><val>11</val><val>28</val></magnitude>
    </node>
    <node name="Beta-carotene ketolase (EC 1.14.-.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-23</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Geranylgeranyl pyrophosphate synthetase (EC 2.5.1.29)">
     <rank><val>function</val></rank>
     <magnitude><val>11</val><val>28</val></magnitude>
     <score><val>-19.3333</val><val>-49.85</val></score>
    </node>
    <node name="Phytoene dehydrogenase (EC 1.14.99.-)">
     <rank><val>function</val></rank>
     <score><val>-2</val><val>-23</val></score>
     <magnitude><val>1</val><val>1</val></magnitude>
    </node>
    <node name="Regulator of carotenoid biosynthesis">
     <magnitude><val>1</val></magnitude>
     <score><val>-18</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Phytoene synthase (EC 2.5.1.32)">
     <rank><val>function</val></rank>
     <score><val>-24</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
   </node>
   <node name="Isoprenoid Biosynthesis">
    <score><val>-33.5474</val><val>-67.7206</val></score>
    <magnitude><val>100</val><val>347</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Isopentenyl-diphosphate delta-isomerase, FMN-dependent (EC 5.3.3.2)">
     <magnitude><val>12</val><val>44</val></magnitude>
     <score><val>-33.1818</val><val>-76.2647</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Undecaprenyl pyrophosphate synthetase (EC 2.5.1.31)">
     <score><val>-29.7143</val><val>-58.65</val></score>
     <magnitude><val>9</val><val>28</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Isopentenyl phosphate kinase">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>24</val></magnitude>
     <score><val>-6.5</val><val>-49.5833</val></score>
    </node>
    <node name="Acetyl-CoA acetyltransferase (EC 2.3.1.9)">
     <magnitude><val>28</val><val>112</val></magnitude>
     <score><val>-49.3462</val><val>-76.8312</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Hydroxymethylglutaryl-CoA synthase (EC 2.3.3.10)">
     <score><val>-116</val><val>-91.2105</val></score>
     <magnitude><val>4</val><val>31</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Geranyltranstransferase (farnesyldiphosphate synthase) (EC 2.5.1.10)">
     <magnitude><val>11</val><val>28</val></magnitude>
     <score><val>-19.3333</val><val>-49.85</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="1-deoxy-D-xylulose 5-phosphate synthase (EC 2.2.1.7)">
     <score><val>-25</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Dimethylallyltransferase (EC 2.5.1.1)">
     <rank><val>function</val></rank>
     <magnitude><val>11</val><val>28</val></magnitude>
     <score><val>-19.3333</val><val>-49.85</val></score>
    </node>
    <node name="4-hydroxy-3-methylbut-2-enyl diphosphate reductase (EC 1.17.1.2)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-51</val></score>
    </node>
    <node name="Candidate gene for the hypothesized phosphomevalonate decarboxylase">
     <score><val>-11.6</val><val>-96.7143</val></score>
     <magnitude><val>6</val><val>13</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="1-deoxy-D-xylulose 5-phosphate reductoisomerase (EC 1.1.1.267)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-22</val></score>
    </node>
    <node name="Hydroxymethylglutaryl-CoA reductase (EC 1.1.1.34)">
     <rank><val>function</val></rank>
     <score><val>-20.3333</val><val>-47.6154</val></score>
     <magnitude><val>5</val><val>17</val></magnitude>
    </node>
    <node name="Diphosphomevalonate decarboxylase (EC 4.1.1.33)">
     <rank><val>function</val></rank>
     <score><val>-16.1667</val><val>-46.375</val></score>
     <magnitude><val>8</val><val>22</val></magnitude>
    </node>
   </node>
  </node>
 </node>
 <node name="Respiration">
  <rank><val>level 1</val></rank>
  <magnitude><val>302</val><val>1262</val></magnitude>
  <score><val>-40.619</val><val>-74.1187</val></score>
  <node name="Electron accepting reactions">
   <rank><val>level 2</val></rank>
   <magnitude><val>81</val><val>437</val></magnitude>
   <score><val>-43.7778</val><val>-79.4664</val></score>
   <node name="Anaerobic respiratory reductases">
    <score><val>-45.12</val><val>-56.4444</val></score>
    <magnitude><val>30</val><val>25</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Arsenate reductase (EC 1.20.4.1)">
     <magnitude><val>1</val></magnitude>
     <score><val>-3</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Polysulfide reductase, subunit B, putative">
     <magnitude><val>1</val></magnitude>
     <score><val>-43</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Ferredoxin reductase">
     <rank><val>function</val></rank>
     <score><val>-9</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="Anaerobic dehydrogenases, typically selenocysteine-containing">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-85</val></score>
    </node>
    <node name="Butyryl-CoA dehydrogenase (EC 1.3.99.2)">
     <magnitude><val>21</val><val>25</val></magnitude>
     <score><val>-54.6</val><val>-56.4444</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Coenzyme F420-dependent oxidoreductase">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-26</val></score>
    </node>
    <node name="Heterodisulfide reductase, cytochrome reductase subunit">
     <score><val>-5</val></score>
     <magnitude><val>3</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Tetrathionate respiration">
    <magnitude><val></val><val>2</val></magnitude>
    <score><val>0</val><val>-12.5</val></score>
    <rank><val>level 3</val></rank>
    <node name="Tetrathionate reductase subunit C">
     <score><val>0</val><val>-14</val></score>
     <magnitude><val></val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Tetrathionate reductase subunit A">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-11</val></score>
     <magnitude><val></val><val>1</val></magnitude>
    </node>
   </node>
   <node name="Terminal AA3-600 quinol oxidase">
    <rank><val>level 3</val></rank>
    <score><val>-79</val></score>
    <magnitude><val>1</val></magnitude>
    <node name="AA3-600 quinol oxidase subunit I">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-79</val></score>
    </node>
   </node>
   <node name="Terminal cytochrome oxidases">
    <magnitude><val>15</val><val>160</val></magnitude>
    <score><val>-42.5467</val><val>-77.4548</val></score>
    <rank><val>level 3</val></rank>
    <node name="putative Cytochrome bd2, subunit I">
     <magnitude><val>7</val><val>111</val></magnitude>
     <score><val>-33.6</val><val>-77.7439</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Transport ATP-binding protein CydC">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-6.5</val></score>
    </node>
    <node name="Cytochrome d ubiquinol oxidase subunit II (EC 1.10.3.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-53</val></score>
    </node>
    <node name="AA3-600 quinol oxidase subunit I">
     <score><val>-79</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Cytochrome O ubiquinol oxidase subunit I (EC 1.10.3.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-165</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cytochrome d ubiquinol oxidase subunit I (EC 1.10.3.-)">
     <magnitude><val>3</val></magnitude>
     <score><val>-31</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="putative Cytochrome bd2, subunit II">
     <rank><val>function</val></rank>
     <magnitude><val></val><val>49</val></magnitude>
     <score><val>0</val><val>-76.8</val></score>
    </node>
   </node>
   <node name="Ubiquinone Menaquinone-cytochrome c reductase complexes">
    <score><val>-51.125</val><val>-42.8095</val></score>
    <magnitude><val>16</val><val>21</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Quinol oxidase-2, sulfocyanin (blue copper protein) (SoxE)">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-81.5</val></score>
     <magnitude><val></val><val>4</val></magnitude>
    </node>
    <node name="Ubiquinol-cytochrome C reductase iron-sulfur subunit (EC 1.10.2.2)">
     <rank><val>function</val></rank>
     <score><val>-24.6667</val><val>-30</val></score>
     <magnitude><val>3</val><val>3</val></magnitude>
    </node>
    <node name="Ubiquinol--cytochrome c reductase, cytochrome B subunit (EC 1.10.2.2)">
     <magnitude><val>13</val><val>14</val></magnitude>
     <score><val>-57.2308</val><val>-34.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Terminal cytochrome d ubiquinol oxidases">
    <rank><val>level 3</val></rank>
    <magnitude><val>13</val><val>160</val></magnitude>
    <score><val>-30.3231</val><val>-77.4548</val></score>
    <node name="Transport ATP-binding protein CydC">
     <magnitude><val>2</val></magnitude>
     <score><val>-6.5</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="putative Cytochrome bd2, subunit I">
     <magnitude><val>7</val><val>111</val></magnitude>
     <score><val>-33.6</val><val>-77.7439</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cytochrome d ubiquinol oxidase subunit II (EC 1.10.3.-)">
     <magnitude><val>1</val></magnitude>
     <score><val>-53</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cytochrome d ubiquinol oxidase subunit I (EC 1.10.3.-)">
     <magnitude><val>3</val></magnitude>
     <score><val>-31</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="putative Cytochrome bd2, subunit II">
     <score><val>0</val><val>-76.8</val></score>
     <magnitude><val></val><val>49</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Terminal cytochrome O ubiquinol oxidase">
    <magnitude><val>1</val></magnitude>
    <score><val>-165</val></score>
    <rank><val>level 3</val></rank>
    <node name="Cytochrome O ubiquinol oxidase subunit I (EC 1.10.3.-)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-165</val></score>
    </node>
   </node>
   <node name="Terminal cytochrome C oxidases">
    <rank><val>level 3</val></rank>
    <magnitude><val>5</val><val>69</val></magnitude>
    <score><val>-19.6</val><val>-110.234</val></score>
    <node name="Cytochrome c oxidase polypeptide I (EC 1.9.3.1)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>29</val></magnitude>
     <score><val>-15.3333</val><val>-115.652</val></score>
    </node>
    <node name="Cytochrome c oxidase polypeptide II (EC 1.9.3.1)">
     <magnitude><val>1</val><val>11</val></magnitude>
     <score><val>-16</val><val>-81.6667</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Cytochrome c oxidase polypeptide III (EC 1.9.3.1)">
     <rank><val>function</val></rank>
     <score><val>-36</val><val>-115.652</val></score>
     <magnitude><val>1</val><val>29</val></magnitude>
    </node>
   </node>
  </node>
  <node name="Soluble cytochromes and functionally related electron carriers">
   <magnitude><val>10</val><val>61</val></magnitude>
   <score><val>-16.3</val><val>-61.9091</val></score>
   <rank><val>level 2</val></rank>
   <node name="Ferredoxin, 2Fe-2S">
    <rank><val>function</val></rank>
    <score><val>-12</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Ferredoxin">
    <score><val>-16.7778</val><val>-61.9091</val></score>
    <magnitude><val>9</val><val>61</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Biogenesis of cytochrome c oxidases">
   <rank><val>level 2</val></rank>
   <magnitude><val>7</val><val>39</val></magnitude>
   <score><val>-21</val><val>-45.4783</val></score>
   <node name="Heme O synthase, protoheme IX farnesyltransferase (EC 2.5.1.-) COX10-CtaB">
    <rank><val>function</val></rank>
    <score><val>-21</val><val>-45.4783</val></score>
    <magnitude><val>7</val><val>39</val></magnitude>
   </node>
  </node>
  <node name="Formate hydrogenase">
   <magnitude><val>43</val><val>182</val></magnitude>
   <score><val>-45.588</val><val>-75.6358</val></score>
   <rank><val>level 2</val></rank>
   <node name="Hydrogenase-4 component E (EC 1.-.-.-)">
    <rank><val>function</val></rank>
    <score><val>-47</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Putative formate dehydrogenase oxidoreductase protein">
    <score><val>-42</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Hydrogenase-4 component B (EC 1.-.-.-)">
    <score><val>-52</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Formate dehydrogenase related protein">
    <score><val>-46</val><val>-87</val></score>
    <magnitude><val>2</val><val>75</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Formate hydrogenlyase hypothetical subunit">
    <magnitude><val>1</val><val>9</val></magnitude>
    <score><val>-17</val><val>-32.2222</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Hydrogenase-4 component F (EC 1.-.-.-)">
    <score><val>-35.3333</val></score>
    <magnitude><val>3</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Formate hydrogenlyase subunit 3">
    <rank><val>function</val></rank>
    <magnitude><val>13</val><val>38</val></magnitude>
    <score><val>-60.0833</val><val>-86.2903</val></score>
   </node>
   <node name="Formate hydrogenlyase subunit 7">
    <score><val>-33.25</val><val>-105</val></score>
    <magnitude><val>4</val><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Hypothetical protein YrhD">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>5</val></magnitude>
    <score><val>0</val><val>-54</val></score>
   </node>
   <node name="Formate hydrogenlyase subunit 5">
    <magnitude><val></val><val>9</val></magnitude>
    <score><val>0</val><val>-76.375</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Formate hydrogenlyase subunit 3/Multisubunit Na+/H+ antiporter, MnhD subunit">
    <magnitude><val>3</val><val>17</val></magnitude>
    <score><val>-108</val><val>-60.6429</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Formate hydrogenlyase subunit 4">
    <magnitude><val>6</val><val>9</val></magnitude>
    <score><val>-33.3333</val><val>-63</val></score>
    <rank><val>function</val></rank>
   </node>
   <node name="Formate dehydrogenase O alpha subunit (EC 1.2.1.2)">
    <score><val>-10</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="NAD-dependent formate dehydrogenase alpha subunit">
    <rank><val>function</val></rank>
    <score><val>-65</val></score>
    <magnitude><val>1</val></magnitude>
   </node>
   <node name="Formate dehydrogenase chain D (EC 1.2.1.2)">
    <magnitude><val>6</val><val>19</val></magnitude>
    <score><val>-15.2</val><val>-53.2308</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Electron donating reactions">
   <rank><val>level 2</val></rank>
   <magnitude><val>107</val><val>312</val></magnitude>
   <score><val>-41.8454</val><val>-88.0812</val></score>
   <node name="Respiratory dehydrogenases 1">
    <magnitude><val>26</val><val>80</val></magnitude>
    <score><val>-29.2863</val><val>-80.7481</val></score>
    <rank><val>level 3</val></rank>
    <node name="Glycerol-3-phosphate dehydrogenase (EC 1.1.5.3)">
     <score><val>-31</val><val>-101.833</val></score>
     <magnitude><val>2</val><val>27</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Aerobic glycerol-3-phosphate dehydrogenase (EC 1.1.5.3)">
     <score><val>-17</val><val>-7</val></score>
     <magnitude><val>2</val><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycerol-1-phosphate dehydrogenase [NAD(P)] (EC 1.1.1.261)">
     <magnitude><val>10</val><val>43</val></magnitude>
     <score><val>-50.4444</val><val>-75.52</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Proline dehydrogenase (EC 1.5.99.8) (Proline oxidase)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val><val>1</val></magnitude>
     <score><val>-15.5</val><val>-25</val></score>
    </node>
    <node name="NADH dehydrogenase (EC 1.6.99.3)">
     <magnitude><val>8</val><val>8</val></magnitude>
     <score><val>-10</val><val>-53.875</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="Glycerol dehydrogenase (EC 1.1.1.6)">
     <rank><val>function</val></rank>
     <score><val>-15</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="L-lactate dehydrogenase (EC 1.1.2.3)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-35</val></score>
    </node>
   </node>
   <node name="Hydrogenases">
    <score><val>-116.5</val></score>
    <magnitude><val>2</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="Uptake hydrogenase large subunit (EC 1.12.99.6)">
     <rank><val>function</val></rank>
     <magnitude><val>2</val></magnitude>
     <score><val>-116.5</val></score>
    </node>
   </node>
   <node name="Formate dehydrogenase">
    <rank><val>level 3</val></rank>
    <score><val>-10</val><val>-106.375</val></score>
    <magnitude><val>1</val><val>19</val></magnitude>
    <node name="Formate dehydrogenase H (EC 1.2.1.2)">
     <score><val>0</val><val>-106.375</val></score>
     <magnitude><val></val><val>19</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Formate dehydrogenase O alpha subunit (EC 1.2.1.2)">
     <magnitude><val>1</val></magnitude>
     <score><val>-10</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Succinate dehydrogenase">
    <rank><val>level 3</val></rank>
    <score><val>-59.1875</val><val>-108.827</val></score>
    <magnitude><val>16</val><val>93</val></magnitude>
    <node name="Succinate dehydrogenase hydrophobic membrane anchor protein">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-37.75</val></score>
     <magnitude><val></val><val>7</val></magnitude>
    </node>
    <node name="Succinate dehydrogenase flavoprotein subunit (EC 1.3.99.1)">
     <rank><val>function</val></rank>
     <score><val>-65.7692</val><val>-125.667</val></score>
     <magnitude><val>13</val><val>53</val></magnitude>
    </node>
    <node name="Succinate dehydrogenase iron-sulfur protein (EC 1.3.99.1)">
     <score><val>-30.6667</val><val>-94.9231</val></score>
     <magnitude><val>3</val><val>18</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="Putative succinate dehydrogenase cytochrome b subunit">
     <rank><val>function</val></rank>
     <score><val>0</val><val>-99.1818</val></score>
     <magnitude><val></val><val>15</val></magnitude>
    </node>
   </node>
   <node name="CO Dehydrogenase">
    <magnitude><val>14</val><val>32</val></magnitude>
    <score><val>-34.2857</val><val>-57.4643</val></score>
    <rank><val>level 3</val></rank>
    <node name="Carbon monoxide dehydrogenase small chain (EC 1.2.99.2)">
     <rank><val>function</val></rank>
     <magnitude><val>4</val><val>8</val></magnitude>
     <score><val>-38</val><val>-28.4286</val></score>
    </node>
    <node name="Carbon monoxide dehydrogenase medium chain (EC 1.2.99.2)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val><val>24</val></magnitude>
     <score><val>-23</val><val>-67.1429</val></score>
    </node>
    <node name="Carbon monoxide dehydrogenase large chain (EC 1.2.99.2)">
     <rank><val>function</val></rank>
     <score><val>-52.6667</val></score>
     <magnitude><val>3</val></magnitude>
    </node>
    <node name="carbon monoxide dehydrogenase E protein">
     <rank><val>function</val></rank>
     <score><val>-42.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="Carbon monoxide dehydrogenase large chain (EC 1.2.99.2) parolog without usual motifs">
     <score><val>-8.5</val></score>
     <magnitude><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="carbon monoxide dehydrogenase D protein">
     <rank><val>function</val></rank>
     <score><val>-28</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="carbon monoxide dehydrogenase G protein">
     <score><val>-17</val></score>
     <magnitude><val>1</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="Respiratory Complex I">
    <rank><val>level 3</val></rank>
    <score><val>-41.5559</val><val>-80.0064</val></score>
    <magnitude><val>45</val><val>88</val></magnitude>
    <node name="NADH-ubiquinone oxidoreductase chain F (EC 1.6.5.3)">
     <magnitude><val>1</val></magnitude>
     <score><val>-43</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="NADH-ubiquinone oxidoreductase chain K (EC 1.6.5.3)">
     <magnitude><val>1</val></magnitude>
     <score><val>-9</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="NADH-ubiquinone oxidoreductase chain J (EC 1.6.5.3)">
     <magnitude><val>2</val></magnitude>
     <score><val>-17</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="NADH-ubiquinone oxidoreductase chain C (EC 1.6.5.3)">
     <rank><val>function</val></rank>
     <score><val>-41</val><val>-61.1429</val></score>
     <magnitude><val>4</val><val>9</val></magnitude>
    </node>
    <node name="NADH-ubiquinone oxidoreductase chain M (EC 1.6.5.3)">
     <magnitude><val>4</val><val>11</val></magnitude>
     <score><val>-30.25</val><val>-103.3</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="NADH ubiquinone oxidoreductase chain A (EC 1.6.5.3)">
     <rank><val>function</val></rank>
     <score><val>-40</val><val>-43.625</val></score>
     <magnitude><val>1</val><val>12</val></magnitude>
    </node>
    <node name="NADH-ubiquinone oxidoreductase chain L (EC 1.6.5.3)">
     <score><val>-30.8571</val><val>-123.2</val></score>
     <magnitude><val>7</val><val>6</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="NADH-ubiquinone oxidoreductase chain D (EC 1.6.5.3)">
     <score><val>-64.7143</val><val>-96.0667</val></score>
     <magnitude><val>8</val><val>18</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="NADH-ubiquinone oxidoreductase chain G (EC 1.6.5.3)">
     <rank><val>function</val></rank>
     <magnitude><val>1</val></magnitude>
     <score><val>-5</val></score>
    </node>
    <node name="NADH-ubiquinone oxidoreductase chain N (EC 1.6.5.3)">
     <rank><val>function</val></rank>
     <magnitude><val>3</val><val>11</val></magnitude>
     <score><val>-19.6667</val><val>-99.2222</val></score>
    </node>
    <node name="NADH-ubiquinone oxidoreductase chain B (EC 1.6.5.3)">
     <rank><val>function</val></rank>
     <magnitude><val>6</val><val>6</val></magnitude>
     <score><val>-47.8</val><val>-71</val></score>
    </node>
    <node name="NADH-ubiquinone oxidoreductase chain H (EC 1.6.5.3)">
     <magnitude><val>5</val><val>13</val></magnitude>
     <score><val>-67.5</val><val>-61.8182</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="NADH dehydrogenase subunit 5">
     <rank><val>function</val></rank>
     <score><val>-2</val></score>
     <magnitude><val>1</val></magnitude>
    </node>
    <node name="NADH-ubiquinone oxidoreductase chain I (EC 1.6.5.3)">
     <score><val>-35</val><val>-20.5</val></score>
     <magnitude><val>1</val><val>2</val></magnitude>
     <rank><val>function</val></rank>
    </node>
   </node>
   <node name="NiFe hydrogenase maturation">
    <magnitude><val>3</val></magnitude>
    <score><val>-58.6667</val></score>
    <rank><val>level 3</val></rank>
    <node name="[NiFe] hydrogenase metallocenter assembly protein HypF">
     <rank><val>function</val></rank>
     <score><val>-44.5</val></score>
     <magnitude><val>2</val></magnitude>
    </node>
    <node name="[NiFe] hydrogenase metallocenter assembly protein HypE">
     <magnitude><val>1</val></magnitude>
     <score><val>-87</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Quinone oxidoreductase family">
   <magnitude><val>8</val><val>1</val></magnitude>
   <score><val>-39.0417</val><val>-53</val></score>
   <rank><val>level 2</val></rank>
   <node name="YhdH, a putative quinone oxidoreductase">
    <rank><val>function</val></rank>
    <magnitude><val>3</val><val>1</val></magnitude>
    <score><val>-68</val><val>-53</val></score>
   </node>
   <node name="Quinone oxidoreductase (EC 1.6.5.5)">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-7</val></score>
   </node>
   <node name="Putative NADP-dependent oxidoreductase PA1648">
    <score><val>-25.3333</val></score>
    <magnitude><val>4</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Biogenesis of cbb3-type cytochrome c oxidases">
   <rank><val>level 2</val></rank>
   <score><val>-24</val></score>
   <magnitude><val>1</val></magnitude>
   <node name="Type cbb3 cytochrome oxidase biogenesis protein CcoI">
    <rank><val>function</val></rank>
    <magnitude><val>1</val></magnitude>
    <score><val>-24</val></score>
   </node>
  </node>
  <node name="ATP synthases">
   <magnitude><val>36</val><val>196</val></magnitude>
   <score><val>-39.691</val><val>-58.3454</val></score>
   <rank><val>level 2</val></rank>
   <node name="V-Type ATP synthase">
    <score><val>-39.691</val><val>-58.3454</val></score>
    <magnitude><val>36</val><val>196</val></magnitude>
    <rank><val>level 3</val></rank>
    <node name="V-type ATP synthase subunit A (EC 3.6.3.14)">
     <rank><val>function</val></rank>
     <score><val>-52.8182</val><val>-58.3077</val></score>
     <magnitude><val>11</val><val>54</val></magnitude>
    </node>
    <node name="V-type ATP synthase subunit I (EC 3.6.3.14)">
     <score><val>-18.875</val><val>-65.9048</val></score>
     <magnitude><val>9</val><val>23</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="V-type ATP synthase subunit F (EC 3.6.3.14)">
     <magnitude><val></val><val>10</val></magnitude>
     <score><val>0</val><val>-18.4444</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="V-type ATP synthase subunit C (EC 3.6.3.14)">
     <rank><val>function</val></rank>
     <score><val>-6</val><val>-81.6667</val></score>
     <magnitude><val>3</val><val>25</val></magnitude>
    </node>
    <node name="V-type ATP synthase subunit B (EC 3.6.3.14)">
     <rank><val>function</val></rank>
     <score><val>-69.875</val><val>-70.8621</val></score>
     <magnitude><val>8</val><val>37</val></magnitude>
    </node>
    <node name="V-type ATP synthase subunit E (EC 3.6.3.14)">
     <score><val>0</val><val>-56.3846</val></score>
     <magnitude><val></val><val>28</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="V-type ATP synthase subunit D (EC 3.6.3.14)">
     <score><val>-18.25</val><val>-22</val></score>
     <magnitude><val>4</val><val>4</val></magnitude>
     <rank><val>function</val></rank>
    </node>
    <node name="V-type ATP synthase subunit G (EC 3.6.3.14)">
     <magnitude><val></val><val>6</val></magnitude>
     <score><val>0</val><val>-30</val></score>
     <rank><val>function</val></rank>
    </node>
    <node name="V-type ATP synthase subunit K (EC 3.6.3.14)">
     <magnitude><val>1</val><val>9</val></magnitude>
     <score><val>-28</val><val>-8.5</val></score>
     <rank><val>function</val></rank>
    </node>
   </node>
  </node>
  <node name="Biogenesis of c-type cytochromes">
   <score><val>-24.5</val></score>
   <magnitude><val>2</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="ABC transporter involved in cytochrome c biogenesis, CcmB subunit">
    <score><val>-12</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Cytochrome c heme lyase subunit CcmF">
    <score><val>-37</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Carbon monoxide dehydrogenase maturation factors">
   <rank><val>level 2</val></rank>
   <score><val>-32.5</val></score>
   <magnitude><val>4</val></magnitude>
   <node name="carbon monoxide dehydrogenase E protein">
    <rank><val>function</val></rank>
    <magnitude><val>2</val></magnitude>
    <score><val>-42.5</val></score>
   </node>
   <node name="carbon monoxide dehydrogenase D protein">
    <score><val>-28</val></score>
    <magnitude><val>1</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="carbon monoxide dehydrogenase G protein">
    <magnitude><val>1</val></magnitude>
    <score><val>-17</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Methanogenesis strays">
   <rank><val>level 2</val></rank>
   <magnitude><val>3</val><val>8</val></magnitude>
   <score><val>-9.66667</val><val>-10</val></score>
   <node name="UPF0129 protein MJ1474">
    <magnitude><val>3</val><val>8</val></magnitude>
    <score><val>-9.66667</val><val>-10</val></score>
    <rank><val>function</val></rank>
   </node>
  </node>
  <node name="Carbon monoxide induced hydrogenase">
   <score><val>0</val><val>-17.1183</val></score>
   <magnitude><val></val><val>26</val></magnitude>
   <rank><val>level 2</val></rank>
   <node name="CO dehydrogenase accessory protein CooC (nickel insertion)">
    <score><val>0</val><val>-16</val></score>
    <magnitude><val></val><val>5</val></magnitude>
    <rank><val>function</val></rank>
   </node>
   <node name="Carbon monoxide dehydrogenase CooS subunit (EC 1.2.99.2)">
    <rank><val>function</val></rank>
    <magnitude><val></val><val>21</val></magnitude>
    <score><val>0</val><val>-17.3846</val></score>
   </node>
  </node>
 </node>
</node>
  </krona>
</div></body></html>
